// akeyless/dist/model/ActiveDirectoryPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ActiveDirectoryPayload model module.
 * @module model/ActiveDirectoryPayload
 * @version 2.20.0
 */
var ActiveDirectoryPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActiveDirectoryPayload</code>.
   * @alias module:model/ActiveDirectoryPayload
   */
  function ActiveDirectoryPayload() {
    _classCallCheck(this, ActiveDirectoryPayload);

    ActiveDirectoryPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ActiveDirectoryPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActiveDirectoryPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ActiveDirectoryPayload} obj Optional instance to populate.
     * @return {module:model/ActiveDirectoryPayload} The populated <code>ActiveDirectoryPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActiveDirectoryPayload();

        if (data.hasOwnProperty('active_directory_target_id')) {
          obj['active_directory_target_id'] = _ApiClient["default"].convertToType(data['active_directory_target_id'], 'Number');
        }

        if (data.hasOwnProperty('auto_rotate')) {
          obj['auto_rotate'] = _ApiClient["default"].convertToType(data['auto_rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('auto_rotate_interval_in_days')) {
          obj['auto_rotate_interval_in_days'] = _ApiClient["default"].convertToType(data['auto_rotate_interval_in_days'], 'Number');
        }

        if (data.hasOwnProperty('auto_rotate_rotation_hour')) {
          obj['auto_rotate_rotation_hour'] = _ApiClient["default"].convertToType(data['auto_rotate_rotation_hour'], 'Number');
        }

        if (data.hasOwnProperty('computer_base_dn')) {
          obj['computer_base_dn'] = _ApiClient["default"].convertToType(data['computer_base_dn'], 'String');
        }

        if (data.hasOwnProperty('discover_local_users')) {
          obj['discover_local_users'] = _ApiClient["default"].convertToType(data['discover_local_users'], 'Boolean');
        }

        if (data.hasOwnProperty('domain_name')) {
          obj['domain_name'] = _ApiClient["default"].convertToType(data['domain_name'], 'String');
        }

        if (data.hasOwnProperty('domain_server_targets_path_template')) {
          obj['domain_server_targets_path_template'] = _ApiClient["default"].convertToType(data['domain_server_targets_path_template'], 'String');
        }

        if (data.hasOwnProperty('domain_users_rotated_secrets_path_template')) {
          obj['domain_users_rotated_secrets_path_template'] = _ApiClient["default"].convertToType(data['domain_users_rotated_secrets_path_template'], 'String');
        }

        if (data.hasOwnProperty('enable_rdp_sra')) {
          obj['enable_rdp_sra'] = _ApiClient["default"].convertToType(data['enable_rdp_sra'], 'Boolean');
        }

        if (data.hasOwnProperty('local_users_ignore_list')) {
          obj['local_users_ignore_list'] = _ApiClient["default"].convertToType(data['local_users_ignore_list'], {
            'String': 'Boolean'
          });
        }

        if (data.hasOwnProperty('local_users_rotated_secrets_path_template')) {
          obj['local_users_rotated_secrets_path_template'] = _ApiClient["default"].convertToType(data['local_users_rotated_secrets_path_template'], 'String');
        }

        if (data.hasOwnProperty('ssh_port')) {
          obj['ssh_port'] = _ApiClient["default"].convertToType(data['ssh_port'], 'String');
        }

        if (data.hasOwnProperty('user_base_dn')) {
          obj['user_base_dn'] = _ApiClient["default"].convertToType(data['user_base_dn'], 'String');
        }

        if (data.hasOwnProperty('user_groups')) {
          obj['user_groups'] = _ApiClient["default"].convertToType(data['user_groups'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return ActiveDirectoryPayload;
}();
/**
 * @member {Number} active_directory_target_id
 */


ActiveDirectoryPayload.prototype['active_directory_target_id'] = undefined;
/**
 * @member {Boolean} auto_rotate
 */

ActiveDirectoryPayload.prototype['auto_rotate'] = undefined;
/**
 * @member {Number} auto_rotate_interval_in_days
 */

ActiveDirectoryPayload.prototype['auto_rotate_interval_in_days'] = undefined;
/**
 * @member {Number} auto_rotate_rotation_hour
 */

ActiveDirectoryPayload.prototype['auto_rotate_rotation_hour'] = undefined;
/**
 * @member {String} computer_base_dn
 */

ActiveDirectoryPayload.prototype['computer_base_dn'] = undefined;
/**
 * @member {Boolean} discover_local_users
 */

ActiveDirectoryPayload.prototype['discover_local_users'] = undefined;
/**
 * @member {String} domain_name
 */

ActiveDirectoryPayload.prototype['domain_name'] = undefined;
/**
 * @member {String} domain_server_targets_path_template
 */

ActiveDirectoryPayload.prototype['domain_server_targets_path_template'] = undefined;
/**
 * @member {String} domain_users_rotated_secrets_path_template
 */

ActiveDirectoryPayload.prototype['domain_users_rotated_secrets_path_template'] = undefined;
/**
 * @member {Boolean} enable_rdp_sra
 */

ActiveDirectoryPayload.prototype['enable_rdp_sra'] = undefined;
/**
 * @member {Object.<String, Boolean>} local_users_ignore_list
 */

ActiveDirectoryPayload.prototype['local_users_ignore_list'] = undefined;
/**
 * @member {String} local_users_rotated_secrets_path_template
 */

ActiveDirectoryPayload.prototype['local_users_rotated_secrets_path_template'] = undefined;
/**
 * @member {String} ssh_port
 */

ActiveDirectoryPayload.prototype['ssh_port'] = undefined;
/**
 * @member {String} user_base_dn
 */

ActiveDirectoryPayload.prototype['user_base_dn'] = undefined;
/**
 * @member {Array.<String>} user_groups
 */

ActiveDirectoryPayload.prototype['user_groups'] = undefined;
var _default = ActiveDirectoryPayload;
exports["default"] = _default;
