// akeyless/dist/model/AkeylessGatewayConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AdminsConfigPart = _interopRequireDefault(require("./AdminsConfigPart"));

var _CFConfigPart = _interopRequireDefault(require("./CFConfigPart"));

var _CacheConfigPart = _interopRequireDefault(require("./CacheConfigPart"));

var _DefaultConfigPart = _interopRequireDefault(require("./DefaultConfigPart"));

var _GatewayMessageQueueInfo = _interopRequireDefault(require("./GatewayMessageQueueInfo"));

var _GeneralConfigPart = _interopRequireDefault(require("./GeneralConfigPart"));

var _K8SAuthsConfigPart = _interopRequireDefault(require("./K8SAuthsConfigPart"));

var _KMIPConfigPart = _interopRequireDefault(require("./KMIPConfigPart"));

var _LdapConfigPart = _interopRequireDefault(require("./LdapConfigPart"));

var _LeadershipConfigPart = _interopRequireDefault(require("./LeadershipConfigPart"));

var _LogForwardingConfigPart = _interopRequireDefault(require("./LogForwardingConfigPart"));

var _MigrationsConfigPart = _interopRequireDefault(require("./MigrationsConfigPart"));

var _ProducersConfigPart = _interopRequireDefault(require("./ProducersConfigPart"));

var _RotatorsConfigPart = _interopRequireDefault(require("./RotatorsConfigPart"));

var _UIdentityConfigPart = _interopRequireDefault(require("./UIdentityConfigPart"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AkeylessGatewayConfig model module.
 * @module model/AkeylessGatewayConfig
 * @version 2.15.24
 */
var AkeylessGatewayConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AkeylessGatewayConfig</code>.
   * @alias module:model/AkeylessGatewayConfig
   */
  function AkeylessGatewayConfig() {
    _classCallCheck(this, AkeylessGatewayConfig);

    AkeylessGatewayConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AkeylessGatewayConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AkeylessGatewayConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AkeylessGatewayConfig} obj Optional instance to populate.
     * @return {module:model/AkeylessGatewayConfig} The populated <code>AkeylessGatewayConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AkeylessGatewayConfig();

        if (data.hasOwnProperty('admins')) {
          obj['admins'] = _AdminsConfigPart["default"].constructFromObject(data['admins']);
        }

        if (data.hasOwnProperty('cache')) {
          obj['cache'] = _CacheConfigPart["default"].constructFromObject(data['cache']);
        }

        if (data.hasOwnProperty('cf')) {
          obj['cf'] = _CFConfigPart["default"].constructFromObject(data['cf']);
        }

        if (data.hasOwnProperty('config_protection_key_name')) {
          obj['config_protection_key_name'] = _ApiClient["default"].convertToType(data['config_protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('general')) {
          obj['general'] = _GeneralConfigPart["default"].constructFromObject(data['general']);
        }

        if (data.hasOwnProperty('k8s_auths')) {
          obj['k8s_auths'] = _K8SAuthsConfigPart["default"].constructFromObject(data['k8s_auths']);
        }

        if (data.hasOwnProperty('kmip_clients')) {
          obj['kmip_clients'] = _KMIPConfigPart["default"].constructFromObject(data['kmip_clients']);
        }

        if (data.hasOwnProperty('ldap')) {
          obj['ldap'] = _LdapConfigPart["default"].constructFromObject(data['ldap']);
        }

        if (data.hasOwnProperty('leadership')) {
          obj['leadership'] = _LeadershipConfigPart["default"].constructFromObject(data['leadership']);
        }

        if (data.hasOwnProperty('log_forwarding')) {
          obj['log_forwarding'] = _LogForwardingConfigPart["default"].constructFromObject(data['log_forwarding']);
        }

        if (data.hasOwnProperty('message_queue_info')) {
          obj['message_queue_info'] = _GatewayMessageQueueInfo["default"].constructFromObject(data['message_queue_info']);
        }

        if (data.hasOwnProperty('migrations')) {
          obj['migrations'] = _MigrationsConfigPart["default"].constructFromObject(data['migrations']);
        }

        if (data.hasOwnProperty('producers')) {
          obj['producers'] = _ProducersConfigPart["default"].constructFromObject(data['producers']);
        }

        if (data.hasOwnProperty('rotators')) {
          obj['rotators'] = _RotatorsConfigPart["default"].constructFromObject(data['rotators']);
        }

        if (data.hasOwnProperty('saml')) {
          obj['saml'] = _DefaultConfigPart["default"].constructFromObject(data['saml']);
        }

        if (data.hasOwnProperty('uidentity')) {
          obj['uidentity'] = _UIdentityConfigPart["default"].constructFromObject(data['uidentity']);
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return AkeylessGatewayConfig;
}();
/**
 * @member {module:model/AdminsConfigPart} admins
 */


AkeylessGatewayConfig.prototype['admins'] = undefined;
/**
 * @member {module:model/CacheConfigPart} cache
 */

AkeylessGatewayConfig.prototype['cache'] = undefined;
/**
 * @member {module:model/CFConfigPart} cf
 */

AkeylessGatewayConfig.prototype['cf'] = undefined;
/**
 * @member {String} config_protection_key_name
 */

AkeylessGatewayConfig.prototype['config_protection_key_name'] = undefined;
/**
 * @member {module:model/GeneralConfigPart} general
 */

AkeylessGatewayConfig.prototype['general'] = undefined;
/**
 * @member {module:model/K8SAuthsConfigPart} k8s_auths
 */

AkeylessGatewayConfig.prototype['k8s_auths'] = undefined;
/**
 * @member {module:model/KMIPConfigPart} kmip_clients
 */

AkeylessGatewayConfig.prototype['kmip_clients'] = undefined;
/**
 * @member {module:model/LdapConfigPart} ldap
 */

AkeylessGatewayConfig.prototype['ldap'] = undefined;
/**
 * @member {module:model/LeadershipConfigPart} leadership
 */

AkeylessGatewayConfig.prototype['leadership'] = undefined;
/**
 * @member {module:model/LogForwardingConfigPart} log_forwarding
 */

AkeylessGatewayConfig.prototype['log_forwarding'] = undefined;
/**
 * @member {module:model/GatewayMessageQueueInfo} message_queue_info
 */

AkeylessGatewayConfig.prototype['message_queue_info'] = undefined;
/**
 * @member {module:model/MigrationsConfigPart} migrations
 */

AkeylessGatewayConfig.prototype['migrations'] = undefined;
/**
 * @member {module:model/ProducersConfigPart} producers
 */

AkeylessGatewayConfig.prototype['producers'] = undefined;
/**
 * @member {module:model/RotatorsConfigPart} rotators
 */

AkeylessGatewayConfig.prototype['rotators'] = undefined;
/**
 * @member {module:model/DefaultConfigPart} saml
 */

AkeylessGatewayConfig.prototype['saml'] = undefined;
/**
 * @member {module:model/UIdentityConfigPart} uidentity
 */

AkeylessGatewayConfig.prototype['uidentity'] = undefined;
/**
 * @member {Number} version
 */

AkeylessGatewayConfig.prototype['version'] = undefined;
var _default = AkeylessGatewayConfig;
exports["default"] = _default;
