// akeyless/dist/model/AssocRoleAuthMethod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AssocRoleAuthMethod model module.
 * @module model/AssocRoleAuthMethod
 * @version 2.15.24
 */
var AssocRoleAuthMethod = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AssocRoleAuthMethod</code>.
   * assocRoleAuthMethod is a command that creates an association between role and auth method.
   * @alias module:model/AssocRoleAuthMethod
   * @param amName {String} The auth method to associate
   * @param roleName {String} The role to associate
   */
  function AssocRoleAuthMethod(amName, roleName) {
    _classCallCheck(this, AssocRoleAuthMethod);

    AssocRoleAuthMethod.initialize(this, amName, roleName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AssocRoleAuthMethod, null, [{
    key: "initialize",
    value: function initialize(obj, amName, roleName) {
      obj['am-name'] = amName;
      obj['role-name'] = roleName;
    }
    /**
     * Constructs a <code>AssocRoleAuthMethod</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AssocRoleAuthMethod} obj Optional instance to populate.
     * @return {module:model/AssocRoleAuthMethod} The populated <code>AssocRoleAuthMethod</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AssocRoleAuthMethod();

        if (data.hasOwnProperty('am-name')) {
          obj['am-name'] = _ApiClient["default"].convertToType(data['am-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('role-name')) {
          obj['role-name'] = _ApiClient["default"].convertToType(data['role-name'], 'String');
        }

        if (data.hasOwnProperty('sub-claims')) {
          obj['sub-claims'] = _ApiClient["default"].convertToType(data['sub-claims'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AssocRoleAuthMethod;
}();
/**
 * The auth method to associate
 * @member {String} am-name
 */


AssocRoleAuthMethod.prototype['am-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

AssocRoleAuthMethod.prototype['password'] = undefined;
/**
 * The role to associate
 * @member {String} role-name
 */

AssocRoleAuthMethod.prototype['role-name'] = undefined;
/**
 * key/val of sub claims, e.g group=admins,developers
 * @member {Object.<String, String>} sub-claims
 */

AssocRoleAuthMethod.prototype['sub-claims'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

AssocRoleAuthMethod.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

AssocRoleAuthMethod.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

AssocRoleAuthMethod.prototype['username'] = undefined;
var _default = AssocRoleAuthMethod;
exports["default"] = _default;
