// akeyless/dist/model/AuthMethod.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AuthMethodAccessInfo = _interopRequireDefault(require("./AuthMethodAccessInfo"));

var _AuthMethodRoleAssociation = _interopRequireDefault(require("./AuthMethodRoleAssociation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethod model module.
 * @module model/AuthMethod
 * @version 2.20.0
 */
var AuthMethod = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethod</code>.
   * @alias module:model/AuthMethod
   */
  function AuthMethod() {
    _classCallCheck(this, AuthMethod);

    AuthMethod.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethod, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AuthMethod</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethod} obj Optional instance to populate.
     * @return {module:model/AuthMethod} The populated <code>AuthMethod</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethod();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('access_info')) {
          obj['access_info'] = _AuthMethodAccessInfo["default"].constructFromObject(data['access_info']);
        }

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_access_id')) {
          obj['auth_method_access_id'] = _ApiClient["default"].convertToType(data['auth_method_access_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_name')) {
          obj['auth_method_name'] = _ApiClient["default"].convertToType(data['auth_method_name'], 'String');
        }

        if (data.hasOwnProperty('auth_method_roles_assoc')) {
          obj['auth_method_roles_assoc'] = _ApiClient["default"].convertToType(data['auth_method_roles_assoc'], [_AuthMethodRoleAssociation["default"]]);
        }

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return AuthMethod;
}();
/**
 * @member {Date} access_date
 */


AuthMethod.prototype['access_date'] = undefined;
/**
 * @member {module:model/AuthMethodAccessInfo} access_info
 */

AuthMethod.prototype['access_info'] = undefined;
/**
 * @member {String} account_id
 */

AuthMethod.prototype['account_id'] = undefined;
/**
 * @member {String} auth_method_access_id
 */

AuthMethod.prototype['auth_method_access_id'] = undefined;
/**
 * @member {String} auth_method_name
 */

AuthMethod.prototype['auth_method_name'] = undefined;
/**
 * @member {Array.<module:model/AuthMethodRoleAssociation>} auth_method_roles_assoc
 */

AuthMethod.prototype['auth_method_roles_assoc'] = undefined;
/**
 * @member {Array.<String>} client_permissions
 */

AuthMethod.prototype['client_permissions'] = undefined;
/**
 * @member {Date} creation_date
 */

AuthMethod.prototype['creation_date'] = undefined;
/**
 * @member {Date} modification_date
 */

AuthMethod.prototype['modification_date'] = undefined;
var _default = AuthMethod;
exports["default"] = _default;
