// akeyless/dist/model/AuthMethodAccessInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _APIKeyAccessRules = _interopRequireDefault(require("./APIKeyAccessRules"));

var _AWSIAMAccessRules = _interopRequireDefault(require("./AWSIAMAccessRules"));

var _AzureADAccessRules = _interopRequireDefault(require("./AzureADAccessRules"));

var _CertAccessRules = _interopRequireDefault(require("./CertAccessRules"));

var _EmailPassAccessRules = _interopRequireDefault(require("./EmailPassAccessRules"));

var _GCPAccessRules = _interopRequireDefault(require("./GCPAccessRules"));

var _HuaweiAccessRules = _interopRequireDefault(require("./HuaweiAccessRules"));

var _KubernetesAccessRules = _interopRequireDefault(require("./KubernetesAccessRules"));

var _LDAPAccessRules = _interopRequireDefault(require("./LDAPAccessRules"));

var _OAuth2AccessRules = _interopRequireDefault(require("./OAuth2AccessRules"));

var _OIDCAccessRules = _interopRequireDefault(require("./OIDCAccessRules"));

var _SAMLAccessRules = _interopRequireDefault(require("./SAMLAccessRules"));

var _UniversalIdentityAccessRules = _interopRequireDefault(require("./UniversalIdentityAccessRules"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodAccessInfo model module.
 * @module model/AuthMethodAccessInfo
 * @version 2.20.0
 */
var AuthMethodAccessInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodAccessInfo</code>.
   * @alias module:model/AuthMethodAccessInfo
   */
  function AuthMethodAccessInfo() {
    _classCallCheck(this, AuthMethodAccessInfo);

    AuthMethodAccessInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodAccessInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AuthMethodAccessInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodAccessInfo} obj Optional instance to populate.
     * @return {module:model/AuthMethodAccessInfo} The populated <code>AuthMethodAccessInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodAccessInfo();

        if (data.hasOwnProperty('access_expires')) {
          obj['access_expires'] = _ApiClient["default"].convertToType(data['access_expires'], 'Number');
        }

        if (data.hasOwnProperty('access_id_alias')) {
          obj['access_id_alias'] = _ApiClient["default"].convertToType(data['access_id_alias'], 'String');
        }

        if (data.hasOwnProperty('api_key_access_rules')) {
          obj['api_key_access_rules'] = _APIKeyAccessRules["default"].constructFromObject(data['api_key_access_rules']);
        }

        if (data.hasOwnProperty('aws_iam_access_rules')) {
          obj['aws_iam_access_rules'] = _AWSIAMAccessRules["default"].constructFromObject(data['aws_iam_access_rules']);
        }

        if (data.hasOwnProperty('azure_ad_access_rules')) {
          obj['azure_ad_access_rules'] = _AzureADAccessRules["default"].constructFromObject(data['azure_ad_access_rules']);
        }

        if (data.hasOwnProperty('cert_access_rules')) {
          obj['cert_access_rules'] = _CertAccessRules["default"].constructFromObject(data['cert_access_rules']);
        }

        if (data.hasOwnProperty('cidr_whitelist')) {
          obj['cidr_whitelist'] = _ApiClient["default"].convertToType(data['cidr_whitelist'], 'String');
        }

        if (data.hasOwnProperty('email_pass_access_rules')) {
          obj['email_pass_access_rules'] = _EmailPassAccessRules["default"].constructFromObject(data['email_pass_access_rules']);
        }

        if (data.hasOwnProperty('force_sub_claims')) {
          obj['force_sub_claims'] = _ApiClient["default"].convertToType(data['force_sub_claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gcp_access_rules')) {
          obj['gcp_access_rules'] = _GCPAccessRules["default"].constructFromObject(data['gcp_access_rules']);
        }

        if (data.hasOwnProperty('gw_cidr_whitelist')) {
          obj['gw_cidr_whitelist'] = _ApiClient["default"].convertToType(data['gw_cidr_whitelist'], 'String');
        }

        if (data.hasOwnProperty('huawei_access_rules')) {
          obj['huawei_access_rules'] = _HuaweiAccessRules["default"].constructFromObject(data['huawei_access_rules']);
        }

        if (data.hasOwnProperty('jwt_ttl')) {
          obj['jwt_ttl'] = _ApiClient["default"].convertToType(data['jwt_ttl'], 'Number');
        }

        if (data.hasOwnProperty('k8s_access_rules')) {
          obj['k8s_access_rules'] = _KubernetesAccessRules["default"].constructFromObject(data['k8s_access_rules']);
        }

        if (data.hasOwnProperty('ldap_access_rules')) {
          obj['ldap_access_rules'] = _LDAPAccessRules["default"].constructFromObject(data['ldap_access_rules']);
        }

        if (data.hasOwnProperty('oauth2_access_rules')) {
          obj['oauth2_access_rules'] = _OAuth2AccessRules["default"].constructFromObject(data['oauth2_access_rules']);
        }

        if (data.hasOwnProperty('oidc_access_rules')) {
          obj['oidc_access_rules'] = _OIDCAccessRules["default"].constructFromObject(data['oidc_access_rules']);
        }

        if (data.hasOwnProperty('rules_type')) {
          obj['rules_type'] = _ApiClient["default"].convertToType(data['rules_type'], 'String');
        }

        if (data.hasOwnProperty('saml_access_rules')) {
          obj['saml_access_rules'] = _SAMLAccessRules["default"].constructFromObject(data['saml_access_rules']);
        }

        if (data.hasOwnProperty('universal_identity_access_rules')) {
          obj['universal_identity_access_rules'] = _UniversalIdentityAccessRules["default"].constructFromObject(data['universal_identity_access_rules']);
        }
      }

      return obj;
    }
  }]);

  return AuthMethodAccessInfo;
}();
/**
 * @member {Number} access_expires
 */


AuthMethodAccessInfo.prototype['access_expires'] = undefined;
/**
 * for accounts where AccessId holds encrypted email this field will hold generated AccessId, for accounts based on regular AccessId it will be equal to accessId itself
 * @member {String} access_id_alias
 */

AuthMethodAccessInfo.prototype['access_id_alias'] = undefined;
/**
 * @member {module:model/APIKeyAccessRules} api_key_access_rules
 */

AuthMethodAccessInfo.prototype['api_key_access_rules'] = undefined;
/**
 * @member {module:model/AWSIAMAccessRules} aws_iam_access_rules
 */

AuthMethodAccessInfo.prototype['aws_iam_access_rules'] = undefined;
/**
 * @member {module:model/AzureADAccessRules} azure_ad_access_rules
 */

AuthMethodAccessInfo.prototype['azure_ad_access_rules'] = undefined;
/**
 * @member {module:model/CertAccessRules} cert_access_rules
 */

AuthMethodAccessInfo.prototype['cert_access_rules'] = undefined;
/**
 * @member {String} cidr_whitelist
 */

AuthMethodAccessInfo.prototype['cidr_whitelist'] = undefined;
/**
 * @member {module:model/EmailPassAccessRules} email_pass_access_rules
 */

AuthMethodAccessInfo.prototype['email_pass_access_rules'] = undefined;
/**
 * if true the role associated with this auth method must include sub claims
 * @member {Boolean} force_sub_claims
 */

AuthMethodAccessInfo.prototype['force_sub_claims'] = undefined;
/**
 * @member {module:model/GCPAccessRules} gcp_access_rules
 */

AuthMethodAccessInfo.prototype['gcp_access_rules'] = undefined;
/**
 * @member {String} gw_cidr_whitelist
 */

AuthMethodAccessInfo.prototype['gw_cidr_whitelist'] = undefined;
/**
 * @member {module:model/HuaweiAccessRules} huawei_access_rules
 */

AuthMethodAccessInfo.prototype['huawei_access_rules'] = undefined;
/**
 * @member {Number} jwt_ttl
 */

AuthMethodAccessInfo.prototype['jwt_ttl'] = undefined;
/**
 * @member {module:model/KubernetesAccessRules} k8s_access_rules
 */

AuthMethodAccessInfo.prototype['k8s_access_rules'] = undefined;
/**
 * @member {module:model/LDAPAccessRules} ldap_access_rules
 */

AuthMethodAccessInfo.prototype['ldap_access_rules'] = undefined;
/**
 * @member {module:model/OAuth2AccessRules} oauth2_access_rules
 */

AuthMethodAccessInfo.prototype['oauth2_access_rules'] = undefined;
/**
 * @member {module:model/OIDCAccessRules} oidc_access_rules
 */

AuthMethodAccessInfo.prototype['oidc_access_rules'] = undefined;
/**
 * @member {String} rules_type
 */

AuthMethodAccessInfo.prototype['rules_type'] = undefined;
/**
 * @member {module:model/SAMLAccessRules} saml_access_rules
 */

AuthMethodAccessInfo.prototype['saml_access_rules'] = undefined;
/**
 * @member {module:model/UniversalIdentityAccessRules} universal_identity_access_rules
 */

AuthMethodAccessInfo.prototype['universal_identity_access_rules'] = undefined;
var _default = AuthMethodAccessInfo;
exports["default"] = _default;
