// akeyless/dist/model/Auth.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Auth model module.
 * @module model/Auth
 * @version 2.20.0
 */
var Auth = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Auth</code>.
   * @alias module:model/Auth
   */
  function Auth() {
    _classCallCheck(this, Auth);

    Auth.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Auth, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Auth</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Auth} obj Optional instance to populate.
     * @return {module:model/Auth} The populated <code>Auth</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Auth();

        if (data.hasOwnProperty('access-id')) {
          obj['access-id'] = _ApiClient["default"].convertToType(data['access-id'], 'String');
        }

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('access-type')) {
          obj['access-type'] = _ApiClient["default"].convertToType(data['access-type'], 'String');
        }

        if (data.hasOwnProperty('admin-email')) {
          obj['admin-email'] = _ApiClient["default"].convertToType(data['admin-email'], 'String');
        }

        if (data.hasOwnProperty('admin-password')) {
          obj['admin-password'] = _ApiClient["default"].convertToType(data['admin-password'], 'String');
        }

        if (data.hasOwnProperty('cert-data')) {
          obj['cert-data'] = _ApiClient["default"].convertToType(data['cert-data'], 'String');
        }

        if (data.hasOwnProperty('cloud-id')) {
          obj['cloud-id'] = _ApiClient["default"].convertToType(data['cloud-id'], 'String');
        }

        if (data.hasOwnProperty('debug')) {
          obj['debug'] = _ApiClient["default"].convertToType(data['debug'], 'Boolean');
        }

        if (data.hasOwnProperty('gcp-audience')) {
          obj['gcp-audience'] = _ApiClient["default"].convertToType(data['gcp-audience'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt')) {
          obj['jwt'] = _ApiClient["default"].convertToType(data['jwt'], 'String');
        }

        if (data.hasOwnProperty('k8s-auth-config-name')) {
          obj['k8s-auth-config-name'] = _ApiClient["default"].convertToType(data['k8s-auth-config-name'], 'String');
        }

        if (data.hasOwnProperty('k8s-service-account-token')) {
          obj['k8s-service-account-token'] = _ApiClient["default"].convertToType(data['k8s-service-account-token'], 'String');
        }

        if (data.hasOwnProperty('key-data')) {
          obj['key-data'] = _ApiClient["default"].convertToType(data['key-data'], 'String');
        }

        if (data.hasOwnProperty('ldap_password')) {
          obj['ldap_password'] = _ApiClient["default"].convertToType(data['ldap_password'], 'String');
        }

        if (data.hasOwnProperty('ldap_username')) {
          obj['ldap_username'] = _ApiClient["default"].convertToType(data['ldap_username'], 'String');
        }

        if (data.hasOwnProperty('uid_token')) {
          obj['uid_token'] = _ApiClient["default"].convertToType(data['uid_token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Auth;
}();
/**
 * Access ID
 * @member {String} access-id
 */


Auth.prototype['access-id'] = undefined;
/**
 * Access key (relevant only for access-type=access_key)
 * @member {String} access-key
 */

Auth.prototype['access-key'] = undefined;
/**
 * Access Type (access_key/password/saml/ldap/k8s/azure_ad/oidc/aws_iam/universal_identity/jwt/gcp/cert)
 * @member {String} access-type
 * @default 'access_key'
 */

Auth.prototype['access-type'] = 'access_key';
/**
 * Email (relevant only for access-type=password)
 * @member {String} admin-email
 */

Auth.prototype['admin-email'] = undefined;
/**
 * Password (relevant only for access-type=password)
 * @member {String} admin-password
 */

Auth.prototype['admin-password'] = undefined;
/**
 * Certificate data encoded in base64. Used if file was not provided. (relevant only for access-type=cert)
 * @member {String} cert-data
 */

Auth.prototype['cert-data'] = undefined;
/**
 * The cloud identity (relevant only for access-type=azure_ad,aws_iam,gcp)
 * @member {String} cloud-id
 */

Auth.prototype['cloud-id'] = undefined;
/**
 * @member {Boolean} debug
 */

Auth.prototype['debug'] = undefined;
/**
 * GCP JWT audience
 * @member {String} gcp-audience
 */

Auth.prototype['gcp-audience'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

Auth.prototype['json'] = undefined;
/**
 * The Json Web Token (relevant only for access-type=jwt/oidc)
 * @member {String} jwt
 */

Auth.prototype['jwt'] = undefined;
/**
 * The K8S Auth config name (relevant only for access-type=k8s)
 * @member {String} k8s-auth-config-name
 */

Auth.prototype['k8s-auth-config-name'] = undefined;
/**
 * The K8S service account token. (relevant only for access-type=k8s)
 * @member {String} k8s-service-account-token
 */

Auth.prototype['k8s-service-account-token'] = undefined;
/**
 * Private key data encoded in base64. Used if file was not provided.(relevant only for access-type=cert)
 * @member {String} key-data
 */

Auth.prototype['key-data'] = undefined;
/**
 * LDAP password (relevant only for access-type=ldap)
 * @member {String} ldap_password
 */

Auth.prototype['ldap_password'] = undefined;
/**
 * LDAP username (relevant only for access-type=ldap)
 * @member {String} ldap_username
 */

Auth.prototype['ldap_username'] = undefined;
/**
 * The universal_identity token (relevant only for access-type=universal_identity)
 * @member {String} uid_token
 */

Auth.prototype['uid_token'] = undefined;
var _default = Auth;
exports["default"] = _default;
