// akeyless/dist/model/AuthMethodRoleAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Rules = _interopRequireDefault(require("./Rules"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AuthMethodRoleAssociation model module.
 * @module model/AuthMethodRoleAssociation
 * @version 2.15.24
 */
var AuthMethodRoleAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AuthMethodRoleAssociation</code>.
   * AuthMethodRoleAssociation includes details of an association between an auth method and a role.
   * @alias module:model/AuthMethodRoleAssociation
   */
  function AuthMethodRoleAssociation() {
    _classCallCheck(this, AuthMethodRoleAssociation);

    AuthMethodRoleAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AuthMethodRoleAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AuthMethodRoleAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AuthMethodRoleAssociation} obj Optional instance to populate.
     * @return {module:model/AuthMethodRoleAssociation} The populated <code>AuthMethodRoleAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AuthMethodRoleAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_sub_claims')) {
          obj['auth_method_sub_claims'] = _ApiClient["default"].convertToType(data['auth_method_sub_claims'], {
            'String': ['String']
          });
        }

        if (data.hasOwnProperty('role_name')) {
          obj['role_name'] = _ApiClient["default"].convertToType(data['role_name'], 'String');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _Rules["default"].constructFromObject(data['rules']);
        }
      }

      return obj;
    }
  }]);

  return AuthMethodRoleAssociation;
}();
/**
 * @member {String} assoc_id
 */


AuthMethodRoleAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} auth_method_sub_claims
 */

AuthMethodRoleAssociation.prototype['auth_method_sub_claims'] = undefined;
/**
 * @member {String} role_name
 */

AuthMethodRoleAssociation.prototype['role_name'] = undefined;
/**
 * @member {module:model/Rules} rules
 */

AuthMethodRoleAssociation.prototype['rules'] = undefined;
var _default = AuthMethodRoleAssociation;
exports["default"] = _default;
