// akeyless/dist/model/AWSIAMAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AWSIAMAccessRules model module.
 * @module model/AWSIAMAccessRules
 * @version 2.15.24
 */
var AWSIAMAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AWSIAMAccessRules</code>.
   * @alias module:model/AWSIAMAccessRules
   */
  function AWSIAMAccessRules() {
    _classCallCheck(this, AWSIAMAccessRules);

    AWSIAMAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AWSIAMAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AWSIAMAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AWSIAMAccessRules} obj Optional instance to populate.
     * @return {module:model/AWSIAMAccessRules} The populated <code>AWSIAMAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AWSIAMAccessRules();

        if (data.hasOwnProperty('account_id')) {
          obj['account_id'] = _ApiClient["default"].convertToType(data['account_id'], ['String']);
        }

        if (data.hasOwnProperty('arn')) {
          obj['arn'] = _ApiClient["default"].convertToType(data['arn'], ['String']);
        }

        if (data.hasOwnProperty('resource_id')) {
          obj['resource_id'] = _ApiClient["default"].convertToType(data['resource_id'], ['String']);
        }

        if (data.hasOwnProperty('role_id')) {
          obj['role_id'] = _ApiClient["default"].convertToType(data['role_id'], ['String']);
        }

        if (data.hasOwnProperty('role_name')) {
          obj['role_name'] = _ApiClient["default"].convertToType(data['role_name'], ['String']);
        }

        if (data.hasOwnProperty('sts_endpoint')) {
          obj['sts_endpoint'] = _ApiClient["default"].convertToType(data['sts_endpoint'], 'String');
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], ['String']);
        }

        if (data.hasOwnProperty('user_name')) {
          obj['user_name'] = _ApiClient["default"].convertToType(data['user_name'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return AWSIAMAccessRules;
}();
/**
 * The list of account ids that the login is restricted to.
 * @member {Array.<String>} account_id
 */


AWSIAMAccessRules.prototype['account_id'] = undefined;
/**
 * The list of ARNs that the login is restricted to.
 * @member {Array.<String>} arn
 */

AWSIAMAccessRules.prototype['arn'] = undefined;
/**
 * The list of resource ids that the login is restricted to.
 * @member {Array.<String>} resource_id
 */

AWSIAMAccessRules.prototype['resource_id'] = undefined;
/**
 * The list of role ids that the login is restricted to.
 * @member {Array.<String>} role_id
 */

AWSIAMAccessRules.prototype['role_id'] = undefined;
/**
 * The list of role names that the login is restricted to.
 * @member {Array.<String>} role_name
 */

AWSIAMAccessRules.prototype['role_name'] = undefined;
/**
 * The sts URL.
 * @member {String} sts_endpoint
 */

AWSIAMAccessRules.prototype['sts_endpoint'] = undefined;
/**
 * The list of user ids that the login is restricted to.
 * @member {Array.<String>} user_id
 */

AWSIAMAccessRules.prototype['user_id'] = undefined;
/**
 * The list of user names that the login is restricted to.
 * @member {Array.<String>} user_name
 */

AWSIAMAccessRules.prototype['user_name'] = undefined;
var _default = AWSIAMAccessRules;
exports["default"] = _default;
