// akeyless/dist/model/AzurePayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AzurePayload model module.
 * @module model/AzurePayload
 * @version 2.15.24
 */
var AzurePayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AzurePayload</code>.
   * @alias module:model/AzurePayload
   */
  function AzurePayload() {
    _classCallCheck(this, AzurePayload);

    AzurePayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AzurePayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AzurePayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AzurePayload} obj Optional instance to populate.
     * @return {module:model/AzurePayload} The populated <code>AzurePayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AzurePayload();

        if (data.hasOwnProperty('client')) {
          obj['client'] = _ApiClient["default"].convertToType(data['client'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('secret')) {
          obj['secret'] = _ApiClient["default"].convertToType(data['secret'], 'String');
        }

        if (data.hasOwnProperty('tenant')) {
          obj['tenant'] = _ApiClient["default"].convertToType(data['tenant'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AzurePayload;
}();
/**
 * @member {String} client
 */


AzurePayload.prototype['client'] = undefined;
/**
 * @member {String} name
 */

AzurePayload.prototype['name'] = undefined;
/**
 * @member {String} secret
 */

AzurePayload.prototype['secret'] = undefined;
/**
 * @member {String} tenant
 */

AzurePayload.prototype['tenant'] = undefined;
var _default = AzurePayload;
exports["default"] = _default;
