// akeyless/dist/model/BastionListEntry.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BastionListEntry model module.
 * @module model/BastionListEntry
 * @version 2.20.0
 */
var BastionListEntry = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BastionListEntry</code>.
   * @alias module:model/BastionListEntry
   */
  function BastionListEntry() {
    _classCallCheck(this, BastionListEntry);

    BastionListEntry.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BastionListEntry, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>BastionListEntry</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BastionListEntry} obj Optional instance to populate.
     * @return {module:model/BastionListEntry} The populated <code>BastionListEntry</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BastionListEntry();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('allowed_access_ids')) {
          obj['allowed_access_ids'] = _ApiClient["default"].convertToType(data['allowed_access_ids'], ['String']);
        }

        if (data.hasOwnProperty('allowed_urls')) {
          obj['allowed_urls'] = _ApiClient["default"].convertToType(data['allowed_urls'], ['String']);
        }

        if (data.hasOwnProperty('cluster_name')) {
          obj['cluster_name'] = _ApiClient["default"].convertToType(data['cluster_name'], 'String');
        }

        if (data.hasOwnProperty('display_name')) {
          obj['display_name'] = _ApiClient["default"].convertToType(data['display_name'], 'String');
        }

        if (data.hasOwnProperty('last_report')) {
          obj['last_report'] = _ApiClient["default"].convertToType(data['last_report'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return BastionListEntry;
}();
/**
 * @member {String} access_id
 */


BastionListEntry.prototype['access_id'] = undefined;
/**
 * @member {Array.<String>} allowed_access_ids
 */

BastionListEntry.prototype['allowed_access_ids'] = undefined;
/**
 * @member {Array.<String>} allowed_urls
 */

BastionListEntry.prototype['allowed_urls'] = undefined;
/**
 * @member {String} cluster_name
 */

BastionListEntry.prototype['cluster_name'] = undefined;
/**
 * @member {String} display_name
 */

BastionListEntry.prototype['display_name'] = undefined;
/**
 * @member {Date} last_report
 */

BastionListEntry.prototype['last_report'] = undefined;
var _default = BastionListEntry;
exports["default"] = _default;
