// akeyless/dist/model/CacheConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CacheConfigPart model module.
 * @module model/CacheConfigPart
 * @version 2.15.24
 */
var CacheConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CacheConfigPart</code>.
   * @alias module:model/CacheConfigPart
   */
  function CacheConfigPart() {
    _classCallCheck(this, CacheConfigPart);

    CacheConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CacheConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CacheConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CacheConfigPart} obj Optional instance to populate.
     * @return {module:model/CacheConfigPart} The populated <code>CacheConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CacheConfigPart();

        if (data.hasOwnProperty('cache_enable')) {
          obj['cache_enable'] = _ApiClient["default"].convertToType(data['cache_enable'], 'Boolean');
        }

        if (data.hasOwnProperty('cache_ttl')) {
          obj['cache_ttl'] = _ApiClient["default"].convertToType(data['cache_ttl'], 'String');
        }

        if (data.hasOwnProperty('proactive_cache_dump_interval')) {
          obj['proactive_cache_dump_interval'] = _ApiClient["default"].convertToType(data['proactive_cache_dump_interval'], 'String');
        }

        if (data.hasOwnProperty('proactive_cache_enable')) {
          obj['proactive_cache_enable'] = _ApiClient["default"].convertToType(data['proactive_cache_enable'], 'Boolean');
        }

        if (data.hasOwnProperty('proactive_cache_minimum_fetching_time')) {
          obj['proactive_cache_minimum_fetching_time'] = _ApiClient["default"].convertToType(data['proactive_cache_minimum_fetching_time'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CacheConfigPart;
}();
/**
 * @member {Boolean} cache_enable
 */


CacheConfigPart.prototype['cache_enable'] = undefined;
/**
 * @member {String} cache_ttl
 */

CacheConfigPart.prototype['cache_ttl'] = undefined;
/**
 * @member {String} proactive_cache_dump_interval
 */

CacheConfigPart.prototype['proactive_cache_dump_interval'] = undefined;
/**
 * @member {Boolean} proactive_cache_enable
 */

CacheConfigPart.prototype['proactive_cache_enable'] = undefined;
/**
 * @member {String} proactive_cache_minimum_fetching_time
 */

CacheConfigPart.prototype['proactive_cache_minimum_fetching_time'] = undefined;
var _default = CacheConfigPart;
exports["default"] = _default;
