// akeyless/dist/model/CertificateIssueInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _PKICertificateIssueDetails = _interopRequireDefault(require("./PKICertificateIssueDetails"));

var _SSHCertificateIssueDetails = _interopRequireDefault(require("./SSHCertificateIssueDetails"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CertificateIssueInfo model module.
 * @module model/CertificateIssueInfo
 * @version 2.15.24
 */
var CertificateIssueInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CertificateIssueInfo</code>.
   * @alias module:model/CertificateIssueInfo
   */
  function CertificateIssueInfo() {
    _classCallCheck(this, CertificateIssueInfo);

    CertificateIssueInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CertificateIssueInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CertificateIssueInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CertificateIssueInfo} obj Optional instance to populate.
     * @return {module:model/CertificateIssueInfo} The populated <code>CertificateIssueInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CertificateIssueInfo();

        if (data.hasOwnProperty('cert_issuer_type')) {
          obj['cert_issuer_type'] = _ApiClient["default"].convertToType(data['cert_issuer_type'], 'String');
        }

        if (data.hasOwnProperty('max_ttl')) {
          obj['max_ttl'] = _ApiClient["default"].convertToType(data['max_ttl'], 'Number');
        }

        if (data.hasOwnProperty('pki_cert_issuer_details')) {
          obj['pki_cert_issuer_details'] = _PKICertificateIssueDetails["default"].constructFromObject(data['pki_cert_issuer_details']);
        }

        if (data.hasOwnProperty('ssh_cert_issuer_details')) {
          obj['ssh_cert_issuer_details'] = _SSHCertificateIssueDetails["default"].constructFromObject(data['ssh_cert_issuer_details']);
        }
      }

      return obj;
    }
  }]);

  return CertificateIssueInfo;
}();
/**
 * @member {String} cert_issuer_type
 */


CertificateIssueInfo.prototype['cert_issuer_type'] = undefined;
/**
 * @member {Number} max_ttl
 */

CertificateIssueInfo.prototype['max_ttl'] = undefined;
/**
 * @member {module:model/PKICertificateIssueDetails} pki_cert_issuer_details
 */

CertificateIssueInfo.prototype['pki_cert_issuer_details'] = undefined;
/**
 * @member {module:model/SSHCertificateIssueDetails} ssh_cert_issuer_details
 */

CertificateIssueInfo.prototype['ssh_cert_issuer_details'] = undefined;
var _default = CertificateIssueInfo;
exports["default"] = _default;
