// akeyless/dist/model/ClassicKeyTargetInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClassicKeyStatusInfo = _interopRequireDefault(require("./ClassicKeyStatusInfo"));

var _ExternalKMSKeyId = _interopRequireDefault(require("./ExternalKMSKeyId"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClassicKeyTargetInfo model module.
 * @module model/ClassicKeyTargetInfo
 * @version 2.20.0
 */
var ClassicKeyTargetInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClassicKeyTargetInfo</code>.
   * @alias module:model/ClassicKeyTargetInfo
   */
  function ClassicKeyTargetInfo() {
    _classCallCheck(this, ClassicKeyTargetInfo);

    ClassicKeyTargetInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClassicKeyTargetInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClassicKeyTargetInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClassicKeyTargetInfo} obj Optional instance to populate.
     * @return {module:model/ClassicKeyTargetInfo} The populated <code>ClassicKeyTargetInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClassicKeyTargetInfo();

        if (data.hasOwnProperty('external_kms_id')) {
          obj['external_kms_id'] = _ExternalKMSKeyId["default"].constructFromObject(data['external_kms_id']);
        }

        if (data.hasOwnProperty('key_purpose')) {
          obj['key_purpose'] = _ApiClient["default"].convertToType(data['key_purpose'], ['String']);
        }

        if (data.hasOwnProperty('key_status')) {
          obj['key_status'] = _ClassicKeyStatusInfo["default"].constructFromObject(data['key_status']);
        }

        if (data.hasOwnProperty('target_assoc_id')) {
          obj['target_assoc_id'] = _ApiClient["default"].convertToType(data['target_assoc_id'], 'String');
        }

        if (data.hasOwnProperty('target_type')) {
          obj['target_type'] = _ApiClient["default"].convertToType(data['target_type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ClassicKeyTargetInfo;
}();
/**
 * @member {module:model/ExternalKMSKeyId} external_kms_id
 */


ClassicKeyTargetInfo.prototype['external_kms_id'] = undefined;
/**
 * @member {Array.<String>} key_purpose
 */

ClassicKeyTargetInfo.prototype['key_purpose'] = undefined;
/**
 * @member {module:model/ClassicKeyStatusInfo} key_status
 */

ClassicKeyTargetInfo.prototype['key_status'] = undefined;
/**
 * @member {String} target_assoc_id
 */

ClassicKeyTargetInfo.prototype['target_assoc_id'] = undefined;
/**
 * @member {String} target_type
 */

ClassicKeyTargetInfo.prototype['target_type'] = undefined;
var _default = ClassicKeyTargetInfo;
exports["default"] = _default;
