// akeyless/dist/model/ClassicKeyStatusInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClassicKeyStatusInfo model module.
 * @module model/ClassicKeyStatusInfo
 * @version 2.15.24
 */
var ClassicKeyStatusInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClassicKeyStatusInfo</code>.
   * @alias module:model/ClassicKeyStatusInfo
   */
  function ClassicKeyStatusInfo() {
    _classCallCheck(this, ClassicKeyStatusInfo);

    ClassicKeyStatusInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClassicKeyStatusInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClassicKeyStatusInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClassicKeyStatusInfo} obj Optional instance to populate.
     * @return {module:model/ClassicKeyStatusInfo} The populated <code>ClassicKeyStatusInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClassicKeyStatusInfo();

        if (data.hasOwnProperty('error_date')) {
          obj['error_date'] = _ApiClient["default"].convertToType(data['error_date'], 'Date');
        }

        if (data.hasOwnProperty('last_error')) {
          obj['last_error'] = _ApiClient["default"].convertToType(data['last_error'], 'String');
        }

        if (data.hasOwnProperty('last_status')) {
          obj['last_status'] = _ApiClient["default"].convertToType(data['last_status'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ClassicKeyStatusInfo;
}();
/**
 * @member {Date} error_date
 */


ClassicKeyStatusInfo.prototype['error_date'] = undefined;
/**
 * @member {String} last_error
 */

ClassicKeyStatusInfo.prototype['last_error'] = undefined;
/**
 * ClassicKeyTargetStatus defines status of classic key target
 * @member {String} last_status
 */

ClassicKeyStatusInfo.prototype['last_status'] = undefined;
/**
 * @member {Number} version
 */

ClassicKeyStatusInfo.prototype['version'] = undefined;
var _default = ClassicKeyStatusInfo;
exports["default"] = _default;
