// akeyless/dist/model/ClassicKeyDetailsInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClassicKeyTargetInfo = _interopRequireDefault(require("./ClassicKeyTargetInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClassicKeyDetailsInfo model module.
 * @module model/ClassicKeyDetailsInfo
 * @version 2.15.24
 */
var ClassicKeyDetailsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClassicKeyDetailsInfo</code>.
   * @alias module:model/ClassicKeyDetailsInfo
   */
  function ClassicKeyDetailsInfo() {
    _classCallCheck(this, ClassicKeyDetailsInfo);

    ClassicKeyDetailsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClassicKeyDetailsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClassicKeyDetailsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClassicKeyDetailsInfo} obj Optional instance to populate.
     * @return {module:model/ClassicKeyDetailsInfo} The populated <code>ClassicKeyDetailsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClassicKeyDetailsInfo();

        if (data.hasOwnProperty('classic_key_attributes')) {
          obj['classic_key_attributes'] = _ApiClient["default"].convertToType(data['classic_key_attributes'], {
            'String': ['String']
          });
        }

        if (data.hasOwnProperty('classic_key_id')) {
          obj['classic_key_id'] = _ApiClient["default"].convertToType(data['classic_key_id'], 'String');
        }

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('is_provided_by_user')) {
          obj['is_provided_by_user'] = _ApiClient["default"].convertToType(data['is_provided_by_user'], 'Boolean');
        }

        if (data.hasOwnProperty('is_unexportable')) {
          obj['is_unexportable'] = _ApiClient["default"].convertToType(data['is_unexportable'], 'Boolean');
        }

        if (data.hasOwnProperty('key_state')) {
          obj['key_state'] = _ApiClient["default"].convertToType(data['key_state'], 'String');
        }

        if (data.hasOwnProperty('key_type')) {
          obj['key_type'] = _ApiClient["default"].convertToType(data['key_type'], 'String');
        }

        if (data.hasOwnProperty('last_error')) {
          obj['last_error'] = _ApiClient["default"].convertToType(data['last_error'], 'String');
        }

        if (data.hasOwnProperty('target_alias_helper')) {
          obj['target_alias_helper'] = _ApiClient["default"].convertToType(data['target_alias_helper'], 'String');
        }

        if (data.hasOwnProperty('target_types')) {
          obj['target_types'] = _ApiClient["default"].convertToType(data['target_types'], ['String']);
        }

        if (data.hasOwnProperty('targets')) {
          obj['targets'] = _ApiClient["default"].convertToType(data['targets'], [_ClassicKeyTargetInfo["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ClassicKeyDetailsInfo;
}();
/**
 * @member {Object.<String, Array.<String>>} classic_key_attributes
 */


ClassicKeyDetailsInfo.prototype['classic_key_attributes'] = undefined;
/**
 * @member {String} classic_key_id
 */

ClassicKeyDetailsInfo.prototype['classic_key_id'] = undefined;
/**
 * @member {Number} gw_cluster_id
 */

ClassicKeyDetailsInfo.prototype['gw_cluster_id'] = undefined;
/**
 * @member {Boolean} is_provided_by_user
 */

ClassicKeyDetailsInfo.prototype['is_provided_by_user'] = undefined;
/**
 * @member {Boolean} is_unexportable
 */

ClassicKeyDetailsInfo.prototype['is_unexportable'] = undefined;
/**
 * ItemState defines the different states an Item can be in
 * @member {String} key_state
 */

ClassicKeyDetailsInfo.prototype['key_state'] = undefined;
/**
 * @member {String} key_type
 */

ClassicKeyDetailsInfo.prototype['key_type'] = undefined;
/**
 * @member {String} last_error
 */

ClassicKeyDetailsInfo.prototype['last_error'] = undefined;
/**
 * @member {String} target_alias_helper
 */

ClassicKeyDetailsInfo.prototype['target_alias_helper'] = undefined;
/**
 * @member {Array.<String>} target_types
 */

ClassicKeyDetailsInfo.prototype['target_types'] = undefined;
/**
 * @member {Array.<module:model/ClassicKeyTargetInfo>} targets
 */

ClassicKeyDetailsInfo.prototype['targets'] = undefined;
var _default = ClassicKeyDetailsInfo;
exports["default"] = _default;
