// akeyless/dist/model/ClientData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientData model module.
 * @module model/ClientData
 * @version 2.15.24
 */
var ClientData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientData</code>.
   * @alias module:model/ClientData
   */
  function ClientData() {
    _classCallCheck(this, ClientData);

    ClientData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ClientData} obj Optional instance to populate.
     * @return {module:model/ClientData} The populated <code>ClientData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientData();

        if (data.hasOwnProperty('clientCertificateData')) {
          obj['clientCertificateData'] = _ApiClient["default"].convertToType(data['clientCertificateData'], 'String');
        }

        if (data.hasOwnProperty('clientKeyData')) {
          obj['clientKeyData'] = _ApiClient["default"].convertToType(data['clientKeyData'], 'String');
        }

        if (data.hasOwnProperty('parentCertificateData')) {
          obj['parentCertificateData'] = _ApiClient["default"].convertToType(data['parentCertificateData'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ClientData;
}();
/**
 * @member {String} clientCertificateData
 */


ClientData.prototype['clientCertificateData'] = undefined;
/**
 * @member {String} clientKeyData
 */

ClientData.prototype['clientKeyData'] = undefined;
/**
 * @member {String} parentCertificateData
 */

ClientData.prototype['parentCertificateData'] = undefined;
var _default = ClientData;
exports["default"] = _default;
