// akeyless/dist/model/ConfigChange.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ConfigHash = _interopRequireDefault(require("./ConfigHash"));

var _LastConfigChange = _interopRequireDefault(require("./LastConfigChange"));

var _LastStatusInfo = _interopRequireDefault(require("./LastStatusInfo"));

var _RequiredActivity = _interopRequireDefault(require("./RequiredActivity"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConfigChange model module.
 * @module model/ConfigChange
 * @version 2.15.24
 */
var ConfigChange = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConfigChange</code>.
   * @alias module:model/ConfigChange
   */
  function ConfigChange() {
    _classCallCheck(this, ConfigChange);

    ConfigChange.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConfigChange, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConfigChange</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConfigChange} obj Optional instance to populate.
     * @return {module:model/ConfigChange} The populated <code>ConfigChange</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConfigChange();

        if (data.hasOwnProperty('config_hash')) {
          obj['config_hash'] = _ConfigHash["default"].constructFromObject(data['config_hash']);
        }

        if (data.hasOwnProperty('last_change')) {
          obj['last_change'] = _LastConfigChange["default"].constructFromObject(data['last_change']);
        }

        if (data.hasOwnProperty('last_status')) {
          obj['last_status'] = _LastStatusInfo["default"].constructFromObject(data['last_status']);
        }

        if (data.hasOwnProperty('required_activity')) {
          obj['required_activity'] = _RequiredActivity["default"].constructFromObject(data['required_activity']);
        }

        if (data.hasOwnProperty('update_stamp')) {
          obj['update_stamp'] = _ApiClient["default"].convertToType(data['update_stamp'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ConfigChange;
}();
/**
 * @member {module:model/ConfigHash} config_hash
 */


ConfigChange.prototype['config_hash'] = undefined;
/**
 * @member {module:model/LastConfigChange} last_change
 */

ConfigChange.prototype['last_change'] = undefined;
/**
 * @member {module:model/LastStatusInfo} last_status
 */

ConfigChange.prototype['last_status'] = undefined;
/**
 * @member {module:model/RequiredActivity} required_activity
 */

ConfigChange.prototype['required_activity'] = undefined;
/**
 * @member {Number} update_stamp
 */

ConfigChange.prototype['update_stamp'] = undefined;
var _default = ConfigChange;
exports["default"] = _default;
