// akeyless/dist/model/ConfigHash.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConfigHash model module.
 * @module model/ConfigHash
 * @version 2.20.0
 */
var ConfigHash = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConfigHash</code>.
   * @alias module:model/ConfigHash
   */
  function ConfigHash() {
    _classCallCheck(this, ConfigHash);

    ConfigHash.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConfigHash, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConfigHash</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConfigHash} obj Optional instance to populate.
     * @return {module:model/ConfigHash} The populated <code>ConfigHash</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConfigHash();

        if (data.hasOwnProperty('admins')) {
          obj['admins'] = _ApiClient["default"].convertToType(data['admins'], 'String');
        }

        if (data.hasOwnProperty('cache')) {
          obj['cache'] = _ApiClient["default"].convertToType(data['cache'], 'String');
        }

        if (data.hasOwnProperty('customer_fragements')) {
          obj['customer_fragements'] = _ApiClient["default"].convertToType(data['customer_fragements'], 'String');
        }

        if (data.hasOwnProperty('general')) {
          obj['general'] = _ApiClient["default"].convertToType(data['general'], 'String');
        }

        if (data.hasOwnProperty('k8s_auths')) {
          obj['k8s_auths'] = _ApiClient["default"].convertToType(data['k8s_auths'], 'String');
        }

        if (data.hasOwnProperty('kmip')) {
          obj['kmip'] = _ApiClient["default"].convertToType(data['kmip'], 'String');
        }

        if (data.hasOwnProperty('ldap')) {
          obj['ldap'] = _ApiClient["default"].convertToType(data['ldap'], 'String');
        }

        if (data.hasOwnProperty('leadership')) {
          obj['leadership'] = _ApiClient["default"].convertToType(data['leadership'], 'String');
        }

        if (data.hasOwnProperty('log_forwarding')) {
          obj['log_forwarding'] = _ApiClient["default"].convertToType(data['log_forwarding'], 'String');
        }

        if (data.hasOwnProperty('m_queue')) {
          obj['m_queue'] = _ApiClient["default"].convertToType(data['m_queue'], 'String');
        }

        if (data.hasOwnProperty('migration_status')) {
          obj['migration_status'] = _ApiClient["default"].convertToType(data['migration_status'], 'String');
        }

        if (data.hasOwnProperty('migrations')) {
          obj['migrations'] = _ApiClient["default"].convertToType(data['migrations'], 'String');
        }

        if (data.hasOwnProperty('producers')) {
          obj['producers'] = _ApiClient["default"].convertToType(data['producers'], Object);
        }

        if (data.hasOwnProperty('producers_status')) {
          obj['producers_status'] = _ApiClient["default"].convertToType(data['producers_status'], 'String');
        }

        if (data.hasOwnProperty('rotators')) {
          obj['rotators'] = _ApiClient["default"].convertToType(data['rotators'], Object);
        }

        if (data.hasOwnProperty('saml')) {
          obj['saml'] = _ApiClient["default"].convertToType(data['saml'], 'String');
        }

        if (data.hasOwnProperty('universal_identity')) {
          obj['universal_identity'] = _ApiClient["default"].convertToType(data['universal_identity'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ConfigHash;
}();
/**
 * @member {String} admins
 */


ConfigHash.prototype['admins'] = undefined;
/**
 * @member {String} cache
 */

ConfigHash.prototype['cache'] = undefined;
/**
 * @member {String} customer_fragements
 */

ConfigHash.prototype['customer_fragements'] = undefined;
/**
 * @member {String} general
 */

ConfigHash.prototype['general'] = undefined;
/**
 * @member {String} k8s_auths
 */

ConfigHash.prototype['k8s_auths'] = undefined;
/**
 * @member {String} kmip
 */

ConfigHash.prototype['kmip'] = undefined;
/**
 * @member {String} ldap
 */

ConfigHash.prototype['ldap'] = undefined;
/**
 * @member {String} leadership
 */

ConfigHash.prototype['leadership'] = undefined;
/**
 * @member {String} log_forwarding
 */

ConfigHash.prototype['log_forwarding'] = undefined;
/**
 * @member {String} m_queue
 */

ConfigHash.prototype['m_queue'] = undefined;
/**
 * @member {String} migration_status
 */

ConfigHash.prototype['migration_status'] = undefined;
/**
 * @member {String} migrations
 */

ConfigHash.prototype['migrations'] = undefined;
/**
 * @member {Object} producers
 */

ConfigHash.prototype['producers'] = undefined;
/**
 * @member {String} producers_status
 */

ConfigHash.prototype['producers_status'] = undefined;
/**
 * @member {Object} rotators
 */

ConfigHash.prototype['rotators'] = undefined;
/**
 * @member {String} saml
 */

ConfigHash.prototype['saml'] = undefined;
/**
 * @member {String} universal_identity
 */

ConfigHash.prototype['universal_identity'] = undefined;
var _default = ConfigHash;
exports["default"] = _default;
