// akeyless/dist/model/CreateAuthMethodOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodOutput model module.
 * @module model/CreateAuthMethodOutput
 * @version 2.20.0
 */
var CreateAuthMethodOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodOutput</code>.
   * @alias module:model/CreateAuthMethodOutput
   */
  function CreateAuthMethodOutput() {
    _classCallCheck(this, CreateAuthMethodOutput);

    CreateAuthMethodOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CreateAuthMethodOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodOutput} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodOutput} The populated <code>CreateAuthMethodOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodOutput();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('access_key')) {
          obj['access_key'] = _ApiClient["default"].convertToType(data['access_key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodOutput;
}();
/**
 * @member {String} access_id
 */


CreateAuthMethodOutput.prototype['access_id'] = undefined;
/**
 * @member {String} access_key
 */

CreateAuthMethodOutput.prototype['access_key'] = undefined;
/**
 * @member {String} name
 */

CreateAuthMethodOutput.prototype['name'] = undefined;
var _default = CreateAuthMethodOutput;
exports["default"] = _default;
