// akeyless/dist/model/CreateRotatedSecret.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateRotatedSecret model module.
 * @module model/CreateRotatedSecret
 * @version 2.20.0
 */
var CreateRotatedSecret = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRotatedSecret</code>.
   * @alias module:model/CreateRotatedSecret
   * @param name {String} Secret name
   * @param rotatorType {String} Rotator Type
   * @param targetName {String} Target name
   */
  function CreateRotatedSecret(name, rotatorType, targetName) {
    _classCallCheck(this, CreateRotatedSecret);

    CreateRotatedSecret.initialize(this, name, rotatorType, targetName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRotatedSecret, null, [{
    key: "initialize",
    value: function initialize(obj, name, rotatorType, targetName) {
      obj['name'] = name;
      obj['rotator-type'] = rotatorType;
      obj['target-name'] = targetName;
    }
    /**
     * Constructs a <code>CreateRotatedSecret</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRotatedSecret} obj Optional instance to populate.
     * @return {module:model/CreateRotatedSecret} The populated <code>CreateRotatedSecret</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRotatedSecret();

        if (data.hasOwnProperty('api-id')) {
          obj['api-id'] = _ApiClient["default"].convertToType(data['api-id'], 'String');
        }

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('application-id')) {
          obj['application-id'] = _ApiClient["default"].convertToType(data['application-id'], 'String');
        }

        if (data.hasOwnProperty('authentication-credentials')) {
          obj['authentication-credentials'] = _ApiClient["default"].convertToType(data['authentication-credentials'], 'String');
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('aws-region')) {
          obj['aws-region'] = _ApiClient["default"].convertToType(data['aws-region'], 'String');
        }

        if (data.hasOwnProperty('custom-payload')) {
          obj['custom-payload'] = _ApiClient["default"].convertToType(data['custom-payload'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rotated-password')) {
          obj['rotated-password'] = _ApiClient["default"].convertToType(data['rotated-password'], 'String');
        }

        if (data.hasOwnProperty('rotated-username')) {
          obj['rotated-username'] = _ApiClient["default"].convertToType(data['rotated-username'], 'String');
        }

        if (data.hasOwnProperty('rotation-hour')) {
          obj['rotation-hour'] = _ApiClient["default"].convertToType(data['rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('rotator-creds-type')) {
          obj['rotator-creds-type'] = _ApiClient["default"].convertToType(data['rotator-creds-type'], 'String');
        }

        if (data.hasOwnProperty('rotator-custom-cmd')) {
          obj['rotator-custom-cmd'] = _ApiClient["default"].convertToType(data['rotator-custom-cmd'], 'String');
        }

        if (data.hasOwnProperty('rotator-type')) {
          obj['rotator-type'] = _ApiClient["default"].convertToType(data['rotator-type'], 'String');
        }

        if (data.hasOwnProperty('secure-access-allow-external-user')) {
          obj['secure-access-allow-external-user'] = _ApiClient["default"].convertToType(data['secure-access-allow-external-user'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-aws-account-id')) {
          obj['secure-access-aws-account-id'] = _ApiClient["default"].convertToType(data['secure-access-aws-account-id'], 'String');
        }

        if (data.hasOwnProperty('secure-access-aws-native-cli')) {
          obj['secure-access-aws-native-cli'] = _ApiClient["default"].convertToType(data['secure-access-aws-native-cli'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-name')) {
          obj['secure-access-db-name'] = _ApiClient["default"].convertToType(data['secure-access-db-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-schema')) {
          obj['secure-access-db-schema'] = _ApiClient["default"].convertToType(data['secure-access-db-schema'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-rdp-domain')) {
          obj['secure-access-rdp-domain'] = _ApiClient["default"].convertToType(data['secure-access-rdp-domain'], 'String');
        }

        if (data.hasOwnProperty('secure-access-rdp-user')) {
          obj['secure-access-rdp-user'] = _ApiClient["default"].convertToType(data['secure-access-rdp-user'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-proxy')) {
          obj['secure-access-web-proxy'] = _ApiClient["default"].convertToType(data['secure-access-web-proxy'], 'Boolean');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('storage-account-key-name')) {
          obj['storage-account-key-name'] = _ApiClient["default"].convertToType(data['storage-account-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-attribute')) {
          obj['user-attribute'] = _ApiClient["default"].convertToType(data['user-attribute'], 'String');
        }

        if (data.hasOwnProperty('user-dn')) {
          obj['user-dn'] = _ApiClient["default"].convertToType(data['user-dn'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateRotatedSecret;
}();
/**
 * @member {String} api-id
 */


CreateRotatedSecret.prototype['api-id'] = undefined;
/**
 * @member {String} api-key
 */

CreateRotatedSecret.prototype['api-key'] = undefined;
/**
 * ApplicationId (used in azure)
 * @member {String} application-id
 */

CreateRotatedSecret.prototype['application-id'] = undefined;
/**
 * @member {String} authentication-credentials
 */

CreateRotatedSecret.prototype['authentication-credentials'] = undefined;
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
 * @member {String} auto-rotate
 */

CreateRotatedSecret.prototype['auto-rotate'] = undefined;
/**
 * Region (used in aws)
 * @member {String} aws-region
 * @default 'us-east-2'
 */

CreateRotatedSecret.prototype['aws-region'] = 'us-east-2';
/**
 * @member {String} custom-payload
 */

CreateRotatedSecret.prototype['custom-payload'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

CreateRotatedSecret.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateRotatedSecret.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateRotatedSecret.prototype['key'] = undefined;
/**
 * Metadata about the secret
 * @member {String} metadata
 */

CreateRotatedSecret.prototype['metadata'] = undefined;
/**
 * Secret name
 * @member {String} name
 */

CreateRotatedSecret.prototype['name'] = undefined;
/**
 * @member {String} rotated-password
 */

CreateRotatedSecret.prototype['rotated-password'] = undefined;
/**
 * @member {String} rotated-username
 */

CreateRotatedSecret.prototype['rotated-username'] = undefined;
/**
 * @member {Number} rotation-hour
 */

CreateRotatedSecret.prototype['rotation-hour'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (1-365)
 * @member {String} rotation-interval
 */

CreateRotatedSecret.prototype['rotation-interval'] = undefined;
/**
 * @member {String} rotator-creds-type
 */

CreateRotatedSecret.prototype['rotator-creds-type'] = undefined;
/**
 * @member {String} rotator-custom-cmd
 */

CreateRotatedSecret.prototype['rotator-custom-cmd'] = undefined;
/**
 * Rotator Type
 * @member {String} rotator-type
 */

CreateRotatedSecret.prototype['rotator-type'] = undefined;
/**
 * Secure Access Allow Providing External User (used in ssh)
 * @member {Boolean} secure-access-allow-external-user
 * @default false
 */

CreateRotatedSecret.prototype['secure-access-allow-external-user'] = false;
/**
 * Secure Access Account Id (used in aws)
 * @member {String} secure-access-aws-account-id
 */

CreateRotatedSecret.prototype['secure-access-aws-account-id'] = undefined;
/**
 * Secure Access Aws Native Cli (used in aws)
 * @member {Boolean} secure-access-aws-native-cli
 */

CreateRotatedSecret.prototype['secure-access-aws-native-cli'] = undefined;
/**
 * Secure Access Bastion Issuer
 * @member {String} secure-access-bastion-issuer
 */

CreateRotatedSecret.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * Secure Access DB Name (used in data bases)
 * @member {String} secure-access-db-name
 */

CreateRotatedSecret.prototype['secure-access-db-name'] = undefined;
/**
 * Secure Access Schema (used in mssql, postgresql)
 * @member {String} secure-access-db-schema
 */

CreateRotatedSecret.prototype['secure-access-db-schema'] = undefined;
/**
 * Secure Access Enabled
 * @member {String} secure-access-enable
 */

CreateRotatedSecret.prototype['secure-access-enable'] = undefined;
/**
 * Secure Access Host
 * @member {Array.<String>} secure-access-host
 */

CreateRotatedSecret.prototype['secure-access-host'] = undefined;
/**
 * Secure Access Domain (used in ssh)
 * @member {String} secure-access-rdp-domain
 */

CreateRotatedSecret.prototype['secure-access-rdp-domain'] = undefined;
/**
 * Secure Access Override User (used in ssh)
 * @member {String} secure-access-rdp-user
 */

CreateRotatedSecret.prototype['secure-access-rdp-user'] = undefined;
/**
 * Secure Access Web
 * @member {Boolean} secure-access-web
 * @default false
 */

CreateRotatedSecret.prototype['secure-access-web'] = false;
/**
 * Secure Access Isolated (used in aws, azure)
 * @member {Boolean} secure-access-web-browsing
 * @default false
 */

CreateRotatedSecret.prototype['secure-access-web-browsing'] = false;
/**
 * Secure Access Web Proxy (used in aws, azure)
 * @member {Boolean} secure-access-web-proxy
 * @default false
 */

CreateRotatedSecret.prototype['secure-access-web-proxy'] = false;
/**
 * Deprecated: use RotatedPassword
 * @member {String} ssh-password
 */

CreateRotatedSecret.prototype['ssh-password'] = undefined;
/**
 * Deprecated: use RotatedUser
 * @member {String} ssh-username
 */

CreateRotatedSecret.prototype['ssh-username'] = undefined;
/**
 * The name of the storage account key to rotate [key1/key2/kerb1/kerb2] (relevat to azure-storage-account)
 * @member {String} storage-account-key-name
 */

CreateRotatedSecret.prototype['storage-account-key-name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

CreateRotatedSecret.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

CreateRotatedSecret.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateRotatedSecret.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateRotatedSecret.prototype['uid-token'] = undefined;
/**
 * LDAP User Attribute, Default value \"cn\"
 * @member {String} user-attribute
 */

CreateRotatedSecret.prototype['user-attribute'] = undefined;
/**
 * LDAP User Base DN
 * @member {String} user-dn
 */

CreateRotatedSecret.prototype['user-dn'] = undefined;
var _default = CreateRotatedSecret;
exports["default"] = _default;
