// akeyless/dist/model/CreateAuthMethodSAML.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodSAML model module.
 * @module model/CreateAuthMethodSAML
 * @version 2.15.24
 */
var CreateAuthMethodSAML = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodSAML</code>.
   * createAuthMethodSAML is a command that creates a new auth method that will be available to authenticate using SAML.
   * @alias module:model/CreateAuthMethodSAML
   * @param name {String} Auth Method name
   * @param uniqueIdentifier {String} A unique identifier (ID) value should be configured for OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
   */
  function CreateAuthMethodSAML(name, uniqueIdentifier) {
    _classCallCheck(this, CreateAuthMethodSAML);

    CreateAuthMethodSAML.initialize(this, name, uniqueIdentifier);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodSAML, null, [{
    key: "initialize",
    value: function initialize(obj, name, uniqueIdentifier) {
      obj['name'] = name;
      obj['unique-identifier'] = uniqueIdentifier;
    }
    /**
     * Constructs a <code>CreateAuthMethodSAML</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodSAML} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodSAML} The populated <code>CreateAuthMethodSAML</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodSAML();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('allowed-redirect-uri')) {
          obj['allowed-redirect-uri'] = _ApiClient["default"].convertToType(data['allowed-redirect-uri'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('idp-metadata-url')) {
          obj['idp-metadata-url'] = _ApiClient["default"].convertToType(data['idp-metadata-url'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodSAML;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodSAML.prototype['access-expires'] = 0;
/**
 * Allowed redirect URIs after the authentication
 * @member {Array.<String>} allowed-redirect-uri
 */

CreateAuthMethodSAML.prototype['allowed-redirect-uri'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodSAML.prototype['bound-ips'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodSAML.prototype['force-sub-claims'] = undefined;
/**
 * IDP metadata url
 * @member {String} idp-metadata-url
 */

CreateAuthMethodSAML.prototype['idp-metadata-url'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodSAML.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

CreateAuthMethodSAML.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodSAML.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodSAML.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value should be configured for OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
 * @member {String} unique-identifier
 */

CreateAuthMethodSAML.prototype['unique-identifier'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

CreateAuthMethodSAML.prototype['username'] = undefined;
var _default = CreateAuthMethodSAML;
exports["default"] = _default;
