// akeyless/dist/model/CreateAzureTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAzureTarget model module.
 * @module model/CreateAzureTarget
 * @version 2.20.0
 */
var CreateAzureTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAzureTarget</code>.
   * @alias module:model/CreateAzureTarget
   * @param name {String} Target name
   */
  function CreateAzureTarget(name) {
    _classCallCheck(this, CreateAzureTarget);

    CreateAzureTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAzureTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateAzureTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAzureTarget} obj Optional instance to populate.
     * @return {module:model/CreateAzureTarget} The populated <code>CreateAzureTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAzureTarget();

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('client-secret')) {
          obj['client-secret'] = _ApiClient["default"].convertToType(data['client-secret'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('resource-group-name')) {
          obj['resource-group-name'] = _ApiClient["default"].convertToType(data['resource-group-name'], 'String');
        }

        if (data.hasOwnProperty('resource-name')) {
          obj['resource-name'] = _ApiClient["default"].convertToType(data['resource-name'], 'String');
        }

        if (data.hasOwnProperty('subscription-id')) {
          obj['subscription-id'] = _ApiClient["default"].convertToType(data['subscription-id'], 'String');
        }

        if (data.hasOwnProperty('tenant-id')) {
          obj['tenant-id'] = _ApiClient["default"].convertToType(data['tenant-id'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return CreateAzureTarget;
}();
/**
 * @member {String} client-id
 */


CreateAzureTarget.prototype['client-id'] = undefined;
/**
 * @member {String} client-secret
 */

CreateAzureTarget.prototype['client-secret'] = undefined;
/**
 * Comment about the target
 * @member {String} comment
 */

CreateAzureTarget.prototype['comment'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateAzureTarget.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateAzureTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

CreateAzureTarget.prototype['name'] = undefined;
/**
 * The Resource Group name in your Azure subscription
 * @member {String} resource-group-name
 */

CreateAzureTarget.prototype['resource-group-name'] = undefined;
/**
 * The name of the relevant Resource
 * @member {String} resource-name
 */

CreateAzureTarget.prototype['resource-name'] = undefined;
/**
 * Azure Subscription Id
 * @member {String} subscription-id
 */

CreateAzureTarget.prototype['subscription-id'] = undefined;
/**
 * @member {String} tenant-id
 */

CreateAzureTarget.prototype['tenant-id'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAzureTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAzureTarget.prototype['uid-token'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

CreateAzureTarget.prototype['use-gw-cloud-identity'] = undefined;
var _default = CreateAzureTarget;
exports["default"] = _default;
