// akeyless/dist/model/CreateAuthMethodOAuth2.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodOAuth2 model module.
 * @module model/CreateAuthMethodOAuth2
 * @version 2.15.24
 */
var CreateAuthMethodOAuth2 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodOAuth2</code>.
   * createAuthMethodOAuth2 is a command that creates a new auth method that will be able to authenticate using Oauth2.
   * @alias module:model/CreateAuthMethodOAuth2
   * @param jwksUri {String} The URL to the JSON Web Key Set (JWKS) that containing the public keys that should be used to verify any JSON Web Token (JWT) issued by the authorization server.
   * @param name {String} Auth Method name
   * @param uniqueIdentifier {String} A unique identifier (ID) value should be configured for OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
   */
  function CreateAuthMethodOAuth2(jwksUri, name, uniqueIdentifier) {
    _classCallCheck(this, CreateAuthMethodOAuth2);

    CreateAuthMethodOAuth2.initialize(this, jwksUri, name, uniqueIdentifier);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodOAuth2, null, [{
    key: "initialize",
    value: function initialize(obj, jwksUri, name, uniqueIdentifier) {
      obj['jwks-uri'] = jwksUri;
      obj['name'] = name;
      obj['unique-identifier'] = uniqueIdentifier;
    }
    /**
     * Constructs a <code>CreateAuthMethodOAuth2</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodOAuth2} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodOAuth2} The populated <code>CreateAuthMethodOAuth2</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodOAuth2();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('bound-client-ids')) {
          obj['bound-client-ids'] = _ApiClient["default"].convertToType(data['bound-client-ids'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('jwks-uri')) {
          obj['jwks-uri'] = _ApiClient["default"].convertToType(data['jwks-uri'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodOAuth2;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodOAuth2.prototype['access-expires'] = 0;
/**
 * The audience in the JWT
 * @member {String} audience
 */

CreateAuthMethodOAuth2.prototype['audience'] = undefined;
/**
 * The clients ids that the access is restricted to
 * @member {Array.<String>} bound-client-ids
 */

CreateAuthMethodOAuth2.prototype['bound-client-ids'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodOAuth2.prototype['bound-ips'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodOAuth2.prototype['force-sub-claims'] = undefined;
/**
 * Issuer URL
 * @member {String} issuer
 */

CreateAuthMethodOAuth2.prototype['issuer'] = undefined;
/**
 * The URL to the JSON Web Key Set (JWKS) that containing the public keys that should be used to verify any JSON Web Token (JWT) issued by the authorization server.
 * @member {String} jwks-uri
 */

CreateAuthMethodOAuth2.prototype['jwks-uri'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodOAuth2.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

CreateAuthMethodOAuth2.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodOAuth2.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodOAuth2.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value should be configured for OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
 * @member {String} unique-identifier
 */

CreateAuthMethodOAuth2.prototype['unique-identifier'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

CreateAuthMethodOAuth2.prototype['username'] = undefined;
var _default = CreateAuthMethodOAuth2;
exports["default"] = _default;
