// akeyless/dist/model/CreateSSHTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateSSHTarget model module.
 * @module model/CreateSSHTarget
 * @version 2.20.0
 */
var CreateSSHTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateSSHTarget</code>.
   * @alias module:model/CreateSSHTarget
   * @param name {String} Target name
   */
  function CreateSSHTarget(name) {
    _classCallCheck(this, CreateSSHTarget);

    CreateSSHTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateSSHTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateSSHTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateSSHTarget} obj Optional instance to populate.
     * @return {module:model/CreateSSHTarget} The populated <code>CreateSSHTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateSSHTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('private-key')) {
          obj['private-key'] = _ApiClient["default"].convertToType(data['private-key'], 'String');
        }

        if (data.hasOwnProperty('private-key-password')) {
          obj['private-key-password'] = _ApiClient["default"].convertToType(data['private-key-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateSSHTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


CreateSSHTarget.prototype['comment'] = undefined;
/**
 * @member {String} host
 */

CreateSSHTarget.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateSSHTarget.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateSSHTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

CreateSSHTarget.prototype['name'] = undefined;
/**
 * @member {String} port
 */

CreateSSHTarget.prototype['port'] = undefined;
/**
 * @member {String} private-key
 */

CreateSSHTarget.prototype['private-key'] = undefined;
/**
 * @member {String} private-key-password
 */

CreateSSHTarget.prototype['private-key-password'] = undefined;
/**
 * @member {String} ssh-password
 */

CreateSSHTarget.prototype['ssh-password'] = undefined;
/**
 * @member {String} ssh-username
 */

CreateSSHTarget.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateSSHTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateSSHTarget.prototype['uid-token'] = undefined;
var _default = CreateSSHTarget;
exports["default"] = _default;
