// akeyless/dist/model/CreateManagedKeyOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateManagedKeyOutput model module.
 * @module model/CreateManagedKeyOutput
 * @version 2.4.0
 */
var CreateManagedKeyOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateManagedKeyOutput</code>.
   * @alias module:model/CreateManagedKeyOutput
   */
  function CreateManagedKeyOutput() {
    _classCallCheck(this, CreateManagedKeyOutput);

    CreateManagedKeyOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateManagedKeyOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CreateManagedKeyOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateManagedKeyOutput} obj Optional instance to populate.
     * @return {module:model/CreateManagedKeyOutput} The populated <code>CreateManagedKeyOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateManagedKeyOutput();

        if (data.hasOwnProperty('managed_key_id')) {
          obj['managed_key_id'] = _ApiClient["default"].convertToType(data['managed_key_id'], 'Number');
        }

        if (data.hasOwnProperty('managed_key_name')) {
          obj['managed_key_name'] = _ApiClient["default"].convertToType(data['managed_key_name'], 'String');
        }

        if (data.hasOwnProperty('managed_key_type')) {
          obj['managed_key_type'] = _ApiClient["default"].convertToType(data['managed_key_type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateManagedKeyOutput;
}();
/**
 * @member {Number} managed_key_id
 */


CreateManagedKeyOutput.prototype['managed_key_id'] = undefined;
/**
 * @member {String} managed_key_name
 */

CreateManagedKeyOutput.prototype['managed_key_name'] = undefined;
/**
 * @member {String} managed_key_type
 */

CreateManagedKeyOutput.prototype['managed_key_type'] = undefined;
var _default = CreateManagedKeyOutput;
exports["default"] = _default;
