// akeyless/dist/model/CreateAuthMethodCert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodCert model module.
 * @module model/CreateAuthMethodCert
 * @version 2.20.0
 */
var CreateAuthMethodCert = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodCert</code>.
   * createAuthMethodCert is a command that creates a new auth method that will be able to authenticate using a client certificae
   * @alias module:model/CreateAuthMethodCert
   * @param name {String} Auth Method name
   * @param uniqueIdentifier {String} A unique identifier (ID) value should be configured, such as common_name or organizational_unit Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
   */
  function CreateAuthMethodCert(name, uniqueIdentifier) {
    _classCallCheck(this, CreateAuthMethodCert);

    CreateAuthMethodCert.initialize(this, name, uniqueIdentifier);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodCert, null, [{
    key: "initialize",
    value: function initialize(obj, name, uniqueIdentifier) {
      obj['name'] = name;
      obj['unique-identifier'] = uniqueIdentifier;
    }
    /**
     * Constructs a <code>CreateAuthMethodCert</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodCert} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodCert} The populated <code>CreateAuthMethodCert</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodCert();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('bound-common-names')) {
          obj['bound-common-names'] = _ApiClient["default"].convertToType(data['bound-common-names'], ['String']);
        }

        if (data.hasOwnProperty('bound-dns-sans')) {
          obj['bound-dns-sans'] = _ApiClient["default"].convertToType(data['bound-dns-sans'], ['String']);
        }

        if (data.hasOwnProperty('bound-email-sans')) {
          obj['bound-email-sans'] = _ApiClient["default"].convertToType(data['bound-email-sans'], ['String']);
        }

        if (data.hasOwnProperty('bound-extensions')) {
          obj['bound-extensions'] = _ApiClient["default"].convertToType(data['bound-extensions'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-organizational-units')) {
          obj['bound-organizational-units'] = _ApiClient["default"].convertToType(data['bound-organizational-units'], ['String']);
        }

        if (data.hasOwnProperty('bound-uri-sans')) {
          obj['bound-uri-sans'] = _ApiClient["default"].convertToType(data['bound-uri-sans'], ['String']);
        }

        if (data.hasOwnProperty('certificate-data')) {
          obj['certificate-data'] = _ApiClient["default"].convertToType(data['certificate-data'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('revoked-cert-ids')) {
          obj['revoked-cert-ids'] = _ApiClient["default"].convertToType(data['revoked-cert-ids'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodCert;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodCert.prototype['access-expires'] = 0;
/**
 * A list of names. At least one must exist in the Common Name. Supports globbing.
 * @member {Array.<String>} bound-common-names
 */

CreateAuthMethodCert.prototype['bound-common-names'] = undefined;
/**
 * A list of DNS names. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-dns-sans
 */

CreateAuthMethodCert.prototype['bound-dns-sans'] = undefined;
/**
 * A list of Email Addresses. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-email-sans
 */

CreateAuthMethodCert.prototype['bound-email-sans'] = undefined;
/**
 * A list of extensions formatted as \"oid:value\". Expects the extension value to be some type of ASN1 encoded string. All values much match. Supports globbing on \"value\".
 * @member {Array.<String>} bound-extensions
 */

CreateAuthMethodCert.prototype['bound-extensions'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodCert.prototype['bound-ips'] = undefined;
/**
 * A list of Organizational Units names. At least one must exist in the OU field.
 * @member {Array.<String>} bound-organizational-units
 */

CreateAuthMethodCert.prototype['bound-organizational-units'] = undefined;
/**
 * A list of URIs. At least one must exist in the SANs. Supports globbing.
 * @member {Array.<String>} bound-uri-sans
 */

CreateAuthMethodCert.prototype['bound-uri-sans'] = undefined;
/**
 * The certificate data in base64, if no file was provided
 * @member {String} certificate-data
 */

CreateAuthMethodCert.prototype['certificate-data'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodCert.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

CreateAuthMethodCert.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateAuthMethodCert.prototype['json'] = undefined;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 */

CreateAuthMethodCert.prototype['jwt-ttl'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodCert.prototype['name'] = undefined;
/**
 * A list of revoked cert ids
 * @member {Array.<String>} revoked-cert-ids
 */

CreateAuthMethodCert.prototype['revoked-cert-ids'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodCert.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodCert.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value should be configured, such as common_name or organizational_unit Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
 * @member {String} unique-identifier
 */

CreateAuthMethodCert.prototype['unique-identifier'] = undefined;
var _default = CreateAuthMethodCert;
exports["default"] = _default;
