// akeyless/dist/model/CreateAuthMethodUniversalIdentity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodUniversalIdentity model module.
 * @module model/CreateAuthMethodUniversalIdentity
 * @version 2.15.24
 */
var CreateAuthMethodUniversalIdentity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodUniversalIdentity</code>.
   * createAuthMethodUniversalIdentity is a command that creates a new auth method that will be able to authenticate using Akeyless Universal Identity.
   * @alias module:model/CreateAuthMethodUniversalIdentity
   * @param name {String} Auth Method name
   */
  function CreateAuthMethodUniversalIdentity(name) {
    _classCallCheck(this, CreateAuthMethodUniversalIdentity);

    CreateAuthMethodUniversalIdentity.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodUniversalIdentity, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateAuthMethodUniversalIdentity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodUniversalIdentity} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodUniversalIdentity} The populated <code>CreateAuthMethodUniversalIdentity</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodUniversalIdentity();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('deny-inheritance')) {
          obj['deny-inheritance'] = _ApiClient["default"].convertToType(data['deny-inheritance'], 'Boolean');
        }

        if (data.hasOwnProperty('deny-rotate')) {
          obj['deny-rotate'] = _ApiClient["default"].convertToType(data['deny-rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodUniversalIdentity;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodUniversalIdentity.prototype['access-expires'] = 0;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodUniversalIdentity.prototype['bound-ips'] = undefined;
/**
 * Deny from root to create children
 * @member {Boolean} deny-inheritance
 */

CreateAuthMethodUniversalIdentity.prototype['deny-inheritance'] = undefined;
/**
 * Deny from the token to rotate
 * @member {Boolean} deny-rotate
 */

CreateAuthMethodUniversalIdentity.prototype['deny-rotate'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodUniversalIdentity.prototype['force-sub-claims'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodUniversalIdentity.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

CreateAuthMethodUniversalIdentity.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodUniversalIdentity.prototype['token'] = undefined;
/**
 * Token ttl
 * @member {Number} ttl
 * @default 60
 */

CreateAuthMethodUniversalIdentity.prototype['ttl'] = 60;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodUniversalIdentity.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

CreateAuthMethodUniversalIdentity.prototype['username'] = undefined;
var _default = CreateAuthMethodUniversalIdentity;
exports["default"] = _default;
