// akeyless/dist/model/CreateTokenizer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateTokenizer model module.
 * @module model/CreateTokenizer
 * @version 2.20.0
 */
var CreateTokenizer = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateTokenizer</code>.
   * createTokenizer is a command that creates a tokenizer item
   * @alias module:model/CreateTokenizer
   * @param name {String} Tokenizer name
   * @param templateType {String} Which template type this tokenizer is used for [SSN,CreditCard,USPhoneNumber,Email,Regexp]
   * @param tokenizerType {String} Tokenizer type
   */
  function CreateTokenizer(name, templateType, tokenizerType) {
    _classCallCheck(this, CreateTokenizer);

    CreateTokenizer.initialize(this, name, templateType, tokenizerType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateTokenizer, null, [{
    key: "initialize",
    value: function initialize(obj, name, templateType, tokenizerType) {
      obj['name'] = name;
      obj['template-type'] = templateType;
      obj['tokenizer-type'] = tokenizerType;
    }
    /**
     * Constructs a <code>CreateTokenizer</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateTokenizer} obj Optional instance to populate.
     * @return {module:model/CreateTokenizer} The populated <code>CreateTokenizer</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateTokenizer();

        if (data.hasOwnProperty('alphabet')) {
          obj['alphabet'] = _ApiClient["default"].convertToType(data['alphabet'], 'String');
        }

        if (data.hasOwnProperty('decoding-template')) {
          obj['decoding-template'] = _ApiClient["default"].convertToType(data['decoding-template'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('encoding-template')) {
          obj['encoding-template'] = _ApiClient["default"].convertToType(data['encoding-template'], 'String');
        }

        if (data.hasOwnProperty('encryption-key-name')) {
          obj['encryption-key-name'] = _ApiClient["default"].convertToType(data['encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('pattern')) {
          obj['pattern'] = _ApiClient["default"].convertToType(data['pattern'], 'String');
        }

        if (data.hasOwnProperty('tag')) {
          obj['tag'] = _ApiClient["default"].convertToType(data['tag'], ['String']);
        }

        if (data.hasOwnProperty('template-type')) {
          obj['template-type'] = _ApiClient["default"].convertToType(data['template-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('tokenizer-type')) {
          obj['tokenizer-type'] = _ApiClient["default"].convertToType(data['tokenizer-type'], 'String');
        }

        if (data.hasOwnProperty('tweak-type')) {
          obj['tweak-type'] = _ApiClient["default"].convertToType(data['tweak-type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateTokenizer;
}();
/**
 * Alphabet to use in regexp vaultless tokenization
 * @member {String} alphabet
 */


CreateTokenizer.prototype['alphabet'] = undefined;
/**
 * The Decoding output template to use in regexp vaultless tokenization
 * @member {String} decoding-template
 */

CreateTokenizer.prototype['decoding-template'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

CreateTokenizer.prototype['delete_protection'] = undefined;
/**
 * The Encoding output template to use in regexp vaultless tokenization
 * @member {String} encoding-template
 */

CreateTokenizer.prototype['encoding-template'] = undefined;
/**
 * AES key name to use in vaultless tokenization
 * @member {String} encryption-key-name
 */

CreateTokenizer.prototype['encryption-key-name'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateTokenizer.prototype['json'] = undefined;
/**
 * A metadata about the tokenizer
 * @member {String} metadata
 */

CreateTokenizer.prototype['metadata'] = undefined;
/**
 * Tokenizer name
 * @member {String} name
 */

CreateTokenizer.prototype['name'] = undefined;
/**
 * Pattern to use in regexp vaultless tokenization
 * @member {String} pattern
 */

CreateTokenizer.prototype['pattern'] = undefined;
/**
 * List of the tags attached to this key
 * @member {Array.<String>} tag
 */

CreateTokenizer.prototype['tag'] = undefined;
/**
 * Which template type this tokenizer is used for [SSN,CreditCard,USPhoneNumber,Email,Regexp]
 * @member {String} template-type
 */

CreateTokenizer.prototype['template-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateTokenizer.prototype['token'] = undefined;
/**
 * Tokenizer type
 * @member {String} tokenizer-type
 */

CreateTokenizer.prototype['tokenizer-type'] = undefined;
/**
 * The tweak type to use in vaultless tokenization [Supplied, Generated, Internal, Masking]
 * @member {String} tweak-type
 */

CreateTokenizer.prototype['tweak-type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateTokenizer.prototype['uid-token'] = undefined;
var _default = CreateTokenizer;
exports["default"] = _default;
