// akeyless/dist/model/CreateManagedKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateManagedKey model module.
 * @module model/CreateManagedKey
 * @version 2.4.0
 */
var CreateManagedKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateManagedKey</code>.
   * CreateManagedKey is a command that creates managed key
   * @alias module:model/CreateManagedKey
   * @param alg {String} Managed Key type; options: [AES256GCM, RSA2048]
   * @param name {String} ManagedKey name
   */
  function CreateManagedKey(alg, name) {
    _classCallCheck(this, CreateManagedKey);

    CreateManagedKey.initialize(this, alg, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateManagedKey, null, [{
    key: "initialize",
    value: function initialize(obj, alg, name) {
      obj['alg'] = alg;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateManagedKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateManagedKey} obj Optional instance to populate.
     * @return {module:model/CreateManagedKey} The populated <code>CreateManagedKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateManagedKey();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('managed-key-value')) {
          obj['managed-key-value'] = _ApiClient["default"].convertToType(data['managed-key-value'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateManagedKey;
}();
/**
 * Managed Key type; options: [AES256GCM, RSA2048]
 * @member {String} alg
 */


CreateManagedKey.prototype['alg'] = undefined;
/**
 * Base64-encoded managed key value
 * @member {String} managed-key-value
 */

CreateManagedKey.prototype['managed-key-value'] = undefined;
/**
 * Metadata about the managed key
 * @member {String} metadata
 */

CreateManagedKey.prototype['metadata'] = undefined;
/**
 * ManagedKey name
 * @member {String} name
 */

CreateManagedKey.prototype['name'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

CreateManagedKey.prototype['protection_key'] = undefined;
/**
 * List of the tags attached to this managed key
 * @member {Array.<String>} tags
 */

CreateManagedKey.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

CreateManagedKey.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateManagedKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateManagedKey.prototype['uid-token'] = undefined;
var _default = CreateManagedKey;
exports["default"] = _default;
