// akeyless/dist/model/CreateRotatedSecret.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateRotatedSecret model module.
 * @module model/CreateRotatedSecret
 * @version 2.15.24
 */
var CreateRotatedSecret = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRotatedSecret</code>.
   * @alias module:model/CreateRotatedSecret
   * @param name {String} Secret name
   * @param rotatorType {String} Rotator Type
   * @param targetName {String} Target name
   */
  function CreateRotatedSecret(name, rotatorType, targetName) {
    _classCallCheck(this, CreateRotatedSecret);

    CreateRotatedSecret.initialize(this, name, rotatorType, targetName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRotatedSecret, null, [{
    key: "initialize",
    value: function initialize(obj, name, rotatorType, targetName) {
      obj['name'] = name;
      obj['rotator-type'] = rotatorType;
      obj['target-name'] = targetName;
    }
    /**
     * Constructs a <code>CreateRotatedSecret</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRotatedSecret} obj Optional instance to populate.
     * @return {module:model/CreateRotatedSecret} The populated <code>CreateRotatedSecret</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRotatedSecret();

        if (data.hasOwnProperty('api-id')) {
          obj['api-id'] = _ApiClient["default"].convertToType(data['api-id'], 'String');
        }

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('authentication-credentials')) {
          obj['authentication-credentials'] = _ApiClient["default"].convertToType(data['authentication-credentials'], 'String');
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('custom-payload')) {
          obj['custom-payload'] = _ApiClient["default"].convertToType(data['custom-payload'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('rotated-password')) {
          obj['rotated-password'] = _ApiClient["default"].convertToType(data['rotated-password'], 'String');
        }

        if (data.hasOwnProperty('rotated-username')) {
          obj['rotated-username'] = _ApiClient["default"].convertToType(data['rotated-username'], 'String');
        }

        if (data.hasOwnProperty('rotation-hour')) {
          obj['rotation-hour'] = _ApiClient["default"].convertToType(data['rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('rotator-creds-type')) {
          obj['rotator-creds-type'] = _ApiClient["default"].convertToType(data['rotator-creds-type'], 'String');
        }

        if (data.hasOwnProperty('rotator-custom-cmd')) {
          obj['rotator-custom-cmd'] = _ApiClient["default"].convertToType(data['rotator-custom-cmd'], 'String');
        }

        if (data.hasOwnProperty('rotator-type')) {
          obj['rotator-type'] = _ApiClient["default"].convertToType(data['rotator-type'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-attribute')) {
          obj['user-attribute'] = _ApiClient["default"].convertToType(data['user-attribute'], 'String');
        }

        if (data.hasOwnProperty('user-dn')) {
          obj['user-dn'] = _ApiClient["default"].convertToType(data['user-dn'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateRotatedSecret;
}();
/**
 * @member {String} api-id
 */


CreateRotatedSecret.prototype['api-id'] = undefined;
/**
 * @member {String} api-key
 */

CreateRotatedSecret.prototype['api-key'] = undefined;
/**
 * @member {String} authentication-credentials
 */

CreateRotatedSecret.prototype['authentication-credentials'] = undefined;
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
 * @member {String} auto-rotate
 */

CreateRotatedSecret.prototype['auto-rotate'] = undefined;
/**
 * @member {String} custom-payload
 */

CreateRotatedSecret.prototype['custom-payload'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateRotatedSecret.prototype['key'] = undefined;
/**
 * Metadata about the secret
 * @member {String} metadata
 */

CreateRotatedSecret.prototype['metadata'] = undefined;
/**
 * Secret name
 * @member {String} name
 */

CreateRotatedSecret.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

CreateRotatedSecret.prototype['password'] = undefined;
/**
 * @member {String} rotated-password
 */

CreateRotatedSecret.prototype['rotated-password'] = undefined;
/**
 * @member {String} rotated-username
 */

CreateRotatedSecret.prototype['rotated-username'] = undefined;
/**
 * @member {Number} rotation-hour
 */

CreateRotatedSecret.prototype['rotation-hour'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (1-365)
 * @member {String} rotation-interval
 */

CreateRotatedSecret.prototype['rotation-interval'] = undefined;
/**
 * @member {String} rotator-creds-type
 */

CreateRotatedSecret.prototype['rotator-creds-type'] = undefined;
/**
 * @member {String} rotator-custom-cmd
 */

CreateRotatedSecret.prototype['rotator-custom-cmd'] = undefined;
/**
 * Rotator Type
 * @member {String} rotator-type
 */

CreateRotatedSecret.prototype['rotator-type'] = undefined;
/**
 * Deprecated: use RotatedPassword
 * @member {String} ssh-password
 */

CreateRotatedSecret.prototype['ssh-password'] = undefined;
/**
 * Deprecated: use RotatedUser
 * @member {String} ssh-username
 */

CreateRotatedSecret.prototype['ssh-username'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

CreateRotatedSecret.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

CreateRotatedSecret.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateRotatedSecret.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateRotatedSecret.prototype['uid-token'] = undefined;
/**
 * User Attribute
 * @member {String} user-attribute
 */

CreateRotatedSecret.prototype['user-attribute'] = undefined;
/**
 * User DN
 * @member {String} user-dn
 */

CreateRotatedSecret.prototype['user-dn'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

CreateRotatedSecret.prototype['username'] = undefined;
var _default = CreateRotatedSecret;
exports["default"] = _default;
