// akeyless/dist/model/CreateRole.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateRole model module.
 * @module model/CreateRole
 * @version 2.20.0
 */
var CreateRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateRole</code>.
   * @alias module:model/CreateRole
   * @param name {String} Role name
   */
  function CreateRole(name) {
    _classCallCheck(this, CreateRole);

    CreateRole.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateRole, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateRole} obj Optional instance to populate.
     * @return {module:model/CreateRole} The populated <code>CreateRole</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateRole();

        if (data.hasOwnProperty('analytics-access')) {
          obj['analytics-access'] = _ApiClient["default"].convertToType(data['analytics-access'], 'String');
        }

        if (data.hasOwnProperty('audit-access')) {
          obj['audit-access'] = _ApiClient["default"].convertToType(data['audit-access'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('gw-analytics-access')) {
          obj['gw-analytics-access'] = _ApiClient["default"].convertToType(data['gw-analytics-access'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('sra-reports-access')) {
          obj['sra-reports-access'] = _ApiClient["default"].convertToType(data['sra-reports-access'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateRole;
}();
/**
 * Allow this role to view analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} analytics-access
 */


CreateRole.prototype['analytics-access'] = undefined;
/**
 * Allow this role to view audit logs. Currently only 'none', 'own' and 'all' values are supported, allowing associated auth methods to view audit logs produced by the same auth methods.
 * @member {String} audit-access
 */

CreateRole.prototype['audit-access'] = undefined;
/**
 * Comment about the role
 * @member {String} comment
 */

CreateRole.prototype['comment'] = undefined;
/**
 * Allow this role to view gw analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} gw-analytics-access
 */

CreateRole.prototype['gw-analytics-access'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateRole.prototype['json'] = undefined;
/**
 * Role name
 * @member {String} name
 */

CreateRole.prototype['name'] = undefined;
/**
 * Allow this role to view SRA Clusters. Currently only 'none', 'own', 'all' values are supported.
 * @member {String} sra-reports-access
 */

CreateRole.prototype['sra-reports-access'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateRole.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateRole.prototype['uid-token'] = undefined;
var _default = CreateRole;
exports["default"] = _default;
