// akeyless/dist/model/CreateKeyOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateKeyOutput model module.
 * @module model/CreateKeyOutput
 * @version 2.20.0
 */
var CreateKeyOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateKeyOutput</code>.
   * @alias module:model/CreateKeyOutput
   */
  function CreateKeyOutput() {
    _classCallCheck(this, CreateKeyOutput);

    CreateKeyOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateKeyOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>CreateKeyOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateKeyOutput} obj Optional instance to populate.
     * @return {module:model/CreateKeyOutput} The populated <code>CreateKeyOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateKeyOutput();

        if (data.hasOwnProperty('display_id')) {
          obj['display_id'] = _ApiClient["default"].convertToType(data['display_id'], 'String');
        }

        if (data.hasOwnProperty('fragment_results')) {
          obj['fragment_results'] = _ApiClient["default"].convertToType(data['fragment_results'], ['Number']);
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return CreateKeyOutput;
}();
/**
 * @member {String} display_id
 */


CreateKeyOutput.prototype['display_id'] = undefined;
/**
 * @member {Array.<Number>} fragment_results
 */

CreateKeyOutput.prototype['fragment_results'] = undefined;
/**
 * @member {Number} item_id
 */

CreateKeyOutput.prototype['item_id'] = undefined;
var _default = CreateKeyOutput;
exports["default"] = _default;
