// akeyless/dist/model/CreateLdapTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateLdapTarget model module.
 * @module model/CreateLdapTarget
 * @version 2.20.0
 */
var CreateLdapTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateLdapTarget</code>.
   * @alias module:model/CreateLdapTarget
   * @param bindDn {String} Bind DN
   * @param bindDnPassword {String} Bind DN Password
   * @param ldapUrl {String} LDAP Server URL
   * @param name {String} Target name
   */
  function CreateLdapTarget(bindDn, bindDnPassword, ldapUrl, name) {
    _classCallCheck(this, CreateLdapTarget);

    CreateLdapTarget.initialize(this, bindDn, bindDnPassword, ldapUrl, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateLdapTarget, null, [{
    key: "initialize",
    value: function initialize(obj, bindDn, bindDnPassword, ldapUrl, name) {
      obj['bind-dn'] = bindDn;
      obj['bind-dn-password'] = bindDnPassword;
      obj['ldap-url'] = ldapUrl;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateLdapTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateLdapTarget} obj Optional instance to populate.
     * @return {module:model/CreateLdapTarget} The populated <code>CreateLdapTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateLdapTarget();

        if (data.hasOwnProperty('bind-dn')) {
          obj['bind-dn'] = _ApiClient["default"].convertToType(data['bind-dn'], 'String');
        }

        if (data.hasOwnProperty('bind-dn-password')) {
          obj['bind-dn-password'] = _ApiClient["default"].convertToType(data['bind-dn-password'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('ldap-ca-cert')) {
          obj['ldap-ca-cert'] = _ApiClient["default"].convertToType(data['ldap-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('ldap-url')) {
          obj['ldap-url'] = _ApiClient["default"].convertToType(data['ldap-url'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('server-type')) {
          obj['server-type'] = _ApiClient["default"].convertToType(data['server-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('token-expiration')) {
          obj['token-expiration'] = _ApiClient["default"].convertToType(data['token-expiration'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateLdapTarget;
}();
/**
 * Bind DN
 * @member {String} bind-dn
 */


CreateLdapTarget.prototype['bind-dn'] = undefined;
/**
 * Bind DN Password
 * @member {String} bind-dn-password
 */

CreateLdapTarget.prototype['bind-dn-password'] = undefined;
/**
 * Comment about the target
 * @member {String} comment
 */

CreateLdapTarget.prototype['comment'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateLdapTarget.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateLdapTarget.prototype['key'] = undefined;
/**
 * CA Certificate File Content
 * @member {String} ldap-ca-cert
 */

CreateLdapTarget.prototype['ldap-ca-cert'] = undefined;
/**
 * LDAP Server URL
 * @member {String} ldap-url
 */

CreateLdapTarget.prototype['ldap-url'] = undefined;
/**
 * Target name
 * @member {String} name
 */

CreateLdapTarget.prototype['name'] = undefined;
/**
 * Set Ldap server type, Options:[OpenLDAP, ActiveDirectory]. Default is OpenLDAP
 * @member {String} server-type
 * @default 'OpenLDAP'
 */

CreateLdapTarget.prototype['server-type'] = 'OpenLDAP';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateLdapTarget.prototype['token'] = undefined;
/**
 * Token expiration
 * @member {String} token-expiration
 */

CreateLdapTarget.prototype['token-expiration'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateLdapTarget.prototype['uid-token'] = undefined;
var _default = CreateLdapTarget;
exports["default"] = _default;
