// akeyless/dist/model/CreateAwsTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAwsTarget model module.
 * @module model/CreateAwsTarget
 * @version 2.4.0
 */
var CreateAwsTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAwsTarget</code>.
   * @alias module:model/CreateAwsTarget
   * @param name {String} Target name
   */
  function CreateAwsTarget(name) {
    _classCallCheck(this, CreateAwsTarget);

    CreateAwsTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAwsTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateAwsTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAwsTarget} obj Optional instance to populate.
     * @return {module:model/CreateAwsTarget} The populated <code>CreateAwsTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAwsTarget();

        if (data.hasOwnProperty('access-key')) {
          obj['access-key'] = _ApiClient["default"].convertToType(data['access-key'], 'String');
        }

        if (data.hasOwnProperty('access-key-id')) {
          obj['access-key-id'] = _ApiClient["default"].convertToType(data['access-key-id'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }

        if (data.hasOwnProperty('session-token')) {
          obj['session-token'] = _ApiClient["default"].convertToType(data['session-token'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAwsTarget;
}();
/**
 * @member {String} access-key
 */


CreateAwsTarget.prototype['access-key'] = undefined;
/**
 * @member {String} access-key-id
 */

CreateAwsTarget.prototype['access-key-id'] = undefined;
/**
 * Comment about the target
 * @member {String} comment
 */

CreateAwsTarget.prototype['comment'] = undefined;
/**
 * Target name
 * @member {String} name
 */

CreateAwsTarget.prototype['name'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

CreateAwsTarget.prototype['protection_key'] = undefined;
/**
 * @member {String} region
 */

CreateAwsTarget.prototype['region'] = undefined;
/**
 * @member {String} session-token
 */

CreateAwsTarget.prototype['session-token'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAwsTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAwsTarget.prototype['uid-token'] = undefined;
var _default = CreateAwsTarget;
exports["default"] = _default;
