// akeyless/dist/model/CreateDynamicSecret.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateDynamicSecret model module.
 * @module model/CreateDynamicSecret
 * @version 2.20.0
 */
var CreateDynamicSecret = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateDynamicSecret</code>.
   * @alias module:model/CreateDynamicSecret
   * @param name {String} Dynamic secret name
   */
  function CreateDynamicSecret(name) {
    _classCallCheck(this, CreateDynamicSecret);

    CreateDynamicSecret.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateDynamicSecret, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateDynamicSecret</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateDynamicSecret} obj Optional instance to populate.
     * @return {module:model/CreateDynamicSecret} The populated <code>CreateDynamicSecret</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateDynamicSecret();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateDynamicSecret;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


CreateDynamicSecret.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateDynamicSecret.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the dynamic secret values (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateDynamicSecret.prototype['key'] = undefined;
/**
 * Metadata about the dynamic secret
 * @member {String} metadata
 * @default 'None'
 */

CreateDynamicSecret.prototype['metadata'] = 'None';
/**
 * Dynamic secret name
 * @member {String} name
 */

CreateDynamicSecret.prototype['name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

CreateDynamicSecret.prototype['tags'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateDynamicSecret.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateDynamicSecret.prototype['uid-token'] = undefined;
var _default = CreateDynamicSecret;
exports["default"] = _default;
