// akeyless/dist/model/CreateAuthMethodK8S.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateAuthMethodK8S model module.
 * @module model/CreateAuthMethodK8S
 * @version 2.20.0
 */
var CreateAuthMethodK8S = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateAuthMethodK8S</code>.
   * createAuthMethodK8S is a command that creates a new auth method that will be able to authenticate using K8S.
   * @alias module:model/CreateAuthMethodK8S
   * @param name {String} Auth Method name
   */
  function CreateAuthMethodK8S(name) {
    _classCallCheck(this, CreateAuthMethodK8S);

    CreateAuthMethodK8S.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateAuthMethodK8S, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateAuthMethodK8S</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateAuthMethodK8S} obj Optional instance to populate.
     * @return {module:model/CreateAuthMethodK8S} The populated <code>CreateAuthMethodK8S</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateAuthMethodK8S();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-namespaces')) {
          obj['bound-namespaces'] = _ApiClient["default"].convertToType(data['bound-namespaces'], ['String']);
        }

        if (data.hasOwnProperty('bound-pod-names')) {
          obj['bound-pod-names'] = _ApiClient["default"].convertToType(data['bound-pod-names'], ['String']);
        }

        if (data.hasOwnProperty('bound-sa-names')) {
          obj['bound-sa-names'] = _ApiClient["default"].convertToType(data['bound-sa-names'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gen-key')) {
          obj['gen-key'] = _ApiClient["default"].convertToType(data['gen-key'], 'String');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('public-key')) {
          obj['public-key'] = _ApiClient["default"].convertToType(data['public-key'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateAuthMethodK8S;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


CreateAuthMethodK8S.prototype['access-expires'] = 0;
/**
 * The audience in the Kubernetes JWT that the access is restricted to
 * @member {String} audience
 */

CreateAuthMethodK8S.prototype['audience'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

CreateAuthMethodK8S.prototype['bound-ips'] = undefined;
/**
 * A list of namespaces that the access is restricted to
 * @member {Array.<String>} bound-namespaces
 */

CreateAuthMethodK8S.prototype['bound-namespaces'] = undefined;
/**
 * A list of pod names that the access is restricted to
 * @member {Array.<String>} bound-pod-names
 */

CreateAuthMethodK8S.prototype['bound-pod-names'] = undefined;
/**
 * A list of service account names that the access is restricted to
 * @member {Array.<String>} bound-sa-names
 */

CreateAuthMethodK8S.prototype['bound-sa-names'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

CreateAuthMethodK8S.prototype['force-sub-claims'] = undefined;
/**
 * Automatically generate key-pair for K8S configuration. If set to false, a public key needs to be provided
 * @member {String} gen-key
 * @default 'true'
 */

CreateAuthMethodK8S.prototype['gen-key'] = 'true';
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

CreateAuthMethodK8S.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateAuthMethodK8S.prototype['json'] = undefined;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 */

CreateAuthMethodK8S.prototype['jwt-ttl'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

CreateAuthMethodK8S.prototype['name'] = undefined;
/**
 * Base64-encoded or PEM formatted public key data for K8S authentication method is required [RSA2048]
 * @member {String} public-key
 */

CreateAuthMethodK8S.prototype['public-key'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateAuthMethodK8S.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateAuthMethodK8S.prototype['uid-token'] = undefined;
var _default = CreateAuthMethodK8S;
exports["default"] = _default;
