// akeyless/dist/model/CreateDockerhubTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateDockerhubTarget model module.
 * @module model/CreateDockerhubTarget
 * @version 2.20.0
 */
var CreateDockerhubTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CreateDockerhubTarget</code>.
   * @alias module:model/CreateDockerhubTarget
   * @param name {String} Target name
   */
  function CreateDockerhubTarget(name) {
    _classCallCheck(this, CreateDockerhubTarget);

    CreateDockerhubTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateDockerhubTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateDockerhubTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateDockerhubTarget} obj Optional instance to populate.
     * @return {module:model/CreateDockerhubTarget} The populated <code>CreateDockerhubTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateDockerhubTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('dockerhub-password')) {
          obj['dockerhub-password'] = _ApiClient["default"].convertToType(data['dockerhub-password'], 'String');
        }

        if (data.hasOwnProperty('dockerhub-username')) {
          obj['dockerhub-username'] = _ApiClient["default"].convertToType(data['dockerhub-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateDockerhubTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


CreateDockerhubTarget.prototype['comment'] = undefined;
/**
 * DockerhubPassword is either the user's password to manage the repository
 * @member {String} dockerhub-password
 */

CreateDockerhubTarget.prototype['dockerhub-password'] = undefined;
/**
 * DockerhubUsername is the name of the user in dockerhub
 * @member {String} dockerhub-username
 */

CreateDockerhubTarget.prototype['dockerhub-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

CreateDockerhubTarget.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

CreateDockerhubTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

CreateDockerhubTarget.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

CreateDockerhubTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

CreateDockerhubTarget.prototype['uid-token'] = undefined;
var _default = CreateDockerhubTarget;
exports["default"] = _default;
