// akeyless/dist/model/Decrypt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Decrypt model module.
 * @module model/Decrypt
 * @version 2.15.24
 */
var Decrypt = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Decrypt</code>.
   * decrypt is a command that decrypts ciphertext into plaintext by using an AES key.
   * @alias module:model/Decrypt
   * @param ciphertext {String} Ciphertext to be decrypted in base64 encoded format
   */
  function Decrypt(ciphertext) {
    _classCallCheck(this, Decrypt);

    Decrypt.initialize(this, ciphertext);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Decrypt, null, [{
    key: "initialize",
    value: function initialize(obj, ciphertext) {
      obj['ciphertext'] = ciphertext;
    }
    /**
     * Constructs a <code>Decrypt</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Decrypt} obj Optional instance to populate.
     * @return {module:model/Decrypt} The populated <code>Decrypt</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Decrypt();

        if (data.hasOwnProperty('ciphertext')) {
          obj['ciphertext'] = _ApiClient["default"].convertToType(data['ciphertext'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('encryption-context')) {
          obj['encryption-context'] = _ApiClient["default"].convertToType(data['encryption-context'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Decrypt;
}();
/**
 * Ciphertext to be decrypted in base64 encoded format
 * @member {String} ciphertext
 */


Decrypt.prototype['ciphertext'] = undefined;
/**
 * The display id of the key to use in the decryption process
 * @member {String} display-id
 */

Decrypt.prototype['display-id'] = undefined;
/**
 * The encryption context. If this was specified in the encrypt command, it must be specified here or the decryption operation will fail
 * @member {Object.<String, String>} encryption-context
 */

Decrypt.prototype['encryption-context'] = undefined;
/**
 * The name of the key to use in the decryption process
 * @member {String} key-name
 */

Decrypt.prototype['key-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

Decrypt.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

Decrypt.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

Decrypt.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

Decrypt.prototype['username'] = undefined;
var _default = Decrypt;
exports["default"] = _default;
