// akeyless/dist/model/DecryptPKCS1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DecryptPKCS1 model module.
 * @module model/DecryptPKCS1
 * @version 2.20.0
 */
var DecryptPKCS1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DecryptPKCS1</code>.
   * @alias module:model/DecryptPKCS1
   * @param ciphertext {String} Ciphertext to be decrypted in base64 encoded format
   * @param keyName {String} The name of the key to use in the decryption process
   */
  function DecryptPKCS1(ciphertext, keyName) {
    _classCallCheck(this, DecryptPKCS1);

    DecryptPKCS1.initialize(this, ciphertext, keyName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DecryptPKCS1, null, [{
    key: "initialize",
    value: function initialize(obj, ciphertext, keyName) {
      obj['ciphertext'] = ciphertext;
      obj['key-name'] = keyName;
    }
    /**
     * Constructs a <code>DecryptPKCS1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DecryptPKCS1} obj Optional instance to populate.
     * @return {module:model/DecryptPKCS1} The populated <code>DecryptPKCS1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DecryptPKCS1();

        if (data.hasOwnProperty('ciphertext')) {
          obj['ciphertext'] = _ApiClient["default"].convertToType(data['ciphertext'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DecryptPKCS1;
}();
/**
 * Ciphertext to be decrypted in base64 encoded format
 * @member {String} ciphertext
 */


DecryptPKCS1.prototype['ciphertext'] = undefined;
/**
 * The display id of the key to use in the decryption process
 * @member {String} display-id
 */

DecryptPKCS1.prototype['display-id'] = undefined;
/**
 * The item id of the key to use in the decryption process
 * @member {Number} item-id
 */

DecryptPKCS1.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

DecryptPKCS1.prototype['json'] = undefined;
/**
 * The name of the key to use in the decryption process
 * @member {String} key-name
 */

DecryptPKCS1.prototype['key-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DecryptPKCS1.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DecryptPKCS1.prototype['uid-token'] = undefined;
var _default = DecryptPKCS1;
exports["default"] = _default;
