// akeyless/dist/model/DecryptWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DecryptWithClassicKey model module.
 * @module model/DecryptWithClassicKey
 * @version 2.20.0
 */
var DecryptWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DecryptWithClassicKey</code>.
   * @alias module:model/DecryptWithClassicKey
   * @param ciphertext {String} Ciphertext to be decrypted in base64 encoded format
   * @param displayId {String} The name of the key to use in the encryption process
   * @param version {Number} classic key version
   */
  function DecryptWithClassicKey(ciphertext, displayId, version) {
    _classCallCheck(this, DecryptWithClassicKey);

    DecryptWithClassicKey.initialize(this, ciphertext, displayId, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DecryptWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, ciphertext, displayId, version) {
      obj['ciphertext'] = ciphertext;
      obj['display-id'] = displayId;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>DecryptWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DecryptWithClassicKey} obj Optional instance to populate.
     * @return {module:model/DecryptWithClassicKey} The populated <code>DecryptWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DecryptWithClassicKey();

        if (data.hasOwnProperty('ciphertext')) {
          obj['ciphertext'] = _ApiClient["default"].convertToType(data['ciphertext'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return DecryptWithClassicKey;
}();
/**
 * Ciphertext to be decrypted in base64 encoded format
 * @member {String} ciphertext
 */


DecryptWithClassicKey.prototype['ciphertext'] = undefined;
/**
 * The name of the key to use in the encryption process
 * @member {String} display-id
 */

DecryptWithClassicKey.prototype['display-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

DecryptWithClassicKey.prototype['json'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DecryptWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DecryptWithClassicKey.prototype['uid-token'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

DecryptWithClassicKey.prototype['version'] = undefined;
var _default = DecryptWithClassicKey;
exports["default"] = _default;
