// akeyless/dist/model/DeleteTargets.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DeleteTargets model module.
 * @module model/DeleteTargets
 * @version 2.15.24
 */
var DeleteTargets = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteTargets</code>.
   * @alias module:model/DeleteTargets
   * @param path {String} Path to delete the targets from
   */
  function DeleteTargets(path) {
    _classCallCheck(this, DeleteTargets);

    DeleteTargets.initialize(this, path);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DeleteTargets, null, [{
    key: "initialize",
    value: function initialize(obj, path) {
      obj['path'] = path;
    }
    /**
     * Constructs a <code>DeleteTargets</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteTargets} obj Optional instance to populate.
     * @return {module:model/DeleteTargets} The populated <code>DeleteTargets</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteTargets();

        if (data.hasOwnProperty('force-deletion')) {
          obj['force-deletion'] = _ApiClient["default"].convertToType(data['force-deletion'], 'Boolean');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DeleteTargets;
}();
/**
 * Enforce deletion
 * @member {Boolean} force-deletion
 * @default false
 */


DeleteTargets.prototype['force-deletion'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

DeleteTargets.prototype['password'] = undefined;
/**
 * Path to delete the targets from
 * @member {String} path
 */

DeleteTargets.prototype['path'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DeleteTargets.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DeleteTargets.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

DeleteTargets.prototype['username'] = undefined;
var _default = DeleteTargets;
exports["default"] = _default;
