// akeyless/dist/model/DeleteTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DeleteTarget model module.
 * @module model/DeleteTarget
 * @version 2.20.0
 */
var DeleteTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteTarget</code>.
   * @alias module:model/DeleteTarget
   * @param name {String} Target name
   */
  function DeleteTarget(name) {
    _classCallCheck(this, DeleteTarget);

    DeleteTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DeleteTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>DeleteTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteTarget} obj Optional instance to populate.
     * @return {module:model/DeleteTarget} The populated <code>DeleteTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteTarget();

        if (data.hasOwnProperty('force-deletion')) {
          obj['force-deletion'] = _ApiClient["default"].convertToType(data['force-deletion'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('target-version')) {
          obj['target-version'] = _ApiClient["default"].convertToType(data['target-version'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DeleteTarget;
}();
/**
 * Enforce deletion
 * @member {Boolean} force-deletion
 * @default false
 */


DeleteTarget.prototype['force-deletion'] = false;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

DeleteTarget.prototype['json'] = undefined;
/**
 * Target name
 * @member {String} name
 */

DeleteTarget.prototype['name'] = undefined;
/**
 * Target version
 * @member {Number} target-version
 */

DeleteTarget.prototype['target-version'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DeleteTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DeleteTarget.prototype['uid-token'] = undefined;
var _default = DeleteTarget;
exports["default"] = _default;
