// akeyless/dist/model/DeleteRoleRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DeleteRoleRule model module.
 * @module model/DeleteRoleRule
 * @version 2.15.24
 */
var DeleteRoleRule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteRoleRule</code>.
   * @alias module:model/DeleteRoleRule
   * @param path {String} The path the rule refers to
   * @param roleName {String} The role name to be updated
   */
  function DeleteRoleRule(path, roleName) {
    _classCallCheck(this, DeleteRoleRule);

    DeleteRoleRule.initialize(this, path, roleName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DeleteRoleRule, null, [{
    key: "initialize",
    value: function initialize(obj, path, roleName) {
      obj['path'] = path;
      obj['role-name'] = roleName;
    }
    /**
     * Constructs a <code>DeleteRoleRule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteRoleRule} obj Optional instance to populate.
     * @return {module:model/DeleteRoleRule} The populated <code>DeleteRoleRule</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteRoleRule();

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('role-name')) {
          obj['role-name'] = _ApiClient["default"].convertToType(data['role-name'], 'String');
        }

        if (data.hasOwnProperty('rule-type')) {
          obj['rule-type'] = _ApiClient["default"].convertToType(data['rule-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DeleteRoleRule;
}();
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */


DeleteRoleRule.prototype['password'] = undefined;
/**
 * The path the rule refers to
 * @member {String} path
 */

DeleteRoleRule.prototype['path'] = undefined;
/**
 * The role name to be updated
 * @member {String} role-name
 */

DeleteRoleRule.prototype['role-name'] = undefined;
/**
 * item-rule, role-rule, auth-method-rule, search-rule or reports-rule
 * @member {String} rule-type
 * @default 'item-rule'
 */

DeleteRoleRule.prototype['rule-type'] = 'item-rule';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DeleteRoleRule.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DeleteRoleRule.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

DeleteRoleRule.prototype['username'] = undefined;
var _default = DeleteRoleRule;
exports["default"] = _default;
