// akeyless/dist/model/DeleteRoleAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DeleteRoleAssociation model module.
 * @module model/DeleteRoleAssociation
 * @version 2.15.24
 */
var DeleteRoleAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteRoleAssociation</code>.
   * deleteRoleAssociation is a command that deletes an association between role and auth method.
   * @alias module:model/DeleteRoleAssociation
   * @param assocId {String} The association id to be deleted
   */
  function DeleteRoleAssociation(assocId) {
    _classCallCheck(this, DeleteRoleAssociation);

    DeleteRoleAssociation.initialize(this, assocId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DeleteRoleAssociation, null, [{
    key: "initialize",
    value: function initialize(obj, assocId) {
      obj['assoc-id'] = assocId;
    }
    /**
     * Constructs a <code>DeleteRoleAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteRoleAssociation} obj Optional instance to populate.
     * @return {module:model/DeleteRoleAssociation} The populated <code>DeleteRoleAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteRoleAssociation();

        if (data.hasOwnProperty('assoc-id')) {
          obj['assoc-id'] = _ApiClient["default"].convertToType(data['assoc-id'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DeleteRoleAssociation;
}();
/**
 * The association id to be deleted
 * @member {String} assoc-id
 */


DeleteRoleAssociation.prototype['assoc-id'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

DeleteRoleAssociation.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DeleteRoleAssociation.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DeleteRoleAssociation.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

DeleteRoleAssociation.prototype['username'] = undefined;
var _default = DeleteRoleAssociation;
exports["default"] = _default;
