// akeyless/dist/model/DescribePermissions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DescribePermissions model module.
 * @module model/DescribePermissions
 * @version 2.20.0
 */
var DescribePermissions = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DescribePermissions</code>.
   * describePermissions is a command that shows which permissions your have to a particular path.
   * @alias module:model/DescribePermissions
   * @param path {String} Path to an object
   * @param type {String} Type of object (item, am, role, target)
   */
  function DescribePermissions(path, type) {
    _classCallCheck(this, DescribePermissions);

    DescribePermissions.initialize(this, path, type);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DescribePermissions, null, [{
    key: "initialize",
    value: function initialize(obj, path, type) {
      obj['path'] = path;
      obj['type'] = type;
    }
    /**
     * Constructs a <code>DescribePermissions</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DescribePermissions} obj Optional instance to populate.
     * @return {module:model/DescribePermissions} The populated <code>DescribePermissions</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DescribePermissions();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DescribePermissions;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 */


DescribePermissions.prototype['json'] = undefined;
/**
 * Path to an object
 * @member {String} path
 */

DescribePermissions.prototype['path'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

DescribePermissions.prototype['token'] = undefined;
/**
 * Type of object (item, am, role, target)
 * @member {String} type
 */

DescribePermissions.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

DescribePermissions.prototype['uid-token'] = undefined;
var _default = DescribePermissions;
exports["default"] = _default;
