// akeyless/dist/model/Detokenize.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Detokenize model module.
 * @module model/Detokenize
 * @version 2.20.0
 */
var Detokenize = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Detokenize</code>.
   * detokenize is a command that decrypts text with a tokenizer
   * @alias module:model/Detokenize
   * @param ciphertext {String} Data to be decrypted
   * @param tokenizerName {String} The name of the tokenizer to use in the decryption process
   */
  function Detokenize(ciphertext, tokenizerName) {
    _classCallCheck(this, Detokenize);

    Detokenize.initialize(this, ciphertext, tokenizerName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Detokenize, null, [{
    key: "initialize",
    value: function initialize(obj, ciphertext, tokenizerName) {
      obj['ciphertext'] = ciphertext;
      obj['tokenizer-name'] = tokenizerName;
    }
    /**
     * Constructs a <code>Detokenize</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Detokenize} obj Optional instance to populate.
     * @return {module:model/Detokenize} The populated <code>Detokenize</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Detokenize();

        if (data.hasOwnProperty('ciphertext')) {
          obj['ciphertext'] = _ApiClient["default"].convertToType(data['ciphertext'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('tokenizer-name')) {
          obj['tokenizer-name'] = _ApiClient["default"].convertToType(data['tokenizer-name'], 'String');
        }

        if (data.hasOwnProperty('tweak')) {
          obj['tweak'] = _ApiClient["default"].convertToType(data['tweak'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Detokenize;
}();
/**
 * Data to be decrypted
 * @member {String} ciphertext
 */


Detokenize.prototype['ciphertext'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

Detokenize.prototype['json'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

Detokenize.prototype['token'] = undefined;
/**
 * The name of the tokenizer to use in the decryption process
 * @member {String} tokenizer-name
 */

Detokenize.prototype['tokenizer-name'] = undefined;
/**
 * Base64 encoded tweak for vaultless encryption
 * @member {String} tweak
 */

Detokenize.prototype['tweak'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

Detokenize.prototype['uid-token'] = undefined;
var _default = Detokenize;
exports["default"] = _default;
