// akeyless/dist/model/DynamicSecretProducerInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The DynamicSecretProducerInfo model module.
 * @module model/DynamicSecretProducerInfo
 * @version 2.15.24
 */
var DynamicSecretProducerInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DynamicSecretProducerInfo</code>.
   * DynamicSecretProducerInfo The dynamic secret producer info This parameter relevant and required only in case of create update dynamic secret.
   * @alias module:model/DynamicSecretProducerInfo
   */
  function DynamicSecretProducerInfo() {
    _classCallCheck(this, DynamicSecretProducerInfo);

    DynamicSecretProducerInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(DynamicSecretProducerInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>DynamicSecretProducerInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DynamicSecretProducerInfo} obj Optional instance to populate.
     * @return {module:model/DynamicSecretProducerInfo} The populated <code>DynamicSecretProducerInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DynamicSecretProducerInfo();

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('producer_last_keep_alive')) {
          obj['producer_last_keep_alive'] = _ApiClient["default"].convertToType(data['producer_last_keep_alive'], 'String');
        }

        if (data.hasOwnProperty('producer_metadata')) {
          obj['producer_metadata'] = _ApiClient["default"].convertToType(data['producer_metadata'], 'String');
        }

        if (data.hasOwnProperty('producer_status')) {
          obj['producer_status'] = _ApiClient["default"].convertToType(data['producer_status'], 'String');
        }

        if (data.hasOwnProperty('producer_type')) {
          obj['producer_type'] = _ApiClient["default"].convertToType(data['producer_type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return DynamicSecretProducerInfo;
}();
/**
 * @member {Number} gw_cluster_id
 */


DynamicSecretProducerInfo.prototype['gw_cluster_id'] = undefined;
/**
 * @member {String} producer_last_keep_alive
 */

DynamicSecretProducerInfo.prototype['producer_last_keep_alive'] = undefined;
/**
 * @member {String} producer_metadata
 */

DynamicSecretProducerInfo.prototype['producer_metadata'] = undefined;
/**
 * RotationStatus defines types of rotation Status
 * @member {String} producer_status
 */

DynamicSecretProducerInfo.prototype['producer_status'] = undefined;
/**
 * @member {String} producer_type
 */

DynamicSecretProducerInfo.prototype['producer_type'] = undefined;
var _default = DynamicSecretProducerInfo;
exports["default"] = _default;
