// akeyless/dist/model/ElasticsearchLogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ElasticsearchLogForwardingConfig model module.
 * @module model/ElasticsearchLogForwardingConfig
 * @version 2.20.0
 */
var ElasticsearchLogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ElasticsearchLogForwardingConfig</code>.
   * @alias module:model/ElasticsearchLogForwardingConfig
   */
  function ElasticsearchLogForwardingConfig() {
    _classCallCheck(this, ElasticsearchLogForwardingConfig);

    ElasticsearchLogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ElasticsearchLogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ElasticsearchLogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ElasticsearchLogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/ElasticsearchLogForwardingConfig} The populated <code>ElasticsearchLogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ElasticsearchLogForwardingConfig();

        if (data.hasOwnProperty('elasticsearch_api_key')) {
          obj['elasticsearch_api_key'] = _ApiClient["default"].convertToType(data['elasticsearch_api_key'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_auth_type')) {
          obj['elasticsearch_auth_type'] = _ApiClient["default"].convertToType(data['elasticsearch_auth_type'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_cloud_id')) {
          obj['elasticsearch_cloud_id'] = _ApiClient["default"].convertToType(data['elasticsearch_cloud_id'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_index')) {
          obj['elasticsearch_index'] = _ApiClient["default"].convertToType(data['elasticsearch_index'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_nodes')) {
          obj['elasticsearch_nodes'] = _ApiClient["default"].convertToType(data['elasticsearch_nodes'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_password')) {
          obj['elasticsearch_password'] = _ApiClient["default"].convertToType(data['elasticsearch_password'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_server_type')) {
          obj['elasticsearch_server_type'] = _ApiClient["default"].convertToType(data['elasticsearch_server_type'], 'String');
        }

        if (data.hasOwnProperty('elasticsearch_user_name')) {
          obj['elasticsearch_user_name'] = _ApiClient["default"].convertToType(data['elasticsearch_user_name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ElasticsearchLogForwardingConfig;
}();
/**
 * @member {String} elasticsearch_api_key
 */


ElasticsearchLogForwardingConfig.prototype['elasticsearch_api_key'] = undefined;
/**
 * @member {String} elasticsearch_auth_type
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_auth_type'] = undefined;
/**
 * @member {String} elasticsearch_cloud_id
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_cloud_id'] = undefined;
/**
 * @member {String} elasticsearch_index
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_index'] = undefined;
/**
 * @member {String} elasticsearch_nodes
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_nodes'] = undefined;
/**
 * @member {String} elasticsearch_password
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_password'] = undefined;
/**
 * @member {String} elasticsearch_server_type
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_server_type'] = undefined;
/**
 * @member {String} elasticsearch_user_name
 */

ElasticsearchLogForwardingConfig.prototype['elasticsearch_user_name'] = undefined;
var _default = ElasticsearchLogForwardingConfig;
exports["default"] = _default;
