// akeyless/dist/model/EmailTokenizerInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EmailTokenizerInfo model module.
 * @module model/EmailTokenizerInfo
 * @version 2.20.0
 */
var EmailTokenizerInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EmailTokenizerInfo</code>.
   * EmailTokenizerInfo represents a tokenizer that specifically tokenizes emails
   * @alias module:model/EmailTokenizerInfo
   */
  function EmailTokenizerInfo() {
    _classCallCheck(this, EmailTokenizerInfo);

    EmailTokenizerInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EmailTokenizerInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>EmailTokenizerInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EmailTokenizerInfo} obj Optional instance to populate.
     * @return {module:model/EmailTokenizerInfo} The populated <code>EmailTokenizerInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailTokenizerInfo();

        if (data.hasOwnProperty('domain_suffix_length')) {
          obj['domain_suffix_length'] = _ApiClient["default"].convertToType(data['domain_suffix_length'], 'Number');
        }

        if (data.hasOwnProperty('fixed_domain_suffix')) {
          obj['fixed_domain_suffix'] = _ApiClient["default"].convertToType(data['fixed_domain_suffix'], 'String');
        }

        if (data.hasOwnProperty('keep_prefix_length')) {
          obj['keep_prefix_length'] = _ApiClient["default"].convertToType(data['keep_prefix_length'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return EmailTokenizerInfo;
}();
/**
 * What length of a random domain suffix to generate used only if FixedDomainSuffix is empty
 * @member {Number} domain_suffix_length
 */


EmailTokenizerInfo.prototype['domain_suffix_length'] = undefined;
/**
 * if FixedDomainSuffix isn't empty, it will be appended to the output
 * @member {String} fixed_domain_suffix
 */

EmailTokenizerInfo.prototype['fixed_domain_suffix'] = undefined;
/**
 * How many letters of the plaintext to keep in the output
 * @member {Number} keep_prefix_length
 */

EmailTokenizerInfo.prototype['keep_prefix_length'] = undefined;
var _default = EmailTokenizerInfo;
exports["default"] = _default;
