// akeyless/dist/model/EmailPassAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EmailPassAccessRules model module.
 * @module model/EmailPassAccessRules
 * @version 2.20.0
 */
var EmailPassAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EmailPassAccessRules</code>.
   * @alias module:model/EmailPassAccessRules
   */
  function EmailPassAccessRules() {
    _classCallCheck(this, EmailPassAccessRules);

    EmailPassAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EmailPassAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>EmailPassAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EmailPassAccessRules} obj Optional instance to populate.
     * @return {module:model/EmailPassAccessRules} The populated <code>EmailPassAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EmailPassAccessRules();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('hash_pass')) {
          obj['hash_pass'] = _ApiClient["default"].convertToType(data['hash_pass'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EmailPassAccessRules;
}();
/**
 * @member {String} alg
 */


EmailPassAccessRules.prototype['alg'] = undefined;
/**
 * The Email value
 * @member {String} email
 */

EmailPassAccessRules.prototype['email'] = undefined;
/**
 * The password value
 * @member {String} hash_pass
 */

EmailPassAccessRules.prototype['hash_pass'] = undefined;
var _default = EmailPassAccessRules;
exports["default"] = _default;
