// akeyless/dist/model/EncryptPKCS1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EncryptPKCS1 model module.
 * @module model/EncryptPKCS1
 * @version 2.15.24
 */
var EncryptPKCS1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EncryptPKCS1</code>.
   * @alias module:model/EncryptPKCS1
   * @param keyName {String} The name of the RSA key to use in the encryption process
   * @param plaintext {String} Data to be encrypted
   */
  function EncryptPKCS1(keyName, plaintext) {
    _classCallCheck(this, EncryptPKCS1);

    EncryptPKCS1.initialize(this, keyName, plaintext);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EncryptPKCS1, null, [{
    key: "initialize",
    value: function initialize(obj, keyName, plaintext) {
      obj['key-name'] = keyName;
      obj['plaintext'] = plaintext;
    }
    /**
     * Constructs a <code>EncryptPKCS1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EncryptPKCS1} obj Optional instance to populate.
     * @return {module:model/EncryptPKCS1} The populated <code>EncryptPKCS1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EncryptPKCS1();

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('plaintext')) {
          obj['plaintext'] = _ApiClient["default"].convertToType(data['plaintext'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EncryptPKCS1;
}();
/**
 * The name of the RSA key to use in the encryption process
 * @member {String} key-name
 */


EncryptPKCS1.prototype['key-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

EncryptPKCS1.prototype['password'] = undefined;
/**
 * Data to be encrypted
 * @member {String} plaintext
 */

EncryptPKCS1.prototype['plaintext'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EncryptPKCS1.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EncryptPKCS1.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

EncryptPKCS1.prototype['username'] = undefined;
var _default = EncryptPKCS1;
exports["default"] = _default;
