// akeyless/dist/model/Encrypt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Encrypt model module.
 * @module model/Encrypt
 * @version 2.15.24
 */
var Encrypt = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Encrypt</code>.
   * @alias module:model/Encrypt
   * @param plaintext {String} Data to be encrypted
   */
  function Encrypt(plaintext) {
    _classCallCheck(this, Encrypt);

    Encrypt.initialize(this, plaintext);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Encrypt, null, [{
    key: "initialize",
    value: function initialize(obj, plaintext) {
      obj['plaintext'] = plaintext;
    }
    /**
     * Constructs a <code>Encrypt</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Encrypt} obj Optional instance to populate.
     * @return {module:model/Encrypt} The populated <code>Encrypt</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Encrypt();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('encryption-context')) {
          obj['encryption-context'] = _ApiClient["default"].convertToType(data['encryption-context'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('plaintext')) {
          obj['plaintext'] = _ApiClient["default"].convertToType(data['plaintext'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Encrypt;
}();
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */


Encrypt.prototype['display-id'] = undefined;
/**
 * name-value pair that specifies the encryption context to be used for authenticated encryption. If used here, the same value must be supplied to the decrypt command or decryption will fail
 * @member {Object.<String, String>} encryption-context
 */

Encrypt.prototype['encryption-context'] = undefined;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

Encrypt.prototype['key-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

Encrypt.prototype['password'] = undefined;
/**
 * Data to be encrypted
 * @member {String} plaintext
 */

Encrypt.prototype['plaintext'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

Encrypt.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

Encrypt.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

Encrypt.prototype['username'] = undefined;
var _default = Encrypt;
exports["default"] = _default;
