// akeyless/dist/model/EncryptWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EncryptWithClassicKey model module.
 * @module model/EncryptWithClassicKey
 * @version 2.15.24
 */
var EncryptWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EncryptWithClassicKey</code>.
   * @alias module:model/EncryptWithClassicKey
   * @param displayId {String} The name of the key to use in the encryption process
   * @param plaintext {String} Data to be encrypted
   * @param version {Number} classic key version
   */
  function EncryptWithClassicKey(displayId, plaintext, version) {
    _classCallCheck(this, EncryptWithClassicKey);

    EncryptWithClassicKey.initialize(this, displayId, plaintext, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EncryptWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, displayId, plaintext, version) {
      obj['display-id'] = displayId;
      obj['plaintext'] = plaintext;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>EncryptWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EncryptWithClassicKey} obj Optional instance to populate.
     * @return {module:model/EncryptWithClassicKey} The populated <code>EncryptWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EncryptWithClassicKey();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('plaintext')) {
          obj['plaintext'] = _ApiClient["default"].convertToType(data['plaintext'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return EncryptWithClassicKey;
}();
/**
 * The name of the key to use in the encryption process
 * @member {String} display-id
 */


EncryptWithClassicKey.prototype['display-id'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

EncryptWithClassicKey.prototype['password'] = undefined;
/**
 * Data to be encrypted
 * @member {String} plaintext
 */

EncryptWithClassicKey.prototype['plaintext'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EncryptWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EncryptWithClassicKey.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

EncryptWithClassicKey.prototype['username'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

EncryptWithClassicKey.prototype['version'] = undefined;
var _default = EncryptWithClassicKey;
exports["default"] = _default;
