// akeyless/dist/model/EncryptFile.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The EncryptFile model module.
 * @module model/EncryptFile
 * @version 2.20.0
 */
var EncryptFile = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>EncryptFile</code>.
   * @alias module:model/EncryptFile
   * @param _in {String} Path to the file to be encrypted. If not provided, the content will be taken from stdin
   * @param keyName {String} The name of the key to use in the encryption process
   */
  function EncryptFile(_in, keyName) {
    _classCallCheck(this, EncryptFile);

    EncryptFile.initialize(this, _in, keyName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(EncryptFile, null, [{
    key: "initialize",
    value: function initialize(obj, _in, keyName) {
      obj['in'] = _in;
      obj['key-name'] = keyName;
    }
    /**
     * Constructs a <code>EncryptFile</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/EncryptFile} obj Optional instance to populate.
     * @return {module:model/EncryptFile} The populated <code>EncryptFile</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new EncryptFile();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('encryption-context')) {
          obj['encryption-context'] = _ApiClient["default"].convertToType(data['encryption-context'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('in')) {
          obj['in'] = _ApiClient["default"].convertToType(data['in'], 'String');
        }

        if (data.hasOwnProperty('item-id')) {
          obj['item-id'] = _ApiClient["default"].convertToType(data['item-id'], 'Number');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('out')) {
          obj['out'] = _ApiClient["default"].convertToType(data['out'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return EncryptFile;
}();
/**
 * The display id of the key to use in the encryption process
 * @member {String} display-id
 */


EncryptFile.prototype['display-id'] = undefined;
/**
 * name-value pair that specifies the encryption context to be used for authenticated encryption. If used here, the same value must be supplied to the decrypt command or decryption will fail
 * @member {Object.<String, String>} encryption-context
 */

EncryptFile.prototype['encryption-context'] = undefined;
/**
 * Path to the file to be encrypted. If not provided, the content will be taken from stdin
 * @member {String} in
 */

EncryptFile.prototype['in'] = undefined;
/**
 * The item id of the key to use in the encryption process
 * @member {Number} item-id
 */

EncryptFile.prototype['item-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

EncryptFile.prototype['json'] = undefined;
/**
 * The name of the key to use in the encryption process
 * @member {String} key-name
 */

EncryptFile.prototype['key-name'] = undefined;
/**
 * Path to the output file. If not provided, the output will be sent to stdout
 * @member {String} out
 */

EncryptFile.prototype['out'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

EncryptFile.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

EncryptFile.prototype['uid-token'] = undefined;
var _default = EncryptFile;
exports["default"] = _default;
