// akeyless/dist/model/GatewayMessageQueueInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayMessageQueueInfo model module.
 * @module model/GatewayMessageQueueInfo
 * @version 2.15.24
 */
var GatewayMessageQueueInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayMessageQueueInfo</code>.
   * @alias module:model/GatewayMessageQueueInfo
   */
  function GatewayMessageQueueInfo() {
    _classCallCheck(this, GatewayMessageQueueInfo);

    GatewayMessageQueueInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayMessageQueueInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayMessageQueueInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayMessageQueueInfo} obj Optional instance to populate.
     * @return {module:model/GatewayMessageQueueInfo} The populated <code>GatewayMessageQueueInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayMessageQueueInfo();

        if (data.hasOwnProperty('mq_type')) {
          obj['mq_type'] = _ApiClient["default"].convertToType(data['mq_type'], 'String');
        }

        if (data.hasOwnProperty('queue_name')) {
          obj['queue_name'] = _ApiClient["default"].convertToType(data['queue_name'], 'String');
        }

        if (data.hasOwnProperty('queue_url')) {
          obj['queue_url'] = _ApiClient["default"].convertToType(data['queue_url'], 'String');
        }

        if (data.hasOwnProperty('use_new_queue')) {
          obj['use_new_queue'] = _ApiClient["default"].convertToType(data['use_new_queue'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return GatewayMessageQueueInfo;
}();
/**
 * @member {String} mq_type
 */


GatewayMessageQueueInfo.prototype['mq_type'] = undefined;
/**
 * @member {String} queue_name
 */

GatewayMessageQueueInfo.prototype['queue_name'] = undefined;
/**
 * @member {String} queue_url
 */

GatewayMessageQueueInfo.prototype['queue_url'] = undefined;
/**
 * @member {Boolean} use_new_queue
 */

GatewayMessageQueueInfo.prototype['use_new_queue'] = undefined;
var _default = GatewayMessageQueueInfo;
exports["default"] = _default;
