// akeyless/dist/model/GatewayUpdateProducerSnowflake.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerSnowflake model module.
 * @module model/GatewayUpdateProducerSnowflake
 * @version 2.20.0
 */
var GatewayUpdateProducerSnowflake = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerSnowflake</code>.
   * gatewayUpdateProducerSnowflakeCmd is a command that updates a Snowflake producer
   * @alias module:model/GatewayUpdateProducerSnowflake
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerSnowflake(name) {
    _classCallCheck(this, GatewayUpdateProducerSnowflake);

    GatewayUpdateProducerSnowflake.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerSnowflake, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerSnowflake</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerSnowflake} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerSnowflake} The populated <code>GatewayUpdateProducerSnowflake</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerSnowflake();

        if (data.hasOwnProperty('account')) {
          obj['account'] = _ApiClient["default"].convertToType(data['account'], 'String');
        }

        if (data.hasOwnProperty('account-password')) {
          obj['account-password'] = _ApiClient["default"].convertToType(data['account-password'], 'String');
        }

        if (data.hasOwnProperty('account-username')) {
          obj['account-username'] = _ApiClient["default"].convertToType(data['account-username'], 'String');
        }

        if (data.hasOwnProperty('db-name')) {
          obj['db-name'] = _ApiClient["default"].convertToType(data['db-name'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('role')) {
          obj['role'] = _ApiClient["default"].convertToType(data['role'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('warehouse')) {
          obj['warehouse'] = _ApiClient["default"].convertToType(data['warehouse'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerSnowflake;
}();
/**
 * Account name
 * @member {String} account
 */


GatewayUpdateProducerSnowflake.prototype['account'] = undefined;
/**
 * Database Password
 * @member {String} account-password
 */

GatewayUpdateProducerSnowflake.prototype['account-password'] = undefined;
/**
 * Database Username
 * @member {String} account-username
 */

GatewayUpdateProducerSnowflake.prototype['account-username'] = undefined;
/**
 * Database name
 * @member {String} db-name
 */

GatewayUpdateProducerSnowflake.prototype['db-name'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

GatewayUpdateProducerSnowflake.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerSnowflake.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerSnowflake.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerSnowflake.prototype['new-name'] = undefined;
/**
 * User role
 * @member {String} role
 */

GatewayUpdateProducerSnowflake.prototype['role'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerSnowflake.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerSnowflake.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerSnowflake.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerSnowflake.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '24h'
 */

GatewayUpdateProducerSnowflake.prototype['user-ttl'] = '24h';
/**
 * Warehouse name
 * @member {String} warehouse
 */

GatewayUpdateProducerSnowflake.prototype['warehouse'] = undefined;
var _default = GatewayUpdateProducerSnowflake;
exports["default"] = _default;
