// akeyless/dist/model/GatewayUpdateProducerArtifactory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerArtifactory model module.
 * @module model/GatewayUpdateProducerArtifactory
 * @version 2.15.24
 */
var GatewayUpdateProducerArtifactory = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerArtifactory</code>.
   * gatewayUpdateProducerArtifactory is a command that updates artifactory producer
   * @alias module:model/GatewayUpdateProducerArtifactory
   * @param artifactoryTokenAudience {String} Token Audience
   * @param artifactoryTokenScope {String} Token Scope
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerArtifactory(artifactoryTokenAudience, artifactoryTokenScope, name) {
    _classCallCheck(this, GatewayUpdateProducerArtifactory);

    GatewayUpdateProducerArtifactory.initialize(this, artifactoryTokenAudience, artifactoryTokenScope, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerArtifactory, null, [{
    key: "initialize",
    value: function initialize(obj, artifactoryTokenAudience, artifactoryTokenScope, name) {
      obj['artifactory-token-audience'] = artifactoryTokenAudience;
      obj['artifactory-token-scope'] = artifactoryTokenScope;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerArtifactory</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerArtifactory} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerArtifactory} The populated <code>GatewayUpdateProducerArtifactory</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerArtifactory();

        if (data.hasOwnProperty('artifactory-admin-name')) {
          obj['artifactory-admin-name'] = _ApiClient["default"].convertToType(data['artifactory-admin-name'], 'String');
        }

        if (data.hasOwnProperty('artifactory-admin-pwd')) {
          obj['artifactory-admin-pwd'] = _ApiClient["default"].convertToType(data['artifactory-admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('artifactory-token-audience')) {
          obj['artifactory-token-audience'] = _ApiClient["default"].convertToType(data['artifactory-token-audience'], 'String');
        }

        if (data.hasOwnProperty('artifactory-token-scope')) {
          obj['artifactory-token-scope'] = _ApiClient["default"].convertToType(data['artifactory-token-scope'], 'String');
        }

        if (data.hasOwnProperty('base-url')) {
          obj['base-url'] = _ApiClient["default"].convertToType(data['base-url'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerArtifactory;
}();
/**
 * Artifactory Admin Name
 * @member {String} artifactory-admin-name
 */


GatewayUpdateProducerArtifactory.prototype['artifactory-admin-name'] = undefined;
/**
 * Artifactory Admin password
 * @member {String} artifactory-admin-pwd
 */

GatewayUpdateProducerArtifactory.prototype['artifactory-admin-pwd'] = undefined;
/**
 * Token Audience
 * @member {String} artifactory-token-audience
 */

GatewayUpdateProducerArtifactory.prototype['artifactory-token-audience'] = undefined;
/**
 * Token Scope
 * @member {String} artifactory-token-scope
 */

GatewayUpdateProducerArtifactory.prototype['artifactory-token-scope'] = undefined;
/**
 * Base URL
 * @member {String} base-url
 */

GatewayUpdateProducerArtifactory.prototype['base-url'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerArtifactory.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerArtifactory.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayUpdateProducerArtifactory.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerArtifactory.prototype['producer-encryption-key-name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerArtifactory.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerArtifactory.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerArtifactory.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerArtifactory.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerArtifactory.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayUpdateProducerArtifactory.prototype['username'] = undefined;
var _default = GatewayUpdateProducerArtifactory;
exports["default"] = _default;
