// akeyless/dist/model/GatewayUpdateProducerRedshift.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerRedshift model module.
 * @module model/GatewayUpdateProducerRedshift
 * @version 2.20.0
 */
var GatewayUpdateProducerRedshift = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerRedshift</code>.
   * gatewayUpdateProducerRedshift is a command that updates redshift producer
   * @alias module:model/GatewayUpdateProducerRedshift
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerRedshift(name) {
    _classCallCheck(this, GatewayUpdateProducerRedshift);

    GatewayUpdateProducerRedshift.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerRedshift, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerRedshift</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerRedshift} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerRedshift} The populated <code>GatewayUpdateProducerRedshift</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerRedshift();

        if (data.hasOwnProperty('creation-statements')) {
          obj['creation-statements'] = _ApiClient["default"].convertToType(data['creation-statements'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key')) {
          obj['producer-encryption-key'] = _ApiClient["default"].convertToType(data['producer-encryption-key'], 'String');
        }

        if (data.hasOwnProperty('redshift-db-name')) {
          obj['redshift-db-name'] = _ApiClient["default"].convertToType(data['redshift-db-name'], 'String');
        }

        if (data.hasOwnProperty('redshift-host')) {
          obj['redshift-host'] = _ApiClient["default"].convertToType(data['redshift-host'], 'String');
        }

        if (data.hasOwnProperty('redshift-password')) {
          obj['redshift-password'] = _ApiClient["default"].convertToType(data['redshift-password'], 'String');
        }

        if (data.hasOwnProperty('redshift-port')) {
          obj['redshift-port'] = _ApiClient["default"].convertToType(data['redshift-port'], 'String');
        }

        if (data.hasOwnProperty('redshift-username')) {
          obj['redshift-username'] = _ApiClient["default"].convertToType(data['redshift-username'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('ssl')) {
          obj['ssl'] = _ApiClient["default"].convertToType(data['ssl'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerRedshift;
}();
/**
 * Redshift Creation statements
 * @member {String} creation-statements
 */


GatewayUpdateProducerRedshift.prototype['creation-statements'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

GatewayUpdateProducerRedshift.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerRedshift.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerRedshift.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerRedshift.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key
 */

GatewayUpdateProducerRedshift.prototype['producer-encryption-key'] = undefined;
/**
 * Redshift DB Name
 * @member {String} redshift-db-name
 */

GatewayUpdateProducerRedshift.prototype['redshift-db-name'] = undefined;
/**
 * Redshift Host
 * @member {String} redshift-host
 * @default '127.0.0.1'
 */

GatewayUpdateProducerRedshift.prototype['redshift-host'] = '127.0.0.1';
/**
 * Redshift Password
 * @member {String} redshift-password
 */

GatewayUpdateProducerRedshift.prototype['redshift-password'] = undefined;
/**
 * Redshift Port
 * @member {String} redshift-port
 * @default '5439'
 */

GatewayUpdateProducerRedshift.prototype['redshift-port'] = '5439';
/**
 * Redshift Username
 * @member {String} redshift-username
 */

GatewayUpdateProducerRedshift.prototype['redshift-username'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerRedshift.prototype['secure-access-enable'] = undefined;
/**
 * @member {Array.<String>} secure-access-host
 */

GatewayUpdateProducerRedshift.prototype['secure-access-host'] = undefined;
/**
 * SSL connection mode
 * @member {Boolean} ssl
 */

GatewayUpdateProducerRedshift.prototype['ssl'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerRedshift.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerRedshift.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerRedshift.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerRedshift.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerRedshift.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerRedshift;
exports["default"] = _default;
