// akeyless/dist/model/GatewayUpdateProducerGcp.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerGcp model module.
 * @module model/GatewayUpdateProducerGcp
 * @version 2.20.0
 */
var GatewayUpdateProducerGcp = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerGcp</code>.
   * gatewayUpdateProducerGcp is a command that updates a GCP producer
   * @alias module:model/GatewayUpdateProducerGcp
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerGcp(name) {
    _classCallCheck(this, GatewayUpdateProducerGcp);

    GatewayUpdateProducerGcp.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerGcp, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerGcp</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerGcp} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerGcp} The populated <code>GatewayUpdateProducerGcp</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerGcp();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('gcp-cred-type')) {
          obj['gcp-cred-type'] = _ApiClient["default"].convertToType(data['gcp-cred-type'], 'String');
        }

        if (data.hasOwnProperty('gcp-key')) {
          obj['gcp-key'] = _ApiClient["default"].convertToType(data['gcp-key'], 'String');
        }

        if (data.hasOwnProperty('gcp-key-algo')) {
          obj['gcp-key-algo'] = _ApiClient["default"].convertToType(data['gcp-key-algo'], 'String');
        }

        if (data.hasOwnProperty('gcp-sa-email')) {
          obj['gcp-sa-email'] = _ApiClient["default"].convertToType(data['gcp-sa-email'], 'String');
        }

        if (data.hasOwnProperty('gcp-token-scopes')) {
          obj['gcp-token-scopes'] = _ApiClient["default"].convertToType(data['gcp-token-scopes'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerGcp;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayUpdateProducerGcp.prototype['delete_protection'] = undefined;
/**
 * @member {String} gcp-cred-type
 */

GatewayUpdateProducerGcp.prototype['gcp-cred-type'] = undefined;
/**
 * Base64-encoded service account private key text
 * @member {String} gcp-key
 */

GatewayUpdateProducerGcp.prototype['gcp-key'] = undefined;
/**
 * Service account key algorithm, e.g. KEY_ALG_RSA_1024
 * @member {String} gcp-key-algo
 */

GatewayUpdateProducerGcp.prototype['gcp-key-algo'] = undefined;
/**
 * GCP service account email
 * @member {String} gcp-sa-email
 */

GatewayUpdateProducerGcp.prototype['gcp-sa-email'] = undefined;
/**
 * Access token scopes list, e.g. scope1,scope2
 * @member {String} gcp-token-scopes
 */

GatewayUpdateProducerGcp.prototype['gcp-token-scopes'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerGcp.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerGcp.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerGcp.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerGcp.prototype['producer-encryption-key-name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerGcp.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerGcp.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerGcp.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerGcp.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerGcp.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerGcp;
exports["default"] = _default;
