// akeyless/dist/model/GatewayUpdateProducerDockerhub.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerDockerhub model module.
 * @module model/GatewayUpdateProducerDockerhub
 * @version 2.20.0
 */
var GatewayUpdateProducerDockerhub = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerDockerhub</code>.
   * gatewayUpdateProducerDockerhub is a command that updates a DOCKERHUB producer
   * @alias module:model/GatewayUpdateProducerDockerhub
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerDockerhub(name) {
    _classCallCheck(this, GatewayUpdateProducerDockerhub);

    GatewayUpdateProducerDockerhub.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerDockerhub, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerDockerhub</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerDockerhub} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerDockerhub} The populated <code>GatewayUpdateProducerDockerhub</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerDockerhub();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('dockerhub-password')) {
          obj['dockerhub-password'] = _ApiClient["default"].convertToType(data['dockerhub-password'], 'String');
        }

        if (data.hasOwnProperty('dockerhub-token-scopes')) {
          obj['dockerhub-token-scopes'] = _ApiClient["default"].convertToType(data['dockerhub-token-scopes'], 'String');
        }

        if (data.hasOwnProperty('dockerhub-username')) {
          obj['dockerhub-username'] = _ApiClient["default"].convertToType(data['dockerhub-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerDockerhub;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayUpdateProducerDockerhub.prototype['delete_protection'] = undefined;
/**
 * DockerhubPassword is either the user's password access token to manage the repository
 * @member {String} dockerhub-password
 */

GatewayUpdateProducerDockerhub.prototype['dockerhub-password'] = undefined;
/**
 * Access token scopes list (comma-separated) to give the dynamic secret valid options are in \"repo:admin\", \"repo:write\", \"repo:read\", \"repo:public_read\"
 * @member {String} dockerhub-token-scopes
 */

GatewayUpdateProducerDockerhub.prototype['dockerhub-token-scopes'] = undefined;
/**
 * DockerhubUsername is the name of the user in dockerhub
 * @member {String} dockerhub-username
 */

GatewayUpdateProducerDockerhub.prototype['dockerhub-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerDockerhub.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerDockerhub.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerDockerhub.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerDockerhub.prototype['producer-encryption-key-name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerDockerhub.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerDockerhub.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerDockerhub.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerDockerhub.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerDockerhub.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerDockerhub;
exports["default"] = _default;
