// akeyless/dist/model/GatewayCreateProducerEks.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerEks model module.
 * @module model/GatewayCreateProducerEks
 * @version 2.15.24
 */
var GatewayCreateProducerEks = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerEks</code>.
   * gatewayCreateProducerEks is a command that creates eks producer
   * @alias module:model/GatewayCreateProducerEks
   * @param name {String} Producer name
   */
  function GatewayCreateProducerEks(name) {
    _classCallCheck(this, GatewayCreateProducerEks);

    GatewayCreateProducerEks.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerEks, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerEks</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerEks} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerEks} The populated <code>GatewayCreateProducerEks</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerEks();

        if (data.hasOwnProperty('eks-access-key-id')) {
          obj['eks-access-key-id'] = _ApiClient["default"].convertToType(data['eks-access-key-id'], 'String');
        }

        if (data.hasOwnProperty('eks-assume-role')) {
          obj['eks-assume-role'] = _ApiClient["default"].convertToType(data['eks-assume-role'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-ca-cert')) {
          obj['eks-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['eks-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-endpoint')) {
          obj['eks-cluster-endpoint'] = _ApiClient["default"].convertToType(data['eks-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('eks-cluster-name')) {
          obj['eks-cluster-name'] = _ApiClient["default"].convertToType(data['eks-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('eks-region')) {
          obj['eks-region'] = _ApiClient["default"].convertToType(data['eks-region'], 'String');
        }

        if (data.hasOwnProperty('eks-secret-access-key')) {
          obj['eks-secret-access-key'] = _ApiClient["default"].convertToType(data['eks-secret-access-key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-allow-port-forwading')) {
          obj['secure-access-allow-port-forwading'] = _ApiClient["default"].convertToType(data['secure-access-allow-port-forwading'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-cluster-endpoint')) {
          obj['secure-access-cluster-endpoint'] = _ApiClient["default"].convertToType(data['secure-access-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerEks;
}();
/**
 * Access Key ID
 * @member {String} eks-access-key-id
 */


GatewayCreateProducerEks.prototype['eks-access-key-id'] = undefined;
/**
 * IAM assume role
 * @member {String} eks-assume-role
 */

GatewayCreateProducerEks.prototype['eks-assume-role'] = undefined;
/**
 * EKS cluster CA certificate
 * @member {String} eks-cluster-ca-cert
 */

GatewayCreateProducerEks.prototype['eks-cluster-ca-cert'] = undefined;
/**
 * EKS cluster URL endpoint
 * @member {String} eks-cluster-endpoint
 */

GatewayCreateProducerEks.prototype['eks-cluster-endpoint'] = undefined;
/**
 * EKS cluster name
 * @member {String} eks-cluster-name
 */

GatewayCreateProducerEks.prototype['eks-cluster-name'] = undefined;
/**
 * Region
 * @member {String} eks-region
 * @default 'us-east-2'
 */

GatewayCreateProducerEks.prototype['eks-region'] = 'us-east-2';
/**
 * Secret Access Key
 * @member {String} eks-secret-access-key
 */

GatewayCreateProducerEks.prototype['eks-secret-access-key'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerEks.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayCreateProducerEks.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerEks.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {Boolean} secure-access-allow-port-forwading
 */

GatewayCreateProducerEks.prototype['secure-access-allow-port-forwading'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayCreateProducerEks.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-cluster-endpoint
 */

GatewayCreateProducerEks.prototype['secure-access-cluster-endpoint'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerEks.prototype['secure-access-enable'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayCreateProducerEks.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerEks.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerEks.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerEks.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerEks.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerEks.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayCreateProducerEks.prototype['username'] = undefined;
var _default = GatewayCreateProducerEks;
exports["default"] = _default;
