// akeyless/dist/model/GatewayMigratePersonalItems.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayMigratePersonalItems model module.
 * @module model/GatewayMigratePersonalItems
 * @version 2.20.0
 */
var GatewayMigratePersonalItems = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayMigratePersonalItems</code>.
   * gatewayMigratePersonalItems is a command that migrate personal items from external vault
   * @alias module:model/GatewayMigratePersonalItems
   */
  function GatewayMigratePersonalItems() {
    _classCallCheck(this, GatewayMigratePersonalItems);

    GatewayMigratePersonalItems.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayMigratePersonalItems, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GatewayMigratePersonalItems</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayMigratePersonalItems} obj Optional instance to populate.
     * @return {module:model/GatewayMigratePersonalItems} The populated <code>GatewayMigratePersonalItems</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayMigratePersonalItems();

        if (data.hasOwnProperty('1password-email')) {
          obj['1password-email'] = _ApiClient["default"].convertToType(data['1password-email'], 'String');
        }

        if (data.hasOwnProperty('1password-password')) {
          obj['1password-password'] = _ApiClient["default"].convertToType(data['1password-password'], 'String');
        }

        if (data.hasOwnProperty('1password-secret-key')) {
          obj['1password-secret-key'] = _ApiClient["default"].convertToType(data['1password-secret-key'], 'String');
        }

        if (data.hasOwnProperty('1password-url')) {
          obj['1password-url'] = _ApiClient["default"].convertToType(data['1password-url'], 'String');
        }

        if (data.hasOwnProperty('1password-vaults')) {
          obj['1password-vaults'] = _ApiClient["default"].convertToType(data['1password-vaults'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('protection-key')) {
          obj['protection-key'] = _ApiClient["default"].convertToType(data['protection-key'], 'String');
        }

        if (data.hasOwnProperty('target-location')) {
          obj['target-location'] = _ApiClient["default"].convertToType(data['target-location'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayMigratePersonalItems;
}();
/**
 * 1Password user email to connect to the API
 * @member {String} 1password-email
 */


GatewayMigratePersonalItems.prototype['1password-email'] = undefined;
/**
 * 1Password user password to connect to the API
 * @member {String} 1password-password
 */

GatewayMigratePersonalItems.prototype['1password-password'] = undefined;
/**
 * 1Password user secret key to connect to the API
 * @member {String} 1password-secret-key
 */

GatewayMigratePersonalItems.prototype['1password-secret-key'] = undefined;
/**
 * 1Password api container url
 * @member {String} 1password-url
 */

GatewayMigratePersonalItems.prototype['1password-url'] = undefined;
/**
 * 1Password list of vault to get the items from
 * @member {Array.<String>} 1password-vaults
 */

GatewayMigratePersonalItems.prototype['1password-vaults'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayMigratePersonalItems.prototype['json'] = undefined;
/**
 * The name of a key that used to encrypt the secret value
 * @member {String} protection-key
 */

GatewayMigratePersonalItems.prototype['protection-key'] = undefined;
/**
 * Target location in Akeyless for migrated secrets
 * @member {String} target-location
 */

GatewayMigratePersonalItems.prototype['target-location'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayMigratePersonalItems.prototype['token'] = undefined;
/**
 * Migration type for now only 1password.
 * @member {String} type
 * @default '1password'
 */

GatewayMigratePersonalItems.prototype['type'] = '1password';
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayMigratePersonalItems.prototype['uid-token'] = undefined;
var _default = GatewayMigratePersonalItems;
exports["default"] = _default;
