// akeyless/dist/model/GatewayCreateProducerAzure.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerAzure model module.
 * @module model/GatewayCreateProducerAzure
 * @version 2.15.24
 */
var GatewayCreateProducerAzure = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerAzure</code>.
   * gatewayCreateProducerAzure is a command that creates azure producer
   * @alias module:model/GatewayCreateProducerAzure
   * @param name {String} Producer name
   */
  function GatewayCreateProducerAzure(name) {
    _classCallCheck(this, GatewayCreateProducerAzure);

    GatewayCreateProducerAzure.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerAzure, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerAzure</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerAzure} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerAzure} The populated <code>GatewayCreateProducerAzure</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerAzure();

        if (data.hasOwnProperty('app-obj-id')) {
          obj['app-obj-id'] = _ApiClient["default"].convertToType(data['app-obj-id'], 'String');
        }

        if (data.hasOwnProperty('azure-client-id')) {
          obj['azure-client-id'] = _ApiClient["default"].convertToType(data['azure-client-id'], 'String');
        }

        if (data.hasOwnProperty('azure-client-secret')) {
          obj['azure-client-secret'] = _ApiClient["default"].convertToType(data['azure-client-secret'], 'String');
        }

        if (data.hasOwnProperty('azure-tenant-id')) {
          obj['azure-tenant-id'] = _ApiClient["default"].convertToType(data['azure-tenant-id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-group-obj-id')) {
          obj['user-group-obj-id'] = _ApiClient["default"].convertToType(data['user-group-obj-id'], 'String');
        }

        if (data.hasOwnProperty('user-portal-access')) {
          obj['user-portal-access'] = _ApiClient["default"].convertToType(data['user-portal-access'], 'Boolean');
        }

        if (data.hasOwnProperty('user-principal-name')) {
          obj['user-principal-name'] = _ApiClient["default"].convertToType(data['user-principal-name'], 'String');
        }

        if (data.hasOwnProperty('user-programmatic-access')) {
          obj['user-programmatic-access'] = _ApiClient["default"].convertToType(data['user-programmatic-access'], 'Boolean');
        }

        if (data.hasOwnProperty('user-role-template-id')) {
          obj['user-role-template-id'] = _ApiClient["default"].convertToType(data['user-role-template-id'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerAzure;
}();
/**
 * Azure App Object Id
 * @member {String} app-obj-id
 */


GatewayCreateProducerAzure.prototype['app-obj-id'] = undefined;
/**
 * Azure Client ID
 * @member {String} azure-client-id
 */

GatewayCreateProducerAzure.prototype['azure-client-id'] = undefined;
/**
 * Azure Client Secret
 * @member {String} azure-client-secret
 */

GatewayCreateProducerAzure.prototype['azure-client-secret'] = undefined;
/**
 * Azure Tenant ID
 * @member {String} azure-tenant-id
 */

GatewayCreateProducerAzure.prototype['azure-tenant-id'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerAzure.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayCreateProducerAzure.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerAzure.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerAzure.prototype['secure-access-enable'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayCreateProducerAzure.prototype['secure-access-web'] = undefined;
/**
 * @member {Boolean} secure-access-web-browsing
 */

GatewayCreateProducerAzure.prototype['secure-access-web-browsing'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerAzure.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerAzure.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerAzure.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerAzure.prototype['uid-token'] = undefined;
/**
 * User Group Object Id
 * @member {String} user-group-obj-id
 */

GatewayCreateProducerAzure.prototype['user-group-obj-id'] = undefined;
/**
 * Azure User portal access
 * @member {Boolean} user-portal-access
 * @default false
 */

GatewayCreateProducerAzure.prototype['user-portal-access'] = false;
/**
 * User Principal Name
 * @member {String} user-principal-name
 */

GatewayCreateProducerAzure.prototype['user-principal-name'] = undefined;
/**
 * Azure User programmatic access
 * @member {Boolean} user-programmatic-access
 * @default true
 */

GatewayCreateProducerAzure.prototype['user-programmatic-access'] = true;
/**
 * User Role Template Id
 * @member {String} user-role-template-id
 */

GatewayCreateProducerAzure.prototype['user-role-template-id'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerAzure.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayCreateProducerAzure.prototype['username'] = undefined;
var _default = GatewayCreateProducerAzure;
exports["default"] = _default;
