// akeyless/dist/model/GatewayCreateProducerHanaDb.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerHanaDb model module.
 * @module model/GatewayCreateProducerHanaDb
 * @version 2.20.0
 */
var GatewayCreateProducerHanaDb = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerHanaDb</code>.
   * gatewayCreateProducerHanaDb is a command that creates hanadb producer
   * @alias module:model/GatewayCreateProducerHanaDb
   * @param name {String} Producer name
   */
  function GatewayCreateProducerHanaDb(name) {
    _classCallCheck(this, GatewayCreateProducerHanaDb);

    GatewayCreateProducerHanaDb.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerHanaDb, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerHanaDb</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerHanaDb} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerHanaDb} The populated <code>GatewayCreateProducerHanaDb</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerHanaDb();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('hana-dbname')) {
          obj['hana-dbname'] = _ApiClient["default"].convertToType(data['hana-dbname'], 'String');
        }

        if (data.hasOwnProperty('hanadb-create-statements')) {
          obj['hanadb-create-statements'] = _ApiClient["default"].convertToType(data['hanadb-create-statements'], 'String');
        }

        if (data.hasOwnProperty('hanadb-host')) {
          obj['hanadb-host'] = _ApiClient["default"].convertToType(data['hanadb-host'], 'String');
        }

        if (data.hasOwnProperty('hanadb-password')) {
          obj['hanadb-password'] = _ApiClient["default"].convertToType(data['hanadb-password'], 'String');
        }

        if (data.hasOwnProperty('hanadb-port')) {
          obj['hanadb-port'] = _ApiClient["default"].convertToType(data['hanadb-port'], 'String');
        }

        if (data.hasOwnProperty('hanadb-revocation-statements')) {
          obj['hanadb-revocation-statements'] = _ApiClient["default"].convertToType(data['hanadb-revocation-statements'], 'String');
        }

        if (data.hasOwnProperty('hanadb-username')) {
          obj['hanadb-username'] = _ApiClient["default"].convertToType(data['hanadb-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-schema')) {
          obj['secure-access-db-schema'] = _ApiClient["default"].convertToType(data['secure-access-db-schema'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerHanaDb;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayCreateProducerHanaDb.prototype['delete_protection'] = undefined;
/**
 * HanaDb Name
 * @member {String} hana-dbname
 */

GatewayCreateProducerHanaDb.prototype['hana-dbname'] = undefined;
/**
 * HanaDb Creation statements
 * @member {String} hanadb-create-statements
 */

GatewayCreateProducerHanaDb.prototype['hanadb-create-statements'] = undefined;
/**
 * HanaDb Host
 * @member {String} hanadb-host
 * @default '127.0.0.1'
 */

GatewayCreateProducerHanaDb.prototype['hanadb-host'] = '127.0.0.1';
/**
 * HanaDb Password
 * @member {String} hanadb-password
 */

GatewayCreateProducerHanaDb.prototype['hanadb-password'] = undefined;
/**
 * HanaDb Port
 * @member {String} hanadb-port
 * @default '443'
 */

GatewayCreateProducerHanaDb.prototype['hanadb-port'] = '443';
/**
 * HanaDb Revocation statements
 * @member {String} hanadb-revocation-statements
 */

GatewayCreateProducerHanaDb.prototype['hanadb-revocation-statements'] = undefined;
/**
 * HanaDb Username
 * @member {String} hanadb-username
 */

GatewayCreateProducerHanaDb.prototype['hanadb-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayCreateProducerHanaDb.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerHanaDb.prototype['name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerHanaDb.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayCreateProducerHanaDb.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-db-schema
 */

GatewayCreateProducerHanaDb.prototype['secure-access-db-schema'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerHanaDb.prototype['secure-access-enable'] = undefined;
/**
 * @member {Array.<String>} secure-access-host
 */

GatewayCreateProducerHanaDb.prototype['secure-access-host'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayCreateProducerHanaDb.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerHanaDb.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerHanaDb.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerHanaDb.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerHanaDb.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerHanaDb.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerHanaDb;
exports["default"] = _default;
