// akeyless/dist/model/GatewayCreateProducerGithub.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerGithub model module.
 * @module model/GatewayCreateProducerGithub
 * @version 2.20.0
 */
var GatewayCreateProducerGithub = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerGithub</code>.
   * gatewayCreateProducerGithub is a command that creates github producer
   * @alias module:model/GatewayCreateProducerGithub
   * @param name {String} Producer name
   */
  function GatewayCreateProducerGithub(name) {
    _classCallCheck(this, GatewayCreateProducerGithub);

    GatewayCreateProducerGithub.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerGithub, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerGithub</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerGithub} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerGithub} The populated <code>GatewayCreateProducerGithub</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerGithub();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('github-app-id')) {
          obj['github-app-id'] = _ApiClient["default"].convertToType(data['github-app-id'], 'Number');
        }

        if (data.hasOwnProperty('github-app-private-key')) {
          obj['github-app-private-key'] = _ApiClient["default"].convertToType(data['github-app-private-key'], 'String');
        }

        if (data.hasOwnProperty('github-base-url')) {
          obj['github-base-url'] = _ApiClient["default"].convertToType(data['github-base-url'], 'String');
        }

        if (data.hasOwnProperty('installation-id')) {
          obj['installation-id'] = _ApiClient["default"].convertToType(data['installation-id'], 'Number');
        }

        if (data.hasOwnProperty('installation-repository')) {
          obj['installation-repository'] = _ApiClient["default"].convertToType(data['installation-repository'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('token-permissions')) {
          obj['token-permissions'] = _ApiClient["default"].convertToType(data['token-permissions'], ['String']);
        }

        if (data.hasOwnProperty('token-repositories')) {
          obj['token-repositories'] = _ApiClient["default"].convertToType(data['token-repositories'], ['String']);
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerGithub;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayCreateProducerGithub.prototype['delete_protection'] = undefined;
/**
 * Github app id
 * @member {Number} github-app-id
 */

GatewayCreateProducerGithub.prototype['github-app-id'] = undefined;
/**
 * App private key
 * @member {String} github-app-private-key
 */

GatewayCreateProducerGithub.prototype['github-app-private-key'] = undefined;
/**
 * Base URL
 * @member {String} github-base-url
 */

GatewayCreateProducerGithub.prototype['github-base-url'] = undefined;
/**
 * Github app installation id
 * @member {Number} installation-id
 */

GatewayCreateProducerGithub.prototype['installation-id'] = undefined;
/**
 * Repository that the app installation has access to
 * @member {String} installation-repository
 */

GatewayCreateProducerGithub.prototype['installation-repository'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayCreateProducerGithub.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerGithub.prototype['name'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerGithub.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerGithub.prototype['token'] = undefined;
/**
 * Optional - installation token's allowed permissions
 * @member {Array.<String>} token-permissions
 */

GatewayCreateProducerGithub.prototype['token-permissions'] = undefined;
/**
 * Optional - installation token's allowed repositories
 * @member {Array.<String>} token-repositories
 */

GatewayCreateProducerGithub.prototype['token-repositories'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerGithub.prototype['uid-token'] = undefined;
var _default = GatewayCreateProducerGithub;
exports["default"] = _default;
