// akeyless/dist/model/GatewayCreateProducerGke.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerGke model module.
 * @module model/GatewayCreateProducerGke
 * @version 2.20.0
 */
var GatewayCreateProducerGke = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerGke</code>.
   * gatewayCreateProducerGke is a command that creates gke producer
   * @alias module:model/GatewayCreateProducerGke
   * @param name {String} Producer name
   */
  function GatewayCreateProducerGke(name) {
    _classCallCheck(this, GatewayCreateProducerGke);

    GatewayCreateProducerGke.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerGke, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerGke</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerGke} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerGke} The populated <code>GatewayCreateProducerGke</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerGke();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('gke-account-key')) {
          obj['gke-account-key'] = _ApiClient["default"].convertToType(data['gke-account-key'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-cert')) {
          obj['gke-cluster-cert'] = _ApiClient["default"].convertToType(data['gke-cluster-cert'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-endpoint')) {
          obj['gke-cluster-endpoint'] = _ApiClient["default"].convertToType(data['gke-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-name')) {
          obj['gke-cluster-name'] = _ApiClient["default"].convertToType(data['gke-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('gke-service-account-email')) {
          obj['gke-service-account-email'] = _ApiClient["default"].convertToType(data['gke-service-account-email'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-allow-port-forwading')) {
          obj['secure-access-allow-port-forwading'] = _ApiClient["default"].convertToType(data['secure-access-allow-port-forwading'], 'Boolean');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-cluster-endpoint')) {
          obj['secure-access-cluster-endpoint'] = _ApiClient["default"].convertToType(data['secure-access-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerGke;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayCreateProducerGke.prototype['delete_protection'] = undefined;
/**
 * GKE Service Account key file path
 * @member {String} gke-account-key
 */

GatewayCreateProducerGke.prototype['gke-account-key'] = undefined;
/**
 * GKE cluster CA certificate
 * @member {String} gke-cluster-cert
 */

GatewayCreateProducerGke.prototype['gke-cluster-cert'] = undefined;
/**
 * GKE cluster URL endpoint
 * @member {String} gke-cluster-endpoint
 */

GatewayCreateProducerGke.prototype['gke-cluster-endpoint'] = undefined;
/**
 * GKE cluster name
 * @member {String} gke-cluster-name
 */

GatewayCreateProducerGke.prototype['gke-cluster-name'] = undefined;
/**
 * GKE service account email
 * @member {String} gke-service-account-email
 */

GatewayCreateProducerGke.prototype['gke-service-account-email'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayCreateProducerGke.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerGke.prototype['name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerGke.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {Boolean} secure-access-allow-port-forwading
 */

GatewayCreateProducerGke.prototype['secure-access-allow-port-forwading'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayCreateProducerGke.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-cluster-endpoint
 */

GatewayCreateProducerGke.prototype['secure-access-cluster-endpoint'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerGke.prototype['secure-access-enable'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayCreateProducerGke.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerGke.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerGke.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerGke.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerGke.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerGke.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerGke;
exports["default"] = _default;
