// akeyless/dist/model/GatewayCreateProducerRabbitMQ.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerRabbitMQ model module.
 * @module model/GatewayCreateProducerRabbitMQ
 * @version 2.15.24
 */
var GatewayCreateProducerRabbitMQ = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerRabbitMQ</code>.
   * gatewayCreateProducerRabbitMQ is a command that creates rabbitmq producer
   * @alias module:model/GatewayCreateProducerRabbitMQ
   * @param name {String} Producer name
   */
  function GatewayCreateProducerRabbitMQ(name) {
    _classCallCheck(this, GatewayCreateProducerRabbitMQ);

    GatewayCreateProducerRabbitMQ.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerRabbitMQ, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerRabbitMQ</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerRabbitMQ} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerRabbitMQ} The populated <code>GatewayCreateProducerRabbitMQ</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerRabbitMQ();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-admin-pwd')) {
          obj['rabbitmq-admin-pwd'] = _ApiClient["default"].convertToType(data['rabbitmq-admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-admin-user')) {
          obj['rabbitmq-admin-user'] = _ApiClient["default"].convertToType(data['rabbitmq-admin-user'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-uri')) {
          obj['rabbitmq-server-uri'] = _ApiClient["default"].convertToType(data['rabbitmq-server-uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-user-conf-permission')) {
          obj['rabbitmq-user-conf-permission'] = _ApiClient["default"].convertToType(data['rabbitmq-user-conf-permission'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-user-read-permission')) {
          obj['rabbitmq-user-read-permission'] = _ApiClient["default"].convertToType(data['rabbitmq-user-read-permission'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-user-tags')) {
          obj['rabbitmq-user-tags'] = _ApiClient["default"].convertToType(data['rabbitmq-user-tags'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-user-vhost')) {
          obj['rabbitmq-user-vhost'] = _ApiClient["default"].convertToType(data['rabbitmq-user-vhost'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-user-write-permission')) {
          obj['rabbitmq-user-write-permission'] = _ApiClient["default"].convertToType(data['rabbitmq-user-write-permission'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-url')) {
          obj['secure-access-url'] = _ApiClient["default"].convertToType(data['secure-access-url'], 'String');
        }

        if (data.hasOwnProperty('secure-access-web-browsing')) {
          obj['secure-access-web-browsing'] = _ApiClient["default"].convertToType(data['secure-access-web-browsing'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerRabbitMQ;
}();
/**
 * Producer name
 * @member {String} name
 */


GatewayCreateProducerRabbitMQ.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayCreateProducerRabbitMQ.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerRabbitMQ.prototype['producer-encryption-key-name'] = undefined;
/**
 * RabbitMQ Admin password
 * @member {String} rabbitmq-admin-pwd
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-admin-pwd'] = undefined;
/**
 * RabbitMQ Admin User
 * @member {String} rabbitmq-admin-user
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-admin-user'] = undefined;
/**
 * Server URI
 * @member {String} rabbitmq-server-uri
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-server-uri'] = undefined;
/**
 * User configuration permission
 * @member {String} rabbitmq-user-conf-permission
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-user-conf-permission'] = undefined;
/**
 * User read permission
 * @member {String} rabbitmq-user-read-permission
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-user-read-permission'] = undefined;
/**
 * User Tags
 * @member {String} rabbitmq-user-tags
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-user-tags'] = undefined;
/**
 * User Virtual Host
 * @member {String} rabbitmq-user-vhost
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-user-vhost'] = undefined;
/**
 * User write permission
 * @member {String} rabbitmq-user-write-permission
 */

GatewayCreateProducerRabbitMQ.prototype['rabbitmq-user-write-permission'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerRabbitMQ.prototype['secure-access-enable'] = undefined;
/**
 * @member {String} secure-access-url
 */

GatewayCreateProducerRabbitMQ.prototype['secure-access-url'] = undefined;
/**
 * @member {Boolean} secure-access-web-browsing
 */

GatewayCreateProducerRabbitMQ.prototype['secure-access-web-browsing'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerRabbitMQ.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerRabbitMQ.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerRabbitMQ.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerRabbitMQ.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerRabbitMQ.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayCreateProducerRabbitMQ.prototype['username'] = undefined;
var _default = GatewayCreateProducerRabbitMQ;
exports["default"] = _default;
