// akeyless/dist/model/GatewayUpdateProducerCassandra.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerCassandra model module.
 * @module model/GatewayUpdateProducerCassandra
 * @version 2.15.24
 */
var GatewayUpdateProducerCassandra = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerCassandra</code>.
   * gatewayUpdateProducerCassandra is a command that updates a Cassandra producer
   * @alias module:model/GatewayUpdateProducerCassandra
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerCassandra(name) {
    _classCallCheck(this, GatewayUpdateProducerCassandra);

    GatewayUpdateProducerCassandra.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerCassandra, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerCassandra</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerCassandra} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerCassandra} The populated <code>GatewayUpdateProducerCassandra</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerCassandra();

        if (data.hasOwnProperty('cassandra-creation-statements')) {
          obj['cassandra-creation-statements'] = _ApiClient["default"].convertToType(data['cassandra-creation-statements'], 'String');
        }

        if (data.hasOwnProperty('cassandra-hosts')) {
          obj['cassandra-hosts'] = _ApiClient["default"].convertToType(data['cassandra-hosts'], 'String');
        }

        if (data.hasOwnProperty('cassandra-password')) {
          obj['cassandra-password'] = _ApiClient["default"].convertToType(data['cassandra-password'], 'String');
        }

        if (data.hasOwnProperty('cassandra-port')) {
          obj['cassandra-port'] = _ApiClient["default"].convertToType(data['cassandra-port'], 'String');
        }

        if (data.hasOwnProperty('cassandra-username')) {
          obj['cassandra-username'] = _ApiClient["default"].convertToType(data['cassandra-username'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerCassandra;
}();
/**
 * Cassandra creation statements
 * @member {String} cassandra-creation-statements
 */


GatewayUpdateProducerCassandra.prototype['cassandra-creation-statements'] = undefined;
/**
 * Cassandra hosts IP or addresses, comma separated
 * @member {String} cassandra-hosts
 */

GatewayUpdateProducerCassandra.prototype['cassandra-hosts'] = undefined;
/**
 * Cassandra superuser password
 * @member {String} cassandra-password
 */

GatewayUpdateProducerCassandra.prototype['cassandra-password'] = undefined;
/**
 * Cassandra port
 * @member {String} cassandra-port
 * @default '9042'
 */

GatewayUpdateProducerCassandra.prototype['cassandra-port'] = '9042';
/**
 * Cassandra superuser username
 * @member {String} cassandra-username
 */

GatewayUpdateProducerCassandra.prototype['cassandra-username'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerCassandra.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerCassandra.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayUpdateProducerCassandra.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerCassandra.prototype['producer-encryption-key-name'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerCassandra.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerCassandra.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerCassandra.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerCassandra.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerCassandra.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayUpdateProducerCassandra.prototype['username'] = undefined;
var _default = GatewayUpdateProducerCassandra;
exports["default"] = _default;
