// akeyless/dist/model/GatewayUpdateProducerChef.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerChef model module.
 * @module model/GatewayUpdateProducerChef
 * @version 2.15.24
 */
var GatewayUpdateProducerChef = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerChef</code>.
   * gatewayUpdateProducerChef is a command that updates chef producer
   * @alias module:model/GatewayUpdateProducerChef
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerChef(name) {
    _classCallCheck(this, GatewayUpdateProducerChef);

    GatewayUpdateProducerChef.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerChef, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerChef</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerChef} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerChef} The populated <code>GatewayUpdateProducerChef</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerChef();

        if (data.hasOwnProperty('chef-orgs')) {
          obj['chef-orgs'] = _ApiClient["default"].convertToType(data['chef-orgs'], 'String');
        }

        if (data.hasOwnProperty('chef-server-key')) {
          obj['chef-server-key'] = _ApiClient["default"].convertToType(data['chef-server-key'], 'String');
        }

        if (data.hasOwnProperty('chef-server-url')) {
          obj['chef-server-url'] = _ApiClient["default"].convertToType(data['chef-server-url'], 'String');
        }

        if (data.hasOwnProperty('chef-server-username')) {
          obj['chef-server-username'] = _ApiClient["default"].convertToType(data['chef-server-username'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('skip-ssl')) {
          obj['skip-ssl'] = _ApiClient["default"].convertToType(data['skip-ssl'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerChef;
}();
/**
 * Organizations
 * @member {String} chef-orgs
 */


GatewayUpdateProducerChef.prototype['chef-orgs'] = undefined;
/**
 * Server key
 * @member {String} chef-server-key
 */

GatewayUpdateProducerChef.prototype['chef-server-key'] = undefined;
/**
 * Server URL
 * @member {String} chef-server-url
 */

GatewayUpdateProducerChef.prototype['chef-server-url'] = undefined;
/**
 * Server username
 * @member {String} chef-server-username
 */

GatewayUpdateProducerChef.prototype['chef-server-username'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerChef.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerChef.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GatewayUpdateProducerChef.prototype['password'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerChef.prototype['producer-encryption-key-name'] = undefined;
/**
 * Skip SSL
 * @member {Boolean} skip-ssl
 * @default true
 */

GatewayUpdateProducerChef.prototype['skip-ssl'] = true;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerChef.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerChef.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerChef.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerChef.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerChef.prototype['user-ttl'] = '60m';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GatewayUpdateProducerChef.prototype['username'] = undefined;
var _default = GatewayUpdateProducerChef;
exports["default"] = _default;
