// akeyless/dist/model/GatewayUpdateProducerOracleDb.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerOracleDb model module.
 * @module model/GatewayUpdateProducerOracleDb
 * @version 2.20.0
 */
var GatewayUpdateProducerOracleDb = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerOracleDb</code>.
   * gatewayUpdateProducerOracleDb is a command that updates oracle db producer
   * @alias module:model/GatewayUpdateProducerOracleDb
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerOracleDb(name) {
    _classCallCheck(this, GatewayUpdateProducerOracleDb);

    GatewayUpdateProducerOracleDb.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerOracleDb, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerOracleDb</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerOracleDb} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerOracleDb} The populated <code>GatewayUpdateProducerOracleDb</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerOracleDb();

        if (data.hasOwnProperty('db-server-certificates')) {
          obj['db-server-certificates'] = _ApiClient["default"].convertToType(data['db-server-certificates'], 'String');
        }

        if (data.hasOwnProperty('db-server-name')) {
          obj['db-server-name'] = _ApiClient["default"].convertToType(data['db-server-name'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('oracle-host')) {
          obj['oracle-host'] = _ApiClient["default"].convertToType(data['oracle-host'], 'String');
        }

        if (data.hasOwnProperty('oracle-password')) {
          obj['oracle-password'] = _ApiClient["default"].convertToType(data['oracle-password'], 'String');
        }

        if (data.hasOwnProperty('oracle-port')) {
          obj['oracle-port'] = _ApiClient["default"].convertToType(data['oracle-port'], 'String');
        }

        if (data.hasOwnProperty('oracle-screation-statements')) {
          obj['oracle-screation-statements'] = _ApiClient["default"].convertToType(data['oracle-screation-statements'], 'String');
        }

        if (data.hasOwnProperty('oracle-service-name')) {
          obj['oracle-service-name'] = _ApiClient["default"].convertToType(data['oracle-service-name'], 'String');
        }

        if (data.hasOwnProperty('oracle-username')) {
          obj['oracle-username'] = _ApiClient["default"].convertToType(data['oracle-username'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerOracleDb;
}();
/**
 * (Optional) DB server certificates
 * @member {String} db-server-certificates
 */


GatewayUpdateProducerOracleDb.prototype['db-server-certificates'] = undefined;
/**
 * (Optional) Server name for certificate verification
 * @member {String} db-server-name
 */

GatewayUpdateProducerOracleDb.prototype['db-server-name'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

GatewayUpdateProducerOracleDb.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerOracleDb.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerOracleDb.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerOracleDb.prototype['new-name'] = undefined;
/**
 * Oracle Host
 * @member {String} oracle-host
 * @default '127.0.0.1'
 */

GatewayUpdateProducerOracleDb.prototype['oracle-host'] = '127.0.0.1';
/**
 * Oracle Password
 * @member {String} oracle-password
 */

GatewayUpdateProducerOracleDb.prototype['oracle-password'] = undefined;
/**
 * Oracle Port
 * @member {String} oracle-port
 * @default '1521'
 */

GatewayUpdateProducerOracleDb.prototype['oracle-port'] = '1521';
/**
 * Oracle Creation statements
 * @member {String} oracle-screation-statements
 */

GatewayUpdateProducerOracleDb.prototype['oracle-screation-statements'] = undefined;
/**
 * Oracle DB Name
 * @member {String} oracle-service-name
 */

GatewayUpdateProducerOracleDb.prototype['oracle-service-name'] = undefined;
/**
 * Oracle Username
 * @member {String} oracle-username
 */

GatewayUpdateProducerOracleDb.prototype['oracle-username'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerOracleDb.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayUpdateProducerOracleDb.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerOracleDb.prototype['secure-access-enable'] = undefined;
/**
 * @member {Array.<String>} secure-access-host
 */

GatewayUpdateProducerOracleDb.prototype['secure-access-host'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayUpdateProducerOracleDb.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerOracleDb.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerOracleDb.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerOracleDb.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerOracleDb.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerOracleDb.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerOracleDb;
exports["default"] = _default;
