// akeyless/dist/model/GatewayCreateProducerMongo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerMongo model module.
 * @module model/GatewayCreateProducerMongo
 * @version 2.20.0
 */
var GatewayCreateProducerMongo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerMongo</code>.
   * gatewayCreateProducerMongo is a command that creates either mongodb  producer or mongodb atlas producer
   * @alias module:model/GatewayCreateProducerMongo
   * @param name {String} Producer name
   */
  function GatewayCreateProducerMongo(name) {
    _classCallCheck(this, GatewayCreateProducerMongo);

    GatewayCreateProducerMongo.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerMongo, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayCreateProducerMongo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerMongo} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerMongo} The populated <code>GatewayCreateProducerMongo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerMongo();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('mongodb-atlas-api-private-key')) {
          obj['mongodb-atlas-api-private-key'] = _ApiClient["default"].convertToType(data['mongodb-atlas-api-private-key'], 'String');
        }

        if (data.hasOwnProperty('mongodb-atlas-api-public-key')) {
          obj['mongodb-atlas-api-public-key'] = _ApiClient["default"].convertToType(data['mongodb-atlas-api-public-key'], 'String');
        }

        if (data.hasOwnProperty('mongodb-atlas-project-id')) {
          obj['mongodb-atlas-project-id'] = _ApiClient["default"].convertToType(data['mongodb-atlas-project-id'], 'String');
        }

        if (data.hasOwnProperty('mongodb-custom-data')) {
          obj['mongodb-custom-data'] = _ApiClient["default"].convertToType(data['mongodb-custom-data'], 'String');
        }

        if (data.hasOwnProperty('mongodb-default-auth-db')) {
          obj['mongodb-default-auth-db'] = _ApiClient["default"].convertToType(data['mongodb-default-auth-db'], 'String');
        }

        if (data.hasOwnProperty('mongodb-host-port')) {
          obj['mongodb-host-port'] = _ApiClient["default"].convertToType(data['mongodb-host-port'], 'String');
        }

        if (data.hasOwnProperty('mongodb-name')) {
          obj['mongodb-name'] = _ApiClient["default"].convertToType(data['mongodb-name'], 'String');
        }

        if (data.hasOwnProperty('mongodb-password')) {
          obj['mongodb-password'] = _ApiClient["default"].convertToType(data['mongodb-password'], 'String');
        }

        if (data.hasOwnProperty('mongodb-roles')) {
          obj['mongodb-roles'] = _ApiClient["default"].convertToType(data['mongodb-roles'], 'String');
        }

        if (data.hasOwnProperty('mongodb-server-uri')) {
          obj['mongodb-server-uri'] = _ApiClient["default"].convertToType(data['mongodb-server-uri'], 'String');
        }

        if (data.hasOwnProperty('mongodb-uri-options')) {
          obj['mongodb-uri-options'] = _ApiClient["default"].convertToType(data['mongodb-uri-options'], 'String');
        }

        if (data.hasOwnProperty('mongodb-username')) {
          obj['mongodb-username'] = _ApiClient["default"].convertToType(data['mongodb-username'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerMongo;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayCreateProducerMongo.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayCreateProducerMongo.prototype['json'] = undefined;
/**
 * MongoDB Atlas private key
 * @member {String} mongodb-atlas-api-private-key
 */

GatewayCreateProducerMongo.prototype['mongodb-atlas-api-private-key'] = undefined;
/**
 * MongoDB Atlas public key
 * @member {String} mongodb-atlas-api-public-key
 */

GatewayCreateProducerMongo.prototype['mongodb-atlas-api-public-key'] = undefined;
/**
 * MongoDB Atlas project ID
 * @member {String} mongodb-atlas-project-id
 */

GatewayCreateProducerMongo.prototype['mongodb-atlas-project-id'] = undefined;
/**
 * MongoDB custom data
 * @member {String} mongodb-custom-data
 */

GatewayCreateProducerMongo.prototype['mongodb-custom-data'] = undefined;
/**
 * MongoDB server default authentication database
 * @member {String} mongodb-default-auth-db
 */

GatewayCreateProducerMongo.prototype['mongodb-default-auth-db'] = undefined;
/**
 * MongoDB server host and port
 * @member {String} mongodb-host-port
 */

GatewayCreateProducerMongo.prototype['mongodb-host-port'] = undefined;
/**
 * MongoDB Name
 * @member {String} mongodb-name
 */

GatewayCreateProducerMongo.prototype['mongodb-name'] = undefined;
/**
 * MongoDB server password. You will prompted to provide a password if it will not appear in CLI parameters
 * @member {String} mongodb-password
 */

GatewayCreateProducerMongo.prototype['mongodb-password'] = undefined;
/**
 * MongoDB Roles
 * @member {String} mongodb-roles
 * @default '[]'
 */

GatewayCreateProducerMongo.prototype['mongodb-roles'] = '[]';
/**
 * MongoDB server URI
 * @member {String} mongodb-server-uri
 */

GatewayCreateProducerMongo.prototype['mongodb-server-uri'] = undefined;
/**
 * MongoDB server URI options
 * @member {String} mongodb-uri-options
 */

GatewayCreateProducerMongo.prototype['mongodb-uri-options'] = undefined;
/**
 * MongoDB server username
 * @member {String} mongodb-username
 */

GatewayCreateProducerMongo.prototype['mongodb-username'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerMongo.prototype['name'] = undefined;
/**
 * Encrypt producer with following key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerMongo.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayCreateProducerMongo.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayCreateProducerMongo.prototype['secure-access-enable'] = undefined;
/**
 * @member {Array.<String>} secure-access-host
 */

GatewayCreateProducerMongo.prototype['secure-access-host'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayCreateProducerMongo.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerMongo.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayCreateProducerMongo.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerMongo.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerMongo.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerMongo.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerMongo;
exports["default"] = _default;
