// akeyless/dist/model/GatewayCreateProducerCustom.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayCreateProducerCustom model module.
 * @module model/GatewayCreateProducerCustom
 * @version 2.20.0
 */
var GatewayCreateProducerCustom = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayCreateProducerCustom</code>.
   * @alias module:model/GatewayCreateProducerCustom
   * @param createSyncUrl {String} URL of an endpoint that implements /sync/create method, for example https://webhook.example.com/sync/create
   * @param name {String} Producer name
   * @param revokeSyncUrl {String} URL of an endpoint that implements /sync/revoke method, for example https://webhook.example.com/sync/revoke
   */
  function GatewayCreateProducerCustom(createSyncUrl, name, revokeSyncUrl) {
    _classCallCheck(this, GatewayCreateProducerCustom);

    GatewayCreateProducerCustom.initialize(this, createSyncUrl, name, revokeSyncUrl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayCreateProducerCustom, null, [{
    key: "initialize",
    value: function initialize(obj, createSyncUrl, name, revokeSyncUrl) {
      obj['create-sync-url'] = createSyncUrl;
      obj['name'] = name;
      obj['revoke-sync-url'] = revokeSyncUrl;
    }
    /**
     * Constructs a <code>GatewayCreateProducerCustom</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayCreateProducerCustom} obj Optional instance to populate.
     * @return {module:model/GatewayCreateProducerCustom} The populated <code>GatewayCreateProducerCustom</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayCreateProducerCustom();

        if (data.hasOwnProperty('admin_rotation_interval_days')) {
          obj['admin_rotation_interval_days'] = _ApiClient["default"].convertToType(data['admin_rotation_interval_days'], 'Number');
        }

        if (data.hasOwnProperty('create-sync-url')) {
          obj['create-sync-url'] = _ApiClient["default"].convertToType(data['create-sync-url'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('enable_admin_rotation')) {
          obj['enable_admin_rotation'] = _ApiClient["default"].convertToType(data['enable_admin_rotation'], 'Boolean');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ApiClient["default"].convertToType(data['payload'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('revoke-sync-url')) {
          obj['revoke-sync-url'] = _ApiClient["default"].convertToType(data['revoke-sync-url'], 'String');
        }

        if (data.hasOwnProperty('rotate-sync-url')) {
          obj['rotate-sync-url'] = _ApiClient["default"].convertToType(data['rotate-sync-url'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('timeout-sec')) {
          obj['timeout-sec'] = _ApiClient["default"].convertToType(data['timeout-sec'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayCreateProducerCustom;
}();
/**
 * Define rotation interval in days
 * @member {Number} admin_rotation_interval_days
 */


GatewayCreateProducerCustom.prototype['admin_rotation_interval_days'] = undefined;
/**
 * URL of an endpoint that implements /sync/create method, for example https://webhook.example.com/sync/create
 * @member {String} create-sync-url
 */

GatewayCreateProducerCustom.prototype['create-sync-url'] = undefined;
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */

GatewayCreateProducerCustom.prototype['delete_protection'] = undefined;
/**
 * Should admin credentials be rotated
 * @member {Boolean} enable_admin_rotation
 * @default false
 */

GatewayCreateProducerCustom.prototype['enable_admin_rotation'] = false;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayCreateProducerCustom.prototype['json'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayCreateProducerCustom.prototype['name'] = undefined;
/**
 * Secret payload to be sent with each create/revoke webhook request
 * @member {String} payload
 */

GatewayCreateProducerCustom.prototype['payload'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayCreateProducerCustom.prototype['producer-encryption-key-name'] = undefined;
/**
 * URL of an endpoint that implements /sync/revoke method, for example https://webhook.example.com/sync/revoke
 * @member {String} revoke-sync-url
 */

GatewayCreateProducerCustom.prototype['revoke-sync-url'] = undefined;
/**
 * URL of an endpoint that implements /sync/rotate method, for example https://webhook.example.com/sync/rotate
 * @member {String} rotate-sync-url
 */

GatewayCreateProducerCustom.prototype['rotate-sync-url'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayCreateProducerCustom.prototype['tags'] = undefined;
/**
 * Maximum allowed time in seconds for the webhook to return the results
 * @member {Number} timeout-sec
 * @default 60
 */

GatewayCreateProducerCustom.prototype['timeout-sec'] = 60;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayCreateProducerCustom.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayCreateProducerCustom.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayCreateProducerCustom.prototype['user-ttl'] = '60m';
var _default = GatewayCreateProducerCustom;
exports["default"] = _default;
