// akeyless/dist/model/GatewayUpdateProducerMSSQL.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GatewayUpdateProducerMSSQL model module.
 * @module model/GatewayUpdateProducerMSSQL
 * @version 2.20.0
 */
var GatewayUpdateProducerMSSQL = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GatewayUpdateProducerMSSQL</code>.
   * gatewayUpdateProducerMSSQL is a command that updates mssql producer
   * @alias module:model/GatewayUpdateProducerMSSQL
   * @param name {String} Producer name
   */
  function GatewayUpdateProducerMSSQL(name) {
    _classCallCheck(this, GatewayUpdateProducerMSSQL);

    GatewayUpdateProducerMSSQL.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GatewayUpdateProducerMSSQL, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GatewayUpdateProducerMSSQL</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GatewayUpdateProducerMSSQL} obj Optional instance to populate.
     * @return {module:model/GatewayUpdateProducerMSSQL} The populated <code>GatewayUpdateProducerMSSQL</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GatewayUpdateProducerMSSQL();

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('mssql-create-statements')) {
          obj['mssql-create-statements'] = _ApiClient["default"].convertToType(data['mssql-create-statements'], 'String');
        }

        if (data.hasOwnProperty('mssql-dbname')) {
          obj['mssql-dbname'] = _ApiClient["default"].convertToType(data['mssql-dbname'], 'String');
        }

        if (data.hasOwnProperty('mssql-host')) {
          obj['mssql-host'] = _ApiClient["default"].convertToType(data['mssql-host'], 'String');
        }

        if (data.hasOwnProperty('mssql-password')) {
          obj['mssql-password'] = _ApiClient["default"].convertToType(data['mssql-password'], 'String');
        }

        if (data.hasOwnProperty('mssql-port')) {
          obj['mssql-port'] = _ApiClient["default"].convertToType(data['mssql-port'], 'String');
        }

        if (data.hasOwnProperty('mssql-revocation-statements')) {
          obj['mssql-revocation-statements'] = _ApiClient["default"].convertToType(data['mssql-revocation-statements'], 'String');
        }

        if (data.hasOwnProperty('mssql-username')) {
          obj['mssql-username'] = _ApiClient["default"].convertToType(data['mssql-username'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('producer-encryption-key-name')) {
          obj['producer-encryption-key-name'] = _ApiClient["default"].convertToType(data['producer-encryption-key-name'], 'String');
        }

        if (data.hasOwnProperty('secure-access-bastion-issuer')) {
          obj['secure-access-bastion-issuer'] = _ApiClient["default"].convertToType(data['secure-access-bastion-issuer'], 'String');
        }

        if (data.hasOwnProperty('secure-access-db-schema')) {
          obj['secure-access-db-schema'] = _ApiClient["default"].convertToType(data['secure-access-db-schema'], 'String');
        }

        if (data.hasOwnProperty('secure-access-enable')) {
          obj['secure-access-enable'] = _ApiClient["default"].convertToType(data['secure-access-enable'], 'String');
        }

        if (data.hasOwnProperty('secure-access-host')) {
          obj['secure-access-host'] = _ApiClient["default"].convertToType(data['secure-access-host'], ['String']);
        }

        if (data.hasOwnProperty('secure-access-web')) {
          obj['secure-access-web'] = _ApiClient["default"].convertToType(data['secure-access-web'], 'Boolean');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], ['String']);
        }

        if (data.hasOwnProperty('target-name')) {
          obj['target-name'] = _ApiClient["default"].convertToType(data['target-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user-ttl')) {
          obj['user-ttl'] = _ApiClient["default"].convertToType(data['user-ttl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GatewayUpdateProducerMSSQL;
}();
/**
 * Protection from accidental deletion of this item
 * @member {String} delete_protection
 */


GatewayUpdateProducerMSSQL.prototype['delete_protection'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GatewayUpdateProducerMSSQL.prototype['json'] = undefined;
/**
 * MSSQL Creation statements
 * @member {String} mssql-create-statements
 */

GatewayUpdateProducerMSSQL.prototype['mssql-create-statements'] = undefined;
/**
 * MSSQL Name
 * @member {String} mssql-dbname
 */

GatewayUpdateProducerMSSQL.prototype['mssql-dbname'] = undefined;
/**
 * MSSQL Host
 * @member {String} mssql-host
 * @default '127.0.0.1'
 */

GatewayUpdateProducerMSSQL.prototype['mssql-host'] = '127.0.0.1';
/**
 * MSSQL Password
 * @member {String} mssql-password
 */

GatewayUpdateProducerMSSQL.prototype['mssql-password'] = undefined;
/**
 * MSSQL Port
 * @member {String} mssql-port
 * @default '1433'
 */

GatewayUpdateProducerMSSQL.prototype['mssql-port'] = '1433';
/**
 * MSSQL Revocation statements
 * @member {String} mssql-revocation-statements
 */

GatewayUpdateProducerMSSQL.prototype['mssql-revocation-statements'] = undefined;
/**
 * MSSQL Username
 * @member {String} mssql-username
 */

GatewayUpdateProducerMSSQL.prototype['mssql-username'] = undefined;
/**
 * Producer name
 * @member {String} name
 */

GatewayUpdateProducerMSSQL.prototype['name'] = undefined;
/**
 * Producer name
 * @member {String} new-name
 */

GatewayUpdateProducerMSSQL.prototype['new-name'] = undefined;
/**
 * Dynamic producer encryption key
 * @member {String} producer-encryption-key-name
 */

GatewayUpdateProducerMSSQL.prototype['producer-encryption-key-name'] = undefined;
/**
 * @member {String} secure-access-bastion-issuer
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-bastion-issuer'] = undefined;
/**
 * @member {String} secure-access-db-schema
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-db-schema'] = undefined;
/**
 * @member {String} secure-access-enable
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-enable'] = undefined;
/**
 * @member {Array.<String>} secure-access-host
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-host'] = undefined;
/**
 * @member {Boolean} secure-access-web
 */

GatewayUpdateProducerMSSQL.prototype['secure-access-web'] = undefined;
/**
 * List of the tags attached to this secret
 * @member {Array.<String>} tags
 */

GatewayUpdateProducerMSSQL.prototype['tags'] = undefined;
/**
 * Target name
 * @member {String} target-name
 */

GatewayUpdateProducerMSSQL.prototype['target-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GatewayUpdateProducerMSSQL.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GatewayUpdateProducerMSSQL.prototype['uid-token'] = undefined;
/**
 * User TTL
 * @member {String} user-ttl
 * @default '60m'
 */

GatewayUpdateProducerMSSQL.prototype['user-ttl'] = '60m';
var _default = GatewayUpdateProducerMSSQL;
exports["default"] = _default;
