// akeyless/dist/model/GetCloudIdentity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetCloudIdentity model module.
 * @module model/GetCloudIdentity
 * @version 2.15.24
 */
var GetCloudIdentity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetCloudIdentity</code>.
   * getCloudIdentity is a command that gets Cloud Identity Token (relevant only for access-type&#x3D;azure_ad, aws_iam, gcp).
   * @alias module:model/GetCloudIdentity
   */
  function GetCloudIdentity() {
    _classCallCheck(this, GetCloudIdentity);

    GetCloudIdentity.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetCloudIdentity, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetCloudIdentity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetCloudIdentity} obj Optional instance to populate.
     * @return {module:model/GetCloudIdentity} The populated <code>GetCloudIdentity</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetCloudIdentity();

        if (data.hasOwnProperty('azure_ad_object_id')) {
          obj['azure_ad_object_id'] = _ApiClient["default"].convertToType(data['azure_ad_object_id'], 'String');
        }

        if (data.hasOwnProperty('debug')) {
          obj['debug'] = _ApiClient["default"].convertToType(data['debug'], 'Boolean');
        }

        if (data.hasOwnProperty('gcp-audience')) {
          obj['gcp-audience'] = _ApiClient["default"].convertToType(data['gcp-audience'], 'String');
        }

        if (data.hasOwnProperty('url_safe')) {
          obj['url_safe'] = _ApiClient["default"].convertToType(data['url_safe'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return GetCloudIdentity;
}();
/**
 * Azure Active Directory ObjectId (relevant only for access-type=azure_ad)
 * @member {String} azure_ad_object_id
 */


GetCloudIdentity.prototype['azure_ad_object_id'] = undefined;
/**
 * @member {Boolean} debug
 */

GetCloudIdentity.prototype['debug'] = undefined;
/**
 * GCP JWT audience
 * @member {String} gcp-audience
 */

GetCloudIdentity.prototype['gcp-audience'] = undefined;
/**
 * Escapes the token so it can be safely placed inside a URL query
 * @member {Boolean} url_safe
 */

GetCloudIdentity.prototype['url_safe'] = undefined;
var _default = GetCloudIdentity;
exports["default"] = _default;
