// akeyless/dist/model/GetDynamicSecretValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetDynamicSecretValue model module.
 * @module model/GetDynamicSecretValue
 * @version 2.20.0
 */
var GetDynamicSecretValue = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetDynamicSecretValue</code>.
   * @alias module:model/GetDynamicSecretValue
   * @param name {String} Dynamic secret name
   */
  function GetDynamicSecretValue(name) {
    _classCallCheck(this, GetDynamicSecretValue);

    GetDynamicSecretValue.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetDynamicSecretValue, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GetDynamicSecretValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetDynamicSecretValue} obj Optional instance to populate.
     * @return {module:model/GetDynamicSecretValue} The populated <code>GetDynamicSecretValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetDynamicSecretValue();

        if (data.hasOwnProperty('args')) {
          obj['args'] = _ApiClient["default"].convertToType(data['args'], ['String']);
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }

        if (data.hasOwnProperty('timeout')) {
          obj['timeout'] = _ApiClient["default"].convertToType(data['timeout'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetDynamicSecretValue;
}();
/**
 * Optional arguments as key=value pairs or JSON strings, e.g - \\\"--args=csr=base64_encoded_csr --args=common_name=bar\\\" or args='{\\\"csr\\\":\\\"base64_encoded_csr\\\"}. It is possible to combine both formats.'
 * @member {Array.<String>} args
 */


GetDynamicSecretValue.prototype['args'] = undefined;
/**
 * Host
 * @member {String} host
 */

GetDynamicSecretValue.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GetDynamicSecretValue.prototype['json'] = undefined;
/**
 * Dynamic secret name
 * @member {String} name
 */

GetDynamicSecretValue.prototype['name'] = undefined;
/**
 * Target Name
 * @member {String} target
 */

GetDynamicSecretValue.prototype['target'] = undefined;
/**
 * Timeout in seconds
 * @member {Number} timeout
 * @default 15
 */

GetDynamicSecretValue.prototype['timeout'] = 15;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetDynamicSecretValue.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetDynamicSecretValue.prototype['uid-token'] = undefined;
var _default = GetDynamicSecretValue;
exports["default"] = _default;
