// akeyless/dist/model/GetPKICertificateOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetPKICertificateOutput model module.
 * @module model/GetPKICertificateOutput
 * @version 2.15.24
 */
var GetPKICertificateOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetPKICertificateOutput</code>.
   * @alias module:model/GetPKICertificateOutput
   */
  function GetPKICertificateOutput() {
    _classCallCheck(this, GetPKICertificateOutput);

    GetPKICertificateOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetPKICertificateOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GetPKICertificateOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetPKICertificateOutput} obj Optional instance to populate.
     * @return {module:model/GetPKICertificateOutput} The populated <code>GetPKICertificateOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetPKICertificateOutput();

        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], 'String');
        }

        if (data.hasOwnProperty('parent_cert')) {
          obj['parent_cert'] = _ApiClient["default"].convertToType(data['parent_cert'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetPKICertificateOutput;
}();
/**
 * @member {String} data
 */


GetPKICertificateOutput.prototype['data'] = undefined;
/**
 * @member {String} parent_cert
 */

GetPKICertificateOutput.prototype['parent_cert'] = undefined;
/**
 * @member {String} path
 */

GetPKICertificateOutput.prototype['path'] = undefined;
var _default = GetPKICertificateOutput;
exports["default"] = _default;
