// akeyless/dist/model/GetPKICertificate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetPKICertificate model module.
 * @module model/GetPKICertificate
 * @version 2.20.0
 */
var GetPKICertificate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetPKICertificate</code>.
   * @alias module:model/GetPKICertificate
   * @param certIssuerName {String} The name of the PKI certificate issuer
   */
  function GetPKICertificate(certIssuerName) {
    _classCallCheck(this, GetPKICertificate);

    GetPKICertificate.initialize(this, certIssuerName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetPKICertificate, null, [{
    key: "initialize",
    value: function initialize(obj, certIssuerName) {
      obj['cert-issuer-name'] = certIssuerName;
    }
    /**
     * Constructs a <code>GetPKICertificate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetPKICertificate} obj Optional instance to populate.
     * @return {module:model/GetPKICertificate} The populated <code>GetPKICertificate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetPKICertificate();

        if (data.hasOwnProperty('alt-names')) {
          obj['alt-names'] = _ApiClient["default"].convertToType(data['alt-names'], 'String');
        }

        if (data.hasOwnProperty('cert-issuer-name')) {
          obj['cert-issuer-name'] = _ApiClient["default"].convertToType(data['cert-issuer-name'], 'String');
        }

        if (data.hasOwnProperty('common-name')) {
          obj['common-name'] = _ApiClient["default"].convertToType(data['common-name'], 'String');
        }

        if (data.hasOwnProperty('extended-key-usage')) {
          obj['extended-key-usage'] = _ApiClient["default"].convertToType(data['extended-key-usage'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('key-data-base64')) {
          obj['key-data-base64'] = _ApiClient["default"].convertToType(data['key-data-base64'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('uri-sans')) {
          obj['uri-sans'] = _ApiClient["default"].convertToType(data['uri-sans'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetPKICertificate;
}();
/**
 * The Subject Alternative Names to be included in the PKI certificate (in a comma-delimited list)
 * @member {String} alt-names
 */


GetPKICertificate.prototype['alt-names'] = undefined;
/**
 * The name of the PKI certificate issuer
 * @member {String} cert-issuer-name
 */

GetPKICertificate.prototype['cert-issuer-name'] = undefined;
/**
 * The common name to be included in the PKI certificate
 * @member {String} common-name
 */

GetPKICertificate.prototype['common-name'] = undefined;
/**
 * A comma-separated list of extended key usage requests which will be used for certificate issuance. Supported values: 'clientauth', 'serverauth'.
 * @member {String} extended-key-usage
 */

GetPKICertificate.prototype['extended-key-usage'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GetPKICertificate.prototype['json'] = undefined;
/**
 * PKI key file contents. If this option is used, the certificate will be printed to stdout
 * @member {String} key-data-base64
 */

GetPKICertificate.prototype['key-data-base64'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetPKICertificate.prototype['token'] = undefined;
/**
 * Updated certificate lifetime in seconds (must be less than the Certificate Issuer default TTL)
 * @member {Number} ttl
 */

GetPKICertificate.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetPKICertificate.prototype['uid-token'] = undefined;
/**
 * The URI Subject Alternative Names to be included in the PKI certificate (in a comma-delimited list)
 * @member {String} uri-sans
 */

GetPKICertificate.prototype['uri-sans'] = undefined;
var _default = GetPKICertificate;
exports["default"] = _default;
