// akeyless/dist/model/GetSSHCertificate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetSSHCertificate model module.
 * @module model/GetSSHCertificate
 * @version 2.20.0
 */
var GetSSHCertificate = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetSSHCertificate</code>.
   * @alias module:model/GetSSHCertificate
   * @param certIssuerName {String} The name of the SSH certificate issuer
   * @param certUsername {String} The username to sign in the SSH certificate
   */
  function GetSSHCertificate(certIssuerName, certUsername) {
    _classCallCheck(this, GetSSHCertificate);

    GetSSHCertificate.initialize(this, certIssuerName, certUsername);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetSSHCertificate, null, [{
    key: "initialize",
    value: function initialize(obj, certIssuerName, certUsername) {
      obj['cert-issuer-name'] = certIssuerName;
      obj['cert-username'] = certUsername;
    }
    /**
     * Constructs a <code>GetSSHCertificate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetSSHCertificate} obj Optional instance to populate.
     * @return {module:model/GetSSHCertificate} The populated <code>GetSSHCertificate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetSSHCertificate();

        if (data.hasOwnProperty('cert-issuer-name')) {
          obj['cert-issuer-name'] = _ApiClient["default"].convertToType(data['cert-issuer-name'], 'String');
        }

        if (data.hasOwnProperty('cert-username')) {
          obj['cert-username'] = _ApiClient["default"].convertToType(data['cert-username'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('legacy-signing-alg-name')) {
          obj['legacy-signing-alg-name'] = _ApiClient["default"].convertToType(data['legacy-signing-alg-name'], 'Boolean');
        }

        if (data.hasOwnProperty('public-key-data')) {
          obj['public-key-data'] = _ApiClient["default"].convertToType(data['public-key-data'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetSSHCertificate;
}();
/**
 * The name of the SSH certificate issuer
 * @member {String} cert-issuer-name
 */


GetSSHCertificate.prototype['cert-issuer-name'] = undefined;
/**
 * The username to sign in the SSH certificate
 * @member {String} cert-username
 */

GetSSHCertificate.prototype['cert-username'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

GetSSHCertificate.prototype['json'] = undefined;
/**
 * Set this option to output legacy ('ssh-rsa-cert-v01@openssh.com') signing algorithm name in the certificate.
 * @member {Boolean} legacy-signing-alg-name
 */

GetSSHCertificate.prototype['legacy-signing-alg-name'] = undefined;
/**
 * SSH public key file contents. If this option is used, the certificate will be printed to stdout
 * @member {String} public-key-data
 */

GetSSHCertificate.prototype['public-key-data'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetSSHCertificate.prototype['token'] = undefined;
/**
 * Updated certificate lifetime in seconds (must be less than the Certificate Issuer default TTL)
 * @member {Number} ttl
 */

GetSSHCertificate.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetSSHCertificate.prototype['uid-token'] = undefined;
var _default = GetSSHCertificate;
exports["default"] = _default;
