// akeyless/dist/model/GetTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GetTargetDetails model module.
 * @module model/GetTargetDetails
 * @version 2.15.24
 */
var GetTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetTargetDetails</code>.
   * @alias module:model/GetTargetDetails
   * @param name {String} Target name
   */
  function GetTargetDetails(name) {
    _classCallCheck(this, GetTargetDetails);

    GetTargetDetails.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GetTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>GetTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetTargetDetails} obj Optional instance to populate.
     * @return {module:model/GetTargetDetails} The populated <code>GetTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetTargetDetails();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('show-versions')) {
          obj['show-versions'] = _ApiClient["default"].convertToType(data['show-versions'], 'Boolean');
        }

        if (data.hasOwnProperty('target-version')) {
          obj['target-version'] = _ApiClient["default"].convertToType(data['target-version'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GetTargetDetails;
}();
/**
 * Target name
 * @member {String} name
 */


GetTargetDetails.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

GetTargetDetails.prototype['password'] = undefined;
/**
 * Include all target versions in reply
 * @member {Boolean} show-versions
 * @default false
 */

GetTargetDetails.prototype['show-versions'] = false;
/**
 * Target version
 * @member {Number} target-version
 */

GetTargetDetails.prototype['target-version'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

GetTargetDetails.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

GetTargetDetails.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

GetTargetDetails.prototype['username'] = undefined;
var _default = GetTargetDetails;
exports["default"] = _default;
