// akeyless/dist/model/GwClusterIdentity.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The GwClusterIdentity model module.
 * @module model/GwClusterIdentity
 * @version 2.20.0
 */
var GwClusterIdentity = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GwClusterIdentity</code>.
   * @alias module:model/GwClusterIdentity
   */
  function GwClusterIdentity() {
    _classCallCheck(this, GwClusterIdentity);

    GwClusterIdentity.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(GwClusterIdentity, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>GwClusterIdentity</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GwClusterIdentity} obj Optional instance to populate.
     * @return {module:model/GwClusterIdentity} The populated <code>GwClusterIdentity</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GwClusterIdentity();

        if (data.hasOwnProperty('allowed')) {
          obj['allowed'] = _ApiClient["default"].convertToType(data['allowed'], 'Boolean');
        }

        if (data.hasOwnProperty('allowed_access_ids')) {
          obj['allowed_access_ids'] = _ApiClient["default"].convertToType(data['allowed_access_ids'], ['String']);
        }

        if (data.hasOwnProperty('cluster_name')) {
          obj['cluster_name'] = _ApiClient["default"].convertToType(data['cluster_name'], 'String');
        }

        if (data.hasOwnProperty('cluster_url')) {
          obj['cluster_url'] = _ApiClient["default"].convertToType(data['cluster_url'], 'String');
        }

        if (data.hasOwnProperty('current_gw')) {
          obj['current_gw'] = _ApiClient["default"].convertToType(data['current_gw'], 'Boolean');
        }

        if (data.hasOwnProperty('customer_fragment_ids')) {
          obj['customer_fragment_ids'] = _ApiClient["default"].convertToType(data['customer_fragment_ids'], ['String']);
        }

        if (data.hasOwnProperty('default_protection_key_id')) {
          obj['default_protection_key_id'] = _ApiClient["default"].convertToType(data['default_protection_key_id'], 'Number');
        }

        if (data.hasOwnProperty('default_secret_location')) {
          obj['default_secret_location'] = _ApiClient["default"].convertToType(data['default_secret_location'], 'String');
        }

        if (data.hasOwnProperty('display_name')) {
          obj['display_name'] = _ApiClient["default"].convertToType(data['display_name'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }

        if (data.hasOwnProperty('status_description')) {
          obj['status_description'] = _ApiClient["default"].convertToType(data['status_description'], 'String');
        }
      }

      return obj;
    }
  }]);

  return GwClusterIdentity;
}();
/**
 * @member {Boolean} allowed
 */


GwClusterIdentity.prototype['allowed'] = undefined;
/**
 * @member {Array.<String>} allowed_access_ids
 */

GwClusterIdentity.prototype['allowed_access_ids'] = undefined;
/**
 * @member {String} cluster_name
 */

GwClusterIdentity.prototype['cluster_name'] = undefined;
/**
 * @member {String} cluster_url
 */

GwClusterIdentity.prototype['cluster_url'] = undefined;
/**
 * @member {Boolean} current_gw
 */

GwClusterIdentity.prototype['current_gw'] = undefined;
/**
 * @member {Array.<String>} customer_fragment_ids
 */

GwClusterIdentity.prototype['customer_fragment_ids'] = undefined;
/**
 * @member {Number} default_protection_key_id
 */

GwClusterIdentity.prototype['default_protection_key_id'] = undefined;
/**
 * @member {String} default_secret_location
 */

GwClusterIdentity.prototype['default_secret_location'] = undefined;
/**
 * @member {String} display_name
 */

GwClusterIdentity.prototype['display_name'] = undefined;
/**
 * @member {Number} id
 */

GwClusterIdentity.prototype['id'] = undefined;
/**
 * @member {String} status
 */

GwClusterIdentity.prototype['status'] = undefined;
/**
 * @member {String} status_description
 */

GwClusterIdentity.prototype['status_description'] = undefined;
var _default = GwClusterIdentity;
exports["default"] = _default;
