// akeyless/dist/model/HashiPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The HashiPayload model module.
 * @module model/HashiPayload
 * @version 2.20.0
 */
var HashiPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HashiPayload</code>.
   * @alias module:model/HashiPayload
   */
  function HashiPayload() {
    _classCallCheck(this, HashiPayload);

    HashiPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(HashiPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>HashiPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HashiPayload} obj Optional instance to populate.
     * @return {module:model/HashiPayload} The populated <code>HashiPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HashiPayload();

        if (data.hasOwnProperty('import_as_json')) {
          obj['import_as_json'] = _ApiClient["default"].convertToType(data['import_as_json'], 'Boolean');
        }

        if (data.hasOwnProperty('namespaces')) {
          obj['namespaces'] = _ApiClient["default"].convertToType(data['namespaces'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }
      }

      return obj;
    }
  }]);

  return HashiPayload;
}();
/**
 * @member {Boolean} import_as_json
 */


HashiPayload.prototype['import_as_json'] = undefined;
/**
 * @member {Array.<String>} namespaces
 */

HashiPayload.prototype['namespaces'] = undefined;
/**
 * @member {String} token
 */

HashiPayload.prototype['token'] = undefined;
/**
 * @member {String} url
 */

HashiPayload.prototype['url'] = undefined;
var _default = HashiPayload;
exports["default"] = _default;
