// akeyless/dist/model/HuaweiAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The HuaweiAccessRules model module.
 * @module model/HuaweiAccessRules
 * @version 2.20.0
 */
var HuaweiAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HuaweiAccessRules</code>.
   * @alias module:model/HuaweiAccessRules
   */
  function HuaweiAccessRules() {
    _classCallCheck(this, HuaweiAccessRules);

    HuaweiAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(HuaweiAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>HuaweiAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HuaweiAccessRules} obj Optional instance to populate.
     * @return {module:model/HuaweiAccessRules} The populated <code>HuaweiAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HuaweiAccessRules();

        if (data.hasOwnProperty('auth_endpoint')) {
          obj['auth_endpoint'] = _ApiClient["default"].convertToType(data['auth_endpoint'], 'String');
        }

        if (data.hasOwnProperty('domain_id')) {
          obj['domain_id'] = _ApiClient["default"].convertToType(data['domain_id'], ['String']);
        }

        if (data.hasOwnProperty('domain_name')) {
          obj['domain_name'] = _ApiClient["default"].convertToType(data['domain_name'], ['String']);
        }

        if (data.hasOwnProperty('tenant_id')) {
          obj['tenant_id'] = _ApiClient["default"].convertToType(data['tenant_id'], ['String']);
        }

        if (data.hasOwnProperty('tenant_name')) {
          obj['tenant_name'] = _ApiClient["default"].convertToType(data['tenant_name'], ['String']);
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], ['String']);
        }

        if (data.hasOwnProperty('user_name')) {
          obj['user_name'] = _ApiClient["default"].convertToType(data['user_name'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return HuaweiAccessRules;
}();
/**
 * The auth URL.
 * @member {String} auth_endpoint
 */


HuaweiAccessRules.prototype['auth_endpoint'] = undefined;
/**
 * The list of domain ids that the login is restricted to.
 * @member {Array.<String>} domain_id
 */

HuaweiAccessRules.prototype['domain_id'] = undefined;
/**
 * The list of domainNames that the login is restricted to.
 * @member {Array.<String>} domain_name
 */

HuaweiAccessRules.prototype['domain_name'] = undefined;
/**
 * The list of tenantIDs  that the login is restricted to.
 * @member {Array.<String>} tenant_id
 */

HuaweiAccessRules.prototype['tenant_id'] = undefined;
/**
 * The list of tenantNames  that the login is restricted to.
 * @member {Array.<String>} tenant_name
 */

HuaweiAccessRules.prototype['tenant_name'] = undefined;
/**
 * The list of user ids that the login is restricted to.
 * @member {Array.<String>} user_id
 */

HuaweiAccessRules.prototype['user_id'] = undefined;
/**
 * The list of user names that the login is restricted to.
 * @member {Array.<String>} user_name
 */

HuaweiAccessRules.prototype['user_name'] = undefined;
var _default = HuaweiAccessRules;
exports["default"] = _default;
