// akeyless/dist/model/ItemGeneralInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateIssueInfo = _interopRequireDefault(require("./CertificateIssueInfo"));

var _ClassicKeyDetailsInfo = _interopRequireDefault(require("./ClassicKeyDetailsInfo"));

var _DynamicSecretProducerInfo = _interopRequireDefault(require("./DynamicSecretProducerInfo"));

var _RotatedSecretDetailsInfo = _interopRequireDefault(require("./RotatedSecretDetailsInfo"));

var _SecureRemoteAccess = _interopRequireDefault(require("./SecureRemoteAccess"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemGeneralInfo model module.
 * @module model/ItemGeneralInfo
 * @version 2.15.24
 */
var ItemGeneralInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemGeneralInfo</code>.
   * @alias module:model/ItemGeneralInfo
   */
  function ItemGeneralInfo() {
    _classCallCheck(this, ItemGeneralInfo);

    ItemGeneralInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemGeneralInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemGeneralInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemGeneralInfo} obj Optional instance to populate.
     * @return {module:model/ItemGeneralInfo} The populated <code>ItemGeneralInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemGeneralInfo();

        if (data.hasOwnProperty('cert_issue_details')) {
          obj['cert_issue_details'] = _CertificateIssueInfo["default"].constructFromObject(data['cert_issue_details']);
        }

        if (data.hasOwnProperty('classic_key_details')) {
          obj['classic_key_details'] = _ClassicKeyDetailsInfo["default"].constructFromObject(data['classic_key_details']);
        }

        if (data.hasOwnProperty('dynamic_secret_producer_details')) {
          obj['dynamic_secret_producer_details'] = _DynamicSecretProducerInfo["default"].constructFromObject(data['dynamic_secret_producer_details']);
        }

        if (data.hasOwnProperty('rotated_secret_details')) {
          obj['rotated_secret_details'] = _RotatedSecretDetailsInfo["default"].constructFromObject(data['rotated_secret_details']);
        }

        if (data.hasOwnProperty('secure_remote_access_details')) {
          obj['secure_remote_access_details'] = _SecureRemoteAccess["default"].constructFromObject(data['secure_remote_access_details']);
        }
      }

      return obj;
    }
  }]);

  return ItemGeneralInfo;
}();
/**
 * @member {module:model/CertificateIssueInfo} cert_issue_details
 */


ItemGeneralInfo.prototype['cert_issue_details'] = undefined;
/**
 * @member {module:model/ClassicKeyDetailsInfo} classic_key_details
 */

ItemGeneralInfo.prototype['classic_key_details'] = undefined;
/**
 * @member {module:model/DynamicSecretProducerInfo} dynamic_secret_producer_details
 */

ItemGeneralInfo.prototype['dynamic_secret_producer_details'] = undefined;
/**
 * @member {module:model/RotatedSecretDetailsInfo} rotated_secret_details
 */

ItemGeneralInfo.prototype['rotated_secret_details'] = undefined;
/**
 * @member {module:model/SecureRemoteAccess} secure_remote_access_details
 */

ItemGeneralInfo.prototype['secure_remote_access_details'] = undefined;
var _default = ItemGeneralInfo;
exports["default"] = _default;
