// akeyless/dist/model/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _CertificateIssueInfo = _interopRequireDefault(require("./CertificateIssueInfo"));

var _ItemGeneralInfo = _interopRequireDefault(require("./ItemGeneralInfo"));

var _ItemTargetAssociation = _interopRequireDefault(require("./ItemTargetAssociation"));

var _ItemVersion = _interopRequireDefault(require("./ItemVersion"));

var _RuleAssigner = _interopRequireDefault(require("./RuleAssigner"));

var _TargetItemVersion = _interopRequireDefault(require("./TargetItemVersion"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Item model module.
 * @module model/Item
 * @version 2.20.0
 */
var Item = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Item</code>.
   * @alias module:model/Item
   */
  function Item() {
    _classCallCheck(this, Item);

    Item.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Item, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Item</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Item} obj Optional instance to populate.
     * @return {module:model/Item} The populated <code>Item</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Item();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('auto_rotate')) {
          obj['auto_rotate'] = _ApiClient["default"].convertToType(data['auto_rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('cert_issuer_signer_key_name')) {
          obj['cert_issuer_signer_key_name'] = _ApiClient["default"].convertToType(data['cert_issuer_signer_key_name'], 'String');
        }

        if (data.hasOwnProperty('certificate_issue_details')) {
          obj['certificate_issue_details'] = _CertificateIssueInfo["default"].constructFromObject(data['certificate_issue_details']);
        }

        if (data.hasOwnProperty('certificates')) {
          obj['certificates'] = _ApiClient["default"].convertToType(data['certificates'], 'String');
        }

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('customer_fragment_id')) {
          obj['customer_fragment_id'] = _ApiClient["default"].convertToType(data['customer_fragment_id'], 'String');
        }

        if (data.hasOwnProperty('delete_protection')) {
          obj['delete_protection'] = _ApiClient["default"].convertToType(data['delete_protection'], 'Boolean');
        }

        if (data.hasOwnProperty('deletion_date')) {
          obj['deletion_date'] = _ApiClient["default"].convertToType(data['deletion_date'], 'Date');
        }

        if (data.hasOwnProperty('display_id')) {
          obj['display_id'] = _ApiClient["default"].convertToType(data['display_id'], 'String');
        }

        if (data.hasOwnProperty('is_enabled')) {
          obj['is_enabled'] = _ApiClient["default"].convertToType(data['is_enabled'], 'Boolean');
        }

        if (data.hasOwnProperty('item_accessibility')) {
          obj['item_accessibility'] = _ApiClient["default"].convertToType(data['item_accessibility'], 'Number');
        }

        if (data.hasOwnProperty('item_general_info')) {
          obj['item_general_info'] = _ItemGeneralInfo["default"].constructFromObject(data['item_general_info']);
        }

        if (data.hasOwnProperty('item_id')) {
          obj['item_id'] = _ApiClient["default"].convertToType(data['item_id'], 'Number');
        }

        if (data.hasOwnProperty('item_metadata')) {
          obj['item_metadata'] = _ApiClient["default"].convertToType(data['item_metadata'], 'String');
        }

        if (data.hasOwnProperty('item_name')) {
          obj['item_name'] = _ApiClient["default"].convertToType(data['item_name'], 'String');
        }

        if (data.hasOwnProperty('item_size')) {
          obj['item_size'] = _ApiClient["default"].convertToType(data['item_size'], 'Number');
        }

        if (data.hasOwnProperty('item_state')) {
          obj['item_state'] = _ApiClient["default"].convertToType(data['item_state'], 'String');
        }

        if (data.hasOwnProperty('item_sub_type')) {
          obj['item_sub_type'] = _ApiClient["default"].convertToType(data['item_sub_type'], 'String');
        }

        if (data.hasOwnProperty('item_tags')) {
          obj['item_tags'] = _ApiClient["default"].convertToType(data['item_tags'], ['String']);
        }

        if (data.hasOwnProperty('item_targets_assoc')) {
          obj['item_targets_assoc'] = _ApiClient["default"].convertToType(data['item_targets_assoc'], [_ItemTargetAssociation["default"]]);
        }

        if (data.hasOwnProperty('item_type')) {
          obj['item_type'] = _ApiClient["default"].convertToType(data['item_type'], 'String');
        }

        if (data.hasOwnProperty('item_versions')) {
          obj['item_versions'] = _ApiClient["default"].convertToType(data['item_versions'], [_ItemVersion["default"]]);
        }

        if (data.hasOwnProperty('last_version')) {
          obj['last_version'] = _ApiClient["default"].convertToType(data['last_version'], 'Number');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('next_rotation_date')) {
          obj['next_rotation_date'] = _ApiClient["default"].convertToType(data['next_rotation_date'], 'Date');
        }

        if (data.hasOwnProperty('protection_key_name')) {
          obj['protection_key_name'] = _ApiClient["default"].convertToType(data['protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('protection_key_type')) {
          obj['protection_key_type'] = _ApiClient["default"].convertToType(data['protection_key_type'], 'String');
        }

        if (data.hasOwnProperty('public_value')) {
          obj['public_value'] = _ApiClient["default"].convertToType(data['public_value'], 'String');
        }

        if (data.hasOwnProperty('rotation_interval')) {
          obj['rotation_interval'] = _ApiClient["default"].convertToType(data['rotation_interval'], 'Number');
        }

        if (data.hasOwnProperty('shared_by')) {
          obj['shared_by'] = _RuleAssigner["default"].constructFromObject(data['shared_by']);
        }

        if (data.hasOwnProperty('target_versions')) {
          obj['target_versions'] = _ApiClient["default"].convertToType(data['target_versions'], [_TargetItemVersion["default"]]);
        }

        if (data.hasOwnProperty('with_customer_fragment')) {
          obj['with_customer_fragment'] = _ApiClient["default"].convertToType(data['with_customer_fragment'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return Item;
}();
/**
 * @member {Date} access_date
 */


Item.prototype['access_date'] = undefined;
/**
 * @member {Boolean} auto_rotate
 */

Item.prototype['auto_rotate'] = undefined;
/**
 * @member {String} cert_issuer_signer_key_name
 */

Item.prototype['cert_issuer_signer_key_name'] = undefined;
/**
 * @member {module:model/CertificateIssueInfo} certificate_issue_details
 */

Item.prototype['certificate_issue_details'] = undefined;
/**
 * @member {String} certificates
 */

Item.prototype['certificates'] = undefined;
/**
 * @member {Array.<String>} client_permissions
 */

Item.prototype['client_permissions'] = undefined;
/**
 * @member {Date} creation_date
 */

Item.prototype['creation_date'] = undefined;
/**
 * @member {String} customer_fragment_id
 */

Item.prototype['customer_fragment_id'] = undefined;
/**
 * @member {Boolean} delete_protection
 */

Item.prototype['delete_protection'] = undefined;
/**
 * @member {Date} deletion_date
 */

Item.prototype['deletion_date'] = undefined;
/**
 * @member {String} display_id
 */

Item.prototype['display_id'] = undefined;
/**
 * @member {Boolean} is_enabled
 */

Item.prototype['is_enabled'] = undefined;
/**
 * @member {Number} item_accessibility
 */

Item.prototype['item_accessibility'] = undefined;
/**
 * @member {module:model/ItemGeneralInfo} item_general_info
 */

Item.prototype['item_general_info'] = undefined;
/**
 * @member {Number} item_id
 */

Item.prototype['item_id'] = undefined;
/**
 * @member {String} item_metadata
 */

Item.prototype['item_metadata'] = undefined;
/**
 * @member {String} item_name
 */

Item.prototype['item_name'] = undefined;
/**
 * @member {Number} item_size
 */

Item.prototype['item_size'] = undefined;
/**
 * ItemState defines the different states an Item can be in
 * @member {String} item_state
 */

Item.prototype['item_state'] = undefined;
/**
 * @member {String} item_sub_type
 */

Item.prototype['item_sub_type'] = undefined;
/**
 * @member {Array.<String>} item_tags
 */

Item.prototype['item_tags'] = undefined;
/**
 * @member {Array.<module:model/ItemTargetAssociation>} item_targets_assoc
 */

Item.prototype['item_targets_assoc'] = undefined;
/**
 * @member {String} item_type
 */

Item.prototype['item_type'] = undefined;
/**
 * @member {Array.<module:model/ItemVersion>} item_versions
 */

Item.prototype['item_versions'] = undefined;
/**
 * @member {Number} last_version
 */

Item.prototype['last_version'] = undefined;
/**
 * @member {Date} modification_date
 */

Item.prototype['modification_date'] = undefined;
/**
 * @member {Date} next_rotation_date
 */

Item.prototype['next_rotation_date'] = undefined;
/**
 * @member {String} protection_key_name
 */

Item.prototype['protection_key_name'] = undefined;
/**
 * @member {String} protection_key_type
 */

Item.prototype['protection_key_type'] = undefined;
/**
 * @member {String} public_value
 */

Item.prototype['public_value'] = undefined;
/**
 * @member {Number} rotation_interval
 */

Item.prototype['rotation_interval'] = undefined;
/**
 * @member {module:model/RuleAssigner} shared_by
 */

Item.prototype['shared_by'] = undefined;
/**
 * @member {Array.<module:model/TargetItemVersion>} target_versions
 */

Item.prototype['target_versions'] = undefined;
/**
 * @member {Boolean} with_customer_fragment
 */

Item.prototype['with_customer_fragment'] = undefined;
var _default = Item;
exports["default"] = _default;
