// akeyless/dist/model/ItemVersion.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemVersion model module.
 * @module model/ItemVersion
 * @version 2.20.0
 */
var ItemVersion = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemVersion</code>.
   * @alias module:model/ItemVersion
   */
  function ItemVersion() {
    _classCallCheck(this, ItemVersion);

    ItemVersion.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemVersion, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemVersion</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemVersion} obj Optional instance to populate.
     * @return {module:model/ItemVersion} The populated <code>ItemVersion</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemVersion();

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('customer_fragment_id')) {
          obj['customer_fragment_id'] = _ApiClient["default"].convertToType(data['customer_fragment_id'], 'String');
        }

        if (data.hasOwnProperty('deletion_date')) {
          obj['deletion_date'] = _ApiClient["default"].convertToType(data['deletion_date'], 'Date');
        }

        if (data.hasOwnProperty('item_version_state')) {
          obj['item_version_state'] = _ApiClient["default"].convertToType(data['item_version_state'], 'String');
        }

        if (data.hasOwnProperty('protection_key_name')) {
          obj['protection_key_name'] = _ApiClient["default"].convertToType(data['protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }

        if (data.hasOwnProperty('with_customer_fragment')) {
          obj['with_customer_fragment'] = _ApiClient["default"].convertToType(data['with_customer_fragment'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return ItemVersion;
}();
/**
 * @member {Date} creation_date
 */


ItemVersion.prototype['creation_date'] = undefined;
/**
 * @member {String} customer_fragment_id
 */

ItemVersion.prototype['customer_fragment_id'] = undefined;
/**
 * @member {Date} deletion_date
 */

ItemVersion.prototype['deletion_date'] = undefined;
/**
 * ItemState defines the different states an Item can be in
 * @member {String} item_version_state
 */

ItemVersion.prototype['item_version_state'] = undefined;
/**
 * @member {String} protection_key_name
 */

ItemVersion.prototype['protection_key_name'] = undefined;
/**
 * @member {Number} version
 */

ItemVersion.prototype['version'] = undefined;
/**
 * @member {Boolean} with_customer_fragment
 */

ItemVersion.prototype['with_customer_fragment'] = undefined;
var _default = ItemVersion;
exports["default"] = _default;
