// akeyless/dist/model/ItemTargetAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ItemTargetAssociation model module.
 * @module model/ItemTargetAssociation
 * @version 2.20.0
 */
var ItemTargetAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ItemTargetAssociation</code>.
   * ItemTargetAssociation includes details of an association between an item and a target.
   * @alias module:model/ItemTargetAssociation
   */
  function ItemTargetAssociation() {
    _classCallCheck(this, ItemTargetAssociation);

    ItemTargetAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ItemTargetAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ItemTargetAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ItemTargetAssociation} obj Optional instance to populate.
     * @return {module:model/ItemTargetAssociation} The populated <code>ItemTargetAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ItemTargetAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('target_id')) {
          obj['target_id'] = _ApiClient["default"].convertToType(data['target_id'], 'Number');
        }

        if (data.hasOwnProperty('target_name')) {
          obj['target_name'] = _ApiClient["default"].convertToType(data['target_name'], 'String');
        }

        if (data.hasOwnProperty('target_type')) {
          obj['target_type'] = _ApiClient["default"].convertToType(data['target_type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ItemTargetAssociation;
}();
/**
 * @member {String} assoc_id
 */


ItemTargetAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {Number} target_id
 */

ItemTargetAssociation.prototype['target_id'] = undefined;
/**
 * @member {String} target_name
 */

ItemTargetAssociation.prototype['target_name'] = undefined;
/**
 * @member {String} target_type
 */

ItemTargetAssociation.prototype['target_type'] = undefined;
var _default = ItemTargetAssociation;
exports["default"] = _default;
