// akeyless/dist/model/K8SAuthsConfigLastChange.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The K8SAuthsConfigLastChange model module.
 * @module model/K8SAuthsConfigLastChange
 * @version 2.20.0
 */
var K8SAuthsConfigLastChange = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>K8SAuthsConfigLastChange</code>.
   * @alias module:model/K8SAuthsConfigLastChange
   */
  function K8SAuthsConfigLastChange() {
    _classCallCheck(this, K8SAuthsConfigLastChange);

    K8SAuthsConfigLastChange.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(K8SAuthsConfigLastChange, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>K8SAuthsConfigLastChange</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/K8SAuthsConfigLastChange} obj Optional instance to populate.
     * @return {module:model/K8SAuthsConfigLastChange} The populated <code>K8SAuthsConfigLastChange</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new K8SAuthsConfigLastChange();

        if (data.hasOwnProperty('changed_k8s_auths_ids')) {
          obj['changed_k8s_auths_ids'] = _ApiClient["default"].convertToType(data['changed_k8s_auths_ids'], ['String']);
        }

        if (data.hasOwnProperty('created_k8s_auths_ids')) {
          obj['created_k8s_auths_ids'] = _ApiClient["default"].convertToType(data['created_k8s_auths_ids'], ['String']);
        }

        if (data.hasOwnProperty('deleted_k8s_auths_ids')) {
          obj['deleted_k8s_auths_ids'] = _ApiClient["default"].convertToType(data['deleted_k8s_auths_ids'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return K8SAuthsConfigLastChange;
}();
/**
 * @member {Array.<String>} changed_k8s_auths_ids
 */


K8SAuthsConfigLastChange.prototype['changed_k8s_auths_ids'] = undefined;
/**
 * @member {Array.<String>} created_k8s_auths_ids
 */

K8SAuthsConfigLastChange.prototype['created_k8s_auths_ids'] = undefined;
/**
 * @member {Array.<String>} deleted_k8s_auths_ids
 */

K8SAuthsConfigLastChange.prototype['deleted_k8s_auths_ids'] = undefined;
var _default = K8SAuthsConfigLastChange;
exports["default"] = _default;
