// akeyless/dist/model/K8SPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The K8SPayload model module.
 * @module model/K8SPayload
 * @version 2.15.24
 */
var K8SPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>K8SPayload</code>.
   * @alias module:model/K8SPayload
   */
  function K8SPayload() {
    _classCallCheck(this, K8SPayload);

    K8SPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(K8SPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>K8SPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/K8SPayload} obj Optional instance to populate.
     * @return {module:model/K8SPayload} The populated <code>K8SPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new K8SPayload();

        if (data.hasOwnProperty('ca')) {
          obj['ca'] = _ApiClient["default"].convertToType(data['ca'], ['Number']);
        }

        if (data.hasOwnProperty('client_cert')) {
          obj['client_cert'] = _ApiClient["default"].convertToType(data['client_cert'], ['Number']);
        }

        if (data.hasOwnProperty('client_key')) {
          obj['client_key'] = _ApiClient["default"].convertToType(data['client_key'], ['Number']);
        }

        if (data.hasOwnProperty('namespace')) {
          obj['namespace'] = _ApiClient["default"].convertToType(data['namespace'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('server')) {
          obj['server'] = _ApiClient["default"].convertToType(data['server'], 'String');
        }

        if (data.hasOwnProperty('skip_system')) {
          obj['skip_system'] = _ApiClient["default"].convertToType(data['skip_system'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return K8SPayload;
}();
/**
 * @member {Array.<Number>} ca
 */


K8SPayload.prototype['ca'] = undefined;
/**
 * @member {Array.<Number>} client_cert
 */

K8SPayload.prototype['client_cert'] = undefined;
/**
 * @member {Array.<Number>} client_key
 */

K8SPayload.prototype['client_key'] = undefined;
/**
 * @member {String} namespace
 */

K8SPayload.prototype['namespace'] = undefined;
/**
 * @member {String} password
 */

K8SPayload.prototype['password'] = undefined;
/**
 * @member {String} server
 */

K8SPayload.prototype['server'] = undefined;
/**
 * @member {Boolean} skip_system
 */

K8SPayload.prototype['skip_system'] = undefined;
/**
 * @member {String} token
 */

K8SPayload.prototype['token'] = undefined;
/**
 * @member {String} username
 */

K8SPayload.prototype['username'] = undefined;
var _default = K8SPayload;
exports["default"] = _default;
