// akeyless/dist/model/KmipClientSetRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipClientSetRule model module.
 * @module model/KmipClientSetRule
 * @version 2.20.0
 */
var KmipClientSetRule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipClientSetRule</code>.
   * @alias module:model/KmipClientSetRule
   * @param capability {Array.<String>} Access capabilities
   * @param path {String} Access path
   */
  function KmipClientSetRule(capability, path) {
    _classCallCheck(this, KmipClientSetRule);

    KmipClientSetRule.initialize(this, capability, path);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipClientSetRule, null, [{
    key: "initialize",
    value: function initialize(obj, capability, path) {
      obj['capability'] = capability;
      obj['path'] = path;
    }
    /**
     * Constructs a <code>KmipClientSetRule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipClientSetRule} obj Optional instance to populate.
     * @return {module:model/KmipClientSetRule} The populated <code>KmipClientSetRule</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipClientSetRule();

        if (data.hasOwnProperty('capability')) {
          obj['capability'] = _ApiClient["default"].convertToType(data['capability'], ['String']);
        }

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KmipClientSetRule;
}();
/**
 * Access capabilities
 * @member {Array.<String>} capability
 */


KmipClientSetRule.prototype['capability'] = undefined;
/**
 * @member {String} client-id
 */

KmipClientSetRule.prototype['client-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

KmipClientSetRule.prototype['json'] = undefined;
/**
 * @member {String} name
 */

KmipClientSetRule.prototype['name'] = undefined;
/**
 * Access path
 * @member {String} path
 */

KmipClientSetRule.prototype['path'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

KmipClientSetRule.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

KmipClientSetRule.prototype['uid-token'] = undefined;
var _default = KmipClientSetRule;
exports["default"] = _default;
