// akeyless/dist/model/KmipDeleteClient.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipDeleteClient model module.
 * @module model/KmipDeleteClient
 * @version 2.20.0
 */
var KmipDeleteClient = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipDeleteClient</code>.
   * @alias module:model/KmipDeleteClient
   */
  function KmipDeleteClient() {
    _classCallCheck(this, KmipDeleteClient);

    KmipDeleteClient.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipDeleteClient, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KmipDeleteClient</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipDeleteClient} obj Optional instance to populate.
     * @return {module:model/KmipDeleteClient} The populated <code>KmipDeleteClient</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipDeleteClient();

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KmipDeleteClient;
}();
/**
 * @member {String} client-id
 */


KmipDeleteClient.prototype['client-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

KmipDeleteClient.prototype['json'] = undefined;
/**
 * @member {String} name
 */

KmipDeleteClient.prototype['name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

KmipDeleteClient.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

KmipDeleteClient.prototype['uid-token'] = undefined;
var _default = KmipDeleteClient;
exports["default"] = _default;
