// akeyless/dist/model/KmipDescribeServerOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipDescribeServerOutput model module.
 * @module model/KmipDescribeServerOutput
 * @version 2.15.24
 */
var KmipDescribeServerOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipDescribeServerOutput</code>.
   * @alias module:model/KmipDescribeServerOutput
   */
  function KmipDescribeServerOutput() {
    _classCallCheck(this, KmipDescribeServerOutput);

    KmipDescribeServerOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipDescribeServerOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KmipDescribeServerOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipDescribeServerOutput} obj Optional instance to populate.
     * @return {module:model/KmipDescribeServerOutput} The populated <code>KmipDescribeServerOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipDescribeServerOutput();

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }

        if (data.hasOwnProperty('ca_cert')) {
          obj['ca_cert'] = _ApiClient["default"].convertToType(data['ca_cert'], ['Number']);
        }

        if (data.hasOwnProperty('hostname')) {
          obj['hostname'] = _ApiClient["default"].convertToType(data['hostname'], 'String');
        }

        if (data.hasOwnProperty('root')) {
          obj['root'] = _ApiClient["default"].convertToType(data['root'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KmipDescribeServerOutput;
}();
/**
 * @member {Boolean} active
 */


KmipDescribeServerOutput.prototype['active'] = undefined;
/**
 * @member {Array.<Number>} ca_cert
 */

KmipDescribeServerOutput.prototype['ca_cert'] = undefined;
/**
 * @member {String} hostname
 */

KmipDescribeServerOutput.prototype['hostname'] = undefined;
/**
 * @member {String} root
 */

KmipDescribeServerOutput.prototype['root'] = undefined;
var _default = KmipDescribeServerOutput;
exports["default"] = _default;
