// akeyless/dist/model/KmipServerSetup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipServerSetup model module.
 * @module model/KmipServerSetup
 * @version 2.15.24
 */
var KmipServerSetup = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipServerSetup</code>.
   * @alias module:model/KmipServerSetup
   * @param hostname {String} Hostname
   */
  function KmipServerSetup(hostname) {
    _classCallCheck(this, KmipServerSetup);

    KmipServerSetup.initialize(this, hostname);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipServerSetup, null, [{
    key: "initialize",
    value: function initialize(obj, hostname) {
      obj['hostname'] = hostname;
    }
    /**
     * Constructs a <code>KmipServerSetup</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipServerSetup} obj Optional instance to populate.
     * @return {module:model/KmipServerSetup} The populated <code>KmipServerSetup</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipServerSetup();

        if (data.hasOwnProperty('certificate-ttl')) {
          obj['certificate-ttl'] = _ApiClient["default"].convertToType(data['certificate-ttl'], 'Number');
        }

        if (data.hasOwnProperty('hostname')) {
          obj['hostname'] = _ApiClient["default"].convertToType(data['hostname'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('root')) {
          obj['root'] = _ApiClient["default"].convertToType(data['root'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KmipServerSetup;
}();
/**
 * @member {Number} certificate-ttl
 */


KmipServerSetup.prototype['certificate-ttl'] = undefined;
/**
 * Hostname
 * @member {String} hostname
 */

KmipServerSetup.prototype['hostname'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

KmipServerSetup.prototype['password'] = undefined;
/**
 * @member {String} root
 */

KmipServerSetup.prototype['root'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

KmipServerSetup.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

KmipServerSetup.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

KmipServerSetup.prototype['username'] = undefined;
var _default = KmipServerSetup;
exports["default"] = _default;
