// akeyless/dist/model/KmipSetServerState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KmipSetServerState model module.
 * @module model/KmipSetServerState
 * @version 2.20.0
 */
var KmipSetServerState = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KmipSetServerState</code>.
   * kmipSetServerState is a command that sets the environment state to active/inactive.
   * @alias module:model/KmipSetServerState
   * @param state {String} 
   */
  function KmipSetServerState(state) {
    _classCallCheck(this, KmipSetServerState);

    KmipSetServerState.initialize(this, state);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KmipSetServerState, null, [{
    key: "initialize",
    value: function initialize(obj, state) {
      obj['state'] = state;
    }
    /**
     * Constructs a <code>KmipSetServerState</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KmipSetServerState} obj Optional instance to populate.
     * @return {module:model/KmipSetServerState} The populated <code>KmipSetServerState</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KmipSetServerState();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('state')) {
          obj['state'] = _ApiClient["default"].convertToType(data['state'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KmipSetServerState;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 */


KmipSetServerState.prototype['json'] = undefined;
/**
 * @member {String} state
 */

KmipSetServerState.prototype['state'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

KmipSetServerState.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

KmipSetServerState.prototype['uid-token'] = undefined;
var _default = KmipSetServerState;
exports["default"] = _default;
