// akeyless/dist/model/KubernetesAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The KubernetesAccessRules model module.
 * @module model/KubernetesAccessRules
 * @version 2.15.24
 */
var KubernetesAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>KubernetesAccessRules</code>.
   * @alias module:model/KubernetesAccessRules
   */
  function KubernetesAccessRules() {
    _classCallCheck(this, KubernetesAccessRules);

    KubernetesAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(KubernetesAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>KubernetesAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/KubernetesAccessRules} obj Optional instance to populate.
     * @return {module:model/KubernetesAccessRules} The populated <code>KubernetesAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new KubernetesAccessRules();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('audience')) {
          obj['audience'] = _ApiClient["default"].convertToType(data['audience'], 'String');
        }

        if (data.hasOwnProperty('bound_namespaces')) {
          obj['bound_namespaces'] = _ApiClient["default"].convertToType(data['bound_namespaces'], ['String']);
        }

        if (data.hasOwnProperty('bound_pod_names')) {
          obj['bound_pod_names'] = _ApiClient["default"].convertToType(data['bound_pod_names'], ['String']);
        }

        if (data.hasOwnProperty('bound_service_account_names')) {
          obj['bound_service_account_names'] = _ApiClient["default"].convertToType(data['bound_service_account_names'], ['String']);
        }

        if (data.hasOwnProperty('pub_key')) {
          obj['pub_key'] = _ApiClient["default"].convertToType(data['pub_key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return KubernetesAccessRules;
}();
/**
 * @member {String} alg
 */


KubernetesAccessRules.prototype['alg'] = undefined;
/**
 * Audience is an optional Kubernetes jwt claim to verify
 * @member {String} audience
 */

KubernetesAccessRules.prototype['audience'] = undefined;
/**
 * A list of namespaces that the authentication is restricted to.
 * @member {Array.<String>} bound_namespaces
 */

KubernetesAccessRules.prototype['bound_namespaces'] = undefined;
/**
 * A list of pods names that the authentication is restricted to.
 * @member {Array.<String>} bound_pod_names
 */

KubernetesAccessRules.prototype['bound_pod_names'] = undefined;
/**
 * A list of service account names that the authentication is restricted to.
 * @member {Array.<String>} bound_service_account_names
 */

KubernetesAccessRules.prototype['bound_service_account_names'] = undefined;
/**
 * The public key value of the Kubernetes auth method configuration in the Akeyless Gateway.
 * @member {String} pub_key
 */

KubernetesAccessRules.prototype['pub_key'] = undefined;
var _default = KubernetesAccessRules;
exports["default"] = _default;
