// akeyless/dist/model/ListItemsInPathOutput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Item = _interopRequireDefault(require("./Item"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListItemsInPathOutput model module.
 * @module model/ListItemsInPathOutput
 * @version 2.15.24
 */
var ListItemsInPathOutput = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListItemsInPathOutput</code>.
   * @alias module:model/ListItemsInPathOutput
   */
  function ListItemsInPathOutput() {
    _classCallCheck(this, ListItemsInPathOutput);

    ListItemsInPathOutput.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListItemsInPathOutput, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListItemsInPathOutput</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListItemsInPathOutput} obj Optional instance to populate.
     * @return {module:model/ListItemsInPathOutput} The populated <code>ListItemsInPathOutput</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListItemsInPathOutput();

        if (data.hasOwnProperty('folders')) {
          obj['folders'] = _ApiClient["default"].convertToType(data['folders'], ['String']);
        }

        if (data.hasOwnProperty('items')) {
          obj['items'] = _ApiClient["default"].convertToType(data['items'], [_Item["default"]]);
        }

        if (data.hasOwnProperty('next_page')) {
          obj['next_page'] = _ApiClient["default"].convertToType(data['next_page'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ListItemsInPathOutput;
}();
/**
 * @member {Array.<String>} folders
 */


ListItemsInPathOutput.prototype['folders'] = undefined;
/**
 * @member {Array.<module:model/Item>} items
 */

ListItemsInPathOutput.prototype['items'] = undefined;
/**
 * @member {String} next_page
 */

ListItemsInPathOutput.prototype['next_page'] = undefined;
var _default = ListItemsInPathOutput;
exports["default"] = _default;
