// akeyless/dist/model/ListAuthMethods.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ListAuthMethods model module.
 * @module model/ListAuthMethods
 * @version 2.20.0
 */
var ListAuthMethods = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ListAuthMethods</code>.
   * listAuthMethods is a command that returns a list of all auth methods in the account.
   * @alias module:model/ListAuthMethods
   */
  function ListAuthMethods() {
    _classCallCheck(this, ListAuthMethods);

    ListAuthMethods.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ListAuthMethods, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ListAuthMethods</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ListAuthMethods} obj Optional instance to populate.
     * @return {module:model/ListAuthMethods} The populated <code>ListAuthMethods</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ListAuthMethods();

        if (data.hasOwnProperty('filter')) {
          obj['filter'] = _ApiClient["default"].convertToType(data['filter'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('pagination-token')) {
          obj['pagination-token'] = _ApiClient["default"].convertToType(data['pagination-token'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], ['String']);
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ListAuthMethods;
}();
/**
 * Filter by auth method name or part of it
 * @member {String} filter
 */


ListAuthMethods.prototype['filter'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

ListAuthMethods.prototype['json'] = undefined;
/**
 * Next page reference
 * @member {String} pagination-token
 */

ListAuthMethods.prototype['pagination-token'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ListAuthMethods.prototype['token'] = undefined;
/**
 * The Auth method types list of the requested method. In case it is empty, all types of auth methods will be returned. options: [api_key, azure_ad, oauth2/jwt, saml2, ldap, aws_iam, oidc, universal_identity, gcp, k8s, cert]
 * @member {Array.<String>} type
 */

ListAuthMethods.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ListAuthMethods.prototype['uid-token'] = undefined;
var _default = ListAuthMethods;
exports["default"] = _default;
