// akeyless/dist/model/ManagedKeyDetailsInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ManagedKeyTargetInfo = _interopRequireDefault(require("./ManagedKeyTargetInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ManagedKeyDetailsInfo model module.
 * @module model/ManagedKeyDetailsInfo
 * @version 2.4.0
 */
var ManagedKeyDetailsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ManagedKeyDetailsInfo</code>.
   * @alias module:model/ManagedKeyDetailsInfo
   */
  function ManagedKeyDetailsInfo() {
    _classCallCheck(this, ManagedKeyDetailsInfo);

    ManagedKeyDetailsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ManagedKeyDetailsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ManagedKeyDetailsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ManagedKeyDetailsInfo} obj Optional instance to populate.
     * @return {module:model/ManagedKeyDetailsInfo} The populated <code>ManagedKeyDetailsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ManagedKeyDetailsInfo();

        if (data.hasOwnProperty('is_provided_by_user')) {
          obj['is_provided_by_user'] = _ApiClient["default"].convertToType(data['is_provided_by_user'], 'Boolean');
        }

        if (data.hasOwnProperty('is_unexportable')) {
          obj['is_unexportable'] = _ApiClient["default"].convertToType(data['is_unexportable'], 'Boolean');
        }

        if (data.hasOwnProperty('key_state')) {
          obj['key_state'] = _ApiClient["default"].convertToType(data['key_state'], 'String');
        }

        if (data.hasOwnProperty('key_type')) {
          obj['key_type'] = _ApiClient["default"].convertToType(data['key_type'], 'String');
        }

        if (data.hasOwnProperty('managed_key_id')) {
          obj['managed_key_id'] = _ApiClient["default"].convertToType(data['managed_key_id'], 'String');
        }

        if (data.hasOwnProperty('target_alias_helper')) {
          obj['target_alias_helper'] = _ApiClient["default"].convertToType(data['target_alias_helper'], 'String');
        }

        if (data.hasOwnProperty('targets')) {
          obj['targets'] = _ApiClient["default"].convertToType(data['targets'], [_ManagedKeyTargetInfo["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ManagedKeyDetailsInfo;
}();
/**
 * @member {Boolean} is_provided_by_user
 */


ManagedKeyDetailsInfo.prototype['is_provided_by_user'] = undefined;
/**
 * @member {Boolean} is_unexportable
 */

ManagedKeyDetailsInfo.prototype['is_unexportable'] = undefined;
/**
 * ItemState defines the different states an Item can be in
 * @member {String} key_state
 */

ManagedKeyDetailsInfo.prototype['key_state'] = undefined;
/**
 * @member {String} key_type
 */

ManagedKeyDetailsInfo.prototype['key_type'] = undefined;
/**
 * @member {String} managed_key_id
 */

ManagedKeyDetailsInfo.prototype['managed_key_id'] = undefined;
/**
 * @member {String} target_alias_helper
 */

ManagedKeyDetailsInfo.prototype['target_alias_helper'] = undefined;
/**
 * @member {Array.<module:model/ManagedKeyTargetInfo>} targets
 */

ManagedKeyDetailsInfo.prototype['targets'] = undefined;
var _default = ManagedKeyDetailsInfo;
exports["default"] = _default;
