// akeyless/dist/model/ManagedKeyStatusInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ManagedKeyStatusInfo model module.
 * @module model/ManagedKeyStatusInfo
 * @version 2.4.0
 */
var ManagedKeyStatusInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ManagedKeyStatusInfo</code>.
   * @alias module:model/ManagedKeyStatusInfo
   */
  function ManagedKeyStatusInfo() {
    _classCallCheck(this, ManagedKeyStatusInfo);

    ManagedKeyStatusInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ManagedKeyStatusInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ManagedKeyStatusInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ManagedKeyStatusInfo} obj Optional instance to populate.
     * @return {module:model/ManagedKeyStatusInfo} The populated <code>ManagedKeyStatusInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ManagedKeyStatusInfo();

        if (data.hasOwnProperty('key_id')) {
          obj['key_id'] = _ApiClient["default"].convertToType(data['key_id'], 'Number');
        }

        if (data.hasOwnProperty('last_error')) {
          obj['last_error'] = _ApiClient["default"].convertToType(data['last_error'], 'String');
        }

        if (data.hasOwnProperty('last_status')) {
          obj['last_status'] = _ApiClient["default"].convertToType(data['last_status'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ManagedKeyStatusInfo;
}();
/**
 * @member {Number} key_id
 */


ManagedKeyStatusInfo.prototype['key_id'] = undefined;
/**
 * @member {String} last_error
 */

ManagedKeyStatusInfo.prototype['last_error'] = undefined;
/**
 * ManagedKeyTargetStatus defines status of managed key target
 * @member {String} last_status
 */

ManagedKeyStatusInfo.prototype['last_status'] = undefined;
var _default = ManagedKeyStatusInfo;
exports["default"] = _default;
