// akeyless/dist/model/ManagedKeyTargetInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ExternalKMSKeyId = _interopRequireDefault(require("./ExternalKMSKeyId"));

var _ManagedKeyStatusInfo = _interopRequireDefault(require("./ManagedKeyStatusInfo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ManagedKeyTargetInfo model module.
 * @module model/ManagedKeyTargetInfo
 * @version 2.4.0
 */
var ManagedKeyTargetInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ManagedKeyTargetInfo</code>.
   * @alias module:model/ManagedKeyTargetInfo
   */
  function ManagedKeyTargetInfo() {
    _classCallCheck(this, ManagedKeyTargetInfo);

    ManagedKeyTargetInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ManagedKeyTargetInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ManagedKeyTargetInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ManagedKeyTargetInfo} obj Optional instance to populate.
     * @return {module:model/ManagedKeyTargetInfo} The populated <code>ManagedKeyTargetInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ManagedKeyTargetInfo();

        if (data.hasOwnProperty('external_kms_id')) {
          obj['external_kms_id'] = _ExternalKMSKeyId["default"].constructFromObject(data['external_kms_id']);
        }

        if (data.hasOwnProperty('key_purpose')) {
          obj['key_purpose'] = _ApiClient["default"].convertToType(data['key_purpose'], ['String']);
        }

        if (data.hasOwnProperty('key_statuses')) {
          obj['key_statuses'] = _ApiClient["default"].convertToType(data['key_statuses'], [_ManagedKeyStatusInfo["default"]]);
        }

        if (data.hasOwnProperty('target_assoc_id')) {
          obj['target_assoc_id'] = _ApiClient["default"].convertToType(data['target_assoc_id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ManagedKeyTargetInfo;
}();
/**
 * @member {module:model/ExternalKMSKeyId} external_kms_id
 */


ManagedKeyTargetInfo.prototype['external_kms_id'] = undefined;
/**
 * @member {Array.<String>} key_purpose
 */

ManagedKeyTargetInfo.prototype['key_purpose'] = undefined;
/**
 * @member {Array.<module:model/ManagedKeyStatusInfo>} key_statuses
 */

ManagedKeyTargetInfo.prototype['key_statuses'] = undefined;
/**
 * @member {String} target_assoc_id
 */

ManagedKeyTargetInfo.prototype['target_assoc_id'] = undefined;
var _default = ManagedKeyTargetInfo;
exports["default"] = _default;
