// akeyless/dist/model/MigrationStatusReplyObj.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MigrationItems = _interopRequireDefault(require("./MigrationItems"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationStatusReplyObj model module.
 * @module model/MigrationStatusReplyObj
 * @version 2.20.0
 */
var MigrationStatusReplyObj = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationStatusReplyObj</code>.
   * @alias module:model/MigrationStatusReplyObj
   */
  function MigrationStatusReplyObj() {
    _classCallCheck(this, MigrationStatusReplyObj);

    MigrationStatusReplyObj.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationStatusReplyObj, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationStatusReplyObj</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationStatusReplyObj} obj Optional instance to populate.
     * @return {module:model/MigrationStatusReplyObj} The populated <code>MigrationStatusReplyObj</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationStatusReplyObj();

        if (data.hasOwnProperty('duration_time')) {
          obj['duration_time'] = _ApiClient["default"].convertToType(data['duration_time'], 'String');
        }

        if (data.hasOwnProperty('last_status_message')) {
          obj['last_status_message'] = _ApiClient["default"].convertToType(data['last_status_message'], 'String');
        }

        if (data.hasOwnProperty('max_name_length')) {
          obj['max_name_length'] = _ApiClient["default"].convertToType(data['max_name_length'], 'Number');
        }

        if (data.hasOwnProperty('max_value_length')) {
          obj['max_value_length'] = _ApiClient["default"].convertToType(data['max_value_length'], 'Number');
        }

        if (data.hasOwnProperty('migration_id')) {
          obj['migration_id'] = _ApiClient["default"].convertToType(data['migration_id'], 'String');
        }

        if (data.hasOwnProperty('migration_items')) {
          obj['migration_items'] = _MigrationItems["default"].constructFromObject(data['migration_items']);
        }

        if (data.hasOwnProperty('migration_name')) {
          obj['migration_name'] = _ApiClient["default"].convertToType(data['migration_name'], 'String');
        }

        if (data.hasOwnProperty('migration_state')) {
          obj['migration_state'] = _ApiClient["default"].convertToType(data['migration_state'], 'String');
        }

        if (data.hasOwnProperty('migration_type')) {
          obj['migration_type'] = _ApiClient["default"].convertToType(data['migration_type'], 'String');
        }

        if (data.hasOwnProperty('start_time')) {
          obj['start_time'] = _ApiClient["default"].convertToType(data['start_time'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MigrationStatusReplyObj;
}();
/**
 * @member {String} duration_time
 */


MigrationStatusReplyObj.prototype['duration_time'] = undefined;
/**
 * @member {String} last_status_message
 */

MigrationStatusReplyObj.prototype['last_status_message'] = undefined;
/**
 * @member {Number} max_name_length
 */

MigrationStatusReplyObj.prototype['max_name_length'] = undefined;
/**
 * @member {Number} max_value_length
 */

MigrationStatusReplyObj.prototype['max_value_length'] = undefined;
/**
 * @member {String} migration_id
 */

MigrationStatusReplyObj.prototype['migration_id'] = undefined;
/**
 * @member {module:model/MigrationItems} migration_items
 */

MigrationStatusReplyObj.prototype['migration_items'] = undefined;
/**
 * @member {String} migration_name
 */

MigrationStatusReplyObj.prototype['migration_name'] = undefined;
/**
 * @member {String} migration_state
 */

MigrationStatusReplyObj.prototype['migration_state'] = undefined;
/**
 * @member {String} migration_type
 */

MigrationStatusReplyObj.prototype['migration_type'] = undefined;
/**
 * @member {String} start_time
 */

MigrationStatusReplyObj.prototype['start_time'] = undefined;
var _default = MigrationStatusReplyObj;
exports["default"] = _default;
