// akeyless/dist/model/MigrationsConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AWSSecretsMigration = _interopRequireDefault(require("./AWSSecretsMigration"));

var _ActiveDirectoryMigration = _interopRequireDefault(require("./ActiveDirectoryMigration"));

var _AzureKeyVaultMigration = _interopRequireDefault(require("./AzureKeyVaultMigration"));

var _GCPSecretsMigration = _interopRequireDefault(require("./GCPSecretsMigration"));

var _HashiMigration = _interopRequireDefault(require("./HashiMigration"));

var _K8SMigration = _interopRequireDefault(require("./K8SMigration"));

var _OnePasswordMigration = _interopRequireDefault(require("./OnePasswordMigration"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationsConfigPart model module.
 * @module model/MigrationsConfigPart
 * @version 2.20.0
 */
var MigrationsConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationsConfigPart</code>.
   * @alias module:model/MigrationsConfigPart
   */
  function MigrationsConfigPart() {
    _classCallCheck(this, MigrationsConfigPart);

    MigrationsConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationsConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationsConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationsConfigPart} obj Optional instance to populate.
     * @return {module:model/MigrationsConfigPart} The populated <code>MigrationsConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationsConfigPart();

        if (data.hasOwnProperty('active_directory_migrations')) {
          obj['active_directory_migrations'] = _ApiClient["default"].convertToType(data['active_directory_migrations'], [_ActiveDirectoryMigration["default"]]);
        }

        if (data.hasOwnProperty('aws_secrets_migrations')) {
          obj['aws_secrets_migrations'] = _ApiClient["default"].convertToType(data['aws_secrets_migrations'], [_AWSSecretsMigration["default"]]);
        }

        if (data.hasOwnProperty('azure_kv_migrations')) {
          obj['azure_kv_migrations'] = _ApiClient["default"].convertToType(data['azure_kv_migrations'], [_AzureKeyVaultMigration["default"]]);
        }

        if (data.hasOwnProperty('gcp_secrets_migrations')) {
          obj['gcp_secrets_migrations'] = _ApiClient["default"].convertToType(data['gcp_secrets_migrations'], [_GCPSecretsMigration["default"]]);
        }

        if (data.hasOwnProperty('hashi_migrations')) {
          obj['hashi_migrations'] = _ApiClient["default"].convertToType(data['hashi_migrations'], [_HashiMigration["default"]]);
        }

        if (data.hasOwnProperty('k8s_migrations')) {
          obj['k8s_migrations'] = _ApiClient["default"].convertToType(data['k8s_migrations'], [_K8SMigration["default"]]);
        }

        if (data.hasOwnProperty('one_password_migrations')) {
          obj['one_password_migrations'] = _ApiClient["default"].convertToType(data['one_password_migrations'], [_OnePasswordMigration["default"]]);
        }
      }

      return obj;
    }
  }]);

  return MigrationsConfigPart;
}();
/**
 * @member {Array.<module:model/ActiveDirectoryMigration>} active_directory_migrations
 */


MigrationsConfigPart.prototype['active_directory_migrations'] = undefined;
/**
 * @member {Array.<module:model/AWSSecretsMigration>} aws_secrets_migrations
 */

MigrationsConfigPart.prototype['aws_secrets_migrations'] = undefined;
/**
 * @member {Array.<module:model/AzureKeyVaultMigration>} azure_kv_migrations
 */

MigrationsConfigPart.prototype['azure_kv_migrations'] = undefined;
/**
 * @member {Array.<module:model/GCPSecretsMigration>} gcp_secrets_migrations
 */

MigrationsConfigPart.prototype['gcp_secrets_migrations'] = undefined;
/**
 * @member {Array.<module:model/HashiMigration>} hashi_migrations
 */

MigrationsConfigPart.prototype['hashi_migrations'] = undefined;
/**
 * @member {Array.<module:model/K8SMigration>} k8s_migrations
 */

MigrationsConfigPart.prototype['k8s_migrations'] = undefined;
/**
 * @member {Array.<module:model/OnePasswordMigration>} one_password_migrations
 */

MigrationsConfigPart.prototype['one_password_migrations'] = undefined;
var _default = MigrationsConfigPart;
exports["default"] = _default;
