// akeyless/dist/model/MigrationsConfigLastChange.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationsConfigLastChange model module.
 * @module model/MigrationsConfigLastChange
 * @version 2.20.0
 */
var MigrationsConfigLastChange = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationsConfigLastChange</code>.
   * @alias module:model/MigrationsConfigLastChange
   */
  function MigrationsConfigLastChange() {
    _classCallCheck(this, MigrationsConfigLastChange);

    MigrationsConfigLastChange.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationsConfigLastChange, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationsConfigLastChange</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationsConfigLastChange} obj Optional instance to populate.
     * @return {module:model/MigrationsConfigLastChange} The populated <code>MigrationsConfigLastChange</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationsConfigLastChange();

        if (data.hasOwnProperty('changed_migrations')) {
          obj['changed_migrations'] = _ApiClient["default"].convertToType(data['changed_migrations'], ['String']);
        }

        if (data.hasOwnProperty('created_migrations')) {
          obj['created_migrations'] = _ApiClient["default"].convertToType(data['created_migrations'], ['String']);
        }

        if (data.hasOwnProperty('deleted_migrations')) {
          obj['deleted_migrations'] = _ApiClient["default"].convertToType(data['deleted_migrations'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return MigrationsConfigLastChange;
}();
/**
 * @member {Array.<String>} changed_migrations
 */


MigrationsConfigLastChange.prototype['changed_migrations'] = undefined;
/**
 * @member {Array.<String>} created_migrations
 */

MigrationsConfigLastChange.prototype['created_migrations'] = undefined;
/**
 * @member {Array.<String>} deleted_migrations
 */

MigrationsConfigLastChange.prototype['deleted_migrations'] = undefined;
var _default = MigrationsConfigLastChange;
exports["default"] = _default;
