// akeyless/dist/model/MigrationGeneral.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationGeneral model module.
 * @module model/MigrationGeneral
 * @version 2.15.24
 */
var MigrationGeneral = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationGeneral</code>.
   * @alias module:model/MigrationGeneral
   */
  function MigrationGeneral() {
    _classCallCheck(this, MigrationGeneral);

    MigrationGeneral.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationGeneral, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationGeneral</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationGeneral} obj Optional instance to populate.
     * @return {module:model/MigrationGeneral} The populated <code>MigrationGeneral</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationGeneral();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('prefix')) {
          obj['prefix'] = _ApiClient["default"].convertToType(data['prefix'], 'String');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MigrationGeneral;
}();
/**
 * @member {String} id
 */


MigrationGeneral.prototype['id'] = undefined;
/**
 * @member {String} name
 */

MigrationGeneral.prototype['name'] = undefined;
/**
 * @member {String} prefix
 */

MigrationGeneral.prototype['prefix'] = undefined;
/**
 * @member {String} protection_key
 */

MigrationGeneral.prototype['protection_key'] = undefined;
var _default = MigrationGeneral;
exports["default"] = _default;
