// akeyless/dist/model/MigrationItems.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MigrationItems model module.
 * @module model/MigrationItems
 * @version 2.20.0
 */
var MigrationItems = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MigrationItems</code>.
   * @alias module:model/MigrationItems
   */
  function MigrationItems() {
    _classCallCheck(this, MigrationItems);

    MigrationItems.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MigrationItems, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MigrationItems</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MigrationItems} obj Optional instance to populate.
     * @return {module:model/MigrationItems} The populated <code>MigrationItems</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MigrationItems();

        if (data.hasOwnProperty('failed')) {
          obj['failed'] = _ApiClient["default"].convertToType(data['failed'], 'Number');
        }

        if (data.hasOwnProperty('migrated')) {
          obj['migrated'] = _ApiClient["default"].convertToType(data['migrated'], 'Number');
        }

        if (data.hasOwnProperty('skipped')) {
          obj['skipped'] = _ApiClient["default"].convertToType(data['skipped'], 'Number');
        }

        if (data.hasOwnProperty('total')) {
          obj['total'] = _ApiClient["default"].convertToType(data['total'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return MigrationItems;
}();
/**
 * @member {Number} failed
 */


MigrationItems.prototype['failed'] = undefined;
/**
 * @member {Number} migrated
 */

MigrationItems.prototype['migrated'] = undefined;
/**
 * @member {Number} skipped
 */

MigrationItems.prototype['skipped'] = undefined;
/**
 * @member {Number} total
 */

MigrationItems.prototype['total'] = undefined;
var _default = MigrationItems;
exports["default"] = _default;
