// akeyless/dist/model/MoveObjects.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MoveObjects model module.
 * @module model/MoveObjects
 * @version 2.15.24
 */
var MoveObjects = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MoveObjects</code>.
   * @alias module:model/MoveObjects
   * @param source {String} Source path to move the objects from
   * @param target {String} Target path to move the objects to
   */
  function MoveObjects(source, target) {
    _classCallCheck(this, MoveObjects);

    MoveObjects.initialize(this, source, target);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MoveObjects, null, [{
    key: "initialize",
    value: function initialize(obj, source, target) {
      obj['source'] = source;
      obj['target'] = target;
    }
    /**
     * Constructs a <code>MoveObjects</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MoveObjects} obj Optional instance to populate.
     * @return {module:model/MoveObjects} The populated <code>MoveObjects</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MoveObjects();

        if (data.hasOwnProperty('objects-type')) {
          obj['objects-type'] = _ApiClient["default"].convertToType(data['objects-type'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], 'String');
        }

        if (data.hasOwnProperty('target')) {
          obj['target'] = _ApiClient["default"].convertToType(data['target'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return MoveObjects;
}();
/**
 * The objects type to move (item/auth_method/role)
 * @member {String} objects-type
 * @default 'item'
 */


MoveObjects.prototype['objects-type'] = 'item';
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

MoveObjects.prototype['password'] = undefined;
/**
 * Source path to move the objects from
 * @member {String} source
 */

MoveObjects.prototype['source'] = undefined;
/**
 * Target path to move the objects to
 * @member {String} target
 */

MoveObjects.prototype['target'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

MoveObjects.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

MoveObjects.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

MoveObjects.prototype['username'] = undefined;
var _default = MoveObjects;
exports["default"] = _default;
