// akeyless/dist/model/OIDCAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _OIDCCustomClaim = _interopRequireDefault(require("./OIDCCustomClaim"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OIDCAccessRules model module.
 * @module model/OIDCAccessRules
 * @version 2.20.0
 */
var OIDCAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OIDCAccessRules</code>.
   * OIDCAccessRules contains access rules specific to Open Id Connect authentication method.
   * @alias module:model/OIDCAccessRules
   */
  function OIDCAccessRules() {
    _classCallCheck(this, OIDCAccessRules);

    OIDCAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OIDCAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OIDCAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OIDCAccessRules} obj Optional instance to populate.
     * @return {module:model/OIDCAccessRules} The populated <code>OIDCAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OIDCAccessRules();

        if (data.hasOwnProperty('allowed_redirect_URIs')) {
          obj['allowed_redirect_URIs'] = _ApiClient["default"].convertToType(data['allowed_redirect_URIs'], ['String']);
        }

        if (data.hasOwnProperty('bound_claims')) {
          obj['bound_claims'] = _ApiClient["default"].convertToType(data['bound_claims'], [_OIDCCustomClaim["default"]]);
        }

        if (data.hasOwnProperty('client_id')) {
          obj['client_id'] = _ApiClient["default"].convertToType(data['client_id'], 'String');
        }

        if (data.hasOwnProperty('client_secret')) {
          obj['client_secret'] = _ApiClient["default"].convertToType(data['client_secret'], 'String');
        }

        if (data.hasOwnProperty('is_internal')) {
          obj['is_internal'] = _ApiClient["default"].convertToType(data['is_internal'], 'Boolean');
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('required_scopes')) {
          obj['required_scopes'] = _ApiClient["default"].convertToType(data['required_scopes'], ['String']);
        }

        if (data.hasOwnProperty('required_scopes_prefix')) {
          obj['required_scopes_prefix'] = _ApiClient["default"].convertToType(data['required_scopes_prefix'], 'String');
        }

        if (data.hasOwnProperty('unique_identifier')) {
          obj['unique_identifier'] = _ApiClient["default"].convertToType(data['unique_identifier'], 'String');
        }
      }

      return obj;
    }
  }]);

  return OIDCAccessRules;
}();
/**
 * Allowed redirect URIs after the authentication
 * @member {Array.<String>} allowed_redirect_URIs
 */


OIDCAccessRules.prototype['allowed_redirect_URIs'] = undefined;
/**
 * The claims that login is restricted to.
 * @member {Array.<module:model/OIDCCustomClaim>} bound_claims
 */

OIDCAccessRules.prototype['bound_claims'] = undefined;
/**
 * Client ID
 * @member {String} client_id
 */

OIDCAccessRules.prototype['client_id'] = undefined;
/**
 * Client Secret
 * @member {String} client_secret
 */

OIDCAccessRules.prototype['client_secret'] = undefined;
/**
 * IsInternal indicates whether this is an internal Auth Method where the client has no control over it, or it was created by the client e.g - Sign In with Google will create an OIDC Auth Method with IsInternal=true
 * @member {Boolean} is_internal
 */

OIDCAccessRules.prototype['is_internal'] = undefined;
/**
 * Issuer URL
 * @member {String} issuer
 */

OIDCAccessRules.prototype['issuer'] = undefined;
/**
 * A list of required scopes to request from the oidc provider, and to check on the token
 * @member {Array.<String>} required_scopes
 */

OIDCAccessRules.prototype['required_scopes'] = undefined;
/**
 * A prefix to add to the required scopes (for example, azures' Application ID URI)
 * @member {String} required_scopes_prefix
 */

OIDCAccessRules.prototype['required_scopes_prefix'] = undefined;
/**
 * A unique identifier to distinguish different users
 * @member {String} unique_identifier
 */

OIDCAccessRules.prototype['unique_identifier'] = undefined;
var _default = OIDCAccessRules;
exports["default"] = _default;
