// akeyless/dist/model/OnePasswordPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The OnePasswordPayload model module.
 * @module model/OnePasswordPayload
 * @version 2.20.0
 */
var OnePasswordPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OnePasswordPayload</code>.
   * @alias module:model/OnePasswordPayload
   */
  function OnePasswordPayload() {
    _classCallCheck(this, OnePasswordPayload);

    OnePasswordPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(OnePasswordPayload, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>OnePasswordPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OnePasswordPayload} obj Optional instance to populate.
     * @return {module:model/OnePasswordPayload} The populated <code>OnePasswordPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OnePasswordPayload();

        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('secret_key')) {
          obj['secret_key'] = _ApiClient["default"].convertToType(data['secret_key'], 'String');
        }

        if (data.hasOwnProperty('url')) {
          obj['url'] = _ApiClient["default"].convertToType(data['url'], 'String');
        }

        if (data.hasOwnProperty('vaults')) {
          obj['vaults'] = _ApiClient["default"].convertToType(data['vaults'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return OnePasswordPayload;
}();
/**
 * @member {String} email
 */


OnePasswordPayload.prototype['email'] = undefined;
/**
 * @member {String} password
 */

OnePasswordPayload.prototype['password'] = undefined;
/**
 * @member {String} secret_key
 */

OnePasswordPayload.prototype['secret_key'] = undefined;
/**
 * @member {String} url
 */

OnePasswordPayload.prototype['url'] = undefined;
/**
 * @member {Array.<String>} vaults
 */

OnePasswordPayload.prototype['vaults'] = undefined;
var _default = OnePasswordPayload;
exports["default"] = _default;
