// akeyless/dist/model/PasswordPolicyInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PasswordPolicyInfo model module.
 * @module model/PasswordPolicyInfo
 * @version 2.20.0
 */
var PasswordPolicyInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PasswordPolicyInfo</code>.
   * @alias module:model/PasswordPolicyInfo
   */
  function PasswordPolicyInfo() {
    _classCallCheck(this, PasswordPolicyInfo);

    PasswordPolicyInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PasswordPolicyInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PasswordPolicyInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PasswordPolicyInfo} obj Optional instance to populate.
     * @return {module:model/PasswordPolicyInfo} The populated <code>PasswordPolicyInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PasswordPolicyInfo();

        if (data.hasOwnProperty('password_length')) {
          obj['password_length'] = _ApiClient["default"].convertToType(data['password_length'], 'Number');
        }

        if (data.hasOwnProperty('use_capital_letters')) {
          obj['use_capital_letters'] = _ApiClient["default"].convertToType(data['use_capital_letters'], 'Boolean');
        }

        if (data.hasOwnProperty('use_lower_letters')) {
          obj['use_lower_letters'] = _ApiClient["default"].convertToType(data['use_lower_letters'], 'Boolean');
        }

        if (data.hasOwnProperty('use_numbers')) {
          obj['use_numbers'] = _ApiClient["default"].convertToType(data['use_numbers'], 'Boolean');
        }

        if (data.hasOwnProperty('use_special_characters')) {
          obj['use_special_characters'] = _ApiClient["default"].convertToType(data['use_special_characters'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return PasswordPolicyInfo;
}();
/**
 * @member {Number} password_length
 */


PasswordPolicyInfo.prototype['password_length'] = undefined;
/**
 * @member {Boolean} use_capital_letters
 */

PasswordPolicyInfo.prototype['use_capital_letters'] = undefined;
/**
 * @member {Boolean} use_lower_letters
 */

PasswordPolicyInfo.prototype['use_lower_letters'] = undefined;
/**
 * @member {Boolean} use_numbers
 */

PasswordPolicyInfo.prototype['use_numbers'] = undefined;
/**
 * @member {Boolean} use_special_characters
 */

PasswordPolicyInfo.prototype['use_special_characters'] = undefined;
var _default = PasswordPolicyInfo;
exports["default"] = _default;
