// akeyless/dist/model/PathRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PathRule model module.
 * @module model/PathRule
 * @version 2.15.24
 */
var PathRule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PathRule</code>.
   * @alias module:model/PathRule
   */
  function PathRule() {
    _classCallCheck(this, PathRule);

    PathRule.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PathRule, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>PathRule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PathRule} obj Optional instance to populate.
     * @return {module:model/PathRule} The populated <code>PathRule</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PathRule();

        if (data.hasOwnProperty('capabilities')) {
          obj['capabilities'] = _ApiClient["default"].convertToType(data['capabilities'], ['String']);
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return PathRule;
}();
/**
 * The approved/denied capabilities in the path
 * @member {Array.<String>} capabilities
 */


PathRule.prototype['capabilities'] = undefined;
/**
 * The path the rule refers to
 * @member {String} path
 */

PathRule.prototype['path'] = undefined;
/**
 * @member {String} type
 */

PathRule.prototype['type'] = undefined;
var _default = PathRule;
exports["default"] = _default;
