// akeyless/dist/model/Producer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Producer model module.
 * @module model/Producer
 * @version 2.20.0
 */
var Producer = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Producer</code>.
   * @alias module:model/Producer
   */
  function Producer() {
    _classCallCheck(this, Producer);

    Producer.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Producer, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Producer</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Producer} obj Optional instance to populate.
     * @return {module:model/Producer} The populated <code>Producer</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Producer();

        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }

        if (data.hasOwnProperty('failure_message')) {
          obj['failure_message'] = _ApiClient["default"].convertToType(data['failure_message'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('init')) {
          obj['init'] = _ApiClient["default"].convertToType(data['init'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Producer;
}();
/**
 * @member {Boolean} active
 */


Producer.prototype['active'] = undefined;
/**
 * @member {String} failure_message
 */

Producer.prototype['failure_message'] = undefined;
/**
 * @member {Number} id
 */

Producer.prototype['id'] = undefined;
/**
 * @member {Boolean} init
 */

Producer.prototype['init'] = undefined;
/**
 * @member {String} name
 */

Producer.prototype['name'] = undefined;
/**
 * @member {String} type
 */

Producer.prototype['type'] = undefined;
var _default = Producer;
exports["default"] = _default;
