// akeyless/dist/model/ReverseRBAC.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ReverseRBAC model module.
 * @module model/ReverseRBAC
 * @version 2.15.24
 */
var ReverseRBAC = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ReverseRBAC</code>.
   * reverseRBAC is a command that shows which auth methods have access to a particular object.
   * @alias module:model/ReverseRBAC
   * @param path {String} Path to an object
   * @param type {String} Type of object (item, am, role)
   */
  function ReverseRBAC(path, type) {
    _classCallCheck(this, ReverseRBAC);

    ReverseRBAC.initialize(this, path, type);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ReverseRBAC, null, [{
    key: "initialize",
    value: function initialize(obj, path, type) {
      obj['path'] = path;
      obj['type'] = type;
    }
    /**
     * Constructs a <code>ReverseRBAC</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ReverseRBAC} obj Optional instance to populate.
     * @return {module:model/ReverseRBAC} The populated <code>ReverseRBAC</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ReverseRBAC();

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('path')) {
          obj['path'] = _ApiClient["default"].convertToType(data['path'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ReverseRBAC;
}();
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */


ReverseRBAC.prototype['password'] = undefined;
/**
 * Path to an object
 * @member {String} path
 */

ReverseRBAC.prototype['path'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

ReverseRBAC.prototype['token'] = undefined;
/**
 * Type of object (item, am, role)
 * @member {String} type
 */

ReverseRBAC.prototype['type'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

ReverseRBAC.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

ReverseRBAC.prototype['username'] = undefined;
var _default = ReverseRBAC;
exports["default"] = _default;
