// akeyless/dist/model/ReverseRBACClient.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AuthMethodRoleAssociation = _interopRequireDefault(require("./AuthMethodRoleAssociation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ReverseRBACClient model module.
 * @module model/ReverseRBACClient
 * @version 2.20.0
 */
var ReverseRBACClient = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ReverseRBACClient</code>.
   * @alias module:model/ReverseRBACClient
   */
  function ReverseRBACClient() {
    _classCallCheck(this, ReverseRBACClient);

    ReverseRBACClient.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ReverseRBACClient, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ReverseRBACClient</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ReverseRBACClient} obj Optional instance to populate.
     * @return {module:model/ReverseRBACClient} The populated <code>ReverseRBACClient</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ReverseRBACClient();

        if (data.hasOwnProperty('allowed_ops')) {
          obj['allowed_ops'] = _ApiClient["default"].convertToType(data['allowed_ops'], ['String']);
        }

        if (data.hasOwnProperty('assocs')) {
          obj['assocs'] = _ApiClient["default"].convertToType(data['assocs'], [_AuthMethodRoleAssociation["default"]]);
        }

        if (data.hasOwnProperty('auth_method_name')) {
          obj['auth_method_name'] = _ApiClient["default"].convertToType(data['auth_method_name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ReverseRBACClient;
}();
/**
 * @member {Array.<String>} allowed_ops
 */


ReverseRBACClient.prototype['allowed_ops'] = undefined;
/**
 * @member {Array.<module:model/AuthMethodRoleAssociation>} assocs
 */

ReverseRBACClient.prototype['assocs'] = undefined;
/**
 * @member {String} auth_method_name
 */

ReverseRBACClient.prototype['auth_method_name'] = undefined;
var _default = ReverseRBACClient;
exports["default"] = _default;
