// akeyless/dist/model/Role.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _RoleAuthMethodAssociation = _interopRequireDefault(require("./RoleAuthMethodAssociation"));

var _Rules = _interopRequireDefault(require("./Rules"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Role model module.
 * @module model/Role
 * @version 2.15.24
 */
var Role = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Role</code>.
   * @alias module:model/Role
   */
  function Role() {
    _classCallCheck(this, Role);

    Role.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Role, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Role</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Role} obj Optional instance to populate.
     * @return {module:model/Role} The populated <code>Role</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Role();

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('role_auth_methods_assoc')) {
          obj['role_auth_methods_assoc'] = _ApiClient["default"].convertToType(data['role_auth_methods_assoc'], [_RoleAuthMethodAssociation["default"]]);
        }

        if (data.hasOwnProperty('role_name')) {
          obj['role_name'] = _ApiClient["default"].convertToType(data['role_name'], 'String');
        }

        if (data.hasOwnProperty('rules')) {
          obj['rules'] = _Rules["default"].constructFromObject(data['rules']);
        }
      }

      return obj;
    }
  }]);

  return Role;
}();
/**
 * @member {Array.<String>} client_permissions
 */


Role.prototype['client_permissions'] = undefined;
/**
 * @member {String} comment
 */

Role.prototype['comment'] = undefined;
/**
 * @member {Array.<module:model/RoleAuthMethodAssociation>} role_auth_methods_assoc
 */

Role.prototype['role_auth_methods_assoc'] = undefined;
/**
 * @member {String} role_name
 */

Role.prototype['role_name'] = undefined;
/**
 * @member {module:model/Rules} rules
 */

Role.prototype['rules'] = undefined;
var _default = Role;
exports["default"] = _default;
