// akeyless/dist/model/RoleAuthMethodAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RoleAuthMethodAssociation model module.
 * @module model/RoleAuthMethodAssociation
 * @version 2.20.0
 */
var RoleAuthMethodAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RoleAuthMethodAssociation</code>.
   * RoleAuthMethodAssociation includes details of an association between a role and an auth method.
   * @alias module:model/RoleAuthMethodAssociation
   */
  function RoleAuthMethodAssociation() {
    _classCallCheck(this, RoleAuthMethodAssociation);

    RoleAuthMethodAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RoleAuthMethodAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RoleAuthMethodAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RoleAuthMethodAssociation} obj Optional instance to populate.
     * @return {module:model/RoleAuthMethodAssociation} The populated <code>RoleAuthMethodAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RoleAuthMethodAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_access_id')) {
          obj['auth_method_access_id'] = _ApiClient["default"].convertToType(data['auth_method_access_id'], 'String');
        }

        if (data.hasOwnProperty('auth_method_name')) {
          obj['auth_method_name'] = _ApiClient["default"].convertToType(data['auth_method_name'], 'String');
        }

        if (data.hasOwnProperty('auth_method_sub_claims')) {
          obj['auth_method_sub_claims'] = _ApiClient["default"].convertToType(data['auth_method_sub_claims'], {
            'String': ['String']
          });
        }

        if (data.hasOwnProperty('sub_claims_case_sensitive')) {
          obj['sub_claims_case_sensitive'] = _ApiClient["default"].convertToType(data['sub_claims_case_sensitive'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return RoleAuthMethodAssociation;
}();
/**
 * @member {String} assoc_id
 */


RoleAuthMethodAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {String} auth_method_access_id
 */

RoleAuthMethodAssociation.prototype['auth_method_access_id'] = undefined;
/**
 * @member {String} auth_method_name
 */

RoleAuthMethodAssociation.prototype['auth_method_name'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} auth_method_sub_claims
 */

RoleAuthMethodAssociation.prototype['auth_method_sub_claims'] = undefined;
/**
 * @member {Boolean} sub_claims_case_sensitive
 */

RoleAuthMethodAssociation.prototype['sub_claims_case_sensitive'] = undefined;
var _default = RoleAuthMethodAssociation;
exports["default"] = _default;
