// akeyless/dist/model/RotatedSecretDetailsInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The RotatedSecretDetailsInfo model module.
 * @module model/RotatedSecretDetailsInfo
 * @version 2.15.24
 */
var RotatedSecretDetailsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RotatedSecretDetailsInfo</code>.
   * RotatedSecretDetailsInfo The rotated secret rotator info
   * @alias module:model/RotatedSecretDetailsInfo
   */
  function RotatedSecretDetailsInfo() {
    _classCallCheck(this, RotatedSecretDetailsInfo);

    RotatedSecretDetailsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(RotatedSecretDetailsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>RotatedSecretDetailsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RotatedSecretDetailsInfo} obj Optional instance to populate.
     * @return {module:model/RotatedSecretDetailsInfo} The populated <code>RotatedSecretDetailsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RotatedSecretDetailsInfo();

        if (data.hasOwnProperty('delete_previous_version_in_days')) {
          obj['delete_previous_version_in_days'] = _ApiClient["default"].convertToType(data['delete_previous_version_in_days'], 'Number');
        }

        if (data.hasOwnProperty('gw_cluster_id')) {
          obj['gw_cluster_id'] = _ApiClient["default"].convertToType(data['gw_cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('last_rotation_error')) {
          obj['last_rotation_error'] = _ApiClient["default"].convertToType(data['last_rotation_error'], 'String');
        }

        if (data.hasOwnProperty('number_of_versions_to_save')) {
          obj['number_of_versions_to_save'] = _ApiClient["default"].convertToType(data['number_of_versions_to_save'], 'Number');
        }

        if (data.hasOwnProperty('rotation_hour')) {
          obj['rotation_hour'] = _ApiClient["default"].convertToType(data['rotation_hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation_interval_min')) {
          obj['rotation_interval_min'] = _ApiClient["default"].convertToType(data['rotation_interval_min'], 'Boolean');
        }

        if (data.hasOwnProperty('rotation_statement')) {
          obj['rotation_statement'] = _ApiClient["default"].convertToType(data['rotation_statement'], 'String');
        }

        if (data.hasOwnProperty('rotator_creds_type')) {
          obj['rotator_creds_type'] = _ApiClient["default"].convertToType(data['rotator_creds_type'], 'String');
        }

        if (data.hasOwnProperty('rotator_status')) {
          obj['rotator_status'] = _ApiClient["default"].convertToType(data['rotator_status'], 'String');
        }

        if (data.hasOwnProperty('rotator_type')) {
          obj['rotator_type'] = _ApiClient["default"].convertToType(data['rotator_type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return RotatedSecretDetailsInfo;
}();
/**
 * @member {Number} delete_previous_version_in_days
 */


RotatedSecretDetailsInfo.prototype['delete_previous_version_in_days'] = undefined;
/**
 * @member {Number} gw_cluster_id
 */

RotatedSecretDetailsInfo.prototype['gw_cluster_id'] = undefined;
/**
 * @member {String} last_rotation_error
 */

RotatedSecretDetailsInfo.prototype['last_rotation_error'] = undefined;
/**
 * @member {Number} number_of_versions_to_save
 */

RotatedSecretDetailsInfo.prototype['number_of_versions_to_save'] = undefined;
/**
 * @member {Number} rotation_hour
 */

RotatedSecretDetailsInfo.prototype['rotation_hour'] = undefined;
/**
 * @member {Boolean} rotation_interval_min
 */

RotatedSecretDetailsInfo.prototype['rotation_interval_min'] = undefined;
/**
 * @member {String} rotation_statement
 */

RotatedSecretDetailsInfo.prototype['rotation_statement'] = undefined;
/**
 * @member {String} rotator_creds_type
 */

RotatedSecretDetailsInfo.prototype['rotator_creds_type'] = undefined;
/**
 * RotationStatus defines types of rotation Status
 * @member {String} rotator_status
 */

RotatedSecretDetailsInfo.prototype['rotator_status'] = undefined;
/**
 * @member {String} rotator_type
 */

RotatedSecretDetailsInfo.prototype['rotator_type'] = undefined;
var _default = RotatedSecretDetailsInfo;
exports["default"] = _default;
