// akeyless/dist/model/Rotator.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Rotator model module.
 * @module model/Rotator
 * @version 2.20.0
 */
var Rotator = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Rotator</code>.
   * @alias module:model/Rotator
   */
  function Rotator() {
    _classCallCheck(this, Rotator);

    Rotator.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Rotator, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Rotator</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Rotator} obj Optional instance to populate.
     * @return {module:model/Rotator} The populated <code>Rotator</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Rotator();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }

        if (data.hasOwnProperty('last_error')) {
          obj['last_error'] = _ApiClient["default"].convertToType(data['last_error'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('rotation_interval')) {
          obj['rotation_interval'] = _ApiClient["default"].convertToType(data['rotation_interval'], 'Number');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Rotator;
}();
/**
 * @member {Number} id
 */


Rotator.prototype['id'] = undefined;
/**
 * @member {String} last_error
 */

Rotator.prototype['last_error'] = undefined;
/**
 * @member {String} name
 */

Rotator.prototype['name'] = undefined;
/**
 * @member {Number} rotation_interval
 */

Rotator.prototype['rotation_interval'] = undefined;
/**
 * @member {String} type
 */

Rotator.prototype['type'] = undefined;
var _default = Rotator;
exports["default"] = _default;
