// akeyless/dist/model/SetItemState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SetItemState model module.
 * @module model/SetItemState
 * @version 2.15.24
 */
var SetItemState = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SetItemState</code>.
   * @alias module:model/SetItemState
   * @param desiredState {String} Desired item state (Enabled, Disabled)
   * @param name {String} Current item name
   */
  function SetItemState(desiredState, name) {
    _classCallCheck(this, SetItemState);

    SetItemState.initialize(this, desiredState, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SetItemState, null, [{
    key: "initialize",
    value: function initialize(obj, desiredState, name) {
      obj['desired-state'] = desiredState;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>SetItemState</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SetItemState} obj Optional instance to populate.
     * @return {module:model/SetItemState} The populated <code>SetItemState</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SetItemState();

        if (data.hasOwnProperty('desired-state')) {
          obj['desired-state'] = _ApiClient["default"].convertToType(data['desired-state'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SetItemState;
}();
/**
 * Desired item state (Enabled, Disabled)
 * @member {String} desired-state
 */


SetItemState.prototype['desired-state'] = undefined;
/**
 * Current item name
 * @member {String} name
 */

SetItemState.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

SetItemState.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SetItemState.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SetItemState.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

SetItemState.prototype['username'] = undefined;
/**
 * The specific version you want to update: 0=item level state (default)
 * @member {Number} version
 * @default 0
 */

SetItemState.prototype['version'] = 0;
var _default = SetItemState;
exports["default"] = _default;
