// akeyless/dist/model/SignPKICertWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignPKICertWithClassicKey model module.
 * @module model/SignPKICertWithClassicKey
 * @version 2.15.24
 */
var SignPKICertWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignPKICertWithClassicKey</code>.
   * @alias module:model/SignPKICertWithClassicKey
   * @param displayId {String} The name of the key to use in the sign PKI Cert process
   * @param signingMethod {String} SigningMethod
   * @param ttl {Number} he requested Time To Live for the certificate, in seconds
   * @param version {Number} classic key version
   */
  function SignPKICertWithClassicKey(displayId, signingMethod, ttl, version) {
    _classCallCheck(this, SignPKICertWithClassicKey);

    SignPKICertWithClassicKey.initialize(this, displayId, signingMethod, ttl, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignPKICertWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, displayId, signingMethod, ttl, version) {
      obj['display-id'] = displayId;
      obj['signing-method'] = signingMethod;
      obj['ttl'] = ttl;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>SignPKICertWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignPKICertWithClassicKey} obj Optional instance to populate.
     * @return {module:model/SignPKICertWithClassicKey} The populated <code>SignPKICertWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignPKICertWithClassicKey();

        if (data.hasOwnProperty('common-name')) {
          obj['common-name'] = _ApiClient["default"].convertToType(data['common-name'], 'String');
        }

        if (data.hasOwnProperty('country')) {
          obj['country'] = _ApiClient["default"].convertToType(data['country'], 'String');
        }

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('dns-names')) {
          obj['dns-names'] = _ApiClient["default"].convertToType(data['dns-names'], 'String');
        }

        if (data.hasOwnProperty('key-usage')) {
          obj['key-usage'] = _ApiClient["default"].convertToType(data['key-usage'], 'String');
        }

        if (data.hasOwnProperty('locality')) {
          obj['locality'] = _ApiClient["default"].convertToType(data['locality'], 'String');
        }

        if (data.hasOwnProperty('organizational-units')) {
          obj['organizational-units'] = _ApiClient["default"].convertToType(data['organizational-units'], 'String');
        }

        if (data.hasOwnProperty('organizations')) {
          obj['organizations'] = _ApiClient["default"].convertToType(data['organizations'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('postal-code')) {
          obj['postal-code'] = _ApiClient["default"].convertToType(data['postal-code'], 'String');
        }

        if (data.hasOwnProperty('province')) {
          obj['province'] = _ApiClient["default"].convertToType(data['province'], 'String');
        }

        if (data.hasOwnProperty('public-key-pem-data')) {
          obj['public-key-pem-data'] = _ApiClient["default"].convertToType(data['public-key-pem-data'], 'String');
        }

        if (data.hasOwnProperty('signing-method')) {
          obj['signing-method'] = _ApiClient["default"].convertToType(data['signing-method'], 'String');
        }

        if (data.hasOwnProperty('street-address')) {
          obj['street-address'] = _ApiClient["default"].convertToType(data['street-address'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('uri-sans')) {
          obj['uri-sans'] = _ApiClient["default"].convertToType(data['uri-sans'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SignPKICertWithClassicKey;
}();
/**
 * The common name to be included in the PKI certificate
 * @member {String} common-name
 */


SignPKICertWithClassicKey.prototype['common-name'] = undefined;
/**
 * A comma-separated list of the country that will be set in the issued certificate
 * @member {String} country
 */

SignPKICertWithClassicKey.prototype['country'] = undefined;
/**
 * The name of the key to use in the sign PKI Cert process
 * @member {String} display-id
 */

SignPKICertWithClassicKey.prototype['display-id'] = undefined;
/**
 * DNS Names to be included in the PKI certificate (in a comma-delimited list)
 * @member {String} dns-names
 */

SignPKICertWithClassicKey.prototype['dns-names'] = undefined;
/**
 * key-usage
 * @member {String} key-usage
 * @default 'DigitalSignature,KeyAgreement,KeyEncipherment'
 */

SignPKICertWithClassicKey.prototype['key-usage'] = 'DigitalSignature,KeyAgreement,KeyEncipherment';
/**
 * A comma-separated list of the locality that will be set in the issued certificate
 * @member {String} locality
 */

SignPKICertWithClassicKey.prototype['locality'] = undefined;
/**
 * A comma-separated list of organizational units (OU) that will be set in the issued certificate
 * @member {String} organizational-units
 */

SignPKICertWithClassicKey.prototype['organizational-units'] = undefined;
/**
 * A comma-separated list of organizations (O) that will be set in the issued certificate
 * @member {String} organizations
 */

SignPKICertWithClassicKey.prototype['organizations'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

SignPKICertWithClassicKey.prototype['password'] = undefined;
/**
 * A comma-separated list of the postal code that will be set in the issued certificate
 * @member {String} postal-code
 */

SignPKICertWithClassicKey.prototype['postal-code'] = undefined;
/**
 * A comma-separated list of the province that will be set in the issued certificate
 * @member {String} province
 */

SignPKICertWithClassicKey.prototype['province'] = undefined;
/**
 * PublicKey using for signing in a PEM format.
 * @member {String} public-key-pem-data
 */

SignPKICertWithClassicKey.prototype['public-key-pem-data'] = undefined;
/**
 * SigningMethod
 * @member {String} signing-method
 */

SignPKICertWithClassicKey.prototype['signing-method'] = undefined;
/**
 * A comma-separated list of the street address that will be set in the issued certificate
 * @member {String} street-address
 */

SignPKICertWithClassicKey.prototype['street-address'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignPKICertWithClassicKey.prototype['token'] = undefined;
/**
 * he requested Time To Live for the certificate, in seconds
 * @member {Number} ttl
 */

SignPKICertWithClassicKey.prototype['ttl'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignPKICertWithClassicKey.prototype['uid-token'] = undefined;
/**
 * The URI Subject Alternative Names to be included in the PKI certificate (in a comma-delimited list)
 * @member {String} uri-sans
 */

SignPKICertWithClassicKey.prototype['uri-sans'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

SignPKICertWithClassicKey.prototype['username'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

SignPKICertWithClassicKey.prototype['version'] = undefined;
var _default = SignPKICertWithClassicKey;
exports["default"] = _default;
