// akeyless/dist/model/SignJWTWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignJWTWithClassicKey model module.
 * @module model/SignJWTWithClassicKey
 * @version 2.15.24
 */
var SignJWTWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignJWTWithClassicKey</code>.
   * @alias module:model/SignJWTWithClassicKey
   * @param displayId {String} The name of the key to use in the sign JWT process
   * @param jwtClaims {String} JWTClaims
   * @param signingMethod {String} SigningMethod
   * @param version {Number} classic key version
   */
  function SignJWTWithClassicKey(displayId, jwtClaims, signingMethod, version) {
    _classCallCheck(this, SignJWTWithClassicKey);

    SignJWTWithClassicKey.initialize(this, displayId, jwtClaims, signingMethod, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignJWTWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, displayId, jwtClaims, signingMethod, version) {
      obj['display-id'] = displayId;
      obj['jwt-claims'] = jwtClaims;
      obj['signing-method'] = signingMethod;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>SignJWTWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignJWTWithClassicKey} obj Optional instance to populate.
     * @return {module:model/SignJWTWithClassicKey} The populated <code>SignJWTWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignJWTWithClassicKey();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('jwt-claims')) {
          obj['jwt-claims'] = _ApiClient["default"].convertToType(data['jwt-claims'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('signing-method')) {
          obj['signing-method'] = _ApiClient["default"].convertToType(data['signing-method'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SignJWTWithClassicKey;
}();
/**
 * The name of the key to use in the sign JWT process
 * @member {String} display-id
 */


SignJWTWithClassicKey.prototype['display-id'] = undefined;
/**
 * JWTClaims
 * @member {String} jwt-claims
 */

SignJWTWithClassicKey.prototype['jwt-claims'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

SignJWTWithClassicKey.prototype['password'] = undefined;
/**
 * SigningMethod
 * @member {String} signing-method
 */

SignJWTWithClassicKey.prototype['signing-method'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignJWTWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignJWTWithClassicKey.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

SignJWTWithClassicKey.prototype['username'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

SignJWTWithClassicKey.prototype['version'] = undefined;
var _default = SignJWTWithClassicKey;
exports["default"] = _default;
