// akeyless/dist/model/SignPKCS1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SignPKCS1 model module.
 * @module model/SignPKCS1
 * @version 2.15.24
 */
var SignPKCS1 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SignPKCS1</code>.
   * signPKCS1 is a command that calculates the signature of hashed data using RSASSA-PKCS1-V1_5-SIGN from RSA PKCS#1 v1.5.
   * @alias module:model/SignPKCS1
   * @param keyName {String} The name of the RSA key to use in the signing process
   * @param message {String} The message to be signed
   */
  function SignPKCS1(keyName, message) {
    _classCallCheck(this, SignPKCS1);

    SignPKCS1.initialize(this, keyName, message);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SignPKCS1, null, [{
    key: "initialize",
    value: function initialize(obj, keyName, message) {
      obj['key-name'] = keyName;
      obj['message'] = message;
    }
    /**
     * Constructs a <code>SignPKCS1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SignPKCS1} obj Optional instance to populate.
     * @return {module:model/SignPKCS1} The populated <code>SignPKCS1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SignPKCS1();

        if (data.hasOwnProperty('key-name')) {
          obj['key-name'] = _ApiClient["default"].convertToType(data['key-name'], 'String');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SignPKCS1;
}();
/**
 * The name of the RSA key to use in the signing process
 * @member {String} key-name
 */


SignPKCS1.prototype['key-name'] = undefined;
/**
 * The message to be signed
 * @member {String} message
 */

SignPKCS1.prototype['message'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

SignPKCS1.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

SignPKCS1.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

SignPKCS1.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

SignPKCS1.prototype['username'] = undefined;
var _default = SignPKCS1;
exports["default"] = _default;
