// akeyless/dist/model/SplunkLogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SplunkLogForwardingConfig model module.
 * @module model/SplunkLogForwardingConfig
 * @version 2.20.0
 */
var SplunkLogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SplunkLogForwardingConfig</code>.
   * @alias module:model/SplunkLogForwardingConfig
   */
  function SplunkLogForwardingConfig() {
    _classCallCheck(this, SplunkLogForwardingConfig);

    SplunkLogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SplunkLogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SplunkLogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SplunkLogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/SplunkLogForwardingConfig} The populated <code>SplunkLogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SplunkLogForwardingConfig();

        if (data.hasOwnProperty('splunk_index')) {
          obj['splunk_index'] = _ApiClient["default"].convertToType(data['splunk_index'], 'String');
        }

        if (data.hasOwnProperty('splunk_source')) {
          obj['splunk_source'] = _ApiClient["default"].convertToType(data['splunk_source'], 'String');
        }

        if (data.hasOwnProperty('splunk_sourcetype')) {
          obj['splunk_sourcetype'] = _ApiClient["default"].convertToType(data['splunk_sourcetype'], 'String');
        }

        if (data.hasOwnProperty('splunk_token')) {
          obj['splunk_token'] = _ApiClient["default"].convertToType(data['splunk_token'], 'String');
        }

        if (data.hasOwnProperty('splunk_url')) {
          obj['splunk_url'] = _ApiClient["default"].convertToType(data['splunk_url'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SplunkLogForwardingConfig;
}();
/**
 * @member {String} splunk_index
 */


SplunkLogForwardingConfig.prototype['splunk_index'] = undefined;
/**
 * @member {String} splunk_source
 */

SplunkLogForwardingConfig.prototype['splunk_source'] = undefined;
/**
 * @member {String} splunk_sourcetype
 */

SplunkLogForwardingConfig.prototype['splunk_sourcetype'] = undefined;
/**
 * @member {String} splunk_token
 */

SplunkLogForwardingConfig.prototype['splunk_token'] = undefined;
/**
 * @member {String} splunk_url
 */

SplunkLogForwardingConfig.prototype['splunk_url'] = undefined;
var _default = SplunkLogForwardingConfig;
exports["default"] = _default;
