// akeyless/dist/model/StaticSecretDetailsInfo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The StaticSecretDetailsInfo model module.
 * @module model/StaticSecretDetailsInfo
 * @version 2.20.0
 */
var StaticSecretDetailsInfo = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>StaticSecretDetailsInfo</code>.
   * @alias module:model/StaticSecretDetailsInfo
   */
  function StaticSecretDetailsInfo() {
    _classCallCheck(this, StaticSecretDetailsInfo);

    StaticSecretDetailsInfo.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(StaticSecretDetailsInfo, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>StaticSecretDetailsInfo</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/StaticSecretDetailsInfo} obj Optional instance to populate.
     * @return {module:model/StaticSecretDetailsInfo} The populated <code>StaticSecretDetailsInfo</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new StaticSecretDetailsInfo();

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }

        if (data.hasOwnProperty('website')) {
          obj['website'] = _ApiClient["default"].convertToType(data['website'], 'String');
        }

        if (data.hasOwnProperty('websites')) {
          obj['websites'] = _ApiClient["default"].convertToType(data['websites'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return StaticSecretDetailsInfo;
}();
/**
 * @member {String} username
 */


StaticSecretDetailsInfo.prototype['username'] = undefined;
/**
 * deprecated
 * @member {String} website
 */

StaticSecretDetailsInfo.prototype['website'] = undefined;
/**
 * @member {Array.<String>} websites
 */

StaticSecretDetailsInfo.prototype['websites'] = undefined;
var _default = StaticSecretDetailsInfo;
exports["default"] = _default;
