// akeyless/dist/model/SyslogLogForwardingConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SyslogLogForwardingConfig model module.
 * @module model/SyslogLogForwardingConfig
 * @version 2.20.0
 */
var SyslogLogForwardingConfig = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SyslogLogForwardingConfig</code>.
   * @alias module:model/SyslogLogForwardingConfig
   */
  function SyslogLogForwardingConfig() {
    _classCallCheck(this, SyslogLogForwardingConfig);

    SyslogLogForwardingConfig.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SyslogLogForwardingConfig, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SyslogLogForwardingConfig</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SyslogLogForwardingConfig} obj Optional instance to populate.
     * @return {module:model/SyslogLogForwardingConfig} The populated <code>SyslogLogForwardingConfig</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SyslogLogForwardingConfig();

        if (data.hasOwnProperty('syslog_formatter')) {
          obj['syslog_formatter'] = _ApiClient["default"].convertToType(data['syslog_formatter'], 'String');
        }

        if (data.hasOwnProperty('syslog_host')) {
          obj['syslog_host'] = _ApiClient["default"].convertToType(data['syslog_host'], 'String');
        }

        if (data.hasOwnProperty('syslog_network')) {
          obj['syslog_network'] = _ApiClient["default"].convertToType(data['syslog_network'], 'String');
        }

        if (data.hasOwnProperty('syslog_target_tag')) {
          obj['syslog_target_tag'] = _ApiClient["default"].convertToType(data['syslog_target_tag'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SyslogLogForwardingConfig;
}();
/**
 * @member {String} syslog_formatter
 */


SyslogLogForwardingConfig.prototype['syslog_formatter'] = undefined;
/**
 * @member {String} syslog_host
 */

SyslogLogForwardingConfig.prototype['syslog_host'] = undefined;
/**
 * @member {String} syslog_network
 */

SyslogLogForwardingConfig.prototype['syslog_network'] = undefined;
/**
 * @member {String} syslog_target_tag
 */

SyslogLogForwardingConfig.prototype['syslog_target_tag'] = undefined;
var _default = SyslogLogForwardingConfig;
exports["default"] = _default;
