// akeyless/dist/model/SystemAccessCredsSettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SystemAccessCredsSettings model module.
 * @module model/SystemAccessCredsSettings
 * @version 2.20.0
 */
var SystemAccessCredsSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SystemAccessCredsSettings</code>.
   * SystemAccessCredsSettings describes system access credential settings for account by minutes
   * @alias module:model/SystemAccessCredsSettings
   */
  function SystemAccessCredsSettings() {
    _classCallCheck(this, SystemAccessCredsSettings);

    SystemAccessCredsSettings.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SystemAccessCredsSettings, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SystemAccessCredsSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SystemAccessCredsSettings} obj Optional instance to populate.
     * @return {module:model/SystemAccessCredsSettings} The populated <code>SystemAccessCredsSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SystemAccessCredsSettings();

        if (data.hasOwnProperty('jwt_ttl_default')) {
          obj['jwt_ttl_default'] = _ApiClient["default"].convertToType(data['jwt_ttl_default'], 'Number');
        }

        if (data.hasOwnProperty('jwt_ttl_maximum')) {
          obj['jwt_ttl_maximum'] = _ApiClient["default"].convertToType(data['jwt_ttl_maximum'], 'Number');
        }

        if (data.hasOwnProperty('jwt_ttl_minimum')) {
          obj['jwt_ttl_minimum'] = _ApiClient["default"].convertToType(data['jwt_ttl_minimum'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return SystemAccessCredsSettings;
}();
/**
 * @member {Number} jwt_ttl_default
 */


SystemAccessCredsSettings.prototype['jwt_ttl_default'] = undefined;
/**
 * @member {Number} jwt_ttl_maximum
 */

SystemAccessCredsSettings.prototype['jwt_ttl_maximum'] = undefined;
/**
 * @member {Number} jwt_ttl_minimum
 */

SystemAccessCredsSettings.prototype['jwt_ttl_minimum'] = undefined;
var _default = SystemAccessCredsSettings;
exports["default"] = _default;
