// akeyless/dist/model/SystemAccessCredentialsReplyObj.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SystemAccessCredentialsReplyObj model module.
 * @module model/SystemAccessCredentialsReplyObj
 * @version 2.20.0
 */
var SystemAccessCredentialsReplyObj = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SystemAccessCredentialsReplyObj</code>.
   * @alias module:model/SystemAccessCredentialsReplyObj
   */
  function SystemAccessCredentialsReplyObj() {
    _classCallCheck(this, SystemAccessCredentialsReplyObj);

    SystemAccessCredentialsReplyObj.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SystemAccessCredentialsReplyObj, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SystemAccessCredentialsReplyObj</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SystemAccessCredentialsReplyObj} obj Optional instance to populate.
     * @return {module:model/SystemAccessCredentialsReplyObj} The populated <code>SystemAccessCredentialsReplyObj</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SystemAccessCredentialsReplyObj();

        if (data.hasOwnProperty('auth_creds')) {
          obj['auth_creds'] = _ApiClient["default"].convertToType(data['auth_creds'], 'String');
        }

        if (data.hasOwnProperty('expiry')) {
          obj['expiry'] = _ApiClient["default"].convertToType(data['expiry'], 'Number');
        }

        if (data.hasOwnProperty('kfm_creds')) {
          obj['kfm_creds'] = _ApiClient["default"].convertToType(data['kfm_creds'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uam_creds')) {
          obj['uam_creds'] = _ApiClient["default"].convertToType(data['uam_creds'], 'String');
        }
      }

      return obj;
    }
  }]);

  return SystemAccessCredentialsReplyObj;
}();
/**
 * Temporary credentials for accessing Auth
 * @member {String} auth_creds
 */


SystemAccessCredentialsReplyObj.prototype['auth_creds'] = undefined;
/**
 * Credentials expiration date
 * @member {Number} expiry
 */

SystemAccessCredentialsReplyObj.prototype['expiry'] = undefined;
/**
 * Temporary credentials for accessing the KFMs instances
 * @member {String} kfm_creds
 */

SystemAccessCredentialsReplyObj.prototype['kfm_creds'] = undefined;
/**
 * Credentials tmp token
 * @member {String} token
 */

SystemAccessCredentialsReplyObj.prototype['token'] = undefined;
/**
 * Temporary credentials for accessing the UAM service
 * @member {String} uam_creds
 */

SystemAccessCredentialsReplyObj.prototype['uam_creds'] = undefined;
var _default = SystemAccessCredentialsReplyObj;
exports["default"] = _default;
