// akeyless/dist/model/Target.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ItemVersion = _interopRequireDefault(require("./ItemVersion"));

var _TargetItemAssociation = _interopRequireDefault(require("./TargetItemAssociation"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Target model module.
 * @module model/Target
 * @version 2.20.0
 */
var Target = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Target</code>.
   * @alias module:model/Target
   */
  function Target() {
    _classCallCheck(this, Target);

    Target.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Target, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Target</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Target} obj Optional instance to populate.
     * @return {module:model/Target} The populated <code>Target</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Target();

        if (data.hasOwnProperty('access_date')) {
          obj['access_date'] = _ApiClient["default"].convertToType(data['access_date'], 'Date');
        }

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _ApiClient["default"].convertToType(data['attributes'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('client_permissions')) {
          obj['client_permissions'] = _ApiClient["default"].convertToType(data['client_permissions'], ['String']);
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('credentials_less')) {
          obj['credentials_less'] = _ApiClient["default"].convertToType(data['credentials_less'], 'Boolean');
        }

        if (data.hasOwnProperty('last_version')) {
          obj['last_version'] = _ApiClient["default"].convertToType(data['last_version'], 'Number');
        }

        if (data.hasOwnProperty('modification_date')) {
          obj['modification_date'] = _ApiClient["default"].convertToType(data['modification_date'], 'Date');
        }

        if (data.hasOwnProperty('protection_key_name')) {
          obj['protection_key_name'] = _ApiClient["default"].convertToType(data['protection_key_name'], 'String');
        }

        if (data.hasOwnProperty('target_id')) {
          obj['target_id'] = _ApiClient["default"].convertToType(data['target_id'], 'Number');
        }

        if (data.hasOwnProperty('target_items_assoc')) {
          obj['target_items_assoc'] = _ApiClient["default"].convertToType(data['target_items_assoc'], [_TargetItemAssociation["default"]]);
        }

        if (data.hasOwnProperty('target_name')) {
          obj['target_name'] = _ApiClient["default"].convertToType(data['target_name'], 'String');
        }

        if (data.hasOwnProperty('target_type')) {
          obj['target_type'] = _ApiClient["default"].convertToType(data['target_type'], 'String');
        }

        if (data.hasOwnProperty('target_versions')) {
          obj['target_versions'] = _ApiClient["default"].convertToType(data['target_versions'], [_ItemVersion["default"]]);
        }

        if (data.hasOwnProperty('with_customer_fragment')) {
          obj['with_customer_fragment'] = _ApiClient["default"].convertToType(data['with_customer_fragment'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return Target;
}();
/**
 * @member {Date} access_date
 */


Target.prototype['access_date'] = undefined;
/**
 * this is not \"omitempty\" since an empty value causes no update while an empty map will clear the attributes
 * @member {Object.<String, Object>} attributes
 */

Target.prototype['attributes'] = undefined;
/**
 * @member {Array.<String>} client_permissions
 */

Target.prototype['client_permissions'] = undefined;
/**
 * @member {String} comment
 */

Target.prototype['comment'] = undefined;
/**
 * @member {Date} creation_date
 */

Target.prototype['creation_date'] = undefined;
/**
 * @member {Boolean} credentials_less
 */

Target.prototype['credentials_less'] = undefined;
/**
 * @member {Number} last_version
 */

Target.prototype['last_version'] = undefined;
/**
 * @member {Date} modification_date
 */

Target.prototype['modification_date'] = undefined;
/**
 * @member {String} protection_key_name
 */

Target.prototype['protection_key_name'] = undefined;
/**
 * @member {Number} target_id
 */

Target.prototype['target_id'] = undefined;
/**
 * @member {Array.<module:model/TargetItemAssociation>} target_items_assoc
 */

Target.prototype['target_items_assoc'] = undefined;
/**
 * @member {String} target_name
 */

Target.prototype['target_name'] = undefined;
/**
 * @member {String} target_type
 */

Target.prototype['target_type'] = undefined;
/**
 * @member {Array.<module:model/ItemVersion>} target_versions
 */

Target.prototype['target_versions'] = undefined;
/**
 * @member {Boolean} with_customer_fragment
 */

Target.prototype['with_customer_fragment'] = undefined;
var _default = Target;
exports["default"] = _default;
