// akeyless/dist/model/TargetItemAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetItemAssociation model module.
 * @module model/TargetItemAssociation
 * @version 2.20.0
 */
var TargetItemAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetItemAssociation</code>.
   * TargetItemAssociation includes details of an association between a target and an item.
   * @alias module:model/TargetItemAssociation
   */
  function TargetItemAssociation() {
    _classCallCheck(this, TargetItemAssociation);

    TargetItemAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetItemAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TargetItemAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetItemAssociation} obj Optional instance to populate.
     * @return {module:model/TargetItemAssociation} The populated <code>TargetItemAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetItemAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _ApiClient["default"].convertToType(data['attributes'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('cluster_id')) {
          obj['cluster_id'] = _ApiClient["default"].convertToType(data['cluster_id'], 'Number');
        }

        if (data.hasOwnProperty('item_name')) {
          obj['item_name'] = _ApiClient["default"].convertToType(data['item_name'], 'String');
        }

        if (data.hasOwnProperty('item_type')) {
          obj['item_type'] = _ApiClient["default"].convertToType(data['item_type'], 'String');
        }

        if (data.hasOwnProperty('relationship')) {
          obj['relationship'] = _ApiClient["default"].convertToType(data['relationship'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetItemAssociation;
}();
/**
 * @member {String} assoc_id
 */


TargetItemAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {Object.<String, String>} attributes
 */

TargetItemAssociation.prototype['attributes'] = undefined;
/**
 * @member {Number} cluster_id
 */

TargetItemAssociation.prototype['cluster_id'] = undefined;
/**
 * @member {String} item_name
 */

TargetItemAssociation.prototype['item_name'] = undefined;
/**
 * @member {String} item_type
 */

TargetItemAssociation.prototype['item_type'] = undefined;
/**
 * @member {String} relationship
 */

TargetItemAssociation.prototype['relationship'] = undefined;
var _default = TargetItemAssociation;
exports["default"] = _default;
