// akeyless/dist/model/TargetObjectAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TargetObjectAssociation model module.
 * @module model/TargetObjectAssociation
 * @version 2.19.0
 */
var TargetObjectAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TargetObjectAssociation</code>.
   * TargetObjectAssociation includes details of an association between a target and an object (target, migration_id).
   * @alias module:model/TargetObjectAssociation
   */
  function TargetObjectAssociation() {
    _classCallCheck(this, TargetObjectAssociation);

    TargetObjectAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TargetObjectAssociation, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TargetObjectAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TargetObjectAssociation} obj Optional instance to populate.
     * @return {module:model/TargetObjectAssociation} The populated <code>TargetObjectAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TargetObjectAssociation();

        if (data.hasOwnProperty('assoc_id')) {
          obj['assoc_id'] = _ApiClient["default"].convertToType(data['assoc_id'], 'String');
        }

        if (data.hasOwnProperty('attributes')) {
          obj['attributes'] = _ApiClient["default"].convertToType(data['attributes'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('object_id')) {
          obj['object_id'] = _ApiClient["default"].convertToType(data['object_id'], 'String');
        }

        if (data.hasOwnProperty('object_name')) {
          obj['object_name'] = _ApiClient["default"].convertToType(data['object_name'], 'String');
        }

        if (data.hasOwnProperty('object_type')) {
          obj['object_type'] = _ApiClient["default"].convertToType(data['object_type'], 'String');
        }

        if (data.hasOwnProperty('target_name')) {
          obj['target_name'] = _ApiClient["default"].convertToType(data['target_name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return TargetObjectAssociation;
}();
/**
 * @member {String} assoc_id
 */


TargetObjectAssociation.prototype['assoc_id'] = undefined;
/**
 * @member {Object.<String, String>} attributes
 */

TargetObjectAssociation.prototype['attributes'] = undefined;
/**
 * @member {String} object_id
 */

TargetObjectAssociation.prototype['object_id'] = undefined;
/**
 * @member {String} object_name
 */

TargetObjectAssociation.prototype['object_name'] = undefined;
/**
 * @member {String} object_type
 */

TargetObjectAssociation.prototype['object_type'] = undefined;
/**
 * @member {String} target_name
 */

TargetObjectAssociation.prototype['target_name'] = undefined;
var _default = TargetObjectAssociation;
exports["default"] = _default;
