// akeyless/dist/model/TmpUserData.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The TmpUserData model module.
 * @module model/TmpUserData
 * @version 2.20.0
 */
var TmpUserData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TmpUserData</code>.
   * @alias module:model/TmpUserData
   */
  function TmpUserData() {
    _classCallCheck(this, TmpUserData);

    TmpUserData.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(TmpUserData, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>TmpUserData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TmpUserData} obj Optional instance to populate.
     * @return {module:model/TmpUserData} The populated <code>TmpUserData</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TmpUserData();

        if (data.hasOwnProperty('access_id')) {
          obj['access_id'] = _ApiClient["default"].convertToType(data['access_id'], 'String');
        }

        if (data.hasOwnProperty('creation_date')) {
          obj['creation_date'] = _ApiClient["default"].convertToType(data['creation_date'], 'Date');
        }

        if (data.hasOwnProperty('custom_ttl')) {
          obj['custom_ttl'] = _ApiClient["default"].convertToType(data['custom_ttl'], 'Number');
        }

        if (data.hasOwnProperty('dynamic_secret_type')) {
          obj['dynamic_secret_type'] = _ApiClient["default"].convertToType(data['dynamic_secret_type'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('sub_claims')) {
          obj['sub_claims'] = _ApiClient["default"].convertToType(data['sub_claims'], {
            'String': ['String']
          });
        }
      }

      return obj;
    }
  }]);

  return TmpUserData;
}();
/**
 * @member {String} access_id
 */


TmpUserData.prototype['access_id'] = undefined;
/**
 * @member {Date} creation_date
 */

TmpUserData.prototype['creation_date'] = undefined;
/**
 * @member {Number} custom_ttl
 */

TmpUserData.prototype['custom_ttl'] = undefined;
/**
 * @member {String} dynamic_secret_type
 */

TmpUserData.prototype['dynamic_secret_type'] = undefined;
/**
 * @member {String} host
 */

TmpUserData.prototype['host'] = undefined;
/**
 * @member {String} id
 */

TmpUserData.prototype['id'] = undefined;
/**
 * @member {Object.<String, Array.<String>>} sub_claims
 */

TmpUserData.prototype['sub_claims'] = undefined;
var _default = TmpUserData;
exports["default"] = _default;
