// akeyless/dist/model/Tokenize.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Tokenize model module.
 * @module model/Tokenize
 * @version 2.20.0
 */
var Tokenize = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Tokenize</code>.
   * tokenize is a command that encrypts text with a tokenizer
   * @alias module:model/Tokenize
   * @param plaintext {String} Data to be encrypted
   * @param tokenizerName {String} The name of the tokenizer to use in the encryption process
   */
  function Tokenize(plaintext, tokenizerName) {
    _classCallCheck(this, Tokenize);

    Tokenize.initialize(this, plaintext, tokenizerName);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Tokenize, null, [{
    key: "initialize",
    value: function initialize(obj, plaintext, tokenizerName) {
      obj['plaintext'] = plaintext;
      obj['tokenizer-name'] = tokenizerName;
    }
    /**
     * Constructs a <code>Tokenize</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Tokenize} obj Optional instance to populate.
     * @return {module:model/Tokenize} The populated <code>Tokenize</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Tokenize();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('plaintext')) {
          obj['plaintext'] = _ApiClient["default"].convertToType(data['plaintext'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('tokenizer-name')) {
          obj['tokenizer-name'] = _ApiClient["default"].convertToType(data['tokenizer-name'], 'String');
        }

        if (data.hasOwnProperty('tweak')) {
          obj['tweak'] = _ApiClient["default"].convertToType(data['tweak'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Tokenize;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 */


Tokenize.prototype['json'] = undefined;
/**
 * Data to be encrypted
 * @member {String} plaintext
 */

Tokenize.prototype['plaintext'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

Tokenize.prototype['token'] = undefined;
/**
 * The name of the tokenizer to use in the encryption process
 * @member {String} tokenizer-name
 */

Tokenize.prototype['tokenizer-name'] = undefined;
/**
 * Base64 encoded tweak for vaultless encryption
 * @member {String} tweak
 */

Tokenize.prototype['tweak'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

Tokenize.prototype['uid-token'] = undefined;
var _default = Tokenize;
exports["default"] = _default;
