// akeyless/dist/model/UidCreateChildToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UidCreateChildToken model module.
 * @module model/UidCreateChildToken
 * @version 2.20.0
 */
var UidCreateChildToken = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UidCreateChildToken</code>.
   * uidCreateChildToken is a command that creates a new child token using Akeyless Universal Identity.
   * @alias module:model/UidCreateChildToken
   */
  function UidCreateChildToken() {
    _classCallCheck(this, UidCreateChildToken);

    UidCreateChildToken.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UidCreateChildToken, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UidCreateChildToken</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UidCreateChildToken} obj Optional instance to populate.
     * @return {module:model/UidCreateChildToken} The populated <code>UidCreateChildToken</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UidCreateChildToken();

        if (data.hasOwnProperty('auth-method-name')) {
          obj['auth-method-name'] = _ApiClient["default"].convertToType(data['auth-method-name'], 'String');
        }

        if (data.hasOwnProperty('child-deny-inheritance')) {
          obj['child-deny-inheritance'] = _ApiClient["default"].convertToType(data['child-deny-inheritance'], 'Boolean');
        }

        if (data.hasOwnProperty('child-deny-rotate')) {
          obj['child-deny-rotate'] = _ApiClient["default"].convertToType(data['child-deny-rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('child-ttl')) {
          obj['child-ttl'] = _ApiClient["default"].convertToType(data['child-ttl'], 'Number');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('uid-token-id')) {
          obj['uid-token-id'] = _ApiClient["default"].convertToType(data['uid-token-id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UidCreateChildToken;
}();
/**
 * The universal identity auth method name, required only when uid-token is not provided
 * @member {String} auth-method-name
 */


UidCreateChildToken.prototype['auth-method-name'] = undefined;
/**
 * Deny from new child to create their own children
 * @member {Boolean} child-deny-inheritance
 */

UidCreateChildToken.prototype['child-deny-inheritance'] = undefined;
/**
 * Deny from new child to rotate
 * @member {Boolean} child-deny-rotate
 */

UidCreateChildToken.prototype['child-deny-rotate'] = undefined;
/**
 * New child token ttl
 * @member {Number} child-ttl
 */

UidCreateChildToken.prototype['child-ttl'] = undefined;
/**
 * New Token comment
 * @member {String} comment
 */

UidCreateChildToken.prototype['comment'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UidCreateChildToken.prototype['json'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UidCreateChildToken.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UidCreateChildToken.prototype['uid-token'] = undefined;
/**
 * The ID of the uid-token, required only when uid-token is not provided
 * @member {String} uid-token-id
 */

UidCreateChildToken.prototype['uid-token-id'] = undefined;
var _default = UidCreateChildToken;
exports["default"] = _default;
