// akeyless/dist/model/UIdentityConfigPart.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UIdentityConfigPart model module.
 * @module model/UIdentityConfigPart
 * @version 2.15.27
 */
var UIdentityConfigPart = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UIdentityConfigPart</code>.
   * @alias module:model/UIdentityConfigPart
   */
  function UIdentityConfigPart() {
    _classCallCheck(this, UIdentityConfigPart);

    UIdentityConfigPart.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UIdentityConfigPart, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UIdentityConfigPart</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UIdentityConfigPart} obj Optional instance to populate.
     * @return {module:model/UIdentityConfigPart} The populated <code>UIdentityConfigPart</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UIdentityConfigPart();

        if (data.hasOwnProperty('uid_access_id')) {
          obj['uid_access_id'] = _ApiClient["default"].convertToType(data['uid_access_id'], 'String');
        }

        if (data.hasOwnProperty('uid_enable')) {
          obj['uid_enable'] = _ApiClient["default"].convertToType(data['uid_enable'], 'Boolean');
        }

        if (data.hasOwnProperty('uid_rotate_interval')) {
          obj['uid_rotate_interval'] = _ApiClient["default"].convertToType(data['uid_rotate_interval'], 'String');
        }

        if (data.hasOwnProperty('uid_status')) {
          obj['uid_status'] = _ApiClient["default"].convertToType(data['uid_status'], 'String');
        }

        if (data.hasOwnProperty('uid_token')) {
          obj['uid_token'] = _ApiClient["default"].convertToType(data['uid_token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UIdentityConfigPart;
}();
/**
 * @member {String} uid_access_id
 */


UIdentityConfigPart.prototype['uid_access_id'] = undefined;
/**
 * @member {Boolean} uid_enable
 */

UIdentityConfigPart.prototype['uid_enable'] = undefined;
/**
 * @member {String} uid_rotate_interval
 */

UIdentityConfigPart.prototype['uid_rotate_interval'] = undefined;
/**
 * @member {String} uid_status
 */

UIdentityConfigPart.prototype['uid_status'] = undefined;
/**
 * @member {String} uid_token
 */

UIdentityConfigPart.prototype['uid_token'] = undefined;
var _default = UIdentityConfigPart;
exports["default"] = _default;
