// akeyless/dist/model/UidRevokeToken.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UidRevokeToken model module.
 * @module model/UidRevokeToken
 * @version 2.15.24
 */
var UidRevokeToken = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UidRevokeToken</code>.
   * @alias module:model/UidRevokeToken
   * @param revokeToken {String} the universal identity token/token-id to revoke
   * @param revokeType {String} revokeSelf/revokeAll (delete only this token/this token and his children)
   */
  function UidRevokeToken(revokeToken, revokeType) {
    _classCallCheck(this, UidRevokeToken);

    UidRevokeToken.initialize(this, revokeToken, revokeType);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UidRevokeToken, null, [{
    key: "initialize",
    value: function initialize(obj, revokeToken, revokeType) {
      obj['revoke-token'] = revokeToken;
      obj['revoke-type'] = revokeType;
    }
    /**
     * Constructs a <code>UidRevokeToken</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UidRevokeToken} obj Optional instance to populate.
     * @return {module:model/UidRevokeToken} The populated <code>UidRevokeToken</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UidRevokeToken();

        if (data.hasOwnProperty('auth-method-name')) {
          obj['auth-method-name'] = _ApiClient["default"].convertToType(data['auth-method-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('revoke-token')) {
          obj['revoke-token'] = _ApiClient["default"].convertToType(data['revoke-token'], 'String');
        }

        if (data.hasOwnProperty('revoke-type')) {
          obj['revoke-type'] = _ApiClient["default"].convertToType(data['revoke-type'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UidRevokeToken;
}();
/**
 * The universal identity auth method name
 * @member {String} auth-method-name
 */


UidRevokeToken.prototype['auth-method-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UidRevokeToken.prototype['password'] = undefined;
/**
 * the universal identity token/token-id to revoke
 * @member {String} revoke-token
 */

UidRevokeToken.prototype['revoke-token'] = undefined;
/**
 * revokeSelf/revokeAll (delete only this token/this token and his children)
 * @member {String} revoke-type
 */

UidRevokeToken.prototype['revoke-type'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UidRevokeToken.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UidRevokeToken.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UidRevokeToken.prototype['username'] = undefined;
var _default = UidRevokeToken;
exports["default"] = _default;
