// akeyless/dist/model/UIDTokenDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UIDTokenDetails model module.
 * @module model/UIDTokenDetails
 * @version 2.15.24
 */
var UIDTokenDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UIDTokenDetails</code>.
   * @alias module:model/UIDTokenDetails
   */
  function UIDTokenDetails() {
    _classCallCheck(this, UIDTokenDetails);

    UIDTokenDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UIDTokenDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UIDTokenDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UIDTokenDetails} obj Optional instance to populate.
     * @return {module:model/UIDTokenDetails} The populated <code>UIDTokenDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UIDTokenDetails();

        if (data.hasOwnProperty('children')) {
          obj['children'] = _ApiClient["default"].convertToType(data['children'], {
            'String': UIDTokenDetails
          });
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('deny_inheritance')) {
          obj['deny_inheritance'] = _ApiClient["default"].convertToType(data['deny_inheritance'], 'Boolean');
        }

        if (data.hasOwnProperty('deny_rotate')) {
          obj['deny_rotate'] = _ApiClient["default"].convertToType(data['deny_rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('depth')) {
          obj['depth'] = _ApiClient["default"].convertToType(data['depth'], 'Number');
        }

        if (data.hasOwnProperty('expired_date')) {
          obj['expired_date'] = _ApiClient["default"].convertToType(data['expired_date'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('last_rotate')) {
          obj['last_rotate'] = _ApiClient["default"].convertToType(data['last_rotate'], 'String');
        }

        if (data.hasOwnProperty('revoked')) {
          obj['revoked'] = _ApiClient["default"].convertToType(data['revoked'], 'Boolean');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return UIDTokenDetails;
}();
/**
 * @member {Object.<String, module:model/UIDTokenDetails>} children
 */


UIDTokenDetails.prototype['children'] = undefined;
/**
 * @member {String} comment
 */

UIDTokenDetails.prototype['comment'] = undefined;
/**
 * @member {Boolean} deny_inheritance
 */

UIDTokenDetails.prototype['deny_inheritance'] = undefined;
/**
 * @member {Boolean} deny_rotate
 */

UIDTokenDetails.prototype['deny_rotate'] = undefined;
/**
 * @member {Number} depth
 */

UIDTokenDetails.prototype['depth'] = undefined;
/**
 * @member {String} expired_date
 */

UIDTokenDetails.prototype['expired_date'] = undefined;
/**
 * @member {String} id
 */

UIDTokenDetails.prototype['id'] = undefined;
/**
 * @member {String} last_rotate
 */

UIDTokenDetails.prototype['last_rotate'] = undefined;
/**
 * @member {Boolean} revoked
 */

UIDTokenDetails.prototype['revoked'] = undefined;
/**
 * @member {Number} ttl
 */

UIDTokenDetails.prototype['ttl'] = undefined;
var _default = UIDTokenDetails;
exports["default"] = _default;
