// akeyless/dist/model/UniversalIdentityDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _UIDTokenDetails = _interopRequireDefault(require("./UIDTokenDetails"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UniversalIdentityDetails model module.
 * @module model/UniversalIdentityDetails
 * @version 2.20.0
 */
var UniversalIdentityDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UniversalIdentityDetails</code>.
   * @alias module:model/UniversalIdentityDetails
   */
  function UniversalIdentityDetails() {
    _classCallCheck(this, UniversalIdentityDetails);

    UniversalIdentityDetails.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UniversalIdentityDetails, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UniversalIdentityDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UniversalIdentityDetails} obj Optional instance to populate.
     * @return {module:model/UniversalIdentityDetails} The populated <code>UniversalIdentityDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UniversalIdentityDetails();

        if (data.hasOwnProperty('max_depth')) {
          obj['max_depth'] = _ApiClient["default"].convertToType(data['max_depth'], 'Number');
        }

        if (data.hasOwnProperty('number_of_tokens')) {
          obj['number_of_tokens'] = _ApiClient["default"].convertToType(data['number_of_tokens'], 'Number');
        }

        if (data.hasOwnProperty('root')) {
          obj['root'] = _UIDTokenDetails["default"].constructFromObject(data['root']);
        }
      }

      return obj;
    }
  }]);

  return UniversalIdentityDetails;
}();
/**
 * @member {Number} max_depth
 */


UniversalIdentityDetails.prototype['max_depth'] = undefined;
/**
 * @member {Number} number_of_tokens
 */

UniversalIdentityDetails.prototype['number_of_tokens'] = undefined;
/**
 * @member {module:model/UIDTokenDetails} root
 */

UniversalIdentityDetails.prototype['root'] = undefined;
var _default = UniversalIdentityDetails;
exports["default"] = _default;
