// akeyless/dist/model/UniversalIdentityAccessRules.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UniversalIdentityAccessRules model module.
 * @module model/UniversalIdentityAccessRules
 * @version 2.20.0
 */
var UniversalIdentityAccessRules = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UniversalIdentityAccessRules</code>.
   * @alias module:model/UniversalIdentityAccessRules
   */
  function UniversalIdentityAccessRules() {
    _classCallCheck(this, UniversalIdentityAccessRules);

    UniversalIdentityAccessRules.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UniversalIdentityAccessRules, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UniversalIdentityAccessRules</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UniversalIdentityAccessRules} obj Optional instance to populate.
     * @return {module:model/UniversalIdentityAccessRules} The populated <code>UniversalIdentityAccessRules</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UniversalIdentityAccessRules();

        if (data.hasOwnProperty('deny_inheritance')) {
          obj['deny_inheritance'] = _ApiClient["default"].convertToType(data['deny_inheritance'], 'Boolean');
        }

        if (data.hasOwnProperty('deny_rotate')) {
          obj['deny_rotate'] = _ApiClient["default"].convertToType(data['deny_rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('ttl')) {
          obj['ttl'] = _ApiClient["default"].convertToType(data['ttl'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return UniversalIdentityAccessRules;
}();
/**
 * @member {Boolean} deny_inheritance
 */


UniversalIdentityAccessRules.prototype['deny_inheritance'] = undefined;
/**
 * @member {Boolean} deny_rotate
 */

UniversalIdentityAccessRules.prototype['deny_rotate'] = undefined;
/**
 * @member {Number} ttl
 */

UniversalIdentityAccessRules.prototype['ttl'] = undefined;
var _default = UniversalIdentityAccessRules;
exports["default"] = _default;
