// akeyless/dist/model/UpdateDBTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateDBTarget model module.
 * @module model/UpdateDBTarget
 * @version 2.20.0
 */
var UpdateDBTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateDBTarget</code>.
   * @alias module:model/UpdateDBTarget
   * @param dbType {String} 
   * @param name {String} Target name
   */
  function UpdateDBTarget(dbType, name) {
    _classCallCheck(this, UpdateDBTarget);

    UpdateDBTarget.initialize(this, dbType, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateDBTarget, null, [{
    key: "initialize",
    value: function initialize(obj, dbType, name) {
      obj['db-type'] = dbType;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateDBTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateDBTarget} obj Optional instance to populate.
     * @return {module:model/UpdateDBTarget} The populated <code>UpdateDBTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateDBTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('db-name')) {
          obj['db-name'] = _ApiClient["default"].convertToType(data['db-name'], 'String');
        }

        if (data.hasOwnProperty('db-server-certificates')) {
          obj['db-server-certificates'] = _ApiClient["default"].convertToType(data['db-server-certificates'], 'String');
        }

        if (data.hasOwnProperty('db-server-name')) {
          obj['db-server-name'] = _ApiClient["default"].convertToType(data['db-server-name'], 'String');
        }

        if (data.hasOwnProperty('db-type')) {
          obj['db-type'] = _ApiClient["default"].convertToType(data['db-type'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('mongodb-atlas')) {
          obj['mongodb-atlas'] = _ApiClient["default"].convertToType(data['mongodb-atlas'], 'Boolean');
        }

        if (data.hasOwnProperty('mongodb-atlas-api-private-key')) {
          obj['mongodb-atlas-api-private-key'] = _ApiClient["default"].convertToType(data['mongodb-atlas-api-private-key'], 'String');
        }

        if (data.hasOwnProperty('mongodb-atlas-api-public-key')) {
          obj['mongodb-atlas-api-public-key'] = _ApiClient["default"].convertToType(data['mongodb-atlas-api-public-key'], 'String');
        }

        if (data.hasOwnProperty('mongodb-atlas-project-id')) {
          obj['mongodb-atlas-project-id'] = _ApiClient["default"].convertToType(data['mongodb-atlas-project-id'], 'String');
        }

        if (data.hasOwnProperty('mongodb-default-auth-db')) {
          obj['mongodb-default-auth-db'] = _ApiClient["default"].convertToType(data['mongodb-default-auth-db'], 'String');
        }

        if (data.hasOwnProperty('mongodb-uri-options')) {
          obj['mongodb-uri-options'] = _ApiClient["default"].convertToType(data['mongodb-uri-options'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('oracle-service-name')) {
          obj['oracle-service-name'] = _ApiClient["default"].convertToType(data['oracle-service-name'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('pwd')) {
          obj['pwd'] = _ApiClient["default"].convertToType(data['pwd'], 'String');
        }

        if (data.hasOwnProperty('snowflake-account')) {
          obj['snowflake-account'] = _ApiClient["default"].convertToType(data['snowflake-account'], 'String');
        }

        if (data.hasOwnProperty('ssl')) {
          obj['ssl'] = _ApiClient["default"].convertToType(data['ssl'], 'Boolean');
        }

        if (data.hasOwnProperty('ssl-certificate')) {
          obj['ssl-certificate'] = _ApiClient["default"].convertToType(data['ssl-certificate'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('user-name')) {
          obj['user-name'] = _ApiClient["default"].convertToType(data['user-name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateDBTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateDBTarget.prototype['comment'] = undefined;
/**
 * @member {String} db-name
 */

UpdateDBTarget.prototype['db-name'] = undefined;
/**
 * (Optional) DB server certificates
 * @member {String} db-server-certificates
 */

UpdateDBTarget.prototype['db-server-certificates'] = undefined;
/**
 * (Optional) Server name for certificate verification
 * @member {String} db-server-name
 */

UpdateDBTarget.prototype['db-server-name'] = undefined;
/**
 * @member {String} db-type
 */

UpdateDBTarget.prototype['db-type'] = undefined;
/**
 * @member {String} host
 */

UpdateDBTarget.prototype['host'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateDBTarget.prototype['json'] = undefined;
/**
 * @member {String} keep-prev-version
 */

UpdateDBTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateDBTarget.prototype['key'] = undefined;
/**
 * @member {Boolean} mongodb-atlas
 */

UpdateDBTarget.prototype['mongodb-atlas'] = undefined;
/**
 * MongoDB Atlas private key
 * @member {String} mongodb-atlas-api-private-key
 */

UpdateDBTarget.prototype['mongodb-atlas-api-private-key'] = undefined;
/**
 * MongoDB Atlas public key
 * @member {String} mongodb-atlas-api-public-key
 */

UpdateDBTarget.prototype['mongodb-atlas-api-public-key'] = undefined;
/**
 * MongoDB Atlas project ID
 * @member {String} mongodb-atlas-project-id
 */

UpdateDBTarget.prototype['mongodb-atlas-project-id'] = undefined;
/**
 * MongoDB server default authentication database
 * @member {String} mongodb-default-auth-db
 */

UpdateDBTarget.prototype['mongodb-default-auth-db'] = undefined;
/**
 * MongoDB server URI options
 * @member {String} mongodb-uri-options
 */

UpdateDBTarget.prototype['mongodb-uri-options'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateDBTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateDBTarget.prototype['new-name'] = undefined;
/**
 * @member {String} oracle-service-name
 */

UpdateDBTarget.prototype['oracle-service-name'] = undefined;
/**
 * @member {String} port
 */

UpdateDBTarget.prototype['port'] = undefined;
/**
 * @member {String} pwd
 */

UpdateDBTarget.prototype['pwd'] = undefined;
/**
 * @member {String} snowflake-account
 */

UpdateDBTarget.prototype['snowflake-account'] = undefined;
/**
 * SSL connection mode
 * @member {Boolean} ssl
 */

UpdateDBTarget.prototype['ssl'] = undefined;
/**
 * SSL connection certificate
 * @member {String} ssl-certificate
 */

UpdateDBTarget.prototype['ssl-certificate'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateDBTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateDBTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateDBTarget.prototype['update-version'] = undefined;
/**
 * @member {String} user-name
 */

UpdateDBTarget.prototype['user-name'] = undefined;
var _default = UpdateDBTarget;
exports["default"] = _default;
