// akeyless/dist/model/UpdateSecretVal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateSecretVal model module.
 * @module model/UpdateSecretVal
 * @version 2.20.0
 */
var UpdateSecretVal = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSecretVal</code>.
   * @alias module:model/UpdateSecretVal
   * @param name {String} Secret name
   * @param value {String} The new secret value
   */
  function UpdateSecretVal(name, value) {
    _classCallCheck(this, UpdateSecretVal);

    UpdateSecretVal.initialize(this, name, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateSecretVal, null, [{
    key: "initialize",
    value: function initialize(obj, name, value) {
      obj['name'] = name;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>UpdateSecretVal</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSecretVal} obj Optional instance to populate.
     * @return {module:model/UpdateSecretVal} The populated <code>UpdateSecretVal</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSecretVal();

        if (data.hasOwnProperty('accessibility')) {
          obj['accessibility'] = _ApiClient["default"].convertToType(data['accessibility'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('multiline')) {
          obj['multiline'] = _ApiClient["default"].convertToType(data['multiline'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('password-manager-custom-field')) {
          obj['password-manager-custom-field'] = _ApiClient["default"].convertToType(data['password-manager-custom-field'], {
            'String': 'String'
          });
        }

        if (data.hasOwnProperty('password-manager-inject-url')) {
          obj['password-manager-inject-url'] = _ApiClient["default"].convertToType(data['password-manager-inject-url'], ['String']);
        }

        if (data.hasOwnProperty('password-manager-password')) {
          obj['password-manager-password'] = _ApiClient["default"].convertToType(data['password-manager-password'], 'String');
        }

        if (data.hasOwnProperty('password-manager-username')) {
          obj['password-manager-username'] = _ApiClient["default"].convertToType(data['password-manager-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateSecretVal;
}();
/**
 * for personal password manager
 * @member {String} accessibility
 */


UpdateSecretVal.prototype['accessibility'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateSecretVal.prototype['json'] = undefined;
/**
 * @member {String} keep-prev-version
 */

UpdateSecretVal.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateSecretVal.prototype['key'] = undefined;
/**
 * The provided value is a multiline value (separated by '\\n')
 * @member {Boolean} multiline
 */

UpdateSecretVal.prototype['multiline'] = undefined;
/**
 * Secret name
 * @member {String} name
 */

UpdateSecretVal.prototype['name'] = undefined;
/**
 * Deprecated
 * @member {Boolean} new-version
 */

UpdateSecretVal.prototype['new-version'] = undefined;
/**
 * For Password Management use, additional fields
 * @member {Object.<String, String>} password-manager-custom-field
 */

UpdateSecretVal.prototype['password-manager-custom-field'] = undefined;
/**
 * For Password Management use, reflect the website context
 * @member {Array.<String>} password-manager-inject-url
 */

UpdateSecretVal.prototype['password-manager-inject-url'] = undefined;
/**
 * For Password Management use, additional fields
 * @member {String} password-manager-password
 */

UpdateSecretVal.prototype['password-manager-password'] = undefined;
/**
 * For Password Management use
 * @member {String} password-manager-username
 */

UpdateSecretVal.prototype['password-manager-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateSecretVal.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateSecretVal.prototype['uid-token'] = undefined;
/**
 * The new secret value
 * @member {String} value
 */

UpdateSecretVal.prototype['value'] = undefined;
var _default = UpdateSecretVal;
exports["default"] = _default;
