// akeyless/dist/model/UpdateDBTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateDBTargetDetails model module.
 * @module model/UpdateDBTargetDetails
 * @version 2.15.24
 */
var UpdateDBTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateDBTargetDetails</code>.
   * @alias module:model/UpdateDBTargetDetails
   * @param name {String} Target name
   */
  function UpdateDBTargetDetails(name) {
    _classCallCheck(this, UpdateDBTargetDetails);

    UpdateDBTargetDetails.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateDBTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateDBTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateDBTargetDetails} obj Optional instance to populate.
     * @return {module:model/UpdateDBTargetDetails} The populated <code>UpdateDBTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateDBTargetDetails();

        if (data.hasOwnProperty('db_type')) {
          obj['db_type'] = _ApiClient["default"].convertToType(data['db_type'], 'String');
        }

        if (data.hasOwnProperty('host_name')) {
          obj['host_name'] = _ApiClient["default"].convertToType(data['host_name'], 'String');
        }

        if (data.hasOwnProperty('mongo_db_name')) {
          obj['mongo_db_name'] = _ApiClient["default"].convertToType(data['mongo_db_name'], 'String');
        }

        if (data.hasOwnProperty('mongo_uri')) {
          obj['mongo_uri'] = _ApiClient["default"].convertToType(data['mongo_uri'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('pwd')) {
          obj['pwd'] = _ApiClient["default"].convertToType(data['pwd'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('user_name')) {
          obj['user_name'] = _ApiClient["default"].convertToType(data['user_name'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateDBTargetDetails;
}();
/**
 * @member {String} db_type
 */


UpdateDBTargetDetails.prototype['db_type'] = undefined;
/**
 * @member {String} host_name
 */

UpdateDBTargetDetails.prototype['host_name'] = undefined;
/**
 * @member {String} mongo_db_name
 */

UpdateDBTargetDetails.prototype['mongo_db_name'] = undefined;
/**
 * @member {String} mongo_uri
 */

UpdateDBTargetDetails.prototype['mongo_uri'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateDBTargetDetails.prototype['name'] = undefined;
/**
 * Whether to create a new version of not
 * @member {Boolean} new-version
 * @default false
 */

UpdateDBTargetDetails.prototype['new-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateDBTargetDetails.prototype['password'] = undefined;
/**
 * @member {String} port
 */

UpdateDBTargetDetails.prototype['port'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

UpdateDBTargetDetails.prototype['protection_key'] = undefined;
/**
 * @member {String} pwd
 */

UpdateDBTargetDetails.prototype['pwd'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateDBTargetDetails.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateDBTargetDetails.prototype['uid-token'] = undefined;
/**
 * @member {String} user_name
 */

UpdateDBTargetDetails.prototype['user_name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateDBTargetDetails.prototype['username'] = undefined;
var _default = UpdateDBTargetDetails;
exports["default"] = _default;
