// akeyless/dist/model/UpdateNativeK8STarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateNativeK8STarget model module.
 * @module model/UpdateNativeK8STarget
 * @version 2.15.24
 */
var UpdateNativeK8STarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateNativeK8STarget</code>.
   * @alias module:model/UpdateNativeK8STarget
   * @param k8sClusterCaCert {String} K8S cluster CA certificate
   * @param k8sClusterEndpoint {String} K8S cluster URL endpoint
   * @param k8sClusterToken {String} K8S cluster Bearer token
   * @param name {String} Target name
   */
  function UpdateNativeK8STarget(k8sClusterCaCert, k8sClusterEndpoint, k8sClusterToken, name) {
    _classCallCheck(this, UpdateNativeK8STarget);

    UpdateNativeK8STarget.initialize(this, k8sClusterCaCert, k8sClusterEndpoint, k8sClusterToken, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateNativeK8STarget, null, [{
    key: "initialize",
    value: function initialize(obj, k8sClusterCaCert, k8sClusterEndpoint, k8sClusterToken, name) {
      obj['k8s-cluster-ca-cert'] = k8sClusterCaCert;
      obj['k8s-cluster-endpoint'] = k8sClusterEndpoint;
      obj['k8s-cluster-token'] = k8sClusterToken;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateNativeK8STarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateNativeK8STarget} obj Optional instance to populate.
     * @return {module:model/UpdateNativeK8STarget} The populated <code>UpdateNativeK8STarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateNativeK8STarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-ca-cert')) {
          obj['k8s-cluster-ca-cert'] = _ApiClient["default"].convertToType(data['k8s-cluster-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-endpoint')) {
          obj['k8s-cluster-endpoint'] = _ApiClient["default"].convertToType(data['k8s-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('k8s-cluster-token')) {
          obj['k8s-cluster-token'] = _ApiClient["default"].convertToType(data['k8s-cluster-token'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateNativeK8STarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateNativeK8STarget.prototype['comment'] = undefined;
/**
 * K8S cluster CA certificate
 * @member {String} k8s-cluster-ca-cert
 */

UpdateNativeK8STarget.prototype['k8s-cluster-ca-cert'] = undefined;
/**
 * K8S cluster URL endpoint
 * @member {String} k8s-cluster-endpoint
 */

UpdateNativeK8STarget.prototype['k8s-cluster-endpoint'] = undefined;
/**
 * K8S cluster Bearer token
 * @member {String} k8s-cluster-token
 */

UpdateNativeK8STarget.prototype['k8s-cluster-token'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateNativeK8STarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateNativeK8STarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateNativeK8STarget.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateNativeK8STarget.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateNativeK8STarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateNativeK8STarget.prototype['uid-token'] = undefined;
/**
 * Create new version for the target
 * @member {Boolean} update-version
 * @default false
 */

UpdateNativeK8STarget.prototype['update-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateNativeK8STarget.prototype['username'] = undefined;
var _default = UpdateNativeK8STarget;
exports["default"] = _default;
