// akeyless/dist/model/UpdateGcpTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateGcpTarget model module.
 * @module model/UpdateGcpTarget
 * @version 2.20.0
 */
var UpdateGcpTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateGcpTarget</code>.
   * @alias module:model/UpdateGcpTarget
   * @param gcpSaEmail {String} GCP service account email
   * @param name {String} Target name
   */
  function UpdateGcpTarget(gcpSaEmail, name) {
    _classCallCheck(this, UpdateGcpTarget);

    UpdateGcpTarget.initialize(this, gcpSaEmail, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateGcpTarget, null, [{
    key: "initialize",
    value: function initialize(obj, gcpSaEmail, name) {
      obj['gcp-sa-email'] = gcpSaEmail;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateGcpTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateGcpTarget} obj Optional instance to populate.
     * @return {module:model/UpdateGcpTarget} The populated <code>UpdateGcpTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateGcpTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('gcp-key')) {
          obj['gcp-key'] = _ApiClient["default"].convertToType(data['gcp-key'], 'String');
        }

        if (data.hasOwnProperty('gcp-sa-email')) {
          obj['gcp-sa-email'] = _ApiClient["default"].convertToType(data['gcp-sa-email'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return UpdateGcpTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateGcpTarget.prototype['comment'] = undefined;
/**
 * Base64-encoded service account private key text
 * @member {String} gcp-key
 */

UpdateGcpTarget.prototype['gcp-key'] = undefined;
/**
 * GCP service account email
 * @member {String} gcp-sa-email
 */

UpdateGcpTarget.prototype['gcp-sa-email'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateGcpTarget.prototype['json'] = undefined;
/**
 * @member {String} keep-prev-version
 */

UpdateGcpTarget.prototype['keep-prev-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateGcpTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateGcpTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateGcpTarget.prototype['new-name'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateGcpTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateGcpTarget.prototype['uid-token'] = undefined;
/**
 * Deprecated
 * @member {Boolean} update-version
 */

UpdateGcpTarget.prototype['update-version'] = undefined;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

UpdateGcpTarget.prototype['use-gw-cloud-identity'] = undefined;
var _default = UpdateGcpTarget;
exports["default"] = _default;
