// akeyless/dist/model/UpdateAuthMethodOIDC.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateAuthMethodOIDC model module.
 * @module model/UpdateAuthMethodOIDC
 * @version 2.20.0
 */
var UpdateAuthMethodOIDC = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAuthMethodOIDC</code>.
   * updateAuthMethodOIDC is a command that updates a new auth method that will be available to authenticate using OIDC.
   * @alias module:model/UpdateAuthMethodOIDC
   * @param name {String} Auth Method name
   * @param uniqueIdentifier {String} A unique identifier (ID) value should be configured for OIDC, OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
   */
  function UpdateAuthMethodOIDC(name, uniqueIdentifier) {
    _classCallCheck(this, UpdateAuthMethodOIDC);

    UpdateAuthMethodOIDC.initialize(this, name, uniqueIdentifier);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAuthMethodOIDC, null, [{
    key: "initialize",
    value: function initialize(obj, name, uniqueIdentifier) {
      obj['name'] = name;
      obj['unique-identifier'] = uniqueIdentifier;
    }
    /**
     * Constructs a <code>UpdateAuthMethodOIDC</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAuthMethodOIDC} obj Optional instance to populate.
     * @return {module:model/UpdateAuthMethodOIDC} The populated <code>UpdateAuthMethodOIDC</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAuthMethodOIDC();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('allowed-redirect-uri')) {
          obj['allowed-redirect-uri'] = _ApiClient["default"].convertToType(data['allowed-redirect-uri'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('client-id')) {
          obj['client-id'] = _ApiClient["default"].convertToType(data['client-id'], 'String');
        }

        if (data.hasOwnProperty('client-secret')) {
          obj['client-secret'] = _ApiClient["default"].convertToType(data['client-secret'], 'String');
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('issuer')) {
          obj['issuer'] = _ApiClient["default"].convertToType(data['issuer'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('required-scopes')) {
          obj['required-scopes'] = _ApiClient["default"].convertToType(data['required-scopes'], ['String']);
        }

        if (data.hasOwnProperty('required-scopes-prefix')) {
          obj['required-scopes-prefix'] = _ApiClient["default"].convertToType(data['required-scopes-prefix'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('unique-identifier')) {
          obj['unique-identifier'] = _ApiClient["default"].convertToType(data['unique-identifier'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateAuthMethodOIDC;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


UpdateAuthMethodOIDC.prototype['access-expires'] = 0;
/**
 * Allowed redirect URIs after the authentication
 * @member {Array.<String>} allowed-redirect-uri
 */

UpdateAuthMethodOIDC.prototype['allowed-redirect-uri'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

UpdateAuthMethodOIDC.prototype['bound-ips'] = undefined;
/**
 * Client ID
 * @member {String} client-id
 */

UpdateAuthMethodOIDC.prototype['client-id'] = undefined;
/**
 * Client Secret
 * @member {String} client-secret
 */

UpdateAuthMethodOIDC.prototype['client-secret'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

UpdateAuthMethodOIDC.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

UpdateAuthMethodOIDC.prototype['gw-bound-ips'] = undefined;
/**
 * Issuer URL
 * @member {String} issuer
 */

UpdateAuthMethodOIDC.prototype['issuer'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateAuthMethodOIDC.prototype['json'] = undefined;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 */

UpdateAuthMethodOIDC.prototype['jwt-ttl'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

UpdateAuthMethodOIDC.prototype['name'] = undefined;
/**
 * Auth Method new name
 * @member {String} new-name
 */

UpdateAuthMethodOIDC.prototype['new-name'] = undefined;
/**
 * RequiredScopes is a list of required scopes that the oidc method will request from the oidc provider and the user must approve
 * @member {Array.<String>} required-scopes
 */

UpdateAuthMethodOIDC.prototype['required-scopes'] = undefined;
/**
 * RequiredScopesPrefix is a a prefix to add to all required-scopes when requesting them from the oidc server (for example, azures' Application ID URI)
 * @member {String} required-scopes-prefix
 */

UpdateAuthMethodOIDC.prototype['required-scopes-prefix'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateAuthMethodOIDC.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateAuthMethodOIDC.prototype['uid-token'] = undefined;
/**
 * A unique identifier (ID) value should be configured for OIDC, OAuth2, LDAP and SAML authentication method types and is usually a value such as the email, username, or upn for example. Whenever a user logs in with a token, these authentication types issue a \"sub claim\" that contains details uniquely identifying that user. This sub claim includes a key containing the ID value that you configured, and is used to distinguish between different users from within the same organization.
 * @member {String} unique-identifier
 */

UpdateAuthMethodOIDC.prototype['unique-identifier'] = undefined;
var _default = UpdateAuthMethodOIDC;
exports["default"] = _default;
