// akeyless/dist/model/UpdateLdapTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateLdapTargetDetails model module.
 * @module model/UpdateLdapTargetDetails
 * @version 2.20.0
 */
var UpdateLdapTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateLdapTargetDetails</code>.
   * @alias module:model/UpdateLdapTargetDetails
   * @param name {String} Target name
   */
  function UpdateLdapTargetDetails(name) {
    _classCallCheck(this, UpdateLdapTargetDetails);

    UpdateLdapTargetDetails.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateLdapTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateLdapTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateLdapTargetDetails} obj Optional instance to populate.
     * @return {module:model/UpdateLdapTargetDetails} The populated <code>UpdateLdapTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateLdapTargetDetails();

        if (data.hasOwnProperty('bind-dn')) {
          obj['bind-dn'] = _ApiClient["default"].convertToType(data['bind-dn'], 'String');
        }

        if (data.hasOwnProperty('bind-dn-password')) {
          obj['bind-dn-password'] = _ApiClient["default"].convertToType(data['bind-dn-password'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('ldap-ca-cert')) {
          obj['ldap-ca-cert'] = _ApiClient["default"].convertToType(data['ldap-ca-cert'], 'String');
        }

        if (data.hasOwnProperty('ldap-url')) {
          obj['ldap-url'] = _ApiClient["default"].convertToType(data['ldap-url'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('token-expiration')) {
          obj['token-expiration'] = _ApiClient["default"].convertToType(data['token-expiration'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateLdapTargetDetails;
}();
/**
 * @member {String} bind-dn
 */


UpdateLdapTargetDetails.prototype['bind-dn'] = undefined;
/**
 * @member {String} bind-dn-password
 */

UpdateLdapTargetDetails.prototype['bind-dn-password'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateLdapTargetDetails.prototype['json'] = undefined;
/**
 * @member {String} keep-prev-version
 */

UpdateLdapTargetDetails.prototype['keep-prev-version'] = undefined;
/**
 * @member {String} key
 */

UpdateLdapTargetDetails.prototype['key'] = undefined;
/**
 * @member {String} ldap-ca-cert
 */

UpdateLdapTargetDetails.prototype['ldap-ca-cert'] = undefined;
/**
 * @member {String} ldap-url
 */

UpdateLdapTargetDetails.prototype['ldap-url'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateLdapTargetDetails.prototype['name'] = undefined;
/**
 * Deprecated
 * @member {Boolean} new-version
 */

UpdateLdapTargetDetails.prototype['new-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

UpdateLdapTargetDetails.prototype['protection_key'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateLdapTargetDetails.prototype['token'] = undefined;
/**
 * @member {String} token-expiration
 */

UpdateLdapTargetDetails.prototype['token-expiration'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateLdapTargetDetails.prototype['uid-token'] = undefined;
var _default = UpdateLdapTargetDetails;
exports["default"] = _default;
