// akeyless/dist/model/UpdateSSHTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateSSHTarget model module.
 * @module model/UpdateSSHTarget
 * @version 2.15.24
 */
var UpdateSSHTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateSSHTarget</code>.
   * @alias module:model/UpdateSSHTarget
   * @param name {String} Target name
   */
  function UpdateSSHTarget(name) {
    _classCallCheck(this, UpdateSSHTarget);

    UpdateSSHTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateSSHTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateSSHTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateSSHTarget} obj Optional instance to populate.
     * @return {module:model/UpdateSSHTarget} The populated <code>UpdateSSHTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateSSHTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('host')) {
          obj['host'] = _ApiClient["default"].convertToType(data['host'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('port')) {
          obj['port'] = _ApiClient["default"].convertToType(data['port'], 'String');
        }

        if (data.hasOwnProperty('private-key')) {
          obj['private-key'] = _ApiClient["default"].convertToType(data['private-key'], 'String');
        }

        if (data.hasOwnProperty('private-key-password')) {
          obj['private-key-password'] = _ApiClient["default"].convertToType(data['private-key-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateSSHTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateSSHTarget.prototype['comment'] = undefined;
/**
 * @member {String} host
 */

UpdateSSHTarget.prototype['host'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateSSHTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateSSHTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateSSHTarget.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateSSHTarget.prototype['password'] = undefined;
/**
 * @member {String} port
 */

UpdateSSHTarget.prototype['port'] = undefined;
/**
 * @member {String} private-key
 */

UpdateSSHTarget.prototype['private-key'] = undefined;
/**
 * @member {String} private-key-password
 */

UpdateSSHTarget.prototype['private-key-password'] = undefined;
/**
 * @member {String} ssh-password
 */

UpdateSSHTarget.prototype['ssh-password'] = undefined;
/**
 * @member {String} ssh-username
 */

UpdateSSHTarget.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateSSHTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateSSHTarget.prototype['uid-token'] = undefined;
/**
 * Create new version for the target
 * @member {Boolean} update-version
 * @default false
 */

UpdateSSHTarget.prototype['update-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateSSHTarget.prototype['username'] = undefined;
var _default = UpdateSSHTarget;
exports["default"] = _default;
