// akeyless/dist/model/UpdateRabbitMQTargetDetails.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRabbitMQTargetDetails model module.
 * @module model/UpdateRabbitMQTargetDetails
 * @version 2.20.0
 */
var UpdateRabbitMQTargetDetails = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRabbitMQTargetDetails</code>.
   * @alias module:model/UpdateRabbitMQTargetDetails
   * @param name {String} Target name
   */
  function UpdateRabbitMQTargetDetails(name) {
    _classCallCheck(this, UpdateRabbitMQTargetDetails);

    UpdateRabbitMQTargetDetails.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRabbitMQTargetDetails, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRabbitMQTargetDetails</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRabbitMQTargetDetails} obj Optional instance to populate.
     * @return {module:model/UpdateRabbitMQTargetDetails} The populated <code>UpdateRabbitMQTargetDetails</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRabbitMQTargetDetails();

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('keep-prev-version')) {
          obj['keep-prev-version'] = _ApiClient["default"].convertToType(data['keep-prev-version'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('protection_key')) {
          obj['protection_key'] = _ApiClient["default"].convertToType(data['protection_key'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq_server_password')) {
          obj['rabbitmq_server_password'] = _ApiClient["default"].convertToType(data['rabbitmq_server_password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq_server_uri')) {
          obj['rabbitmq_server_uri'] = _ApiClient["default"].convertToType(data['rabbitmq_server_uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq_server_user')) {
          obj['rabbitmq_server_user'] = _ApiClient["default"].convertToType(data['rabbitmq_server_user'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRabbitMQTargetDetails;
}();
/**
 * Set output format to JSON
 * @member {Boolean} json
 */


UpdateRabbitMQTargetDetails.prototype['json'] = undefined;
/**
 * @member {String} keep-prev-version
 */

UpdateRabbitMQTargetDetails.prototype['keep-prev-version'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateRabbitMQTargetDetails.prototype['name'] = undefined;
/**
 * Deprecated
 * @member {Boolean} new-version
 */

UpdateRabbitMQTargetDetails.prototype['new-version'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} protection_key
 */

UpdateRabbitMQTargetDetails.prototype['protection_key'] = undefined;
/**
 * @member {String} rabbitmq_server_password
 */

UpdateRabbitMQTargetDetails.prototype['rabbitmq_server_password'] = undefined;
/**
 * @member {String} rabbitmq_server_uri
 */

UpdateRabbitMQTargetDetails.prototype['rabbitmq_server_uri'] = undefined;
/**
 * @member {String} rabbitmq_server_user
 */

UpdateRabbitMQTargetDetails.prototype['rabbitmq_server_user'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRabbitMQTargetDetails.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRabbitMQTargetDetails.prototype['uid-token'] = undefined;
var _default = UpdateRabbitMQTargetDetails;
exports["default"] = _default;
