// akeyless/dist/model/UpdateRole.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRole model module.
 * @module model/UpdateRole
 * @version 2.20.0
 */
var UpdateRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRole</code>.
   * @alias module:model/UpdateRole
   * @param name {String} Role name
   */
  function UpdateRole(name) {
    _classCallCheck(this, UpdateRole);

    UpdateRole.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRole, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRole} obj Optional instance to populate.
     * @return {module:model/UpdateRole} The populated <code>UpdateRole</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRole();

        if (data.hasOwnProperty('analytics-access')) {
          obj['analytics-access'] = _ApiClient["default"].convertToType(data['analytics-access'], 'String');
        }

        if (data.hasOwnProperty('audit-access')) {
          obj['audit-access'] = _ApiClient["default"].convertToType(data['audit-access'], 'String');
        }

        if (data.hasOwnProperty('gw-analytics-access')) {
          obj['gw-analytics-access'] = _ApiClient["default"].convertToType(data['gw-analytics-access'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-comment')) {
          obj['new-comment'] = _ApiClient["default"].convertToType(data['new-comment'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('sra-reports-access')) {
          obj['sra-reports-access'] = _ApiClient["default"].convertToType(data['sra-reports-access'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRole;
}();
/**
 * Allow this role to view analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} analytics-access
 */


UpdateRole.prototype['analytics-access'] = undefined;
/**
 * Allow this role to view audit logs. Currently only 'none', 'own' and 'all' values are supported, allowing associated auth methods to view audit logs produced by the same auth methods.
 * @member {String} audit-access
 */

UpdateRole.prototype['audit-access'] = undefined;
/**
 * Allow this role to view gw analytics. Currently only 'none', 'own', 'all' values are supported, allowing associated auth methods to view reports produced by the same auth methods.
 * @member {String} gw-analytics-access
 */

UpdateRole.prototype['gw-analytics-access'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateRole.prototype['json'] = undefined;
/**
 * Role name
 * @member {String} name
 */

UpdateRole.prototype['name'] = undefined;
/**
 * New comment about the role
 * @member {String} new-comment
 * @default 'default_comment'
 */

UpdateRole.prototype['new-comment'] = 'default_comment';
/**
 * New Role name
 * @member {String} new-name
 */

UpdateRole.prototype['new-name'] = undefined;
/**
 * Allow this role to view SRA Clusters. Currently only 'none', 'own', 'all' values are supported.
 * @member {String} sra-reports-access
 */

UpdateRole.prototype['sra-reports-access'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRole.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRole.prototype['uid-token'] = undefined;
var _default = UpdateRole;
exports["default"] = _default;
