// akeyless/dist/model/UpdateRotatedSecret.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRotatedSecret model module.
 * @module model/UpdateRotatedSecret
 * @version 2.15.24
 */
var UpdateRotatedSecret = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRotatedSecret</code>.
   * updateRotatedSecret is a command that updates rotated secret. [Deprecated: Use gateway-update-item command]
   * @alias module:model/UpdateRotatedSecret
   * @param name {String} Secret name
   */
  function UpdateRotatedSecret(name) {
    _classCallCheck(this, UpdateRotatedSecret);

    UpdateRotatedSecret.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRotatedSecret, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRotatedSecret</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRotatedSecret} obj Optional instance to populate.
     * @return {module:model/UpdateRotatedSecret} The populated <code>UpdateRotatedSecret</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRotatedSecret();

        if (data.hasOwnProperty('add-tag')) {
          obj['add-tag'] = _ApiClient["default"].convertToType(data['add-tag'], ['String']);
        }

        if (data.hasOwnProperty('api-id')) {
          obj['api-id'] = _ApiClient["default"].convertToType(data['api-id'], 'String');
        }

        if (data.hasOwnProperty('api-key')) {
          obj['api-key'] = _ApiClient["default"].convertToType(data['api-key'], 'String');
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('custom-payload')) {
          obj['custom-payload'] = _ApiClient["default"].convertToType(data['custom-payload'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-metadata')) {
          obj['new-metadata'] = _ApiClient["default"].convertToType(data['new-metadata'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('rm-tag')) {
          obj['rm-tag'] = _ApiClient["default"].convertToType(data['rm-tag'], ['String']);
        }

        if (data.hasOwnProperty('rotated-password')) {
          obj['rotated-password'] = _ApiClient["default"].convertToType(data['rotated-password'], 'String');
        }

        if (data.hasOwnProperty('rotated-username')) {
          obj['rotated-username'] = _ApiClient["default"].convertToType(data['rotated-username'], 'String');
        }

        if (data.hasOwnProperty('rotation-hour')) {
          obj['rotation-hour'] = _ApiClient["default"].convertToType(data['rotation-hour'], 'Number');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('rotator-creds-type')) {
          obj['rotator-creds-type'] = _ApiClient["default"].convertToType(data['rotator-creds-type'], 'String');
        }

        if (data.hasOwnProperty('rotator-custom-cmd')) {
          obj['rotator-custom-cmd'] = _ApiClient["default"].convertToType(data['rotator-custom-cmd'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRotatedSecret;
}();
/**
 * List of the new tags that will be attached to this item
 * @member {Array.<String>} add-tag
 */


UpdateRotatedSecret.prototype['add-tag'] = undefined;
/**
 * @member {String} api-id
 */

UpdateRotatedSecret.prototype['api-id'] = undefined;
/**
 * @member {String} api-key
 */

UpdateRotatedSecret.prototype['api-key'] = undefined;
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
 * @member {String} auto-rotate
 */

UpdateRotatedSecret.prototype['auto-rotate'] = undefined;
/**
 * @member {String} custom-payload
 */

UpdateRotatedSecret.prototype['custom-payload'] = undefined;
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateRotatedSecret.prototype['key'] = undefined;
/**
 * Secret name
 * @member {String} name
 */

UpdateRotatedSecret.prototype['name'] = undefined;
/**
 * New item metadata
 * @member {String} new-metadata
 * @default 'default_metadata'
 */

UpdateRotatedSecret.prototype['new-metadata'] = 'default_metadata';
/**
 * New item name
 * @member {String} new-name
 */

UpdateRotatedSecret.prototype['new-name'] = undefined;
/**
 * Whether to create a new version of not
 * @member {Boolean} new-version
 * @default false
 */

UpdateRotatedSecret.prototype['new-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateRotatedSecret.prototype['password'] = undefined;
/**
 * List of the existent tags that will be removed from this item
 * @member {Array.<String>} rm-tag
 */

UpdateRotatedSecret.prototype['rm-tag'] = undefined;
/**
 * @member {String} rotated-password
 */

UpdateRotatedSecret.prototype['rotated-password'] = undefined;
/**
 * @member {String} rotated-username
 */

UpdateRotatedSecret.prototype['rotated-username'] = undefined;
/**
 * @member {Number} rotation-hour
 */

UpdateRotatedSecret.prototype['rotation-hour'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (7-365)
 * @member {String} rotation-interval
 */

UpdateRotatedSecret.prototype['rotation-interval'] = undefined;
/**
 * @member {String} rotator-creds-type
 */

UpdateRotatedSecret.prototype['rotator-creds-type'] = undefined;
/**
 * @member {String} rotator-custom-cmd
 */

UpdateRotatedSecret.prototype['rotator-custom-cmd'] = undefined;
/**
 * Deprecated: use RotatedPassword
 * @member {String} ssh-password
 */

UpdateRotatedSecret.prototype['ssh-password'] = undefined;
/**
 * Deprecated: use RotatedUser
 * @member {String} ssh-username
 */

UpdateRotatedSecret.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRotatedSecret.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRotatedSecret.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateRotatedSecret.prototype['username'] = undefined;
var _default = UpdateRotatedSecret;
exports["default"] = _default;
