// akeyless/dist/model/UpdateGKETarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateGKETarget model module.
 * @module model/UpdateGKETarget
 * @version 2.15.24
 */
var UpdateGKETarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateGKETarget</code>.
   * @alias module:model/UpdateGKETarget
   * @param name {String} Target name
   */
  function UpdateGKETarget(name) {
    _classCallCheck(this, UpdateGKETarget);

    UpdateGKETarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateGKETarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateGKETarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateGKETarget} obj Optional instance to populate.
     * @return {module:model/UpdateGKETarget} The populated <code>UpdateGKETarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateGKETarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('gke-account-key')) {
          obj['gke-account-key'] = _ApiClient["default"].convertToType(data['gke-account-key'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-cert')) {
          obj['gke-cluster-cert'] = _ApiClient["default"].convertToType(data['gke-cluster-cert'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-endpoint')) {
          obj['gke-cluster-endpoint'] = _ApiClient["default"].convertToType(data['gke-cluster-endpoint'], 'String');
        }

        if (data.hasOwnProperty('gke-cluster-name')) {
          obj['gke-cluster-name'] = _ApiClient["default"].convertToType(data['gke-cluster-name'], 'String');
        }

        if (data.hasOwnProperty('gke-service-account-email')) {
          obj['gke-service-account-email'] = _ApiClient["default"].convertToType(data['gke-service-account-email'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('use-gw-cloud-identity')) {
          obj['use-gw-cloud-identity'] = _ApiClient["default"].convertToType(data['use-gw-cloud-identity'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateGKETarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateGKETarget.prototype['comment'] = undefined;
/**
 * GKE Service Account key file path
 * @member {String} gke-account-key
 */

UpdateGKETarget.prototype['gke-account-key'] = undefined;
/**
 * GKE cluster CA certificate
 * @member {String} gke-cluster-cert
 */

UpdateGKETarget.prototype['gke-cluster-cert'] = undefined;
/**
 * GKE cluster URL endpoint
 * @member {String} gke-cluster-endpoint
 */

UpdateGKETarget.prototype['gke-cluster-endpoint'] = undefined;
/**
 * GKE cluster name
 * @member {String} gke-cluster-name
 */

UpdateGKETarget.prototype['gke-cluster-name'] = undefined;
/**
 * GKE service account email
 * @member {String} gke-service-account-email
 */

UpdateGKETarget.prototype['gke-service-account-email'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateGKETarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateGKETarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateGKETarget.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateGKETarget.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateGKETarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateGKETarget.prototype['uid-token'] = undefined;
/**
 * Create new version for the target
 * @member {Boolean} update-version
 * @default false
 */

UpdateGKETarget.prototype['update-version'] = false;
/**
 * @member {Boolean} use-gw-cloud-identity
 */

UpdateGKETarget.prototype['use-gw-cloud-identity'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateGKETarget.prototype['username'] = undefined;
var _default = UpdateGKETarget;
exports["default"] = _default;
