// akeyless/dist/model/UpdateAuthMethodAWSIAM.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateAuthMethodAWSIAM model module.
 * @module model/UpdateAuthMethodAWSIAM
 * @version 2.20.0
 */
var UpdateAuthMethodAWSIAM = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateAuthMethodAWSIAM</code>.
   * updateAuthMethodAWSIAM is a command that updates a new Auth Method that will be able to authenticate using AWS IAM credentials.
   * @alias module:model/UpdateAuthMethodAWSIAM
   * @param boundAwsAccountId {Array.<String>} A list of AWS account-IDs that the access is restricted to
   * @param name {String} Auth Method name
   */
  function UpdateAuthMethodAWSIAM(boundAwsAccountId, name) {
    _classCallCheck(this, UpdateAuthMethodAWSIAM);

    UpdateAuthMethodAWSIAM.initialize(this, boundAwsAccountId, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateAuthMethodAWSIAM, null, [{
    key: "initialize",
    value: function initialize(obj, boundAwsAccountId, name) {
      obj['bound-aws-account-id'] = boundAwsAccountId;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateAuthMethodAWSIAM</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateAuthMethodAWSIAM} obj Optional instance to populate.
     * @return {module:model/UpdateAuthMethodAWSIAM} The populated <code>UpdateAuthMethodAWSIAM</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateAuthMethodAWSIAM();

        if (data.hasOwnProperty('access-expires')) {
          obj['access-expires'] = _ApiClient["default"].convertToType(data['access-expires'], 'Number');
        }

        if (data.hasOwnProperty('bound-arn')) {
          obj['bound-arn'] = _ApiClient["default"].convertToType(data['bound-arn'], ['String']);
        }

        if (data.hasOwnProperty('bound-aws-account-id')) {
          obj['bound-aws-account-id'] = _ApiClient["default"].convertToType(data['bound-aws-account-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-ips')) {
          obj['bound-ips'] = _ApiClient["default"].convertToType(data['bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('bound-resource-id')) {
          obj['bound-resource-id'] = _ApiClient["default"].convertToType(data['bound-resource-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-role-id')) {
          obj['bound-role-id'] = _ApiClient["default"].convertToType(data['bound-role-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-role-name')) {
          obj['bound-role-name'] = _ApiClient["default"].convertToType(data['bound-role-name'], ['String']);
        }

        if (data.hasOwnProperty('bound-user-id')) {
          obj['bound-user-id'] = _ApiClient["default"].convertToType(data['bound-user-id'], ['String']);
        }

        if (data.hasOwnProperty('bound-user-name')) {
          obj['bound-user-name'] = _ApiClient["default"].convertToType(data['bound-user-name'], ['String']);
        }

        if (data.hasOwnProperty('force-sub-claims')) {
          obj['force-sub-claims'] = _ApiClient["default"].convertToType(data['force-sub-claims'], 'Boolean');
        }

        if (data.hasOwnProperty('gw-bound-ips')) {
          obj['gw-bound-ips'] = _ApiClient["default"].convertToType(data['gw-bound-ips'], ['String']);
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt-ttl')) {
          obj['jwt-ttl'] = _ApiClient["default"].convertToType(data['jwt-ttl'], 'Number');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('sts-url')) {
          obj['sts-url'] = _ApiClient["default"].convertToType(data['sts-url'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateAuthMethodAWSIAM;
}();
/**
 * Access expiration date in Unix timestamp (select 0 for access without expiry date)
 * @member {Number} access-expires
 * @default 0
 */


UpdateAuthMethodAWSIAM.prototype['access-expires'] = 0;
/**
 * A list of full arns that the access is restricted to
 * @member {Array.<String>} bound-arn
 */

UpdateAuthMethodAWSIAM.prototype['bound-arn'] = undefined;
/**
 * A list of AWS account-IDs that the access is restricted to
 * @member {Array.<String>} bound-aws-account-id
 */

UpdateAuthMethodAWSIAM.prototype['bound-aws-account-id'] = undefined;
/**
 * A CIDR whitelist with the IPs that the access is restricted to
 * @member {Array.<String>} bound-ips
 */

UpdateAuthMethodAWSIAM.prototype['bound-ips'] = undefined;
/**
 * A list of full resource ids that the access is restricted to
 * @member {Array.<String>} bound-resource-id
 */

UpdateAuthMethodAWSIAM.prototype['bound-resource-id'] = undefined;
/**
 * A list of full role ids that the access is restricted to
 * @member {Array.<String>} bound-role-id
 */

UpdateAuthMethodAWSIAM.prototype['bound-role-id'] = undefined;
/**
 * A list of full role-name that the access is restricted to
 * @member {Array.<String>} bound-role-name
 */

UpdateAuthMethodAWSIAM.prototype['bound-role-name'] = undefined;
/**
 * A list of full user ids that the access is restricted to
 * @member {Array.<String>} bound-user-id
 */

UpdateAuthMethodAWSIAM.prototype['bound-user-id'] = undefined;
/**
 * A list of full user-name that the access is restricted to
 * @member {Array.<String>} bound-user-name
 */

UpdateAuthMethodAWSIAM.prototype['bound-user-name'] = undefined;
/**
 * if true: enforce role-association must include sub claims
 * @member {Boolean} force-sub-claims
 */

UpdateAuthMethodAWSIAM.prototype['force-sub-claims'] = undefined;
/**
 * A CIDR whitelist with the GW IPs that the access is restricted to
 * @member {Array.<String>} gw-bound-ips
 */

UpdateAuthMethodAWSIAM.prototype['gw-bound-ips'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

UpdateAuthMethodAWSIAM.prototype['json'] = undefined;
/**
 * Jwt TTL
 * @member {Number} jwt-ttl
 */

UpdateAuthMethodAWSIAM.prototype['jwt-ttl'] = undefined;
/**
 * Auth Method name
 * @member {String} name
 */

UpdateAuthMethodAWSIAM.prototype['name'] = undefined;
/**
 * Auth Method new name
 * @member {String} new-name
 */

UpdateAuthMethodAWSIAM.prototype['new-name'] = undefined;
/**
 * sts URL
 * @member {String} sts-url
 * @default 'https://sts.amazonaws.com'
 */

UpdateAuthMethodAWSIAM.prototype['sts-url'] = 'https://sts.amazonaws.com';
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateAuthMethodAWSIAM.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateAuthMethodAWSIAM.prototype['uid-token'] = undefined;
var _default = UpdateAuthMethodAWSIAM;
exports["default"] = _default;
