// akeyless/dist/model/UpdateRotatedSecretSC.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRotatedSecretSC model module.
 * @module model/UpdateRotatedSecretSC
 * @version 2.4.0
 */
var UpdateRotatedSecretSC = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRotatedSecretSC</code>.
   * @alias module:model/UpdateRotatedSecretSC
   * @param name {String} Secret name
   */
  function UpdateRotatedSecretSC(name) {
    _classCallCheck(this, UpdateRotatedSecretSC);

    UpdateRotatedSecretSC.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRotatedSecretSC, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRotatedSecretSC</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRotatedSecretSC} obj Optional instance to populate.
     * @return {module:model/UpdateRotatedSecretSC} The populated <code>UpdateRotatedSecretSC</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRotatedSecretSC();

        if (data.hasOwnProperty('add-tag')) {
          obj['add-tag'] = _ApiClient["default"].convertToType(data['add-tag'], ['String']);
        }

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'String');
        }

        if (data.hasOwnProperty('gateway-url')) {
          obj['gateway-url'] = _ApiClient["default"].convertToType(data['gateway-url'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-metadata')) {
          obj['new-metadata'] = _ApiClient["default"].convertToType(data['new-metadata'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('new-version')) {
          obj['new-version'] = _ApiClient["default"].convertToType(data['new-version'], 'Boolean');
        }

        if (data.hasOwnProperty('rm-tag')) {
          obj['rm-tag'] = _ApiClient["default"].convertToType(data['rm-tag'], ['String']);
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'String');
        }

        if (data.hasOwnProperty('rotation_hour')) {
          obj['rotation_hour'] = _ApiClient["default"].convertToType(data['rotation_hour'], 'Number');
        }

        if (data.hasOwnProperty('rotator_creds_type')) {
          obj['rotator_creds_type'] = _ApiClient["default"].convertToType(data['rotator_creds_type'], 'String');
        }

        if (data.hasOwnProperty('ssh-password')) {
          obj['ssh-password'] = _ApiClient["default"].convertToType(data['ssh-password'], 'String');
        }

        if (data.hasOwnProperty('ssh-username')) {
          obj['ssh-username'] = _ApiClient["default"].convertToType(data['ssh-username'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRotatedSecretSC;
}();
/**
 * List of the new tags that will be attached to this item
 * @member {Array.<String>} add-tag
 */


UpdateRotatedSecretSC.prototype['add-tag'] = undefined;
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
 * @member {String} auto-rotate
 */

UpdateRotatedSecretSC.prototype['auto-rotate'] = undefined;
/**
 * Gateway url
 * @member {String} gateway-url
 * @default 'http://localhost:8000'
 */

UpdateRotatedSecretSC.prototype['gateway-url'] = 'http://localhost:8000';
/**
 * The name of a key that used to encrypt the secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateRotatedSecretSC.prototype['key'] = undefined;
/**
 * Secret name
 * @member {String} name
 */

UpdateRotatedSecretSC.prototype['name'] = undefined;
/**
 * New item metadata
 * @member {String} new-metadata
 * @default 'default_metadata'
 */

UpdateRotatedSecretSC.prototype['new-metadata'] = 'default_metadata';
/**
 * New item name
 * @member {String} new-name
 */

UpdateRotatedSecretSC.prototype['new-name'] = undefined;
/**
 * Whether to create a new version of not
 * @member {Boolean} new-version
 * @default false
 */

UpdateRotatedSecretSC.prototype['new-version'] = false;
/**
 * List of the existent tags that will be removed from this item
 * @member {Array.<String>} rm-tag
 */

UpdateRotatedSecretSC.prototype['rm-tag'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (7-365)
 * @member {String} rotation-interval
 */

UpdateRotatedSecretSC.prototype['rotation-interval'] = undefined;
/**
 * @member {Number} rotation_hour
 */

UpdateRotatedSecretSC.prototype['rotation_hour'] = undefined;
/**
 * @member {String} rotator_creds_type
 */

UpdateRotatedSecretSC.prototype['rotator_creds_type'] = undefined;
/**
 * @member {String} ssh-password
 */

UpdateRotatedSecretSC.prototype['ssh-password'] = undefined;
/**
 * @member {String} ssh-username
 */

UpdateRotatedSecretSC.prototype['ssh-username'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRotatedSecretSC.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRotatedSecretSC.prototype['uid-token'] = undefined;
var _default = UpdateRotatedSecretSC;
exports["default"] = _default;
