// akeyless/dist/model/UpdateRabbitMQTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRabbitMQTarget model module.
 * @module model/UpdateRabbitMQTarget
 * @version 2.15.24
 */
var UpdateRabbitMQTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRabbitMQTarget</code>.
   * @alias module:model/UpdateRabbitMQTarget
   * @param name {String} Target name
   */
  function UpdateRabbitMQTarget(name) {
    _classCallCheck(this, UpdateRabbitMQTarget);

    UpdateRabbitMQTarget.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRabbitMQTarget, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRabbitMQTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRabbitMQTarget} obj Optional instance to populate.
     * @return {module:model/UpdateRabbitMQTarget} The populated <code>UpdateRabbitMQTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRabbitMQTarget();

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-password')) {
          obj['rabbitmq-server-password'] = _ApiClient["default"].convertToType(data['rabbitmq-server-password'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-uri')) {
          obj['rabbitmq-server-uri'] = _ApiClient["default"].convertToType(data['rabbitmq-server-uri'], 'String');
        }

        if (data.hasOwnProperty('rabbitmq-server-user')) {
          obj['rabbitmq-server-user'] = _ApiClient["default"].convertToType(data['rabbitmq-server-user'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRabbitMQTarget;
}();
/**
 * Comment about the target
 * @member {String} comment
 */


UpdateRabbitMQTarget.prototype['comment'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateRabbitMQTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateRabbitMQTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateRabbitMQTarget.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateRabbitMQTarget.prototype['password'] = undefined;
/**
 * @member {String} rabbitmq-server-password
 */

UpdateRabbitMQTarget.prototype['rabbitmq-server-password'] = undefined;
/**
 * @member {String} rabbitmq-server-uri
 */

UpdateRabbitMQTarget.prototype['rabbitmq-server-uri'] = undefined;
/**
 * @member {String} rabbitmq-server-user
 */

UpdateRabbitMQTarget.prototype['rabbitmq-server-user'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRabbitMQTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRabbitMQTarget.prototype['uid-token'] = undefined;
/**
 * Create new version for the target
 * @member {Boolean} update-version
 * @default false
 */

UpdateRabbitMQTarget.prototype['update-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateRabbitMQTarget.prototype['username'] = undefined;
var _default = UpdateRabbitMQTarget;
exports["default"] = _default;
