// akeyless/dist/model/UpdateRotationSettings.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateRotationSettings model module.
 * @module model/UpdateRotationSettings
 * @version 2.15.24
 */
var UpdateRotationSettings = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateRotationSettings</code>.
   * updateRotationSettings is a command that updates rotations settings of an existing key
   * @alias module:model/UpdateRotationSettings
   * @param autoRotate {Boolean} Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
   * @param name {String} Key name
   */
  function UpdateRotationSettings(autoRotate, name) {
    _classCallCheck(this, UpdateRotationSettings);

    UpdateRotationSettings.initialize(this, autoRotate, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateRotationSettings, null, [{
    key: "initialize",
    value: function initialize(obj, autoRotate, name) {
      obj['auto-rotate'] = autoRotate;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateRotationSettings</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateRotationSettings} obj Optional instance to populate.
     * @return {module:model/UpdateRotationSettings} The populated <code>UpdateRotationSettings</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateRotationSettings();

        if (data.hasOwnProperty('auto-rotate')) {
          obj['auto-rotate'] = _ApiClient["default"].convertToType(data['auto-rotate'], 'Boolean');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('rotation-interval')) {
          obj['rotation-interval'] = _ApiClient["default"].convertToType(data['rotation-interval'], 'Number');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateRotationSettings;
}();
/**
 * Whether to automatically rotate every --rotation-interval days, or disable existing automatic rotation
 * @member {Boolean} auto-rotate
 */


UpdateRotationSettings.prototype['auto-rotate'] = undefined;
/**
 * Key name
 * @member {String} name
 */

UpdateRotationSettings.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateRotationSettings.prototype['password'] = undefined;
/**
 * The number of days to wait between every automatic key rotation (7-365)
 * @member {Number} rotation-interval
 */

UpdateRotationSettings.prototype['rotation-interval'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateRotationSettings.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateRotationSettings.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateRotationSettings.prototype['username'] = undefined;
var _default = UpdateRotationSettings;
exports["default"] = _default;
