// akeyless/dist/model/UpdateArtifactoryTarget.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UpdateArtifactoryTarget model module.
 * @module model/UpdateArtifactoryTarget
 * @version 2.15.24
 */
var UpdateArtifactoryTarget = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateArtifactoryTarget</code>.
   * @alias module:model/UpdateArtifactoryTarget
   * @param artifactoryAdminName {String} Artifactory Admin Name
   * @param artifactoryAdminPwd {String} Artifactory Admin password
   * @param baseUrl {String} Base URL
   * @param name {String} Target name
   */
  function UpdateArtifactoryTarget(artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
    _classCallCheck(this, UpdateArtifactoryTarget);

    UpdateArtifactoryTarget.initialize(this, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UpdateArtifactoryTarget, null, [{
    key: "initialize",
    value: function initialize(obj, artifactoryAdminName, artifactoryAdminPwd, baseUrl, name) {
      obj['artifactory-admin-name'] = artifactoryAdminName;
      obj['artifactory-admin-pwd'] = artifactoryAdminPwd;
      obj['base-url'] = baseUrl;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UpdateArtifactoryTarget</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateArtifactoryTarget} obj Optional instance to populate.
     * @return {module:model/UpdateArtifactoryTarget} The populated <code>UpdateArtifactoryTarget</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateArtifactoryTarget();

        if (data.hasOwnProperty('artifactory-admin-name')) {
          obj['artifactory-admin-name'] = _ApiClient["default"].convertToType(data['artifactory-admin-name'], 'String');
        }

        if (data.hasOwnProperty('artifactory-admin-pwd')) {
          obj['artifactory-admin-pwd'] = _ApiClient["default"].convertToType(data['artifactory-admin-pwd'], 'String');
        }

        if (data.hasOwnProperty('base-url')) {
          obj['base-url'] = _ApiClient["default"].convertToType(data['base-url'], 'String');
        }

        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }

        if (data.hasOwnProperty('key')) {
          obj['key'] = _ApiClient["default"].convertToType(data['key'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('new-name')) {
          obj['new-name'] = _ApiClient["default"].convertToType(data['new-name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('update-version')) {
          obj['update-version'] = _ApiClient["default"].convertToType(data['update-version'], 'Boolean');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UpdateArtifactoryTarget;
}();
/**
 * Artifactory Admin Name
 * @member {String} artifactory-admin-name
 */


UpdateArtifactoryTarget.prototype['artifactory-admin-name'] = undefined;
/**
 * Artifactory Admin password
 * @member {String} artifactory-admin-pwd
 */

UpdateArtifactoryTarget.prototype['artifactory-admin-pwd'] = undefined;
/**
 * Base URL
 * @member {String} base-url
 */

UpdateArtifactoryTarget.prototype['base-url'] = undefined;
/**
 * Comment about the target
 * @member {String} comment
 */

UpdateArtifactoryTarget.prototype['comment'] = undefined;
/**
 * The name of a key that used to encrypt the target secret value (if empty, the account default protectionKey key will be used)
 * @member {String} key
 */

UpdateArtifactoryTarget.prototype['key'] = undefined;
/**
 * Target name
 * @member {String} name
 */

UpdateArtifactoryTarget.prototype['name'] = undefined;
/**
 * New target name
 * @member {String} new-name
 */

UpdateArtifactoryTarget.prototype['new-name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UpdateArtifactoryTarget.prototype['password'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UpdateArtifactoryTarget.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UpdateArtifactoryTarget.prototype['uid-token'] = undefined;
/**
 * Create new version for the target
 * @member {Boolean} update-version
 * @default false
 */

UpdateArtifactoryTarget.prototype['update-version'] = false;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UpdateArtifactoryTarget.prototype['username'] = undefined;
var _default = UpdateArtifactoryTarget;
exports["default"] = _default;
