// akeyless/dist/model/UploadRSA.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UploadRSA model module.
 * @module model/UploadRSA
 * @version 2.15.24
 */
var UploadRSA = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UploadRSA</code>.
   * @alias module:model/UploadRSA
   * @param alg {String} Key type. options: [RSA1024, RSA2048, RSA3072, RSA4096]
   * @param name {String} Name of key to be created
   */
  function UploadRSA(alg, name) {
    _classCallCheck(this, UploadRSA);

    UploadRSA.initialize(this, alg, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UploadRSA, null, [{
    key: "initialize",
    value: function initialize(obj, alg, name) {
      obj['alg'] = alg;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>UploadRSA</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UploadRSA} obj Optional instance to populate.
     * @return {module:model/UploadRSA} The populated <code>UploadRSA</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UploadRSA();

        if (data.hasOwnProperty('alg')) {
          obj['alg'] = _ApiClient["default"].convertToType(data['alg'], 'String');
        }

        if (data.hasOwnProperty('cert-file-data')) {
          obj['cert-file-data'] = _ApiClient["default"].convertToType(data['cert-file-data'], 'String');
        }

        if (data.hasOwnProperty('customer-frg-id')) {
          obj['customer-frg-id'] = _ApiClient["default"].convertToType(data['customer-frg-id'], 'String');
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('password')) {
          obj['password'] = _ApiClient["default"].convertToType(data['password'], 'String');
        }

        if (data.hasOwnProperty('rsa-file-data')) {
          obj['rsa-file-data'] = _ApiClient["default"].convertToType(data['rsa-file-data'], 'String');
        }

        if (data.hasOwnProperty('split-level')) {
          obj['split-level'] = _ApiClient["default"].convertToType(data['split-level'], 'Number');
        }

        if (data.hasOwnProperty('tag')) {
          obj['tag'] = _ApiClient["default"].convertToType(data['tag'], ['String']);
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('username')) {
          obj['username'] = _ApiClient["default"].convertToType(data['username'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UploadRSA;
}();
/**
 * Key type. options: [RSA1024, RSA2048, RSA3072, RSA4096]
 * @member {String} alg
 */


UploadRSA.prototype['alg'] = undefined;
/**
 * Certificate in a PEM format.
 * @member {String} cert-file-data
 */

UploadRSA.prototype['cert-file-data'] = undefined;
/**
 * The customer fragment ID that will be used to split the key (if empty, the key will be created independently of a customer fragment)
 * @member {String} customer-frg-id
 */

UploadRSA.prototype['customer-frg-id'] = undefined;
/**
 * A metadata about the key
 * @member {String} metadata
 */

UploadRSA.prototype['metadata'] = undefined;
/**
 * Name of key to be created
 * @member {String} name
 */

UploadRSA.prototype['name'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} password
 */

UploadRSA.prototype['password'] = undefined;
/**
 * RSA private key data, base64 encoded
 * @member {String} rsa-file-data
 */

UploadRSA.prototype['rsa-file-data'] = undefined;
/**
 * The number of fragments that the item will be split into
 * @member {Number} split-level
 * @default 2
 */

UploadRSA.prototype['split-level'] = 2;
/**
 * List of the tags attached to this key
 * @member {Array.<String>} tag
 */

UploadRSA.prototype['tag'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

UploadRSA.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

UploadRSA.prototype['uid-token'] = undefined;
/**
 * Required only when the authentication process requires a username and password
 * @member {String} username
 */

UploadRSA.prototype['username'] = undefined;
var _default = UploadRSA;
exports["default"] = _default;
