// akeyless/dist/model/VerifyPKICertWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VerifyPKICertWithClassicKey model module.
 * @module model/VerifyPKICertWithClassicKey
 * @version 2.20.0
 */
var VerifyPKICertWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerifyPKICertWithClassicKey</code>.
   * @alias module:model/VerifyPKICertWithClassicKey
   * @param displayId {String} The name of the key to use in the verify PKICert process
   * @param pkiCert {String} PkiCert
   * @param version {Number} classic key version
   */
  function VerifyPKICertWithClassicKey(displayId, pkiCert, version) {
    _classCallCheck(this, VerifyPKICertWithClassicKey);

    VerifyPKICertWithClassicKey.initialize(this, displayId, pkiCert, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VerifyPKICertWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, displayId, pkiCert, version) {
      obj['display-id'] = displayId;
      obj['pki-cert'] = pkiCert;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>VerifyPKICertWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerifyPKICertWithClassicKey} obj Optional instance to populate.
     * @return {module:model/VerifyPKICertWithClassicKey} The populated <code>VerifyPKICertWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerifyPKICertWithClassicKey();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('pki-cert')) {
          obj['pki-cert'] = _ApiClient["default"].convertToType(data['pki-cert'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return VerifyPKICertWithClassicKey;
}();
/**
 * The name of the key to use in the verify PKICert process
 * @member {String} display-id
 */


VerifyPKICertWithClassicKey.prototype['display-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

VerifyPKICertWithClassicKey.prototype['json'] = undefined;
/**
 * PkiCert
 * @member {String} pki-cert
 */

VerifyPKICertWithClassicKey.prototype['pki-cert'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

VerifyPKICertWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

VerifyPKICertWithClassicKey.prototype['uid-token'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

VerifyPKICertWithClassicKey.prototype['version'] = undefined;
var _default = VerifyPKICertWithClassicKey;
exports["default"] = _default;
