// akeyless/dist/model/VerifyJWTWithClassicKey.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The VerifyJWTWithClassicKey model module.
 * @module model/VerifyJWTWithClassicKey
 * @version 2.20.0
 */
var VerifyJWTWithClassicKey = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>VerifyJWTWithClassicKey</code>.
   * @alias module:model/VerifyJWTWithClassicKey
   * @param displayId {String} The name of the key to use in the verify JWT process
   * @param jwt {String} JWT
   * @param requiredClaims {String} RequiredClaims
   * @param version {Number} classic key version
   */
  function VerifyJWTWithClassicKey(displayId, jwt, requiredClaims, version) {
    _classCallCheck(this, VerifyJWTWithClassicKey);

    VerifyJWTWithClassicKey.initialize(this, displayId, jwt, requiredClaims, version);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(VerifyJWTWithClassicKey, null, [{
    key: "initialize",
    value: function initialize(obj, displayId, jwt, requiredClaims, version) {
      obj['display-id'] = displayId;
      obj['jwt'] = jwt;
      obj['required-claims'] = requiredClaims;
      obj['version'] = version;
    }
    /**
     * Constructs a <code>VerifyJWTWithClassicKey</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/VerifyJWTWithClassicKey} obj Optional instance to populate.
     * @return {module:model/VerifyJWTWithClassicKey} The populated <code>VerifyJWTWithClassicKey</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new VerifyJWTWithClassicKey();

        if (data.hasOwnProperty('display-id')) {
          obj['display-id'] = _ApiClient["default"].convertToType(data['display-id'], 'String');
        }

        if (data.hasOwnProperty('json')) {
          obj['json'] = _ApiClient["default"].convertToType(data['json'], 'Boolean');
        }

        if (data.hasOwnProperty('jwt')) {
          obj['jwt'] = _ApiClient["default"].convertToType(data['jwt'], 'String');
        }

        if (data.hasOwnProperty('required-claims')) {
          obj['required-claims'] = _ApiClient["default"].convertToType(data['required-claims'], 'String');
        }

        if (data.hasOwnProperty('token')) {
          obj['token'] = _ApiClient["default"].convertToType(data['token'], 'String');
        }

        if (data.hasOwnProperty('uid-token')) {
          obj['uid-token'] = _ApiClient["default"].convertToType(data['uid-token'], 'String');
        }

        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return VerifyJWTWithClassicKey;
}();
/**
 * The name of the key to use in the verify JWT process
 * @member {String} display-id
 */


VerifyJWTWithClassicKey.prototype['display-id'] = undefined;
/**
 * Set output format to JSON
 * @member {Boolean} json
 */

VerifyJWTWithClassicKey.prototype['json'] = undefined;
/**
 * JWT
 * @member {String} jwt
 */

VerifyJWTWithClassicKey.prototype['jwt'] = undefined;
/**
 * RequiredClaims
 * @member {String} required-claims
 */

VerifyJWTWithClassicKey.prototype['required-claims'] = undefined;
/**
 * Authentication token (see `/auth` and `/configure`)
 * @member {String} token
 */

VerifyJWTWithClassicKey.prototype['token'] = undefined;
/**
 * The universal identity token, Required only for universal_identity authentication
 * @member {String} uid-token
 */

VerifyJWTWithClassicKey.prototype['uid-token'] = undefined;
/**
 * classic key version
 * @member {Number} version
 */

VerifyJWTWithClassicKey.prototype['version'] = undefined;
var _default = VerifyJWTWithClassicKey;
exports["default"] = _default;
